include: "rules/common.smk"

##### Target rules #####

rule all:
    input:
        expand("/project2/yangili1/bjf79/FastqForTony/{PowerAnalysisSample}.fastq.gz", PowerAnalysisSample=PowerAnalysisFastqFrame.index),
        "DataForSubmission/EVA/ToSubmit.vcf.gz",
        "MiscOutput/ChrConversionKey.tab",
        "Misc/EnsemblRef/Ref.fa.fai",
        "DataForSubmission/GEO/FastqChecksums.txt",
        "DataForSubmission/GEO/ChimpanzeesQQNorm_eQTL_Mapping.tsv.gz",
        "../../output/TSP.eQTLs.GTEx.AllTissues.txt.gz",
        expand("../../output/Final/FigS_Power_NumDE_{ReadDepthKey}.pdf", ReadDepthKey=config["PowerAnalysis"]["DepthsToTest"]+["Full"]),
        expand("../../output/Overdispersion{DropSampleList}_P_SE_Combined.txt.gz", DropSampleList=list(samples_dispersion["WildcardIdentifier"])),
        "PowerAnalysis/BootstrapRepsMerged.txt.gz",
        "../../output/OverdispersionEstimatesFromChimp.txt.Pvals.tab",
        "../../output/Final/TableS2.tab",
        "Dispersion/BoostrapSE.Chimp.PermutationsCombined.txt",
        "Dispersion/BoostrapInference.PermutationsCombined.txt",
        "MiscOutput/ExonicRegions.chimp.bed",
        "../../output/QC/RNASeqMultiQC.stats.tab",
        "MiscOutput/CallableSites.summary",
        "../../output/GorillaChimpHumanFixedSubstitutionsCount.gz",
        "../../output/NeutralityIndex/chimp/ChimpsThisStudy/PnPs.txt",
        "../../output/NeutralityIndex/chimp/ChimpsAllTroglodytes/PnPs.txt",
        "../../output/NeutralityIndex/human/AllGtex/PnPs.txt",
        "../../output/NeutralityIndex/human/HumansGtexSubsample/PnPs.txt",
        "MiscOutput/ExonicRegions.human.bed",
        "../../output/LeflerTestedSnps.Chimp.cis.tsv",
        # "../../output/QQPlot.png",
        # "eQTL_mapping/MatrixEQTL/ConfigCovariateModelResults/Results.Permuted.txt",
        "../../output/SharedSnpExampleBoxplot.human.expression.txt",
        "../../output/LeflerTestedSnps.Human.snploc.txt",
        "../../output/LeflerTestedSnps.tsv",
        "../../data/Appris.principal.isoforms.txt",
        # "../../output/GorillaChimpHumanFixedSubstitutionsCount.gz",
        # "vep/vcf/Chimp.Ref_Plus_ThisStudy.vcf.gz",
        "../../output/TissueSpecificity/CellTypeSpecificity.TabulaMurisHeart.z.log.txt",
        "../../output/ChimpEgenes.testsnps.txt.gz",
        "../../data/TSS_annotations/TataAnnotations.txt.gz",
        "../../output/OverdispersionEstimatesFromChimp_NoVirusChallangedIndividuals.NoLengthNorm.txt",
        expand("../../output/GtexTissueMatrix.mu.NoLengthNorm.{listname}.txt.gz", listname=["40EachTissue","AllModeExtractionSamples"]),
        "OverdispersionAnalysis/TableSubset.txt",
        expand("../../output/GTEX_renalysis/SampleSize_{n}.txt.gz", n=config["GTEx_eQTL_mapping"]["variable_sample_sizes"]),
        # expand("GTEX_renalysis/Subsamples/output/Subsample_{subsample}.txt.gz", subsample=range(1, Num_subsamples + 1)),
        expand("PopulationSubstructure/ReferencePanelMerged.annotated.splits/{chromosome}.vcf", chromosome=contigs),
        "../../output/ChimpEgenes.eigenMT.txt.gz",
        # "OverdispersionAnalysis/GtexTissueMatrix.overdispersion.txt",
        "OverdispersionAnalysis/GTEx.genelengths.txt",
        # "eQTL_mapping/SharedPolymorphisms/MatchedSNPsToTest.vcf.gz",
        "OverdispersionAnalysis/FullCountTable.txt.gz",
        "eQTL_mapping/MatrixEQTL/ConfigCovariateModelResults/PermutationsCombined.txt",
        "../../output/TissueSpecificity/tau.log.txt",
        "MiscOutput/HumanVcfs/YRI.hg38.vcf.gz",
        "eQTL_mapping/SharedPolymorphisms/SpeciesSharedSnps.PanTro5.snps",
        "eQTL_mapping/SharedPolymorphisms/SpeciesSharedSnps.PanTro5.snploc",
        "eQTL_mapping/SharedPolymorphisms/SpeciesSharedMatchedSnps.PanTro5.snploc",
        "MiscOutput/allele_encodings.txt.gz",
        # "MiscOutput/HumanLocusZoomPlot",
        "../../output/SharedPolymorphisms.shared.chimpeqtls.txt",
        # "MiscOutput/HumanVcfs/YRI.vcf.gz",
        "MiscOutput/HumanVcfs/GEUVADIS.vcf.gz",
        "../../output/SharedPolymorphisms.shared.chimpeqtls.txt",
        # "MiscOutput/LocusZoomPlot",
        "MiscOutput/ForAssociationTesting.chromsrenamed.vcf.gz",
        "eQTL_mapping/MatrixEQTL/ConfigCovariateModelResults/Results.txt",
        expand("PowerAnalysis/Subread/{species}.{depth}.subread.txt.gz", species=["Human", "Chimp"], depth=config["PowerAnalysis"]["DepthsToTest"]),
        expand(config["gitinclude_output"] + "PowerAnalysisFullCountTable.{species}.subread.txt.gz", species=["Human", "Chimp"]),
        "PowerAnalysis/Multiqc/multiqc_report.html",
        # expand(config["temp_files_prefix"] + "PowerAnalysisBams/{PowerAnalysisSample}/Aligned.sortedByCoord.out.bam", PowerAnalysisSample=PowerAnalysisFastqFrame.index),
        # expand(config["temp_files_prefix"] + "PowerAnalysisFastq/{PowerAnalysisSample}.fastq.gz", PowerAnalysisSample=PowerAnalysisFastqFrame.index),
        "sQTL_mapping/MatrixEQTL/Results/images/PCsVsSQTLs.pdf",
        config["gitinclude_output"] + "MatrixEQTL_sig_genotypes.raw",
        # "logs/eQTL_mapping/collect_Gemma_batchlogs.log",
        "eQTL_mapping/MatrixEQTL/Results/images/PCsVsEQTLs.pdf",
        "eQTL_mapping/Kinship/GRM.cXX.txt",
        "eQTL_mapping/FastQTL/ForAssociationTesting.bed.gz.tbi",
        "eQTL_mapping/MatrixEQTL/ForAssociationTesting.snploc",
        "../../output/RNA_seq_FlowCellInfo.txt",
        # expand ("qc/verifybamid/{sample}.bestSM", sample=samples["sample"] ),
        "eQTL_mapping/MatrixEQTL/ForAssociationTesting.geneloc.txt",
        "eQTL_mapping/MatrixEQTL/ForAssociationTesting.phenotypes.txt",
        # "eQTL_mapping/batchscripts/FullGetCisSNPsJobList.sh",
        # "logs/eQTL_mapping/collect_GetCisSNPs_batchlogs.log",
        "eQTL_mapping/MatrixEQTL/ForAssociationTesting.snps",
        "filtered/all.vcf.gz",
        config['gitinclude_output'] + "ForAssociationTesting.phenotypes.txt",
        # "PopulationSubstructure/ReferencePanelMerged.vcf",
        # "MiscOutput/LowComplexityRegions.bed",
        # "PopulationSubstructure/out.relatedness2",
        # "MiscOutput/filtered.bcftools.stats",
        # "MiscOutput/filtered.bcftools.smplstats",
        # "MiscOutput/STAR_index/chrLength.txt",
        expand("RNASeq/STAR/{sample}/Aligned.sortedByCoord.out.bam", sample=samples["sample"]),
        expand("RNASeq/STAR/{sample}/Aligned.sortedByCoord.out.bam.bai", sample=samples["sample"]),
        #"annotated/all.vcf.gz",
        config["gitinclude_output"] + "CountTable.SeparatedByFastq.tpm.txt.gz",
        "eQTL_mapping/plink/ForAssociationTesting.fam",
        config["gitinclude_output"] + "CountTable.tpm.txt.gz",
        config["gitinclude_output"] + "PopulationStructure/pca.eigenval",
        config["gitinclude_output"] + "PopulationStructure/pca.eigenvec",
        "PopulationSubstructure/plink/plink.eigenvec.var",
        # "qc/multiqc.html",
        # "MiscOutput/CallableSites.bed",
        # "genotyped/all.5.20000000-30000000.vcf",
        # expand("dedup_merged/{sample}.sorted.bam.bai", sample=samples["sample"]),
        # "filtered/all.vcf.gz",
        # "filtered/all.vcf.gz.tbi",
        # "MiscOutput/filtered.bcftools.stats",
        #"tables/calls.tsv.gz",
        #"plots/depths.svg",
        #"plots/allele-freqs.svg"
        # expand ( "PopulationSubstructure/Admixture/MergedForAdmixture.{K}.Q.pdf", K=config["PopulationSubstructure"]["AdmixtureK"] ),
        # expand(config["gitinclude_output"] + "PopulationStructure/Admixture/MergedForAdmixture.{K}.Q.labelled", K=config["PopulationSubstructure"]["AdmixtureK"]),
        "eQTL_mapping/Misc/cis_gene_windows.bed",
        "eQTL_mapping/plink/ForAssociationTesting.bed",
        config['gitinclude_output'] + "ForAssociationTesting.temp.fam",
        "eQTL_mapping/plink/ForAssociationTesting.bim",
        "AddressReviews/plink/GTEX.v8.bed",
        "AddressReviews/GWAS/GTEX.v8.lmm.cov",
        "output/result.assoc.txt.gz",
        "Misc/GtexDispersionGenes.hg38.bed",
        "../../output/STAR.RawCountTable.txt.gz",
        "../../output/CellProportionGWAS.RandomControlloci.closestGenes.txt",
        "../../output/CellTypeDispersion.SE.tsv.gz"




##### Modules #####
include: "rules/make_test_data.smk"
include: "rules/mapping.smk"
include: "rules/RNASeqMapping.smk"
include: "rules/calling.smk"
include: "rules/populationstructure.smk"
#include: "rules/stats.smk"
include: "rules/eqtl_calling.smk"
include: "rules/sqtl_calling.smk"
include: "rules/PowerAnalysis.smk"
include: "rules/eQTL_analysis.smk"
include: "rules/Overdispersion.smk"
include: "rules/GTExReanalysis.smk"
include: "rules/GTExReanalysis_constantSampleSize.smk"
include: "rules/vep.smk"
include: "rules/CalculateSubstitutionsWithGorilla.smk"
include: "rules/qc.smk"
include: "rules/MakeSupplementalTables.smk"
include: "rules/OrganizeDataSubmission.smk"
include: "rules/AddressReviews.smk"
#include: "rules/annotation.smk"
