<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>Check RNA-seq PCs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Check RNA-seq PCs</h1>
<h4 class="author"><em>Ben Fair</em></h4>
<h4 class="date"><em>3/20/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-04-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtreefc37682444bda6468e6954c8273d93fdca385897targetblankfc37682a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/fc37682444bda6468e6954c8273d93fdca385897" target="_blank">fc37682</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtreefc37682444bda6468e6954c8273d93fdca385897targetblankfc37682a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  analysis/20190321_Check-Kinship-And-PopulationStructure.Rmd
    Untracked:  analysis/20190325_MergingRNASeqLanes.Rmd
    Untracked:  analysis/20190326_Admixture.Rmd
    Untracked:  analysis/20190326_PCA.Rmd
    Untracked:  analysis/20190327_MakeFamAndCovariateFiles.Rmd
    Untracked:  analysis/20190327_MakeFamPhenotypeFile.Rmd
    Untracked:  docs/figure/20190321_Check-Kinship-And-PopulationStructure.Rmd/
    Untracked:  docs/figure/20190325_MergingRNASeqLanes.Rmd/
    Untracked:  docs/figure/20190326_PCA.Rmd/

Unstaged changes:
    Deleted:    ._workflowr.yml.swp
    Modified:   analysis/20190320_Check-RNAseq-PCs.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(plyr)
library(reshape2)
library(tidyverse)
library(readxl)
library(knitr)
library(ggrepel)</code></pre>
<pre class="r"><code># Read in kinship data
YerkesKinship_from_pedigree &lt;- as.data.frame(read_excel(&quot;../data/Metadata.xlsx&quot;, sheet=&quot;Yerkes.coan&quot;, col_types = c(&quot;text&quot;, &quot;text&quot;, &quot;numeric&quot;)))
kable(head(YerkesKinship_from_pedigree))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ind1</th>
<th align="left">ind2</th>
<th align="right">coeff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">200</td>
<td align="left">200</td>
<td align="right">0.5</td>
</tr>
<tr class="even">
<td align="left">204</td>
<td align="left">200</td>
<td align="right">0.0</td>
</tr>
<tr class="odd">
<td align="left">295</td>
<td align="left">200</td>
<td align="right">0.0</td>
</tr>
<tr class="even">
<td align="left">299</td>
<td align="left">200</td>
<td align="right">0.0</td>
</tr>
<tr class="odd">
<td align="left">317</td>
<td align="left">200</td>
<td align="right">0.0</td>
</tr>
<tr class="even">
<td align="left">338</td>
<td align="left">200</td>
<td align="right">0.0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>Kinship_from_KING.WGS &lt;- read.table(&quot;../output/PopulationStructure/king.kin&quot;, header=T, stringsAsFactors = F)
kable(head(Kinship_from_KING.WGS))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">FID</th>
<th align="left">ID1</th>
<th align="left">ID2</th>
<th align="right">N_SNP</th>
<th align="right">Z0</th>
<th align="right">Phi</th>
<th align="right">HetHet</th>
<th align="right">IBS0</th>
<th align="right">Kinship</th>
<th align="right">Error</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="left">Little_R</td>
<td align="left">MD_And</td>
<td align="right">4901461</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.042</td>
<td align="right">0.0277</td>
<td align="right">-0.0526</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="left">Little_R</td>
<td align="left">4x373</td>
<td align="right">4901461</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.045</td>
<td align="right">0.0255</td>
<td align="right">-0.0250</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="left">Little_R</td>
<td align="left">4x523</td>
<td align="right">4901461</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.046</td>
<td align="right">0.0269</td>
<td align="right">-0.0206</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="left">Little_R</td>
<td align="left">4x0025</td>
<td align="right">4901461</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.044</td>
<td align="right">0.0262</td>
<td align="right">-0.0325</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="left">Little_R</td>
<td align="left">4x0043</td>
<td align="right">4901461</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.043</td>
<td align="right">0.0281</td>
<td align="right">-0.0548</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="left">Little_R</td>
<td align="left">4X0095</td>
<td align="right">4901461</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.042</td>
<td align="right">0.0296</td>
<td align="right">-0.0702</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Plot the kinship matrix obtained from Yerkes pedigree data, as well as the matrix obtained from whole genome SNP data…</p>
<pre class="r"><code>Yerkes.Matrix &lt;-acast(YerkesKinship_from_pedigree, ind1 ~ ind2, value.var=&quot;coeff&quot;, fill=0)
melt(Yerkes.Matrix) %&gt;%
  ggplot(aes(x=Var1, y=Var2, fill=value)) +
    geom_tile() +
    scale_fill_gradient(low=&quot;blue&quot;, high=&quot;red&quot;, limits=c(-0.5, 0.5)) +
    theme(text = element_text(size=4), axis.text.x = element_text(angle=90, hjust=1))</code></pre>
<p><img src="figure/20190321_Check-Kinship-And-PopulationStructure.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>WGS.Matrix &lt;- acast(Kinship_from_KING.WGS, ID1 ~ ID2, value.var=&quot;Kinship&quot;, fill=0) + acast(Kinship_from_KING.WGS, ID2 ~ ID1, value.var=&quot;Kinship&quot;, fill=0)
melt(WGS.Matrix) %&gt;%
  ggplot(aes(x=Var1, y=Var2, fill=value)) +
    geom_tile() +
    scale_fill_gradient(low=&quot;blue&quot;, high=&quot;red&quot;, limits=c(-0.5, 0.5)) +
    theme(text = element_text(size=7), axis.text.x = element_text(angle=90, hjust=1))</code></pre>
<p><img src="figure/20190321_Check-Kinship-And-PopulationStructure.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Let’s fix a known sample mislabel… 554_2 is actually 554. And 554 is actually unknown.</p>
<pre class="r"><code>Kinship_from_KING.WGS$ID1 &lt;- mapvalues(Kinship_from_KING.WGS$ID1, from=c(&quot;554_2&quot;), to=c(&quot;554&quot;))
Kinship_from_KING.WGS$ID2 &lt;- mapvalues(Kinship_from_KING.WGS$ID2, from=c(&quot;554_2&quot;), to=c(&quot;554&quot;))</code></pre>
<p>Now I want to correlate the kinship-coefficients from Yerkes pedigree to those from the whole genome SNP data (KING algorithm). Easiest way I could think to match up the pairwise kinship coefficients is to make a new field that is the sorted ID pair and then merge the Yerkes coefficient-table to the KING-table by the sorted ID pairs…</p>
<pre class="r"><code>KSort &lt;- Kinship_from_KING.WGS %&gt;%
  mutate(Teams = paste(pmin(ID1, ID2), pmax(ID1, ID2), sep= &quot; - &quot;))
YSort &lt;- YerkesKinship_from_pedigree %&gt;%
  mutate(Teams = paste(pmin(ind1, ind2), pmax(ind1, ind2), sep= &quot; - &quot;)) %&gt;%
  distinct(Teams, .keep_all=T)
# Merge and plot
Merged &lt;- merge(KSort, YSort, by=&quot;Teams&quot;)
Merged$label &lt;- Merged$Team
Merged$label[Merged$coeff==0] &lt;- &quot;&quot;

ggplot(Merged, aes(x=coeff, y=Kinship, label=label)) +
  geom_point() +
  geom_abline(slope=1, intercept=0, color=&quot;red&quot;) +
  geom_text_repel(size=2.5) +
  xlab(&quot;Kinship from Yerkes pedigree&quot;) +
  ylab(&quot;Kinship from WGS (KING algorithm)&quot;)</code></pre>
<p><img src="figure/20190321_Check-Kinship-And-PopulationStructure.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # geom_jitter()</code></pre>
<p>The expected value of Kinships coefficients from KING seem to match pedigree info, with the exception of sample 554 (not to be confused with 554_2 which for purposes of matching KING coefficients to pedigree coefficients we relabelled as 554) which we had prior knowledge to be a mislabelled sample… Hence why 554 is unrelated to a point labelled 554 (actually 554_2). Assuming sample 554 came from Yerkes, we might be able to correlate its kinship coefficients to other Yerkes chimps to help identify it. Bryan Pavlovic already did this for me by looking at the kinship matrices by hand and concluded that 554 is most likely Booka, but here I will repeat that analysis…</p>
<pre class="r"><code># Find who 554 is related to from whole genome SNP data.
WGS.Matrix[&#39;554&#39;,] %&gt;%
  sort(decreasing=T) %&gt;%
  head() %&gt;%
  t() %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">495</th>
<th align="right">4x0430</th>
<th align="right">295</th>
<th align="right">4x0519</th>
<th align="right">529</th>
<th align="right">4x523</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.0936</td>
<td align="right">0.0561</td>
<td align="right">0.0463</td>
<td align="right">0.0356</td>
<td align="right">0.0348</td>
<td align="right">0.0328</td>
</tr>
</tbody>
</table>
<p>…554 looks related to 495 (Amos), perhaps 2nd or 3rd degree relationship</p>
<pre class="r"><code># Find who to Amos is related from Yerkes Pedigree
Yerkes.Matrix[&#39;495&#39;,] %&gt;%
  sort(decreasing=T) %&gt;%
  head(20) %&gt;%
  t() %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">495</th>
<th align="right">724</th>
<th align="right">462</th>
<th align="right">646</th>
<th align="right">650</th>
<th align="right">535</th>
<th align="right">726</th>
<th align="right">200</th>
<th align="right">204</th>
<th align="right">295</th>
<th align="right">299</th>
<th align="right">317</th>
<th align="right">338</th>
<th align="right">380</th>
<th align="right">389</th>
<th align="right">421</th>
<th align="right">425</th>
<th align="right">431</th>
<th align="right">434</th>
<th align="right">438</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.5</td>
<td align="right">0.25</td>
<td align="right">0.125</td>
<td align="right">0.125</td>
<td align="right">0.125</td>
<td align="right">0.0625</td>
<td align="right">0.0625</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
Can eliminate 495, 724, 462 as they are all also part of my cohort, and not related to 554. <br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggrepel_0.8.0   knitr_1.22      readxl_1.1.0    forcats_0.4.0  
 [5] stringr_1.4.0   dplyr_0.8.0.1   purrr_0.3.2     readr_1.3.1    
 [9] tidyr_0.8.2     tibble_2.1.1    ggplot2_3.1.0   tidyverse_1.2.1
[13] reshape2_1.4.3  plyr_1.8.4     

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5 xfun_0.6         haven_2.1.0      lattice_0.20-38 
 [5] colorspace_1.4-1 generics_0.0.2   htmltools_0.3.6  yaml_2.2.0      
 [9] rlang_0.3.3      pillar_1.3.1     glue_1.3.1       withr_2.1.2     
[13] modelr_0.1.4     munsell_0.5.0    gtable_0.3.0     workflowr_1.2.0 
[17] cellranger_1.1.0 rvest_0.3.2      evaluate_0.13    labeling_0.3    
[21] highr_0.8        broom_0.5.1      Rcpp_1.0.1       scales_1.0.0    
[25] backports_1.1.3  jsonlite_1.6     fs_1.2.6         hms_0.4.2       
[29] digest_0.6.18    stringi_1.4.3    grid_3.5.1       rprojroot_1.3-2 
[33] cli_1.1.0        tools_3.5.1      magrittr_1.5     lazyeval_0.2.2  
[37] crayon_1.3.4     pkgconfig_2.0.2  xml2_1.2.0       lubridate_1.7.4 
[41] assertthat_0.2.1 rmarkdown_1.11   httr_1.4.0       rstudioapi_0.10 
[45] R6_2.4.0         nlme_3.1-137     git2r_0.24.0     compiler_3.5.1  </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
