<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>20190627_DiffContactsWithEgenes</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20190627_DiffContactsWithEgenes</h1>
<h4 class="author">Ben Fair</h4>
<h4 class="date">6/27/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-08-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtreee9747fe1160eea2ddd59d851d92d4f18ddd9d35dtargetblanke9747fea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/e9747fe1160eea2ddd59d851d92d4f18ddd9d35d" target="_blank">e9747fe</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtreee9747fe1160eea2ddd59d851d92d4f18ddd9d35dtargetblanke9747fea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/20190521_eQTL_CrossSpeciesEnrichment_cache/
    Ignored:    analysis/figure/
    Ignored:    analysis_temp/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/snakemake_workflow/.DS_Store
    Ignored:    code/snakemake_workflow/rules/.eQTL_analysis.smk.swp
    Ignored:    data/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    docs/assets/.DS_Store

Unstaged changes:
    Modified:   analysis/20190521_eQTL_CrossSpeciesEnrichment.Rmd
    Modified:   analysis/20190627_DiffContactsEgenes.Rmd
    Modified:   analysis/20190708_DiffContactsEgenes_CisWindowControlled.Rmd
    Modified:   analysis/20190716_VarianceInsteadOfEgenes.Rmd
    Modified:   analysis/index.Rmd
    Modified:   code/snakemake_workflow/Session.vim
    Modified:   code/snakemake_workflow/rules/eQTL_analysis.smk

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/34e6b4f4c30464c6c32619d985b9663b392de4fb/analysis/20190627_DiffContactsEgenes.Rmd" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/34e6b4f4c30464c6c32619d985b9663b392de4fb/docs/20190627_DiffContactsEgenes.html" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
<td>
update site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(&quot;edgeR&quot;)
library(corrplot)
library(gplots)
library(pROC)
library(qvalue)
library(reshape2)
library(gridExtra)</code></pre>
<p>As an initial pass at testing the hypothesis that Chimp-specific cis-eGenes have more DNA contacts at those cis-windows than humans in those same cis-windows, I looked at chimp/human differential contact windows from Eres et al and intersected in with my list of Chimp eGene locations (+/- 100kb). Then I looked at the sum of effect sizes within each of those windows.</p>
<pre class="r"><code>DataIn &lt;- read.table(&quot;../data/DCContactsInEgenes.bed&quot;, col.names = c(&quot;chr&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;gene&quot;, &quot;blank&quot;, &quot;strand&quot;, &quot;chrContact&quot;, &quot;startContact&quot;, &quot;beta&quot;, &quot;nameContact&quot;, &quot;beta&quot;, &quot;strandContact&quot;), sep=&#39;\t&#39;, row.names = NULL)

GroupedData &lt;- DataIn %&gt;%
  group_by(stop) %&gt;%
  summarise(BetaSum = sum(beta))

ggplot(GroupedData, aes(x=BetaSum)) +
  stat_ecdf(geom = &quot;step&quot;) +
  xlab(&quot;Differential contact effect size sum over Chimp eGenes cis-window\n(Negative means more contact in chimp)&quot;) +
  ylab(&quot;Cumulative frequency&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/34e6b4f4c30464c6c32619d985b9663b392de4fb/docs/figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-1-1.png" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>wilcox.test(GroupedData$BetaSum, mu = 0, alternative = &quot;less&quot;)</code></pre>
<pre><code>
    Wilcoxon signed rank test

data:  GroupedData$BetaSum
V = 135, p-value = 0.02245
alternative hypothesis: true location is less than 0</code></pre>
<p>Ok it seems there is a subtle but significant shift in that chimp eGenes have more differential contacts a way that slightly favors high connectivity in chimps.</p>
<p>To explore this more carefully, Ittai gave me a list of Homer-normalized contact data (not just significant differential contacts and effect sizes) for all his 8 samples (4 human, 4 chimp) for the cis-window surrounding each eGene (+/- 250kb).</p>
<p>From this I can estimate connectivity within a cis window as the sum of all Homer normalized contact scores within a species. Then I will look at the difference between that sum between species, and ask if it is correlated with some species measure of eGene character.</p>
<p>First, read in Ittai’s Homer normalized contact data for each individual in both chimp and human…</p>
<pre class="r"><code>SampleA&lt;- read.csv(gzfile(&quot;../data/IttaiHomerInteractionScoresInCisWindows/adj_bins_25_A-21792_10kb_norm.gz&quot;), sep=&#39;\t&#39;)
SampleB&lt;- read.csv(gzfile(&quot;../data/IttaiHomerInteractionScoresInCisWindows/adj_bins_25_B-28126_10kb_norm.gz&quot;), sep=&#39;\t&#39;)
SampleC&lt;- read.csv(gzfile(&quot;../data/IttaiHomerInteractionScoresInCisWindows/adj_bins_25_C-3649_10kb_norm.gz&quot;), sep=&#39;\t&#39;)
SampleD&lt;- read.csv(gzfile(&quot;../data/IttaiHomerInteractionScoresInCisWindows/adj_bins_25_D-40300_10kb_norm.gz&quot;), sep=&#39;\t&#39;)
SampleE&lt;- read.csv(gzfile(&quot;../data/IttaiHomerInteractionScoresInCisWindows/adj_bins_25_E-28815_10kb_norm.gz&quot;), sep=&#39;\t&#39;)
SampleF&lt;- read.csv(gzfile(&quot;../data/IttaiHomerInteractionScoresInCisWindows/adj_bins_25_F-28834_10kb_norm.gz&quot;), sep=&#39;\t&#39;)
SampleG&lt;- read.csv(gzfile(&quot;../data/IttaiHomerInteractionScoresInCisWindows/adj_bins_25_G-3624_10kb_norm.gz&quot;), sep=&#39;\t&#39;)
SampleH&lt;- read.csv(gzfile(&quot;../data/IttaiHomerInteractionScoresInCisWindows/adj_bins_25_H-3651_10kb_norm.gz&quot;), sep=&#39;\t&#39;)

ChimpInteractionSums &lt;- SampleC + SampleD + SampleG + SampleH
HumanInteractionSums &lt;- SampleA + SampleB + SampleE + SampleF


HumanInteractions &lt;- data.frame(H.Score = rowSums(cbind(SampleA, SampleB, SampleE, SampleF))) %&gt;%
  rownames_to_column() %&gt;%
  mutate(HumanID = gsub(&quot;(.+?)\\..+?&quot;, &quot;\\1&quot;, rowname, perl=T))


ChimpInteractions &lt;- data.frame(C.Score = rowSums(cbind(SampleC, SampleD, SampleG, SampleH))) %&gt;%
  rownames_to_column(&quot;ChimpID&quot;)</code></pre>
<p>Ok now read in eQTL data…</p>
<pre class="r"><code>eQTLs &lt;- read.table(gzfile(&quot;../data/PastAnalysesDataToKeep/20190521_eQTLs_250kB_10MAF.txt.gz&quot;), header=T)

# List of chimp tested genes
ChimpTestedGenes &lt;- rownames(read.table(&#39;../output/ExpressionMatrix.un-normalized.txt.gz&#39;, header=T, check.names=FALSE, row.names = 1))

ChimpToHumanGeneMap &lt;- read.table(&quot;../data/Biomart_export.Hsap.Ptro.orthologs.txt.gz&quot;, header=T, sep=&#39;\t&#39;, stringsAsFactors = F)
kable(head(ChimpToHumanGeneMap))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Gene.stable.ID</th>
<th align="left">Transcript.stable.ID</th>
<th align="left">Chimpanzee.gene.stable.ID</th>
<th align="left">Chimpanzee.gene.name</th>
<th align="left">Chimpanzee.protein.or.transcript.stable.ID</th>
<th align="left">Chimpanzee.homology.type</th>
<th align="right">X.id..target.Chimpanzee.gene.identical.to.query.gene</th>
<th align="right">X.id..query.gene.identical.to.target.Chimpanzee.gene</th>
<th align="right">dN.with.Chimpanzee</th>
<th align="right">dS.with.Chimpanzee</th>
<th align="right">Chimpanzee.orthology.confidence..0.low..1.high.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000198888</td>
<td align="left">ENST00000361390</td>
<td align="left">ENSPTRG00000042641</td>
<td align="left">MT-ND1</td>
<td align="left">ENSPTRP00000061407</td>
<td align="left">ortholog_one2one</td>
<td align="right">94.6541</td>
<td align="right">94.6541</td>
<td align="right">0.0267</td>
<td align="right">0.5455</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">ENSG00000198763</td>
<td align="left">ENST00000361453</td>
<td align="left">ENSPTRG00000042626</td>
<td align="left">MT-ND2</td>
<td align="left">ENSPTRP00000061406</td>
<td align="left">ortholog_one2one</td>
<td align="right">96.2536</td>
<td align="right">96.2536</td>
<td align="right">0.0185</td>
<td align="right">0.7225</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000210127</td>
<td align="left">ENST00000387392</td>
<td align="left">ENSPTRG00000042642</td>
<td align="left">MT-TA</td>
<td align="left">ENSPTRT00000076396</td>
<td align="left">ortholog_one2one</td>
<td align="right">100.0000</td>
<td align="right">100.0000</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000198804</td>
<td align="left">ENST00000361624</td>
<td align="left">ENSPTRG00000042657</td>
<td align="left">MT-CO1</td>
<td align="left">ENSPTRP00000061408</td>
<td align="left">ortholog_one2one</td>
<td align="right">98.8304</td>
<td align="right">98.8304</td>
<td align="right">0.0065</td>
<td align="right">0.5486</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198712</td>
<td align="left">ENST00000361739</td>
<td align="left">ENSPTRG00000042660</td>
<td align="left">MT-CO2</td>
<td align="left">ENSPTRP00000061402</td>
<td align="left">ortholog_one2one</td>
<td align="right">97.7974</td>
<td align="right">97.7974</td>
<td align="right">0.0106</td>
<td align="right">0.5943</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">ENSG00000228253</td>
<td align="left">ENST00000361851</td>
<td align="left">ENSPTRG00000042653</td>
<td align="left">MT-ATP8</td>
<td align="left">ENSPTRP00000061400</td>
<td align="left">ortholog_one2one</td>
<td align="right">94.1176</td>
<td align="right">94.1176</td>
<td align="right">0.0325</td>
<td align="right">0.3331</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Of this ortholog list, how many genes are one2one
table(ChimpToHumanGeneMap$Chimpanzee.homology.type)</code></pre>
<pre><code>
ortholog_many2many  ortholog_one2many   ortholog_one2one 
              2278              19917             140351 </code></pre>
<pre class="r"><code>OneToOneMap &lt;- ChimpToHumanGeneMap %&gt;%
  filter(Chimpanzee.homology.type==&quot;ortholog_one2one&quot;)

# Read gtex heart egene list
# Only consider those that were tested in both species and are one2one orthologs
GtexHeartEgenes &lt;- read.table(&quot;../data/Heart_Left_Ventricle.v7.egenes.txt.gz&quot;, header=T, sep=&#39;\t&#39;, stringsAsFactors = F) %&gt;%
  mutate(gene_id_stable = gsub(&quot;.\\d+$&quot;,&quot;&quot;,gene_id)) %&gt;%
  filter(gene_id_stable %in% OneToOneMap$Gene.stable.ID) %&gt;%
  mutate(chimp_id = plyr::mapvalues(gene_id_stable, OneToOneMap$Gene.stable.ID,  OneToOneMap$Chimpanzee.gene.stable.ID, warn_missing = F)) %&gt;%
  filter(chimp_id %in% ChimpTestedGenes)

ChimpToHuman.ID &lt;- function(Chimp.ID){
  #function to convert chimp ensembl to human ensembl gene ids
  return(
    plyr::mapvalues(Chimp.ID, OneToOneMap$Chimpanzee.gene.stable.ID, OneToOneMap$Gene.stable.ID, warn_missing = F)
  )}</code></pre>
<p>First question: do the ~300 chimp eGenes have more contacts in their cis-window in chimp</p>
<pre class="r"><code>Chimp_OrderedGenes &lt;- eQTLs %&gt;%
    group_by(gene) %&gt;% 
    dplyr::slice(which.min(qvalue)) %&gt;%
    filter(gene %in% GtexHeartEgenes$chimp_id) %&gt;%
    left_join(GtexHeartEgenes, by=c(&quot;gene&quot;=&quot;chimp_id&quot;)) %&gt;%
    dplyr::select(gene, qvalue, qval) %&gt;% as.data.frame() %&gt;%
    mutate(ChimpRank = dense_rank(qvalue)) %&gt;%
    mutate(HumanRank = dense_rank(qval)) %&gt;%
    mutate(RankDifference = HumanRank-ChimpRank) %&gt;%
    filter(qvalue &lt;0.1) %&gt;%
    mutate(HumanID=ChimpToHuman.ID(gene))

# OneToOneMap %&gt;%
#   inner_join(HumanInteractions, by=c(&quot;Gene.stable.ID&quot;=&quot;HumanId&quot;)) %&gt;% dim()
#   inner_join(ChimpInteractions, by=c(&quot;Chimpanzee.gene.stable.ID&quot;=&quot;ChimpID&quot;)) %&gt;% dim()
#   right_join(Chimp_OrderedGenes, by=c(&quot;Chimpanzee.gene.stable.ID&quot;=&quot;gene&quot;)) %&gt;% dim()

Chimp_OrderedGenes.WithContactInfo &lt;- Chimp_OrderedGenes %&gt;%
  left_join(HumanInteractions, by=c(&quot;HumanID&quot;)) %&gt;%
  left_join(ChimpInteractions, by=c(&quot;gene&quot;=&quot;ChimpID&quot;)) %&gt;% 
  mutate(InteractionDifference=H.Score - C.Score)

  
ggplot(Chimp_OrderedGenes.WithContactInfo, aes(x=InteractionDifference)) +
  stat_ecdf(geom = &quot;step&quot;) +
  xlab(&quot;Difference in contacts over chimp eGene cis-windows\n(Positive means more contact in chimp)&quot;) +
  ylab(&quot;Cumulative frequency&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/34e6b4f4c30464c6c32619d985b9663b392de4fb/docs/figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-1.png" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(Chimp_OrderedGenes.WithContactInfo, aes(x=RankDifference, y=InteractionDifference)) +
  geom_point() +
  theme_bw() +
  xlab(&quot;Rank Difference in eGene significance\nMore in human &lt;--  --&gt; More in chimp&quot;) +
  ylab(&quot;Differential contacts in cis window\nMore in human &lt;--  --&gt; More in chimp&quot;) +
  geom_smooth(method=&#39;lm&#39;,formula=y~x)</code></pre>
<p><img src="figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/34e6b4f4c30464c6c32619d985b9663b392de4fb/docs/figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-2.png" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cor.test(x=Chimp_OrderedGenes.WithContactInfo$RankDifference, y=Chimp_OrderedGenes.WithContactInfo$InteractionDifference, method=&quot;spearman&quot;)</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  Chimp_OrderedGenes.WithContactInfo$RankDifference and Chimp_OrderedGenes.WithContactInfo$InteractionDifference
S = 656450, p-value = 0.01665
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.1839659 </code></pre>
<pre class="r"><code>contacts.v.eGene.lm = lm(InteractionDifference ~ RankDifference, data=Chimp_OrderedGenes.WithContactInfo)
summary(contacts.v.eGene.lm)</code></pre>
<pre><code>
Call:
lm(formula = InteractionDifference ~ RankDifference, data = Chimp_OrderedGenes.WithContactInfo)

Residuals:
     Min       1Q   Median       3Q      Max 
-127.372  -29.715   -2.727   33.539  222.680 

Coefficients:
                Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)    -5.776579   7.953704  -0.726   0.4687  
RankDifference  0.003498   0.001552   2.253   0.0255 *
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 55.08 on 167 degrees of freedom
  (111 observations deleted due to missingness)
Multiple R-squared:  0.02951,   Adjusted R-squared:  0.0237 
F-statistic: 5.078 on 1 and 167 DF,  p-value: 0.02553</code></pre>
<pre class="r"><code>plot(contacts.v.eGene.lm)</code></pre>
<p><img src="figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-3">
Past versions of unnamed-chunk-4-3.png
</button>
</p>
<div id="fig-unnamed-chunk-4-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/34e6b4f4c30464c6c32619d985b9663b392de4fb/docs/figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-3.png" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-4">
Past versions of unnamed-chunk-4-4.png
</button>
</p>
<div id="fig-unnamed-chunk-4-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/34e6b4f4c30464c6c32619d985b9663b392de4fb/docs/figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-4.png" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-5">
Past versions of unnamed-chunk-4-5.png
</button>
</p>
<div id="fig-unnamed-chunk-4-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/34e6b4f4c30464c6c32619d985b9663b392de4fb/docs/figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-5.png" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-6">
Past versions of unnamed-chunk-4-6.png
</button>
</p>
<div id="fig-unnamed-chunk-4-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/34e6b4f4c30464c6c32619d985b9663b392de4fb/docs/figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-4-6.png" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok yes, there is a slight correlation between chimp eGene character, and contacts in chimp. Now I am going to take a different approach to ask a similar question: Do lead chimp eQTL SNPs have more contacts between a gene and its promoter window than human.</p>
<p>First, I will read in genomic corrdinates of chimp genes, so I can identify which the correct hi-C interaction point between the SNP-containing 10kb window and the gene promoter for chimp data. Then, I will need to liftover the top eQTL-SNPs into human coordinates (and their human gene names) and identify the orthologous Hi-C interaction point. Finally, I will test if these distributions are significantly different, with the expectation that the chimp eQTL SNPs will have stronger contacts in chimp. The recipricol analysis (human centric) will also be useful.</p>
<pre class="r"><code>#read in gene coordinates of chimp genes
ChimpGeneLocs &lt;- read.table(&quot;../data/ChimpEgenes.bed&quot;, col.names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;)) %&gt;%
  mutate(TSS.Coord = case_when(
    strand == &quot;+&quot; ~ start,
    strand == &quot;-&quot; ~ stop
  ))

# Get lead snps for eQTLs, find the HiC bin number.
Top.Chimp.eQTL.SNPs &lt;- eQTLs %&gt;%
    group_by(gene) %&gt;%
    filter(qvalue &lt;0.1) %&gt;%
    dplyr::slice(which.min(qvalue)) %&gt;%
    filter(gene %in% GtexHeartEgenes$chimp_id) %&gt;%
    dplyr::select(c(&quot;snps&quot;, &quot;gene&quot;, &quot;qvalue&quot;)) %&gt;%
    mutate(snp.pos = as.numeric(gsub(&quot;ID.[0-9A-B]+.(\\d+).[GCTA]+.[GCTA]+&quot;, &quot;\\1&quot;, snps, perl=T))) %&gt;%
    left_join(ChimpGeneLocs, by=&quot;gene&quot;) %&gt;%
    filter(!is.na(TSS.Coord)) %&gt;%
    mutate(DistanceToTSS = snp.pos - TSS.Coord) %&gt;%
    mutate(ChimpHiC.bin=round(DistanceToTSS/10000)+26) %&gt;%
    filter(ChimpHiC.bin&gt;0 &amp; ChimpHiC.bin&lt;51)
  # #if minus strand to reverse
  #   mutate(ChimpHiC.bin2 = case_when(
  #     strand == &quot;+&quot; ~ ChimpHiC.bin,
  #     strand == &quot;-&quot; ~ 51-ChimpHiC.bin
  #   ))
  

#Check that eQTL snps enriched near TSS as expected
hist(Top.Chimp.eQTL.SNPs$DistanceToTSS, breaks=100, xlim=c(-250000,250000))</code></pre>
<p><img src="figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Without lifting over snp positions to human to get most comparable bins, just compare bins as is (I&#39;m guessing for most snps the lifted over bins will be the same).

ChimpeQTL.Interactions.Chimp &lt;- NULL
ChimpeQTL.Interactions.Human&lt;- NULL
for (i in 1:dim(Top.Chimp.eQTL.SNPs)[1]){
  # print(Top.Chimp.eQTL.SNPs$ChimpHiC.bin[i])
  ChimpInteractionScore &lt;- (ChimpInteractionSums[Top.Chimp.eQTL.SNPs$gene[i], Top.Chimp.eQTL.SNPs$ChimpHiC.bin[i]])
  HumanInteractionScore &lt;- (HumanInteractionSums[ChimpToHuman.ID(Top.Chimp.eQTL.SNPs$gene[i]), Top.Chimp.eQTL.SNPs$ChimpHiC.bin[i]])

  ChimpeQTL.Interactions.Chimp &lt;- append(ChimpeQTL.Interactions.Chimp, ChimpInteractionScore)
  ChimpeQTL.Interactions.Human &lt;- append(ChimpeQTL.Interactions.Human, HumanInteractionScore)

}

hist(ChimpeQTL.Interactions.Chimp - ChimpeQTL.Interactions.Human)</code></pre>
<p><img src="figure/20190627_DiffContactsEgenes.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Test if interactions for chimp lead snps to their promoter are systematically greater in chimp. Paired t-test seems appropriate.
t.test(ChimpeQTL.Interactions.Chimp,ChimpeQTL.Interactions.Human, paired=T)</code></pre>
<pre><code>
    Paired t-test

data:  ChimpeQTL.Interactions.Chimp and ChimpeQTL.Interactions.Human
t = -0.75752, df = 232, p-value = 0.4495
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.5115782  0.2274391
sample estimates:
mean of the differences 
             -0.1420696 </code></pre>
<p>Ok no signal to be seen. Still to do the analysis properly, I should liftover the lead chimp snps to human genome, and identify the relevant Hi-C interaction points from lifted snp positions. Additionally, I need to verify that the Hi-C matrix is always low-coordinate to high-coordinate, regardless of the gene orientation. Alternatively, one explanation is that species-specific eGenes have more contacts in their neghborhood (as the linear association above suggests) but the specific eQTL-promoter contacts are too noisy to detect. Alternatively, the linear association could be a false positive or associated by some alternate explanation that eludes me.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gridExtra_2.3   reshape2_1.4.3  qvalue_2.14.1   pROC_1.15.0    
 [5] gplots_3.0.1.1  corrplot_0.84   edgeR_3.24.3    limma_3.38.3   
 [9] knitr_1.23      forcats_0.4.0   stringr_1.4.0   dplyr_0.8.1    
[13] purrr_0.3.2     readr_1.3.1     tidyr_0.8.3     tibble_2.1.3   
[17] ggplot2_3.1.1   tidyverse_1.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1         locfit_1.5-9.1     lubridate_1.7.4   
 [4] lattice_0.20-38    gtools_3.8.1       assertthat_0.2.1  
 [7] rprojroot_1.3-2    digest_0.6.19      R6_2.4.0          
[10] cellranger_1.1.0   plyr_1.8.4         backports_1.1.4   
[13] evaluate_0.14      httr_1.4.0         highr_0.8         
[16] pillar_1.4.1       rlang_0.3.4        lazyeval_0.2.2    
[19] readxl_1.3.1       rstudioapi_0.10    gdata_2.18.0      
[22] whisker_0.3-2      rmarkdown_1.13     labeling_0.3      
[25] splines_3.5.1      munsell_0.5.0      broom_0.5.2       
[28] compiler_3.5.1     modelr_0.1.4       xfun_0.7          
[31] pkgconfig_2.0.2    htmltools_0.3.6    tidyselect_0.2.5  
[34] workflowr_1.4.0    crayon_1.3.4       withr_2.1.2       
[37] bitops_1.0-6       grid_3.5.1         nlme_3.1-140      
[40] jsonlite_1.6       gtable_0.3.0       git2r_0.25.2      
[43] magrittr_1.5       scales_1.0.0       KernSmooth_2.23-15
[46] cli_1.1.0          stringi_1.4.3      fs_1.3.1          
[49] xml2_1.2.0         generics_0.0.2     tools_3.5.1       
[52] glue_1.3.1         hms_0.4.2          yaml_2.2.0        
[55] colorspace_1.4-1   caTools_1.17.1.2   rvest_0.3.4       
[58] haven_2.1.0       </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
