<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>04272019_CheckRNASeqPCs_2</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">04272019_CheckRNASeqPCs_2</h1>
<h4 class="author">Ben Fair</h4>
<h4 class="date">4/27/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-11
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.3.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree1f5f2ed5edb4cf28b69fe9a9f6ada48ff28bb20dtargetblank1f5f2eda"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/1f5f2ed5edb4cf28b69fe9a9f6ada48ff28bb20d" target="_blank">1f5f2ed</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree1f5f2ed5edb4cf28b69fe9a9f6ada48ff28bb20dtargetblank1f5f2eda" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/20190521_eQTL_CrossSpeciesEnrichment_cache/
    Ignored:    analysis_temp/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/snakemake_workflow/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    docs/assets/.DS_Store

Untracked files:
    Untracked:  analysis/20190502_Check_GemmaKinshipMatrix.Rmd
    Untracked:  analysis/20190502_Check_eQTLs.Rmd
    Untracked:  analysis/20190521_eQTL_CrossSpeciesEnrichment.Rmd
    Untracked:  analysis/20190606_eGene_Conservation.Rmd
    Untracked:  analysis_temp/20190412_Check_eQTLs.Rmd
    Untracked:  analysis_temp/PlayingWithGOAnalysis.R
    Untracked:  data/PastAnalysesDataToKeep/20190428_log10TPM.txt.gz
    Untracked:  docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/
    Untracked:  docs/figure/20190606_eGene_Conservation.Rmd/

Unstaged changes:
    Deleted:    analysis/20190412_Check_eQTLs.Rmd
    Modified:   analysis/20190426_CheckRNASeqPCs_2.Rmd
    Modified:   analysis/20190428_Check_eQTLs.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/20190426_CheckRNASeqPCs_2.html" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/ece3b4adc719867c12bd16f3ed9002e5c076c586/analysis/20190426_CheckRNASeqPCs_2.Rmd" target="_blank">ece3b4a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-30
</td>
<td>
added STAR v kallisto comparison
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/db7c8470f4dc117c727eea6bd8ceff487e983abb/analysis/20190426_CheckRNASeqPCs_2.Rmd" target="_blank">db7c847</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-27
</td>
<td>
Added PCA of normalized data, new covariates R script
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(corrplot)
library(ggfortify)
library(readxl)
library(tidyverse)
library(psych)
library(ggrepel)
library(knitr)
library(reshape2)
library(gplots)</code></pre>
<p>I previously did a <a href="">similar analysis</a>.. But this time I am using different data: Here the N genes by M individulals expression matrix has been summed to gene-level log(TPM) (as opposed to transcript level quantifications), filtered, standardized across individuals, quantile normalized across genes, and is exactly the phenotypes I will use for association testing.</p>
<pre class="r"><code># Read in count table (not raw counts but transformed as described above)
CountTable &lt;- read.table(gzfile(&#39;../data/PastAnalysesDataToKeep/20190428_log10TPM.StandardizedAndNormalized.txt.gz&#39;), header=T, check.names=FALSE, row.names = 1)

kable(CountTable[1:10,1:10])</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">4X0095</th>
<th align="right">4X0212</th>
<th align="right">4X0267</th>
<th align="right">4X0333</th>
<th align="right">4X0339</th>
<th align="right">4X0354</th>
<th align="right">4X0357</th>
<th align="right">4X0550</th>
<th align="right">4x0025</th>
<th align="right">4x0043</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSPTRG00000000001</td>
<td align="right">0.8416212</td>
<td align="right">-0.7904331</td>
<td align="right">1.5144345</td>
<td align="right">0.5872556</td>
<td align="right">-0.2028427</td>
<td align="right">-0.5092311</td>
<td align="right">0.5824408</td>
<td align="right">-0.5291596</td>
<td align="right">0.3792723</td>
<td align="right">-0.1078288</td>
</tr>
<tr class="even">
<td>ENSPTRG00000000008</td>
<td align="right">0.7757291</td>
<td align="right">0.5565898</td>
<td align="right">0.6001624</td>
<td align="right">-2.2634369</td>
<td align="right">1.3151323</td>
<td align="right">1.7291771</td>
<td align="right">0.8583642</td>
<td align="right">-1.4100801</td>
<td align="right">-1.0562368</td>
<td align="right">-1.5091332</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000000009</td>
<td align="right">-1.3483746</td>
<td align="right">-0.3429953</td>
<td align="right">-0.5080765</td>
<td align="right">1.1377030</td>
<td align="right">1.0684250</td>
<td align="right">1.3367160</td>
<td align="right">0.4142126</td>
<td align="right">-1.1517021</td>
<td align="right">-0.6595468</td>
<td align="right">1.4661884</td>
</tr>
<tr class="even">
<td>ENSPTRG00000000021</td>
<td align="right">-0.7569059</td>
<td align="right">-0.4845533</td>
<td align="right">-0.1795951</td>
<td align="right">-1.3572402</td>
<td align="right">0.6238155</td>
<td align="right">-0.4889312</td>
<td align="right">0.8329788</td>
<td align="right">-1.2756167</td>
<td align="right">0.5377367</td>
<td align="right">0.7925146</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000000024</td>
<td align="right">0.8308280</td>
<td align="right">-1.6069711</td>
<td align="right">1.1635983</td>
<td align="right">2.0259921</td>
<td align="right">1.9458773</td>
<td align="right">0.8910849</td>
<td align="right">1.4642099</td>
<td align="right">-0.4564110</td>
<td align="right">0.3076960</td>
<td align="right">-1.2210758</td>
</tr>
<tr class="even">
<td>ENSPTRG00000000025</td>
<td align="right">0.7997104</td>
<td align="right">-1.1851636</td>
<td align="right">1.0160362</td>
<td align="right">1.1025578</td>
<td align="right">0.7643574</td>
<td align="right">-0.9831549</td>
<td align="right">-1.0559415</td>
<td align="right">-0.1501142</td>
<td align="right">0.4671330</td>
<td align="right">-1.5410475</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000000027</td>
<td align="right">0.0181772</td>
<td align="right">1.4756654</td>
<td align="right">1.1422363</td>
<td align="right">-1.0294431</td>
<td align="right">1.0639476</td>
<td align="right">0.3301120</td>
<td align="right">0.6700917</td>
<td align="right">-0.1518244</td>
<td align="right">-0.5623238</td>
<td align="right">0.5159799</td>
</tr>
<tr class="even">
<td>ENSPTRG00000000028</td>
<td align="right">0.1034081</td>
<td align="right">-1.3947115</td>
<td align="right">0.7078222</td>
<td align="right">1.4817199</td>
<td align="right">1.7375314</td>
<td align="right">2.1964366</td>
<td align="right">1.6976258</td>
<td align="right">-1.1134913</td>
<td align="right">-0.9211644</td>
<td align="right">-1.0317436</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000000029</td>
<td align="right">0.4249217</td>
<td align="right">-0.4301090</td>
<td align="right">0.1528507</td>
<td align="right">-0.1381548</td>
<td align="right">0.4234417</td>
<td align="right">0.4874074</td>
<td align="right">0.4114508</td>
<td align="right">0.0234199</td>
<td align="right">0.5856492</td>
<td align="right">0.8435503</td>
</tr>
<tr class="even">
<td>ENSPTRG00000000031</td>
<td align="right">0.8123452</td>
<td align="right">-1.3223937</td>
<td align="right">-1.9747476</td>
<td align="right">1.6411045</td>
<td align="right">2.2196605</td>
<td align="right">1.7062608</td>
<td align="right">1.6008516</td>
<td align="right">-0.5832423</td>
<td align="right">-0.3320767</td>
<td align="right">0.9622599</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Read admixture coefficients (K=4), and first 3 principle components, since some form of population substructure will likely be included in the expression modeling as a covariate.
AdmixtureCoeff &lt;- read.table(&quot;../output/PopulationStructure/Admixture/MergedForAdmixture.4.Q.labelled&quot;) %&gt;% 
  dplyr::rename(Individual.ID=V2) %&gt;%
  select(-V1, -V3, -V4, -V5, -V6) %&gt;%
  dplyr::rename(Admix.Western=V9, Admix.Eastern=V10, Admix.Central=V8, Admix.NigeriaCameroon=V7) #Renaming the admixture clusters after looking at plots with known subspecies
kable(head(AdmixtureCoeff))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Individual.ID</th>
<th align="right">Admix.NigeriaCameroon</th>
<th align="right">Admix.Central</th>
<th align="right">Admix.Western</th>
<th align="right">Admix.Eastern</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">549</td>
<td align="right">1.0e-05</td>
<td align="right">0.000010</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">570</td>
<td align="right">1.3e-05</td>
<td align="right">0.059266</td>
<td align="right">0.940711</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">389</td>
<td align="right">1.0e-05</td>
<td align="right">0.000010</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="right">1.1e-05</td>
<td align="right">0.000010</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">623</td>
<td align="right">1.0e-05</td>
<td align="right">0.000010</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">438</td>
<td align="right">1.0e-05</td>
<td align="right">0.000010</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
</tbody>
</table>
<pre class="r"><code>GenotypePCs &lt;- read.table(&quot;../output/PopulationStructure/pca.eigenvec&quot;, header=T) %&gt;%
  select(IID, PC1, PC2, PC3) %&gt;%
  dplyr::rename(Individual.ID=IID, GenotypePC1=PC1, GenotypePC2=PC2, GenotypePC3=PC3)
kable(head(GenotypePCs))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Individual.ID</th>
<th align="right">GenotypePC1</th>
<th align="right">GenotypePC2</th>
<th align="right">GenotypePC3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">549</td>
<td align="right">-0.1085980</td>
<td align="right">-0.0184943</td>
<td align="right">0.0047314</td>
</tr>
<tr class="even">
<td align="left">570</td>
<td align="right">-0.0945827</td>
<td align="right">-0.0228220</td>
<td align="right">-0.0140236</td>
</tr>
<tr class="odd">
<td align="left">389</td>
<td align="right">-0.1101630</td>
<td align="right">-0.0206061</td>
<td align="right">0.0032451</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="right">-0.1080570</td>
<td align="right">-0.0178108</td>
<td align="right">0.0044597</td>
</tr>
<tr class="odd">
<td align="left">623</td>
<td align="right">-0.1098320</td>
<td align="right">-0.0206678</td>
<td align="right">0.0041903</td>
</tr>
<tr class="even">
<td align="left">438</td>
<td align="right">-0.1081420</td>
<td align="right">-0.0177823</td>
<td align="right">0.0045724</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Read in other metadata
OtherMetadata &lt;- as.data.frame(read_excel(&quot;../data/Metadata.xlsx&quot;))</code></pre>
<pre><code>New names:
* `` -&gt; ...19</code></pre>
<pre class="r"><code>kable(head(OtherMetadata))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Individual.ID</th>
<th align="left">Source</th>
<th align="left">Individual.Name</th>
<th align="right">Yerkes.ID</th>
<th align="left">Label</th>
<th align="left">Notes</th>
<th align="left">FileID.(Library_Species_CellType_FlowCell)</th>
<th align="left">SX</th>
<th align="right">RNA.Library.prep.batch</th>
<th align="right">RNA.Sequencing.Lane</th>
<th align="right">Sequencing.Barcode</th>
<th align="left">RNA.Extract_date</th>
<th align="left">DNASeq_FastqIdentifier</th>
<th align="right">DNA.library.prep.batch</th>
<th align="right">DNA.Sequencing.Lane</th>
<th align="left">DNA.Sequencin.Barcode</th>
<th align="left">DNA.Extract_date</th>
<th align="right">Age</th>
<th align="left">…19</th>
<th align="left">Post.mortem.time.interval</th>
<th align="right">RIN</th>
<th align="left">Viral.status</th>
<th align="right">RNA.total.reads.mapped.to.genome</th>
<th align="right">RNA.total.reads.mapping.to.ortho.exons</th>
<th align="left">Subspecies</th>
<th align="left">DOB</th>
<th align="left">DOD</th>
<th align="left">DOB Estimated</th>
<th align="right">Age (DOD-DOB)</th>
<th align="left">OldLibInfo. RIN,RNA-extractdate,RNAbatch</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">295</td>
<td align="left">Yerkes</td>
<td align="left">Duncan</td>
<td align="right">295</td>
<td align="left">295</td>
<td align="left">NA</td>
<td align="left">24_CM_3_L006.bam</td>
<td align="left">M</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">18</td>
<td align="left">2018-10-10</td>
<td align="left">YG3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">40</td>
<td align="left">NA</td>
<td align="left">0.5</td>
<td align="right">7.3</td>
<td align="left">NA</td>
<td align="right">45.67002</td>
<td align="right">17.51562</td>
<td align="left">verus/ellioti</td>
<td align="left">24731</td>
<td align="left">39386</td>
<td align="left">NA</td>
<td align="right">40</td>
<td align="left">6.3,6/14/2016,2</td>
</tr>
<tr class="even">
<td align="left">317</td>
<td align="left">Yerkes</td>
<td align="left">Iyk</td>
<td align="right">317</td>
<td align="left">317</td>
<td align="left">NA</td>
<td align="left">11_CM_3_L004.bam</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="left">2016-06-07</td>
<td align="left">YG2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">44</td>
<td align="left">NA</td>
<td align="left">2.5</td>
<td align="right">7.6</td>
<td align="left">NA</td>
<td align="right">42.75617</td>
<td align="right">17.18811</td>
<td align="left">verus</td>
<td align="left">22859</td>
<td align="left">38832</td>
<td align="left">NA</td>
<td align="right">43</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">338</td>
<td align="left">Yerkes</td>
<td align="left">Maxine</td>
<td align="right">338</td>
<td align="left">338</td>
<td align="left">NA</td>
<td align="left">8_CF_3_L008.bam</td>
<td align="left">F</td>
<td align="right">3</td>
<td align="right">8</td>
<td align="right">6</td>
<td align="left">2016-06-07</td>
<td align="left">YG1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">53</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">7.2</td>
<td align="left">NA</td>
<td align="right">50.52632</td>
<td align="right">19.49295</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40179</td>
<td align="left">Yes</td>
<td align="right">53</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">389</td>
<td align="left">Yerkes</td>
<td align="left">Rogger</td>
<td align="right">389</td>
<td align="left">389</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">M</td>
<td align="right">4</td>
<td align="right">NA</td>
<td align="right">23</td>
<td align="left">2018-10-10</td>
<td align="left">YG39</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">NA</td>
<td align="left">2018-10-01</td>
<td align="right">45</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.7</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">verus</td>
<td align="left">25204</td>
<td align="left">41656</td>
<td align="left">NA</td>
<td align="right">45</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">438</td>
<td align="left">Yerkes</td>
<td align="left">Cheeta</td>
<td align="right">438</td>
<td align="left">438</td>
<td align="left">NA</td>
<td align="left">155_CF_3_L004.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">8</td>
<td align="left">2016-06-22</td>
<td align="left">YG22</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">55</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.6</td>
<td align="left">NA</td>
<td align="right">55.30614</td>
<td align="right">18.06375</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40909</td>
<td align="left">Yes</td>
<td align="right">55</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="left">Yerkes</td>
<td align="left">Mai</td>
<td align="right">456</td>
<td align="left">456</td>
<td align="left">NA</td>
<td align="left">156_CF_3_L001.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="left">2016-06-22</td>
<td align="left">YG23</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">49</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.5</td>
<td align="left">NA</td>
<td align="right">54.00665</td>
<td align="right">20.13760</td>
<td align="left">verus</td>
<td align="left">23377</td>
<td align="left">41275</td>
<td align="left">Yes</td>
<td align="right">49</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Merge all metadata tables
Metadata &lt;- OtherMetadata %&gt;%
  left_join(GenotypePCs, by=c(&quot;Individual.ID&quot;)) %&gt;%
  left_join(AdmixtureCoeff, by=c(&quot;Individual.ID&quot;))
kable(head(Metadata))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Individual.ID</th>
<th align="left">Source</th>
<th align="left">Individual.Name</th>
<th align="right">Yerkes.ID</th>
<th align="left">Label</th>
<th align="left">Notes</th>
<th align="left">FileID.(Library_Species_CellType_FlowCell)</th>
<th align="left">SX</th>
<th align="right">RNA.Library.prep.batch</th>
<th align="right">RNA.Sequencing.Lane</th>
<th align="right">Sequencing.Barcode</th>
<th align="left">RNA.Extract_date</th>
<th align="left">DNASeq_FastqIdentifier</th>
<th align="right">DNA.library.prep.batch</th>
<th align="right">DNA.Sequencing.Lane</th>
<th align="left">DNA.Sequencin.Barcode</th>
<th align="left">DNA.Extract_date</th>
<th align="right">Age</th>
<th align="left">…19</th>
<th align="left">Post.mortem.time.interval</th>
<th align="right">RIN</th>
<th align="left">Viral.status</th>
<th align="right">RNA.total.reads.mapped.to.genome</th>
<th align="right">RNA.total.reads.mapping.to.ortho.exons</th>
<th align="left">Subspecies</th>
<th align="left">DOB</th>
<th align="left">DOD</th>
<th align="left">DOB Estimated</th>
<th align="right">Age (DOD-DOB)</th>
<th align="left">OldLibInfo. RIN,RNA-extractdate,RNAbatch</th>
<th align="right">GenotypePC1</th>
<th align="right">GenotypePC2</th>
<th align="right">GenotypePC3</th>
<th align="right">Admix.NigeriaCameroon</th>
<th align="right">Admix.Central</th>
<th align="right">Admix.Western</th>
<th align="right">Admix.Eastern</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">295</td>
<td align="left">Yerkes</td>
<td align="left">Duncan</td>
<td align="right">295</td>
<td align="left">295</td>
<td align="left">NA</td>
<td align="left">24_CM_3_L006.bam</td>
<td align="left">M</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">18</td>
<td align="left">2018-10-10</td>
<td align="left">YG3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">40</td>
<td align="left">NA</td>
<td align="left">0.5</td>
<td align="right">7.3</td>
<td align="left">NA</td>
<td align="right">45.67002</td>
<td align="right">17.51562</td>
<td align="left">verus/ellioti</td>
<td align="left">24731</td>
<td align="left">39386</td>
<td align="left">NA</td>
<td align="right">40</td>
<td align="left">6.3,6/14/2016,2</td>
<td align="right">-0.0819553</td>
<td align="right">0.0484042</td>
<td align="right">0.0151794</td>
<td align="right">0.184097</td>
<td align="right">1e-05</td>
<td align="right">0.815883</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">317</td>
<td align="left">Yerkes</td>
<td align="left">Iyk</td>
<td align="right">317</td>
<td align="left">317</td>
<td align="left">NA</td>
<td align="left">11_CM_3_L004.bam</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="left">2016-06-07</td>
<td align="left">YG2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">44</td>
<td align="left">NA</td>
<td align="left">2.5</td>
<td align="right">7.6</td>
<td align="left">NA</td>
<td align="right">42.75617</td>
<td align="right">17.18811</td>
<td align="left">verus</td>
<td align="left">22859</td>
<td align="left">38832</td>
<td align="left">NA</td>
<td align="right">43</td>
<td align="left">NA</td>
<td align="right">-0.1089490</td>
<td align="right">-0.0190248</td>
<td align="right">0.0051753</td>
<td align="right">0.000011</td>
<td align="right">1e-05</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">338</td>
<td align="left">Yerkes</td>
<td align="left">Maxine</td>
<td align="right">338</td>
<td align="left">338</td>
<td align="left">NA</td>
<td align="left">8_CF_3_L008.bam</td>
<td align="left">F</td>
<td align="right">3</td>
<td align="right">8</td>
<td align="right">6</td>
<td align="left">2016-06-07</td>
<td align="left">YG1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">53</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">7.2</td>
<td align="left">NA</td>
<td align="right">50.52632</td>
<td align="right">19.49295</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40179</td>
<td align="left">Yes</td>
<td align="right">53</td>
<td align="left">NA</td>
<td align="right">-0.1081500</td>
<td align="right">-0.0179254</td>
<td align="right">0.0048894</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">389</td>
<td align="left">Yerkes</td>
<td align="left">Rogger</td>
<td align="right">389</td>
<td align="left">389</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">M</td>
<td align="right">4</td>
<td align="right">NA</td>
<td align="right">23</td>
<td align="left">2018-10-10</td>
<td align="left">YG39</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">NA</td>
<td align="left">2018-10-01</td>
<td align="right">45</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.7</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">verus</td>
<td align="left">25204</td>
<td align="left">41656</td>
<td align="left">NA</td>
<td align="right">45</td>
<td align="left">NA</td>
<td align="right">-0.1101630</td>
<td align="right">-0.0206061</td>
<td align="right">0.0032451</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">438</td>
<td align="left">Yerkes</td>
<td align="left">Cheeta</td>
<td align="right">438</td>
<td align="left">438</td>
<td align="left">NA</td>
<td align="left">155_CF_3_L004.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">8</td>
<td align="left">2016-06-22</td>
<td align="left">YG22</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">55</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.6</td>
<td align="left">NA</td>
<td align="right">55.30614</td>
<td align="right">18.06375</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40909</td>
<td align="left">Yes</td>
<td align="right">55</td>
<td align="left">NA</td>
<td align="right">-0.1081420</td>
<td align="right">-0.0177823</td>
<td align="right">0.0045724</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="left">Yerkes</td>
<td align="left">Mai</td>
<td align="right">456</td>
<td align="left">456</td>
<td align="left">NA</td>
<td align="left">156_CF_3_L001.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="left">2016-06-22</td>
<td align="left">YG23</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">49</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.5</td>
<td align="left">NA</td>
<td align="right">54.00665</td>
<td align="right">20.13760</td>
<td align="left">verus</td>
<td align="left">23377</td>
<td align="left">41275</td>
<td align="left">Yes</td>
<td align="right">49</td>
<td align="left">NA</td>
<td align="right">-0.1080570</td>
<td align="right">-0.0178108</td>
<td align="right">0.0044597</td>
<td align="right">0.000011</td>
<td align="right">1e-05</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
</tr>
</tbody>
</table>
<p>First, look at distribution of a couple genes across samples. Plot histogram and normal QQ-plot</p>
<pre class="r"><code>set.seed(0)
# Pick some random row numbers to plot.
for (i in sample(1:nrow(CountTable), 5, replace=F)){
  Phenotype &lt;- as.numeric(CountTable[i,])
  hist(Phenotype, main=rownames(CountTable)[i])
  qqnorm(Phenotype, main=rownames(CountTable)[i])
  qqline(Phenotype)
}</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-1.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-2">
Past versions of unnamed-chunk-1-2.png
</button>
</p>
<div id="fig-unnamed-chunk-1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-2.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-3">
Past versions of unnamed-chunk-1-3.png
</button>
</p>
<div id="fig-unnamed-chunk-1-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-3.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-4">
Past versions of unnamed-chunk-1-4.png
</button>
</p>
<div id="fig-unnamed-chunk-1-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-4.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-5">
Past versions of unnamed-chunk-1-5.png
</button>
</p>
<div id="fig-unnamed-chunk-1-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-5.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-6">
Past versions of unnamed-chunk-1-6.png
</button>
</p>
<div id="fig-unnamed-chunk-1-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-6.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-7">
Past versions of unnamed-chunk-1-7.png
</button>
</p>
<div id="fig-unnamed-chunk-1-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-7.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-8">
Past versions of unnamed-chunk-1-8.png
</button>
</p>
<div id="fig-unnamed-chunk-1-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-8.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-9">
Past versions of unnamed-chunk-1-9.png
</button>
</p>
<div id="fig-unnamed-chunk-1-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-9.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-10">
Past versions of unnamed-chunk-1-10.png
</button>
</p>
<div id="fig-unnamed-chunk-1-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-10.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Pick a some randome column numbers to plot. these distributions (of all expressed genes in an individual) should be normal since they are quantile normalized.
for (i in sample(1:ncol(CountTable), 3, replace=F)){
  Phenotype &lt;- as.numeric(CountTable[,i])
  hist(Phenotype, main=colnames(CountTable)[i])
  qqnorm(Phenotype, main=colnames(CountTable)[i])
  qqline(Phenotype)
}</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-11">
Past versions of unnamed-chunk-1-11.png
</button>
</p>
<div id="fig-unnamed-chunk-1-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-11.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-12">
Past versions of unnamed-chunk-1-12.png
</button>
</p>
<div id="fig-unnamed-chunk-1-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-12.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-13">
Past versions of unnamed-chunk-1-13.png
</button>
</p>
<div id="fig-unnamed-chunk-1-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-13.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-14">
Past versions of unnamed-chunk-1-14.png
</button>
</p>
<div id="fig-unnamed-chunk-1-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-14.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-15">
Past versions of unnamed-chunk-1-15.png
</button>
</p>
<div id="fig-unnamed-chunk-1-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-15.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-16">
Past versions of unnamed-chunk-1-16.png
</button>
</p>
<div id="fig-unnamed-chunk-1-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-1-16.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plot correlation matrix before standardization and qqnorm.</p>
<pre class="r"><code>CountTableNoNorm &lt;- read.table(gzfile(&#39;../data/PastAnalysesDataToKeep/20190428_log10TPM.txt.gz&#39;), header=T, check.names=FALSE, row.names = 1)
CorMatrix &lt;- CountTableNoNorm %&gt;%
  cor(method = c(&quot;spearman&quot;))

RNAExtractionDate &lt;- as.character(unclass(factor(plyr::mapvalues(row.names(CorMatrix), from=Metadata$Individual.ID, to=Metadata$RNA.Extract_date))))
RNA.Library.prep.batch &lt;- as.character(unclass(factor(plyr::mapvalues(row.names(CorMatrix), from=Metadata$Individual.ID, to=Metadata$RNA.Library.prep.batch))))

# Heatmap of correlation. Row colors for RNA extraction batch, column colors for RNA library prep batch
heatmap.2(CorMatrix, trace=&quot;none&quot;, ColSideColors=RNAExtractionDate, RowSideColors = RNA.Library.prep.batch)</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-2-1.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>qplot(CountTableNoNorm$`4X0267`, CountTableNoNorm$`4X0212`)</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-2">
Past versions of unnamed-chunk-2-2.png
</button>
</p>
<div id="fig-unnamed-chunk-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-2-2.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>qplot(CountTableNoNorm$Little_R, CountTableNoNorm$`4X0212`)</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-3">
Past versions of unnamed-chunk-2-3.png
</button>
</p>
<div id="fig-unnamed-chunk-2-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-2-3.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># What is mean correlation
mean(CorMatrix)</code></pre>
<pre><code>[1] 0.9346274</code></pre>
<p>Plot correlation matrix after standardization and qqnorm.</p>
<pre class="r"><code>CorMatrix &lt;- CountTable %&gt;%
  cor(method = c(&quot;spearman&quot;))

RNAExtractionDate &lt;- as.character(unclass(factor(plyr::mapvalues(row.names(CorMatrix), from=Metadata$Individual.ID, to=Metadata$RNA.Extract_date))))
RNA.Library.prep.batch &lt;- as.character(unclass(factor(plyr::mapvalues(row.names(CorMatrix), from=Metadata$Individual.ID, to=Metadata$RNA.Library.prep.batch))))

# Heatmap of correlation. Row colors for RNA extraction batch, column colors for RNA library prep batch
heatmap.2(CorMatrix, trace=&quot;none&quot;, ColSideColors=RNAExtractionDate, RowSideColors = RNA.Library.prep.batch)</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-3-1.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># What is mean correlation
mean(CorMatrix)</code></pre>
<pre><code>[1] 0.0131308</code></pre>
<p>This correlation matrix looks reassuringly similar to the <a href="">previous analysis</a> which had a number of pre-processing differences. Still a within-batch clustering of the same samples in the bottom left.</p>
<p>Now perform PCA, plot a few visualizations…</p>
<pre class="r"><code># pca with log-transformed count table (+ 0.1 pseudocount)
pca_results &lt;- CountTable %&gt;%
  t() %&gt;%
  prcomp(center=T, scale. = T)

summary(pca_results)</code></pre>
<pre><code>Importance of components:
                           PC1      PC2      PC3      PC4      PC5
Standard deviation     55.6245 37.69842 32.66306 29.43659 27.67100
Proportion of Variance  0.2087  0.09587  0.07197  0.05845  0.05165
Cumulative Proportion   0.2087  0.30459  0.37656  0.43501  0.48667
                            PC6      PC7      PC8      PC9     PC10
Standard deviation     25.26349 24.38488 23.86068 21.49625 19.98403
Proportion of Variance  0.04305  0.04011  0.03841  0.03117  0.02694
Cumulative Proportion   0.52972  0.56983  0.60824  0.63941  0.66635
                          PC11     PC12     PC13    PC14     PC15     PC16
Standard deviation     19.2508 17.75043 17.52576 16.5156 15.85069 15.60038
Proportion of Variance  0.0250  0.02125  0.02072  0.0184  0.01695  0.01642
Cumulative Proportion   0.6914  0.71260  0.73332  0.7517  0.76867  0.78509
                           PC17     PC18     PC19     PC20     PC21
Standard deviation     14.97137 14.78349 14.38254 13.68883 13.40536
Proportion of Variance  0.01512  0.01474  0.01395  0.01264  0.01212
Cumulative Proportion   0.80021  0.81495  0.82891  0.84155  0.85367
                           PC22     PC23    PC24     PC25     PC26
Standard deviation     13.25781 13.02614 12.8281 12.61317 12.30077
Proportion of Variance  0.01186  0.01145  0.0111  0.01073  0.01021
Cumulative Proportion   0.86553  0.87697  0.8881  0.89881  0.90901
                           PC27     PC28     PC29    PC30    PC31     PC32
Standard deviation     11.66758 11.43071 11.27978 11.1598 11.0282 10.77957
Proportion of Variance  0.00918  0.00881  0.00858  0.0084  0.0082  0.00784
Cumulative Proportion   0.91820  0.92701  0.93559  0.9440  0.9522  0.96004
                           PC33     PC34    PC35    PC36    PC37    PC38
Standard deviation     10.49005 10.33856 10.1852 9.95047 9.51603 9.06382
Proportion of Variance  0.00742  0.00721  0.0070 0.00668 0.00611 0.00554
Cumulative Proportion   0.96746  0.97467  0.9817 0.98835 0.99446 1.00000
                            PC39
Standard deviation     6.397e-14
Proportion of Variance 0.000e+00
Cumulative Proportion  1.000e+00</code></pre>
<pre class="r"><code>screeplot(pca_results, main=&quot;scree plot of PCA&quot;)</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/pca-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-1">
Past versions of pca-1.png
</button>
</p>
<div id="fig-pca-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/pca-1.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Merge with metadata
Merged &lt;- merge(pca_results$x, Metadata, by.x = &quot;row.names&quot;, by.y = &quot;Individual.ID&quot;, all=TRUE)
kable(head(Merged))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Row.names</th>
<th align="right">PC1</th>
<th align="right">PC2</th>
<th align="right">PC3</th>
<th align="right">PC4</th>
<th align="right">PC5</th>
<th align="right">PC6</th>
<th align="right">PC7</th>
<th align="right">PC8</th>
<th align="right">PC9</th>
<th align="right">PC10</th>
<th align="right">PC11</th>
<th align="right">PC12</th>
<th align="right">PC13</th>
<th align="right">PC14</th>
<th align="right">PC15</th>
<th align="right">PC16</th>
<th align="right">PC17</th>
<th align="right">PC18</th>
<th align="right">PC19</th>
<th align="right">PC20</th>
<th align="right">PC21</th>
<th align="right">PC22</th>
<th align="right">PC23</th>
<th align="right">PC24</th>
<th align="right">PC25</th>
<th align="right">PC26</th>
<th align="right">PC27</th>
<th align="right">PC28</th>
<th align="right">PC29</th>
<th align="right">PC30</th>
<th align="right">PC31</th>
<th align="right">PC32</th>
<th align="right">PC33</th>
<th align="right">PC34</th>
<th align="right">PC35</th>
<th align="right">PC36</th>
<th align="right">PC37</th>
<th align="right">PC38</th>
<th align="right">PC39</th>
<th align="left">Source</th>
<th align="left">Individual.Name</th>
<th align="right">Yerkes.ID</th>
<th align="left">Label</th>
<th align="left">Notes</th>
<th align="left">FileID.(Library_Species_CellType_FlowCell)</th>
<th align="left">SX</th>
<th align="right">RNA.Library.prep.batch</th>
<th align="right">RNA.Sequencing.Lane</th>
<th align="right">Sequencing.Barcode</th>
<th align="left">RNA.Extract_date</th>
<th align="left">DNASeq_FastqIdentifier</th>
<th align="right">DNA.library.prep.batch</th>
<th align="right">DNA.Sequencing.Lane</th>
<th align="left">DNA.Sequencin.Barcode</th>
<th align="left">DNA.Extract_date</th>
<th align="right">Age</th>
<th align="left">…19</th>
<th align="left">Post.mortem.time.interval</th>
<th align="right">RIN</th>
<th align="left">Viral.status</th>
<th align="right">RNA.total.reads.mapped.to.genome</th>
<th align="right">RNA.total.reads.mapping.to.ortho.exons</th>
<th align="left">Subspecies</th>
<th align="left">DOB</th>
<th align="left">DOD</th>
<th align="left">DOB Estimated</th>
<th align="right">Age (DOD-DOB)</th>
<th align="left">OldLibInfo. RIN,RNA-extractdate,RNAbatch</th>
<th align="right">GenotypePC1</th>
<th align="right">GenotypePC2</th>
<th align="right">GenotypePC3</th>
<th align="right">Admix.NigeriaCameroon</th>
<th align="right">Admix.Central</th>
<th align="right">Admix.Western</th>
<th align="right">Admix.Eastern</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">295</td>
<td align="right">-44.25396</td>
<td align="right">39.46117</td>
<td align="right">17.359257</td>
<td align="right">0.2046911</td>
<td align="right">-47.536198</td>
<td align="right">33.86663</td>
<td align="right">20.219863</td>
<td align="right">-6.086724</td>
<td align="right">26.7928698</td>
<td align="right">-11.498029</td>
<td align="right">9.405888</td>
<td align="right">-4.187123</td>
<td align="right">-12.145145</td>
<td align="right">-6.252464</td>
<td align="right">-2.0213575</td>
<td align="right">-11.599748</td>
<td align="right">-14.352268</td>
<td align="right">-3.135726</td>
<td align="right">22.4873184</td>
<td align="right">-14.8178317</td>
<td align="right">9.405017</td>
<td align="right">14.176096</td>
<td align="right">6.0304810</td>
<td align="right">-13.561258</td>
<td align="right">5.4039399</td>
<td align="right">13.4904886</td>
<td align="right">17.272210</td>
<td align="right">-11.298755</td>
<td align="right">17.8043461</td>
<td align="right">-7.452475</td>
<td align="right">11.580727</td>
<td align="right">9.743226</td>
<td align="right">6.855401</td>
<td align="right">12.517849</td>
<td align="right">-25.424186</td>
<td align="right">-0.9077519</td>
<td align="right">-7.5764276</td>
<td align="right">5.8458830</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Duncan</td>
<td align="right">295</td>
<td align="left">295</td>
<td align="left">NA</td>
<td align="left">24_CM_3_L006.bam</td>
<td align="left">M</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">18</td>
<td align="left">2018-10-10</td>
<td align="left">YG3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">40</td>
<td align="left">NA</td>
<td align="left">0.5</td>
<td align="right">7.3</td>
<td align="left">NA</td>
<td align="right">45.67002</td>
<td align="right">17.51562</td>
<td align="left">verus/ellioti</td>
<td align="left">24731</td>
<td align="left">39386</td>
<td align="left">NA</td>
<td align="right">40</td>
<td align="left">6.3,6/14/2016,2</td>
<td align="right">-0.0819553</td>
<td align="right">0.0484042</td>
<td align="right">0.0151794</td>
<td align="right">0.184097</td>
<td align="right">1e-05</td>
<td align="right">0.815883</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">317</td>
<td align="right">-26.89597</td>
<td align="right">-28.73500</td>
<td align="right">13.860490</td>
<td align="right">42.2073257</td>
<td align="right">-69.264050</td>
<td align="right">-6.70541</td>
<td align="right">9.734624</td>
<td align="right">7.936016</td>
<td align="right">-0.6354965</td>
<td align="right">11.864536</td>
<td align="right">-7.933379</td>
<td align="right">24.696077</td>
<td align="right">5.460629</td>
<td align="right">-3.740944</td>
<td align="right">-2.2856360</td>
<td align="right">-17.093822</td>
<td align="right">2.475751</td>
<td align="right">16.771805</td>
<td align="right">-0.6031362</td>
<td align="right">16.6068327</td>
<td align="right">3.485551</td>
<td align="right">22.072873</td>
<td align="right">5.2915842</td>
<td align="right">10.430069</td>
<td align="right">-2.8691931</td>
<td align="right">27.2579562</td>
<td align="right">2.519391</td>
<td align="right">2.724505</td>
<td align="right">-19.0987695</td>
<td align="right">18.870823</td>
<td align="right">2.701881</td>
<td align="right">-1.036678</td>
<td align="right">-12.423519</td>
<td align="right">-18.527419</td>
<td align="right">3.105442</td>
<td align="right">5.5070933</td>
<td align="right">-10.1533966</td>
<td align="right">-2.4582925</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Iyk</td>
<td align="right">317</td>
<td align="left">317</td>
<td align="left">NA</td>
<td align="left">11_CM_3_L004.bam</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="left">2016-06-07</td>
<td align="left">YG2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">44</td>
<td align="left">NA</td>
<td align="left">2.5</td>
<td align="right">7.6</td>
<td align="left">NA</td>
<td align="right">42.75617</td>
<td align="right">17.18811</td>
<td align="left">verus</td>
<td align="left">22859</td>
<td align="left">38832</td>
<td align="left">NA</td>
<td align="right">43</td>
<td align="left">NA</td>
<td align="right">-0.1089490</td>
<td align="right">-0.0190248</td>
<td align="right">0.0051753</td>
<td align="right">0.000011</td>
<td align="right">1e-05</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">338</td>
<td align="right">-29.77820</td>
<td align="right">-25.90570</td>
<td align="right">-17.158137</td>
<td align="right">-45.2377382</td>
<td align="right">11.163235</td>
<td align="right">-31.38944</td>
<td align="right">-63.202548</td>
<td align="right">9.528283</td>
<td align="right">-14.7587833</td>
<td align="right">26.583439</td>
<td align="right">-13.804533</td>
<td align="right">7.950232</td>
<td align="right">12.935148</td>
<td align="right">1.757393</td>
<td align="right">19.7078123</td>
<td align="right">7.968451</td>
<td align="right">-7.033978</td>
<td align="right">16.893201</td>
<td align="right">3.3684154</td>
<td align="right">-3.6439445</td>
<td align="right">15.812871</td>
<td align="right">-2.094309</td>
<td align="right">11.0748962</td>
<td align="right">8.136154</td>
<td align="right">0.9705953</td>
<td align="right">10.6349247</td>
<td align="right">8.182168</td>
<td align="right">-35.849824</td>
<td align="right">16.7763344</td>
<td align="right">-5.653386</td>
<td align="right">-2.946741</td>
<td align="right">3.082690</td>
<td align="right">-5.534944</td>
<td align="right">1.310680</td>
<td align="right">6.145383</td>
<td align="right">10.4990161</td>
<td align="right">-2.2335835</td>
<td align="right">-0.4868652</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Maxine</td>
<td align="right">338</td>
<td align="left">338</td>
<td align="left">NA</td>
<td align="left">8_CF_3_L008.bam</td>
<td align="left">F</td>
<td align="right">3</td>
<td align="right">8</td>
<td align="right">6</td>
<td align="left">2016-06-07</td>
<td align="left">YG1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">53</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">7.2</td>
<td align="left">NA</td>
<td align="right">50.52632</td>
<td align="right">19.49295</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40179</td>
<td align="left">Yes</td>
<td align="right">53</td>
<td align="left">NA</td>
<td align="right">-0.1081500</td>
<td align="right">-0.0179254</td>
<td align="right">0.0048894</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">389</td>
<td align="right">-51.39185</td>
<td align="right">30.16459</td>
<td align="right">16.153499</td>
<td align="right">3.5874022</td>
<td align="right">16.873720</td>
<td align="right">26.88904</td>
<td align="right">-21.859003</td>
<td align="right">17.004977</td>
<td align="right">7.1826759</td>
<td align="right">-30.828229</td>
<td align="right">-27.411625</td>
<td align="right">31.722105</td>
<td align="right">-9.249130</td>
<td align="right">-9.854308</td>
<td align="right">26.1733970</td>
<td align="right">13.571025</td>
<td align="right">-14.070649</td>
<td align="right">-16.612163</td>
<td align="right">-9.4107028</td>
<td align="right">22.2163771</td>
<td align="right">-5.953215</td>
<td align="right">-3.499962</td>
<td align="right">0.6961453</td>
<td align="right">32.852243</td>
<td align="right">-18.3509125</td>
<td align="right">-4.1584074</td>
<td align="right">5.856220</td>
<td align="right">10.802174</td>
<td align="right">0.6790158</td>
<td align="right">5.103287</td>
<td align="right">-3.765796</td>
<td align="right">-1.526496</td>
<td align="right">11.410236</td>
<td align="right">13.192573</td>
<td align="right">-4.075502</td>
<td align="right">2.2978888</td>
<td align="right">-8.9920692</td>
<td align="right">8.8159064</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Rogger</td>
<td align="right">389</td>
<td align="left">389</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">M</td>
<td align="right">4</td>
<td align="right">NA</td>
<td align="right">23</td>
<td align="left">2018-10-10</td>
<td align="left">YG39</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">NA</td>
<td align="left">2018-10-01</td>
<td align="right">45</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.7</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">verus</td>
<td align="left">25204</td>
<td align="left">41656</td>
<td align="left">NA</td>
<td align="right">45</td>
<td align="left">NA</td>
<td align="right">-0.1101630</td>
<td align="right">-0.0206061</td>
<td align="right">0.0032451</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">438</td>
<td align="right">-62.33852</td>
<td align="right">36.92141</td>
<td align="right">39.104542</td>
<td align="right">-19.3222516</td>
<td align="right">2.345069</td>
<td align="right">-14.70660</td>
<td align="right">-22.140914</td>
<td align="right">-19.583242</td>
<td align="right">-5.4979372</td>
<td align="right">1.585372</td>
<td align="right">-8.799630</td>
<td align="right">-29.366938</td>
<td align="right">13.754492</td>
<td align="right">-1.453955</td>
<td align="right">4.9970367</td>
<td align="right">-2.628545</td>
<td align="right">21.676829</td>
<td align="right">8.132961</td>
<td align="right">-18.7010009</td>
<td align="right">-0.7851945</td>
<td align="right">8.684619</td>
<td align="right">3.330513</td>
<td align="right">-2.1922982</td>
<td align="right">2.562032</td>
<td align="right">-9.1400894</td>
<td align="right">0.3411153</td>
<td align="right">17.550933</td>
<td align="right">19.434638</td>
<td align="right">-8.5296090</td>
<td align="right">-20.122647</td>
<td align="right">17.236158</td>
<td align="right">-15.881218</td>
<td align="right">-27.288212</td>
<td align="right">7.557714</td>
<td align="right">-6.862128</td>
<td align="right">-7.4860033</td>
<td align="right">2.3166852</td>
<td align="right">3.9779611</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Cheeta</td>
<td align="right">438</td>
<td align="left">438</td>
<td align="left">NA</td>
<td align="left">155_CF_3_L004.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">8</td>
<td align="left">2016-06-22</td>
<td align="left">YG22</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">55</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.6</td>
<td align="left">NA</td>
<td align="right">55.30614</td>
<td align="right">18.06375</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40909</td>
<td align="left">Yes</td>
<td align="right">55</td>
<td align="left">NA</td>
<td align="right">-0.1081420</td>
<td align="right">-0.0177823</td>
<td align="right">0.0045724</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="right">-68.68134</td>
<td align="right">2.34152</td>
<td align="right">3.310246</td>
<td align="right">-30.2323017</td>
<td align="right">7.652629</td>
<td align="right">-28.02238</td>
<td align="right">-3.335123</td>
<td align="right">-22.198197</td>
<td align="right">-33.6189279</td>
<td align="right">-23.040870</td>
<td align="right">18.494698</td>
<td align="right">-20.952720</td>
<td align="right">11.472863</td>
<td align="right">-6.511476</td>
<td align="right">-0.2133327</td>
<td align="right">-17.377888</td>
<td align="right">14.057399</td>
<td align="right">-5.011164</td>
<td align="right">3.5675026</td>
<td align="right">-20.0366185</td>
<td align="right">-15.274251</td>
<td align="right">4.512365</td>
<td align="right">-3.2263743</td>
<td align="right">8.089678</td>
<td align="right">0.7228579</td>
<td align="right">4.2801799</td>
<td align="right">17.411825</td>
<td align="right">9.877313</td>
<td align="right">16.7347428</td>
<td align="right">35.290192</td>
<td align="right">-9.587897</td>
<td align="right">-8.534385</td>
<td align="right">14.786548</td>
<td align="right">-2.838145</td>
<td align="right">2.797917</td>
<td align="right">1.3409950</td>
<td align="right">-0.1892656</td>
<td align="right">-5.8230831</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Mai</td>
<td align="right">456</td>
<td align="left">456</td>
<td align="left">NA</td>
<td align="left">156_CF_3_L001.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="left">2016-06-22</td>
<td align="left">YG23</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">49</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.5</td>
<td align="left">NA</td>
<td align="right">54.00665</td>
<td align="right">20.13760</td>
<td align="left">verus</td>
<td align="left">23377</td>
<td align="left">41275</td>
<td align="left">Yes</td>
<td align="right">49</td>
<td align="left">NA</td>
<td align="right">-0.1080570</td>
<td align="right">-0.0178108</td>
<td align="right">0.0044597</td>
<td align="right">0.000011</td>
<td align="right">1e-05</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Plot a couple PCs with a couple potential covariates
ggplot(Merged, aes(x=PC2, y=PC1, color=factor(RNA.Extract_date), label=Row.names)) + 
  geom_point() +
  geom_text_repel(size=2.5)</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/pca-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-2">
Past versions of pca-2.png
</button>
</p>
<div id="fig-pca-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/pca-2.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(Merged, aes(x=PC1, y=PC2, color=factor(RNA.Library.prep.batch), label=Row.names)) + 
  geom_point() +
  geom_text_repel(size=2.5)</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/pca-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-3">
Past versions of pca-3.png
</button>
</p>
<div id="fig-pca-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/pca-3.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(Merged, aes(x=PC2, y=PC1, color=Age, label=Row.names)) + 
  geom_point() +
  geom_text_repel(size=2.5)</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/pca-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-4">
Past versions of pca-4.png
</button>
</p>
<div id="fig-pca-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/pca-4.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Can already see maybe something with the RNA extraction batch in first PC… Now I am going to look more systematically for significant correlations between potential observed confounders in the Metadata and the first 10 principle components. Will use Pearsons’s correlation to test continuous continuous confounders, will use anova for categorical confounders.</p>
<p>First the continuous confounders</p>
<pre class="r"><code># Grab first 10 PCs
PCs_to_test &lt;- Merged[,2:15]
kable(head(PCs_to_test))</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">PC1</th>
<th align="right">PC2</th>
<th align="right">PC3</th>
<th align="right">PC4</th>
<th align="right">PC5</th>
<th align="right">PC6</th>
<th align="right">PC7</th>
<th align="right">PC8</th>
<th align="right">PC9</th>
<th align="right">PC10</th>
<th align="right">PC11</th>
<th align="right">PC12</th>
<th align="right">PC13</th>
<th align="right">PC14</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">-44.25396</td>
<td align="right">39.46117</td>
<td align="right">17.359257</td>
<td align="right">0.2046911</td>
<td align="right">-47.536198</td>
<td align="right">33.86663</td>
<td align="right">20.219863</td>
<td align="right">-6.086724</td>
<td align="right">26.7928698</td>
<td align="right">-11.498029</td>
<td align="right">9.405888</td>
<td align="right">-4.187123</td>
<td align="right">-12.145145</td>
<td align="right">-6.252464</td>
</tr>
<tr class="even">
<td align="right">-26.89597</td>
<td align="right">-28.73500</td>
<td align="right">13.860490</td>
<td align="right">42.2073257</td>
<td align="right">-69.264050</td>
<td align="right">-6.70541</td>
<td align="right">9.734624</td>
<td align="right">7.936016</td>
<td align="right">-0.6354965</td>
<td align="right">11.864536</td>
<td align="right">-7.933379</td>
<td align="right">24.696077</td>
<td align="right">5.460629</td>
<td align="right">-3.740944</td>
</tr>
<tr class="odd">
<td align="right">-29.77820</td>
<td align="right">-25.90570</td>
<td align="right">-17.158137</td>
<td align="right">-45.2377382</td>
<td align="right">11.163235</td>
<td align="right">-31.38944</td>
<td align="right">-63.202548</td>
<td align="right">9.528283</td>
<td align="right">-14.7587833</td>
<td align="right">26.583439</td>
<td align="right">-13.804533</td>
<td align="right">7.950232</td>
<td align="right">12.935148</td>
<td align="right">1.757393</td>
</tr>
<tr class="even">
<td align="right">-51.39185</td>
<td align="right">30.16459</td>
<td align="right">16.153499</td>
<td align="right">3.5874022</td>
<td align="right">16.873720</td>
<td align="right">26.88904</td>
<td align="right">-21.859003</td>
<td align="right">17.004977</td>
<td align="right">7.1826759</td>
<td align="right">-30.828229</td>
<td align="right">-27.411625</td>
<td align="right">31.722105</td>
<td align="right">-9.249130</td>
<td align="right">-9.854308</td>
</tr>
<tr class="odd">
<td align="right">-62.33852</td>
<td align="right">36.92141</td>
<td align="right">39.104542</td>
<td align="right">-19.3222516</td>
<td align="right">2.345069</td>
<td align="right">-14.70660</td>
<td align="right">-22.140914</td>
<td align="right">-19.583242</td>
<td align="right">-5.4979372</td>
<td align="right">1.585372</td>
<td align="right">-8.799630</td>
<td align="right">-29.366938</td>
<td align="right">13.754492</td>
<td align="right">-1.453955</td>
</tr>
<tr class="even">
<td align="right">-68.68134</td>
<td align="right">2.34152</td>
<td align="right">3.310246</td>
<td align="right">-30.2323017</td>
<td align="right">7.652629</td>
<td align="right">-28.02238</td>
<td align="right">-3.335123</td>
<td align="right">-22.198197</td>
<td align="right">-33.6189279</td>
<td align="right">-23.040870</td>
<td align="right">18.494698</td>
<td align="right">-20.952720</td>
<td align="right">11.472863</td>
<td align="right">-6.511476</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Grab potential continuous confounders that make sense to test
Continuous_confounders_to_test &lt;- Merged[, c(&quot;RIN&quot;, &quot;Age&quot;, &quot;GenotypePC1&quot;, &quot;GenotypePC2&quot;, &quot;GenotypePC3&quot;, &quot;Admix.Western&quot;, &quot;Admix.Eastern&quot;, &quot;Admix.Central&quot;, &quot;Admix.NigeriaCameroon&quot;)]
kable(head(Continuous_confounders_to_test))</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">RIN</th>
<th align="right">Age</th>
<th align="right">GenotypePC1</th>
<th align="right">GenotypePC2</th>
<th align="right">GenotypePC3</th>
<th align="right">Admix.Western</th>
<th align="right">Admix.Eastern</th>
<th align="right">Admix.Central</th>
<th align="right">Admix.NigeriaCameroon</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">7.3</td>
<td align="right">40</td>
<td align="right">-0.0819553</td>
<td align="right">0.0484042</td>
<td align="right">0.0151794</td>
<td align="right">0.815883</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.184097</td>
</tr>
<tr class="even">
<td align="right">7.6</td>
<td align="right">44</td>
<td align="right">-0.1089490</td>
<td align="right">-0.0190248</td>
<td align="right">0.0051753</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.000011</td>
</tr>
<tr class="odd">
<td align="right">7.2</td>
<td align="right">53</td>
<td align="right">-0.1081500</td>
<td align="right">-0.0179254</td>
<td align="right">0.0048894</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.000010</td>
</tr>
<tr class="even">
<td align="right">5.7</td>
<td align="right">45</td>
<td align="right">-0.1101630</td>
<td align="right">-0.0206061</td>
<td align="right">0.0032451</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.000010</td>
</tr>
<tr class="odd">
<td align="right">5.6</td>
<td align="right">55</td>
<td align="right">-0.1081420</td>
<td align="right">-0.0177823</td>
<td align="right">0.0045724</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.000010</td>
</tr>
<tr class="even">
<td align="right">5.5</td>
<td align="right">49</td>
<td align="right">-0.1080570</td>
<td align="right">-0.0178108</td>
<td align="right">0.0044597</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.000011</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Test
Spearman_test_results &lt;- corr.test(Continuous_confounders_to_test, PCs_to_test, adjust=&quot;none&quot;, method=&quot;spearman&quot;)

MinP_floor &lt;- floor(log10(min(Spearman_test_results$p)))
# Plot
Spearman_test_results$p %&gt;%
  melt() %&gt;%
  dplyr::rename(Pvalue = value, Principle.Component=Var2, Potential.Confounder=Var1) %&gt;%
  ggplot(aes(x=Potential.Confounder, y=Principle.Component, fill=Pvalue)) + 
    geom_tile() +
    geom_text(aes(label = signif(Pvalue, 2))) +
    scale_fill_gradient(limits=c(10**MinP_floor, 1), breaks=10**seq(MinP_floor,0,1), trans = &#39;log&#39;, high=&quot;white&quot;, low=&quot;red&quot; ) +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/continuous_confounders-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-continuous_confounders-1">
Past versions of continuous_confounders-1.png
</button>
</p>
<div id="fig-continuous_confounders-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/continuous_confounders-1.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And the categorical confounders…</p>
<pre class="r"><code># Grab potential categorical confounders that make sense to test
Categorical_confounders_to_test &lt;- Merged[,c(&quot;Viral.status&quot;, &quot;SX&quot;,&quot;RNA.Extract_date&quot;, &quot;RNA.Library.prep.batch&quot;, &quot;Source&quot;)]
kable(head(Categorical_confounders_to_test))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Viral.status</th>
<th align="left">SX</th>
<th align="left">RNA.Extract_date</th>
<th align="right">RNA.Library.prep.batch</th>
<th align="left">Source</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
<td align="left">M</td>
<td align="left">2018-10-10</td>
<td align="right">5</td>
<td align="left">Yerkes</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">M</td>
<td align="left">2016-06-07</td>
<td align="right">3</td>
<td align="left">Yerkes</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">F</td>
<td align="left">2016-06-07</td>
<td align="right">3</td>
<td align="left">Yerkes</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">M</td>
<td align="left">2018-10-10</td>
<td align="right">4</td>
<td align="left">Yerkes</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">F</td>
<td align="left">2016-06-22</td>
<td align="right">2</td>
<td align="left">Yerkes</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">F</td>
<td align="left">2016-06-22</td>
<td align="right">2</td>
<td align="left">Yerkes</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Viral status will need to be reformatted to make factors that make sense for testing (example: HBV+, HBV- HCV+, HCV- are factors that make sense). Let&#39;s assume that NA means negative status.
Categorical_confounders_to_test$HBV_status &lt;- grepl(&quot;HBV+&quot;, Categorical_confounders_to_test$Viral.status)
Categorical_confounders_to_test$HCV_status &lt;- grepl(&quot;HCV+&quot;, Categorical_confounders_to_test$Viral.status)
Categorical_confounders_to_test &lt;- Categorical_confounders_to_test[, -1 ]
kable(head(Categorical_confounders_to_test))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SX</th>
<th align="left">RNA.Extract_date</th>
<th align="right">RNA.Library.prep.batch</th>
<th align="left">Source</th>
<th align="left">HBV_status</th>
<th align="left">HCV_status</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">M</td>
<td align="left">2018-10-10</td>
<td align="right">5</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">M</td>
<td align="left">2016-06-07</td>
<td align="right">3</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">F</td>
<td align="left">2016-06-07</td>
<td align="right">3</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">M</td>
<td align="left">2018-10-10</td>
<td align="right">4</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">F</td>
<td align="left">2016-06-22</td>
<td align="right">2</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="left">2016-06-22</td>
<td align="right">2</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Do one-way anova test as a loop.
# First initialize results matrix
Pvalues &lt;- matrix(ncol = dim(PCs_to_test)[2], nrow = dim(Categorical_confounders_to_test)[2])
colnames(Pvalues) &lt;- colnames(PCs_to_test)
rownames(Pvalues) &lt;- colnames(Categorical_confounders_to_test)
for (confounder in seq_along(Categorical_confounders_to_test)) {
    for (PC in seq_along(PCs_to_test)) {
      res.aov &lt;- aov(PCs_to_test[[PC]] ~ Categorical_confounders_to_test[[confounder]])
      pval &lt;- summary(res.aov)[[1]][[&quot;Pr(&gt;F)&quot;]][1]
      Pvalues[confounder, PC] &lt;- pval
    }
}
# Plot
MinP_floor &lt;- floor(log10(min(Pvalues)))
Pvalues %&gt;%
  melt() %&gt;%
  dplyr::rename(Pvalue = value, Principle.Component=Var2, Potential.Confounder=Var1) %&gt;%
  ggplot(aes(x=Potential.Confounder, y=Principle.Component, fill=Pvalue)) + 
    geom_tile() +
    geom_text(aes(label = signif(Pvalue, 2))) +
    scale_fill_gradient(limits=c(10**MinP_floor, 1), breaks=10**seq(MinP_floor,0,1), trans = &#39;log&#39;, high=&quot;white&quot;, low=&quot;red&quot; ) +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/categorical_confounders-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-categorical_confounders-1">
Past versions of categorical_confounders-1.png
</button>
</p>
<div id="fig-categorical_confounders-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/categorical_confounders-1.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plot some of the significant PC-metadata associations for a better look…</p>
<pre class="r"><code>ggplot(Merged, aes(x=factor(RNA.Extract_date), y=PC2, label=Label)) + 
  geom_boxplot() +
  geom_jitter(position=position_jitter(width=.2, height=0))</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-4-1.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(Merged, aes(x=factor(RNA.Library.prep.batch), y=PC2)) + 
  geom_boxplot() +
  geom_jitter(position=position_jitter(width=.1, height=0))</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-4-2.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(Merged, aes(factor(RNA.Extract_date))) + 
  geom_bar(aes(fill = factor(RNA.Library.prep.batch)))</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-3">
Past versions of unnamed-chunk-4-3.png
</button>
</p>
<div id="fig-unnamed-chunk-4-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-4-3.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Strongest effect seems to be related to batch… RNA library prep batch and RNA extraction batch (date) both covary with a top PC. Have to check with Claudia that the last batch was a different extraction method (Trizol vs RNEasy).</p>
<pre class="r"><code>ggplot(Merged, aes(x=RIN, y=PC5, label=Label, color=factor(Source))) +
  geom_point() +
  geom_text_repel(size=2.5)</code></pre>
<p><img src="figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/figure/20190426_CheckRNASeqPCs_2.Rmd/unnamed-chunk-5-1.png" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PC5 is somewhat correlated with RIN and source.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gplots_3.0.1.1  reshape2_1.4.3  knitr_1.23      ggrepel_0.8.1  
 [5] psych_1.8.12    forcats_0.4.0   stringr_1.4.0   dplyr_0.8.1    
 [9] purrr_0.3.2     readr_1.3.1     tidyr_0.8.3     tibble_2.1.1   
[13] tidyverse_1.2.1 readxl_1.3.1    ggfortify_0.4.7 ggplot2_3.1.1  
[17] corrplot_0.84  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1         lubridate_1.7.4    lattice_0.20-38   
 [4] gtools_3.8.1       assertthat_0.2.1   rprojroot_1.3-2   
 [7] digest_0.6.19      R6_2.4.0           cellranger_1.1.0  
[10] plyr_1.8.4         backports_1.1.4    evaluate_0.13     
[13] highr_0.8          httr_1.4.0         pillar_1.4.0      
[16] rlang_0.3.4        lazyeval_0.2.2     rstudioapi_0.10   
[19] gdata_2.18.0       whisker_0.3-2      rmarkdown_1.13    
[22] labeling_0.3       foreign_0.8-71     munsell_0.5.0     
[25] broom_0.5.2        compiler_3.5.1     modelr_0.1.4      
[28] xfun_0.7           pkgconfig_2.0.2    mnormt_1.5-5      
[31] htmltools_0.3.6    tidyselect_0.2.5   gridExtra_2.3     
[34] workflowr_1.3.0    crayon_1.3.4       withr_2.1.2       
[37] bitops_1.0-6       grid_3.5.1         nlme_3.1-140      
[40] jsonlite_1.6       gtable_0.3.0       git2r_0.25.2      
[43] magrittr_1.5       scales_1.0.0       KernSmooth_2.23-15
[46] cli_1.1.0          stringi_1.4.3      fs_1.3.1          
[49] xml2_1.2.0         generics_0.0.2     tools_3.5.1       
[52] glue_1.3.1         hms_0.4.2          parallel_3.5.1    
[55] yaml_2.2.0         colorspace_1.4-1   caTools_1.17.1.2  
[58] rvest_0.3.4        haven_2.1.0       </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
