<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>20190930_OverdispersionEstimatesBetweenSpecies</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20190930_OverdispersionEstimatesBetweenSpecies</h1>
<h4 class="author">Ben Fair</h4>
<h4 class="date">9/30/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-10-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree65c8360ca271cb584c648059261c789592f5429ftargetblank65c8360a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/65c8360ca271cb584c648059261c789592f5429f" target="_blank">65c8360</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree65c8360ca271cb584c648059261c789592f5429ftargetblank65c8360a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/20190521_eQTL_CrossSpeciesEnrichment_cache/
    Ignored:    analysis_temp/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/snakemake_workflow/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    docs/assets/.DS_Store

Untracked files:
    Untracked:  analysis/20191001_GtexAnnotations.Rmd
    Untracked:  analysis_temp/20190716_VarianceInsteadOfEgenes_2.Rmd

Unstaged changes:
    Deleted:    analysis/20190716_VarianceInsteadOfEgenes_2.Rmd
    Modified:   analysis/20190930_OverdispersionEstimates.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/781081997b0eaf5358342507947137ed99599e0a/analysis/20190930_OverdispersionEstimates.Rmd" target="_blank">7810819</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-10-01
</td>
<td>
added overdispersion analysis, gene lists
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>From my dataset of gene expression measurements across 39 chimp individuals (RNA-seq from heart tissue) and ~39 human individuals (~50 RNA-seq sampels randomly chosen from human GTEx datasets, then culled to exclude outliers and lowly sequenced samples) I am interested in quantifying gene expression dispersion within each population. I had previously made the observation that gene expression dispersion (summarized by statistics like standard deviation, or variance) is well correlated (<span class="math inline">\(R^2\approx0.6\)</span>) between the two species, even after regressing out the fact that variance is correlated to expression (due to both biological and statstical reasons). Abhishek suggested I do this analysis a bit differently. Rather than simply plotting standard deviation for each gene after regressing out the relationship with absolute expression, I should model the raw count data with a negative binomial model and estimate an over dispersion parameter. Abhishek wrote more description <a href="https://users.rcc.uchicago.edu/~aksarkar/singlecell-ideas/species-var.html">here</a> with some snippets of how I might accomplish this.</p>
<pre class="r"><code>library(plyr)
library(tidyverse)
library(knitr)
library(&quot;edgeR&quot;)
library(stats)
library(corrplot)
library(gplots)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(enrichplot)
library(cowplot)
library(&#39;latex2exp&#39;)
library(gdata)

# custom functions to read count tables, estimate overdispersion based on Abhishek&#39;s code snippets
source(&quot;../code/CustomFunctions.R&quot;)</code></pre>
<p>First, I will pick samples and genes to run this analysis on… The choice of which human samples from GTEx to include are the same ones I used for differential expression analysis <a href="20190613_PowerAnalysis.html">here</a>. Basically, I just excluded outlier samples that didn’t cluster with the large majority of samples. The samples were re-mapped to their respective genomes and gene counts were determined using only reads over othologous exons. The genes I will include in this analysis will be all those that were tested for eGene in both species and in the ortho-exon count table.</p>
<pre class="r"><code>HumanSamplesToDrop &lt;- c(c(&quot;SRR1507229&quot;,&quot;SRR603918&quot;, &quot;SRR1478149&quot;, &quot;SRR598509&quot;, &quot;SRR613186&quot;), c(&quot;SRR1489693&quot;, &quot;SRR598148&quot;, &quot;59167&quot;, &quot;SRR1478900&quot;, &quot;SRR1474730&quot;, &quot;61317&quot;))
ChimpSamplesToDrop &lt;- c(&quot;Little_R&quot;)
CountTableChimpFile &lt;- &#39;../output/PowerAnalysisFullCountTable.Chimp.subread.txt.gz&#39;
CountTableHumanFile &lt;- &#39;../output/PowerAnalysisFullCountTable.Human.subread.txt.gz&#39;
eQTLs &lt;- read.table(gzfile(&quot;../data/PastAnalysesDataToKeep/20190521_eQTLs_250kB_10MAF.txt.gz&quot;), header=T)
# List of chimp tested genes
ChimpTestedGenes &lt;- rownames(read.table(&#39;../output/ExpressionMatrix.un-normalized.txt.gz&#39;, header=T, check.names=FALSE, row.names = 1))

ChimpToHumanGeneMap &lt;- read.table(&quot;../data/Biomart_export.Hsap.Ptro.orthologs.txt.gz&quot;, header=T, sep=&#39;\t&#39;, stringsAsFactors = F)

# Of this ortholog list, how many genes are one2one
OneToOneMap &lt;- ChimpToHumanGeneMap %&gt;%
  filter(Chimpanzee.homology.type==&quot;ortholog_one2one&quot;)

# Read gtex heart egene list
# Only consider those that were tested in both species and are one2one orthologs
GtexHeartEgenes &lt;- read.table(&quot;../data/Heart_Left_Ventricle.v7.egenes.txt.gz&quot;, header=T, sep=&#39;\t&#39;, stringsAsFactors = F) %&gt;%
  mutate(gene_id_stable = gsub(&quot;.\\d+$&quot;,&quot;&quot;,gene_id)) %&gt;%
  filter(gene_id_stable %in% OneToOneMap$Gene.stable.ID) %&gt;%
  mutate(chimp_id = plyr::mapvalues(gene_id_stable, OneToOneMap$Gene.stable.ID,  OneToOneMap$Chimpanzee.gene.stable.ID, warn_missing = F)) %&gt;%
  filter(chimp_id %in% ChimpTestedGenes)


EgenesTested &lt;- gsub(&quot;\\..+&quot;, &quot;&quot;, GtexHeartEgenes$gene_id, perl=T)
length(EgenesTested)</code></pre>
<pre><code>[1] 11586</code></pre>
<pre class="r"><code>GenesInDESet &lt;- read.table(gzfile(CountTableChimpFile), header=T, check.names=FALSE, skip=1)$Geneid
length(GenesInDESet)</code></pre>
<pre><code>[1] 44125</code></pre>
<pre class="r"><code>GeneList &lt;- intersect(as.character(GenesInDESet),EgenesTested)</code></pre>
<p>Now, load count tables and calculate overdispersion</p>
<pre class="r"><code>CountTables &lt;- GetCountTables(CountTableChimpFile,
                               CountTableHumanFile,
                               0, GeneList, ChimpSampleDrop=ChimpSamplesToDrop, HumanSampleDrop = HumanSamplesToDrop)


Chimp.NB.fit.parameters&lt;-GetParameterEstimatesOfUnderlyingGamma_lengthAdjusted_FromTable(CountTables$Chimp$Counts, CountTables$Chimp$GeneLengths)
Human.NB.fit.parameters&lt;-GetParameterEstimatesOfUnderlyingGamma_lengthAdjusted_FromTable(CountTables$Human$Counts, CountTables$Human$GeneLengths)</code></pre>
<p>Just to as one simple confirmation that the code is working as expected, here is a plot of the population mean expression estimate <span class="math inline">\(\hat{\mu_j}\)</span> for each gene <span class="math inline">\(j\)</span> from the negative binomial fit, versus when I calculate the expression estimates by a simply averaging across samples (let’s call it <span class="math inline">\(\bar{x_j}\)</span>) for all genes.</p>
<pre class="r"><code>ToPlot &lt;- cbind(Chimp.NB.fit.parameters,
                apply(CountTables$Chimp$log2RPKM, 1, mean),
                Human.NB.fit.parameters,
                apply(CountTables$Human$log2RPKM, 1, mean))

colnames(ToPlot) &lt;- c(&quot;Chimp.Mean.Expression&quot;, &quot;Chimp.Overdispersion&quot;, &quot;Chimp.Mean.Log2RPKM&quot;, &quot;Human.Mean.Expression&quot;, &quot;Human.Overdispersion&quot;, &quot;Human.Mean.Log2RPKM&quot;)

#Plot mu parameter estimate (estimate of log(mean expression)) vs logRPKM
R&lt;-cor(ToPlot$Chimp.Mean.Expression, ToPlot$Chimp.Mean.Log2RPKM, use=&quot;complete.obs&quot;)
lb1 &lt;- paste(&quot;~R^2==~&quot;, round(R**2,2))
ggplot(ToPlot, aes(x=Chimp.Mean.Expression, y=Chimp.Mean.Log2RPKM)) +
  geom_point(alpha=0.05) +
  xlab(TeX(&#39;$\\hat{\\mu}$&#39;)) +
  ylab(expression(bar(&quot;x&quot;))) +
  annotate(&quot;text&quot;,x=Inf,y=-Inf, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok now, let’s see how the estimated overdispersion estimates (<span class="math inline">\(1/\hat{\phi_j}\)</span>) correlate between species:</p>
<pre class="r"><code>R&lt;-cor(log(ToPlot$Chimp.Overdispersion),log(ToPlot$Human.Overdispersion), use=&quot;complete.obs&quot;)
lb1 &lt;- paste(&quot;~R^2==~&quot;, round(R**2,2))
Overdispersion.Species.Correlation &lt;- ggplot(ToPlot, aes(x=Chimp.Overdispersion, y=Human.Overdispersion)) +
  geom_point(alpha=0.05) +
  scale_x_continuous(trans=&#39;log10&#39;, limits=c(1E-2,10), name=expression(paste(&quot;Chimp overdispersion, &quot;, &quot;1&quot;,&quot;/&quot;, hat(phi)))) +
  scale_y_continuous(trans=&#39;log10&#39;, limits=c(1E-2,10), name=expression(paste(&quot;Human overdispersion, &quot;, &quot;1&quot;,&quot;/&quot;, hat(phi)))) +
  annotate(&quot;text&quot;,x=5,y=0.1, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()
Overdispersion.Species.Correlation</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok those correlate quite well. While this model fitting method for estimating overdispersion parameter should properly take into account the statistical counting variance (related to gene length, and expression level), given that gene expression between species correlate so well (<span class="math inline">\(R^2&gt;0.9\)</span>) it is natural to check if there there may be residual expression-related reasons that explain how dispersion is correlated between species. So let’s see if there is still a correlation between overdispersion <span class="math inline">\(1/\hat{\phi_j}\)</span> and expression level <span class="math inline">\(\hat{\mu_j}\)</span>.</p>
<pre class="r"><code>#Plot overdispersion vs mu
LoessPlot &lt;- ggplot(ToPlot, aes(x=Chimp.Mean.Expression, y=Chimp.Overdispersion)) +
  geom_point(alpha=0.2) +
  scale_x_continuous(name=TeX(&#39;Chimp log(RPKM), $\\hat{\\mu}$&#39;)) +
  scale_y_continuous(trans=&quot;log10&quot;, name=expression(paste(&quot;Chimp overdispersion, &quot;, &quot;log(1&quot;,&quot;/&quot;, hat(phi), &quot;)&quot;)), limits=c(0.001,10)) +
  geom_smooth(method=loess, show.legend = FALSE, se=T, method.args=list(degree=1)) +
  theme_bw()
LoessPlot</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, this is similar to what Abhishek has described as seeing, as well as <a href="https://www.cell.com/cell-systems/pdfExtended/S2405-4712(18)30278-3">Eling et al 2018</a>. Yoav still expressed skepticism that the correlation between species overdispersion is still due some technical aspect to expression levels. To rule that out as best as possible, similar to what I have done before, (and similar to Eling et al) I will regress out the expression level and see if the residual overdispersion <span class="math inline">\(\epsilon_j\)</span> is still well corellated.</p>
<pre class="r"><code>ResidualDemoPlot &lt;- LoessPlot +
  geom_segment(aes(x=-21.125,xend=-21.125, y=0.35, yend=2.5),
  lineend = &quot;round&quot;, linejoin = &quot;round&quot;, color=&quot;brown&quot;,
     size = 0.7, arrow = arrow(length = unit(0.1, &quot;inches&quot;))
  ) +
  annotate(&quot;text&quot;,x=-20,y=0.1, label=TeX(&quot;$\\epsilon_j$&quot;), hjust=1, vjust=-1, color=&quot;brown&quot;, size=10)
ResidualDemoPlot</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GetLoessResidual &lt;- function(x, y){
  loess.fit &lt;- loess(y ~ x, degree=1)
  #for our purposes, the fit with the default degree=2 looks like unnatural fit, especially near the edges
  loess.residual &lt;- y - predict(loess.fit, x)
  return(loess.residual)
}

# I got the residual of log transformed overdispersion estimate, then expontiate it and plot it on log-transformed axis. Just a personal preference for plotting, instead of just keeping data log-transformed.
ToPlot$Chimp.Residual &lt;- exp(GetLoessResidual(ToPlot$Chimp.Mean.Expression, log(ToPlot$Chimp.Overdispersion)))
ToPlot$Human.Residual &lt;- exp(GetLoessResidual(ToPlot$Human.Mean.Expression, log(ToPlot$Human.Overdispersion)))

R&lt;-cor(log(ToPlot$Chimp.Residual),log(ToPlot$Human.Residual), use=&quot;complete.obs&quot;)
cor.test(log(ToPlot$Chimp.Residual),log(ToPlot$Human.Residual), use=&quot;complete.obs&quot;)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  log(ToPlot$Chimp.Residual) and log(ToPlot$Human.Residual)
t = 106.99, df = 11411, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.6983673 0.7166890
sample estimates:
      cor 
0.7076471 </code></pre>
<pre class="r"><code>lb1 &lt;- paste(&quot;~R^2==~&quot;, round(R**2,2))
Overdispersion.Species.Correlation.Corrected &lt;- ggplot(ToPlot, aes(x=Chimp.Residual, y=Human.Residual)) +
  geom_point(alpha=0.05) +
  scale_x_continuous(limits=c(0.05,50), trans=&quot;log10&quot;, name=TeX(&quot;Chimp residual, $\\epsilon_j$&quot;)) +
  scale_y_continuous(limits=c(0.05,50), trans=&quot;log10&quot;, name=TeX(&quot;Human residual, $\\epsilon_j$&quot;)) +
  annotate(&quot;text&quot;,x=5,y=0.1, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()
Overdispersion.Species.Correlation.Corrected</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok so even when I regress out the affect of the mean, I still get a strong correlation of overdispersion between species. What is the biological meaning of this? Are certain gene groups more or less overdispersed in one species or another? What is a reasonable way to go about hypothesis testing for differentially overdispersed genes between species? How does this level of correlation of overdispersion compare to between tissues within human GTEx samples?</p>
<p>First, to help with thinking about how to interpret overdispersed genes, let’s do GSEA to get a sense of what kinds of genes are highly overdispersed. My prediction is that overdispersed genes will have negative enrichment for essential and housekeeping genes.</p>
<pre class="r"><code>gsea.full &lt;- function(NamedGeneVector, ...){
  
  SortedNamedGeneVector &lt;- sort(NamedGeneVector, decreasing=T)

  BP&lt;-gseGO(gene=SortedNamedGeneVector,
            ont = &quot;BP&quot;,
            OrgDb=org.Hs.eg.db,
            keyType=&#39;ENSEMBL&#39;,
            nPerm=100000)
  BiologicalProcess&lt;-as.data.frame(simplify(BP))

  MF&lt;-gseGO(gene=SortedNamedGeneVector,
            ont = &quot;MF&quot;,
            OrgDb=org.Hs.eg.db,
            keyType=&#39;ENSEMBL&#39;,
            nPerm=100000)
  MolecularFunction&lt;-as.data.frame(simplify(MF))

  CC&lt;-gseGO(gene=SortedNamedGeneVector,
            ont = &quot;CC&quot;,
            OrgDb=org.Hs.eg.db,
            keyType=&#39;ENSEMBL&#39;,
            nPerm=100000)
  CellularComponent&lt;-as.data.frame(simplify(CC))

  
  Combined.simplified &lt;- combine(BiologicalProcess,MolecularFunction,CellularComponent)
  
  return(list(BP=BP, MF=MF, CC=CC, Simplified=Combined.simplified))
}

# MeanOverdispersion &lt;- (log(ToPlot$Chimp.Overdispersion) + log(ToPlot$Human.Overdispersion))/2
MeanOverdispersion &lt;- log(ToPlot$Human.Overdispersion)

names(MeanOverdispersion) &lt;- rownames(ToPlot)

GSEA.mean &lt;-gsea.full(MeanOverdispersion)

#Plot most enriched categories (absolute value of enrichment score, with p.adjust&gt;0.05)
OverdispersionGSEAPlot &lt;- GSEA.mean$Simplified %&gt;% 
  filter(setSize&gt;=10) %&gt;%
  filter(p.adjust&lt;=0.003) %&gt;%
  group_by(source) %&gt;%
  top_n(n = 5, wt = abs(enrichmentScore)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=enrichmentScore, y=Description, color=p.adjust, size=setSize)) +
  geom_point() +
  xlim(c(-1,1)) +
  facet_grid(source~., scales = &quot;free&quot;) +
  scale_colour_gradient(low=&quot;red&quot;, high=&quot;black&quot;) +
  facet_grid(source~., scales = &quot;free&quot;) +
  xlab(&quot;Enrichment\nLowOverdispersion&lt;--&gt;HighOverdispersion&quot;) +
  scale_y_discrete(labels = function(x) lapply(strwrap(x, width = 40, simplify = FALSE), paste, collapse=&quot;\n&quot;)) +
  labs(color = &quot;Adjusted P-value&quot;) +
  theme_bw()+ theme(legend.position=&quot;top&quot;)

#Plot most significant categories
GSEA.mean$Simplified %&gt;% 
  group_by(source) %&gt;%
  top_n(n = -5, wt = p.adjust) %&gt;%
  ungroup() %&gt;%
  group_by(source) %&gt;%
  sample_n(8) %&gt;%
  ggplot(aes(x=enrichmentScore, y=Description, color=p.adjust, size=setSize)) +
  geom_point() +
  xlim(c(-1,1)) +
  facet_grid(source~., scales = &quot;free&quot;) +
  scale_colour_gradient(low=&quot;red&quot;, high=&quot;black&quot;) +
  facet_grid(source~., scales = &quot;free&quot;) +
  xlab(&quot;Enrichment\nLowOverdispersion&lt;--&gt;HighOverdispersion&quot;) +
  scale_y_discrete(labels = function(x) lapply(strwrap(x, width = 60, simplify = FALSE), paste, collapse=&quot;\n&quot;)) +
  labs(color = &quot;Adjusted P-value&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Consistent with that intuition, I would say the genes with low dispersion (negative enrichment) are words relating to essential processes, while MHC complex and other immune related words make sense in that they are associated with high overdispersion and these gene sets respond to random environmental insults that could contribute to population variation. Continuing with the idea that low overdispersion is related to important housekeeping genes, let’s check if lowly overdispersed genes are related to gene lists of essentiality (based on <a href="https://www.g3journal.org/content/7/8/2719">Hart et al</a>, a cell culture CRISPR screen experiment).</p>
<pre class="r"><code>EssentialGenes &lt;- read.table(&quot;../data/lists/CEGv2_subset_universe.tsv&quot;, stringsAsFactors = F) %&gt;% pull(V1) %&gt;% mapIds(org.Hs.eg.db, keys=., keytype = &quot;SYMBOL&quot;, column=&quot;ENSEMBL&quot;)

NonEssentialGenes &lt;- read.table(&quot;../data/lists/NEGv1_subset_universe.tsv&quot;, stringsAsFactors = F) %&gt;% pull(V1) %&gt;% mapIds(org.Hs.eg.db, keys=., keytype = &quot;SYMBOL&quot;, column=&quot;ENSEMBL&quot;)


Essentiality &lt;- ToPlot %&gt;%
  rownames_to_column() %&gt;%
  mutate(CultureEssentiality=case_when(
    rowname %in% EssentialGenes ~ &quot;Essential&quot;,
    rowname %in% NonEssentialGenes ~ &quot;NonEssential&quot;)) %&gt;%
  filter(CultureEssentiality %in% c(&quot;Essential&quot;, &quot;NonEssential&quot;))

Essentiality$CultureEssentiality %&gt;% table()</code></pre>
<pre><code>.
   Essential NonEssential 
         555            6 </code></pre>
<pre class="r"><code>  ggplot(Essentiality, aes(color=CultureEssentiality,x=Human.Overdispersion)) +
    stat_ecdf(geom = &quot;step&quot;) +
    scale_x_continuous(trans=&#39;log10&#39;) +
    ylab(&quot;Cumulative frequency&quot;) +
    annotate(&quot;text&quot;, x = 0.5, y = 0.2, label = paste(&quot;Mann-Whitney\none-sided P =&quot;, signif(wilcox.test(data=Essentiality, Human.Overdispersion ~ CultureEssentiality, alternative=&quot;less&quot;)$p.value, 2) )) +
    theme_bw()</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Although it seems that the vast majority of genes for which I estimated overdispersion are essential, from the 6 non essential genes, they do seem to have a significantly higher distribution of overdispersion estimates.</p>
<p>Now let’s ask what types of genes are overdispersed in my chimp sample population relative to human sample population (and vice versa).</p>
<pre class="r"><code>OverdispersionSpeciesDifference &lt;- (log(ToPlot$Chimp.Overdispersion) - log(ToPlot$Human.Overdispersion))
names(OverdispersionSpeciesDifference) &lt;- rownames(ToPlot)

GSEA.Difference &lt;-gsea.full(OverdispersionSpeciesDifference)


#Plot most enriched categories (absolute value of enrichment score, with p.adjust&gt;0.05)
GSEA.Difference$Simplified %&gt;% 
  group_by(source) %&gt;%
  top_n(n = 15, wt = abs(enrichmentScore)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=enrichmentScore, y=Description, color=p.adjust, size=setSize)) +
  geom_point() +
  xlim(c(-1,1)) +
  facet_grid(source~., scales = &quot;free&quot;) +
  scale_colour_gradient(low=&quot;red&quot;, high=&quot;black&quot;) +
  facet_grid(source~., scales = &quot;free&quot;) +
  xlab(&quot;Enrichment\nOverdispersedInHuman&lt;--&gt;OverdispersedInChimp&quot;) +
  scale_y_discrete(labels = function(x) lapply(strwrap(x, width = 60, simplify = FALSE), paste, collapse=&quot;\n&quot;)) +
  labs(color = &quot;Adjusted P-value&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Plot most significant categories
GSEA.Difference$Simplified %&gt;% 
  group_by(source) %&gt;%
  top_n(n = -15, wt = p.adjust) %&gt;%
  ungroup() %&gt;%
  group_by(source) %&gt;%
  sample_n(8) %&gt;%
  ggplot(aes(x=enrichmentScore, y=Description, color=p.adjust, size=setSize)) +
  geom_point() +
  xlim(c(-1,1)) +
  facet_grid(source~., scales = &quot;free&quot;) +
  scale_colour_gradient(low=&quot;red&quot;, high=&quot;black&quot;) +
  facet_grid(source~., scales = &quot;free&quot;) +
  xlab(&quot;Enrichment\nOverdispersedInHuman&lt;--&gt;OverdispersedInChimp&quot;) +
  scale_y_discrete(labels = function(x) lapply(strwrap(x, width = 60, simplify = FALSE), paste, collapse=&quot;\n&quot;)) +
  labs(color = &quot;Adjusted P-value&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Immune words seem to pop up as more overdispersed in chimp. Keep in mind that 7 of our chimps were HCV+ or HBV+. One day I will have to repeat this analysis after excluding those virus-challanged chimps.</p>
<p>Next I am interested in comparing the level of correlation of gene-wise overdispersion statistics across GTEx tissues. The fact that I saw a correlation of (<span class="math inline">\(R^2=0.6\)</span>) across chimp and human seemed high to me, and I am interested in seeing how that compares across tissues within humans. Using a different Rscript, I have already generated a table of mu estimates and overdispersion estimates for each gene for each GTEx tissue using 40 samples per tissue. <a href="20191001_GtexAnnotations.html">Here</a> is more info on how I chose those samples (basically I excluded samples with different types of RNA-isolation methods for library prep and low mapping rates or RIN scores, then randomly selected 40 samples from the remaining set).</p>
<pre class="r"><code>MuTable &lt;- read.table(&quot;../output/GTEx.Tissues.Mu.Matrix.txt.gz&quot;, header=T, sep=&#39;\t&#39;, check.names = F, row.names = 1) %&gt;%
  rownames_to_column(&quot;X&quot;) %&gt;%
  mutate(gene = gsub(&quot;\\.\\d+&quot;, &quot;&quot;, X, perl=T)) %&gt;% dplyr::select(-X)
OverdispersionTable &lt;- read.table(&quot;../output/GTEx.Tissues.Overdispersion.Matrix.txt.gz&quot;, header=T, sep=&#39;\t&#39;, check.names = F, row.names = 1) %&gt;%
  rownames_to_column(&quot;X&quot;)  %&gt;% 
  mutate(gene = gsub(&quot;\\.\\d+&quot;, &quot;&quot;, X, perl=T)) %&gt;% dplyr::select(-X)

OverdispersionCorrelationMatrix&lt;-ToPlot %&gt;% rownames_to_column() %&gt;%
  dplyr::select(rowname, Chimp.Overdispersion, Human.Overdispersion) %&gt;% 
  left_join(OverdispersionTable, by=c(&quot;rowname&quot;=&quot;gene&quot;)) %&gt;%
  column_to_rownames() %&gt;%
  as.matrix() %&gt;% scale() %&gt;% cor(use=&quot;na.or.complete&quot;)

GtexColors &lt;- read.table(&quot;../data/GTEx_Analysis_TissueColorCodes.txt&quot;, sep=&#39;\t&#39;, header=T, stringsAsFactors = F) %&gt;% mutate(HEX=paste0(&quot;#&quot;,Color.code))
GtexColors[nrow(GtexColors) + 1,] = c(&quot;Chimp.Mean.Expression&quot;,&quot;black&quot;, &quot;#000000&quot;)
GtexColors[nrow(GtexColors) + 1,] = c(&quot;Human.Mean.Expression&quot;,&quot;black&quot;, &quot;#000000&quot;)
GtexColors[nrow(GtexColors) + 1,] = c(&quot;Chimp.Overdispersion&quot;,&quot;black&quot;, &quot;#000000&quot;)
GtexColors[nrow(GtexColors) + 1,] = c(&quot;Human.Overdispersion&quot;,&quot;black&quot;, &quot;#000000&quot;)

GtexScheme &lt;- plyr::mapvalues(row.names(OverdispersionCorrelationMatrix), from=GtexColors$Tissue, to=GtexColors$HEX)

OverdispersionCorrelationHeatmap &lt;- heatmap.2(OverdispersionCorrelationMatrix, trace=&quot;none&quot;, dendrogram=c(&quot;col&quot;), labCol=F, cexRow = 0.7, ColSideColors=GtexScheme, colRow = GtexScheme, adjRow=c(0,0), offsetRow=-0.6, main=&quot;overdispersion correlation matrix&quot;)</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>MeanCorrelationMatrix&lt;-ToPlot %&gt;% rownames_to_column() %&gt;%
  dplyr::select(rowname, Chimp.Mean.Expression, Human.Mean.Expression) %&gt;% 
  left_join(MuTable, by=c(&quot;rowname&quot;=&quot;gene&quot;)) %&gt;%
  column_to_rownames() %&gt;%
  as.matrix() %&gt;% scale() %&gt;% cor(use=&quot;na.or.complete&quot;)
MeanCorrelationHeatmap &lt;- heatmap.2(MeanCorrelationMatrix, trace=&quot;none&quot;, dendrogram=c(&quot;col&quot;), labCol=F, cexRow = 0.7, ColSideColors=GtexScheme, colRow = GtexScheme, adjRow=c(0,0), offsetRow=-0.6, main=&quot;mean expression correlation matrix&quot;)</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(OverdispersionGSEAPlot,
          Overdispersion.Species.Correlation,
          ResidualDemoPlot,
          Overdispersion.Species.Correlation.Corrected,
          labels = c(&#39;A&#39;, &#39;B&#39;, &#39;C&#39;), label_size = 12, scale=1, rel_heights = c(2,1))</code></pre>
<p><img src="figure/20190930_OverdispersionEstimates.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] MASS_7.3-51.4          gdata_2.18.0           latex2exp_0.4.0       
 [4] cowplot_0.9.4          enrichplot_1.2.0       org.Hs.eg.db_3.7.0    
 [7] AnnotationDbi_1.44.0   IRanges_2.16.0         S4Vectors_0.20.1      
[10] Biobase_2.42.0         BiocGenerics_0.28.0    clusterProfiler_3.10.1
[13] gplots_3.0.1.1         corrplot_0.84          edgeR_3.24.3          
[16] limma_3.38.3           knitr_1.23             forcats_0.4.0         
[19] stringr_1.4.0          dplyr_0.8.1            purrr_0.3.2           
[22] readr_1.3.1            tidyr_0.8.3            tibble_2.1.3          
[25] ggplot2_3.1.1          tidyverse_1.2.1        plyr_1.8.4            

loaded via a namespace (and not attached):
 [1] fgsea_1.8.0         colorspace_1.4-1    ggridges_0.5.1     
 [4] rprojroot_1.3-2     qvalue_2.14.1       fs_1.3.1           
 [7] rstudioapi_0.10     farver_1.1.0        urltools_1.7.3     
[10] ggrepel_0.8.1       bit64_0.9-7         lubridate_1.7.4    
[13] xml2_1.2.0          splines_3.5.1       GOSemSim_2.8.0     
[16] polyclip_1.10-0     jsonlite_1.6        workflowr_1.4.0    
[19] broom_0.5.2         GO.db_3.7.0         ggforce_0.2.2      
[22] compiler_3.5.1      httr_1.4.0          rvcheck_0.1.3      
[25] backports_1.1.4     assertthat_0.2.1    Matrix_1.2-17      
[28] lazyeval_0.2.2      cli_1.1.0           tweenr_1.0.1       
[31] htmltools_0.3.6     prettyunits_1.0.2   tools_3.5.1        
[34] igraph_1.2.4.1      gtable_0.3.0        glue_1.3.1         
[37] reshape2_1.4.3      DO.db_2.9           fastmatch_1.1-0    
[40] Rcpp_1.0.1          cellranger_1.1.0    nlme_3.1-140       
[43] ggraph_1.0.2        xfun_0.7            rvest_0.3.4        
[46] gtools_3.8.1        DOSE_3.8.2          europepmc_0.3      
[49] scales_1.0.0        hms_0.4.2           RColorBrewer_1.1-2 
[52] yaml_2.2.0          memoise_1.1.0       gridExtra_2.3      
[55] UpSetR_1.4.0        triebeard_0.3.0     stringi_1.4.3      
[58] RSQLite_2.1.1       caTools_1.17.1.2    BiocParallel_1.16.6
[61] rlang_0.3.4         pkgconfig_2.0.2     bitops_1.0-6       
[64] evaluate_0.14       lattice_0.20-38     labeling_0.3       
[67] bit_1.1-14          tidyselect_0.2.5    magrittr_1.5       
[70] R6_2.4.0            generics_0.0.2      DBI_1.0.0          
[73] pillar_1.4.1        haven_2.1.0         whisker_0.3-2      
[76] withr_2.1.2         modelr_0.1.4        crayon_1.3.4       
[79] KernSmooth_2.23-15  rmarkdown_1.13      viridis_0.5.1      
[82] progress_1.2.2      locfit_1.5-9.1      grid_3.5.1         
[85] readxl_1.3.1        data.table_1.12.2   blob_1.1.1         
[88] git2r_0.25.2        digest_0.6.19       gridGraphics_0.4-1 
[91] munsell_0.5.0       viridisLite_0.3.0   ggplotify_0.0.3    </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
