<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>20190613_PowerAnalysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20190613_PowerAnalysis</h1>
<h4 class="author">Ben Fair</h4>
<h4 class="date">6/13/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree0a2a4b1c183e1048f882dc8ce51b87a3638b5e1etargetblank0a2a4b1a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/0a2a4b1c183e1048f882dc8ce51b87a3638b5e1e" target="_blank">0a2a4b1</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree0a2a4b1c183e1048f882dc8ce51b87a3638b5e1etargetblank0a2a4b1a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/20190521_eQTL_CrossSpeciesEnrichment_cache/
    Ignored:    analysis/figure/
    Ignored:    analysis_temp/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/snakemake_workflow/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    docs/assets/.DS_Store

Unstaged changes:
    Modified:   analysis/20190613_PowerAnalysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/analysis/20190613_PowerAnalysis.Rmd" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/20190613_PowerAnalysis.html" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/aa8067e373e87c51af6dce8ae46faf3cdbe3971c/analysis/20190613_PowerAnalysis.Rmd" target="_blank">aa8067e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-15
</td>
<td>
added power analysis rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(&quot;edgeR&quot;)
library(corrplot)
library(gplots)
library(pROC)
library(qvalue)
library(reshape2)
library(gridExtra)</code></pre>
<p>39 Chimp heart RNA-seq datasets (from this project as well as from Pavlovic et al 2018) as well as 10 human heart RNA-seq datasets (from Pavlovic et al) and 39 randomly selected GTEx left ventricle heart RNA-seq datasets were trimmed to same read length (single end, 75bp) and aligned to the respective genomes. Gene counts were obtained with subread software using gene annotations based only on orthologous exons (Pavlovic et al 2018). Here I will perform differential gene expression analysis to understand the relationship between read depth and number of individuals (samples) needed to identify cross-species differentially expressed genes.</p>
<pre class="r"><code>CountTableChimpFile &lt;- &#39;../output/PowerAnalysisCountTable.Chimp.25000000.subread.txt.gz&#39;
CountTableChimp &lt;- read.table(gzfile(CountTableChimpFile), header=T, check.names=FALSE, skip=1)
colnames(CountTableChimp) &lt;- paste0(&quot;C.&quot;, colnames(CountTableChimp))
kable(CountTableChimp[1:10,1:10])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">C.Geneid</th>
<th align="left">C.Chr</th>
<th align="left">C.Start</th>
<th align="left">C.End</th>
<th align="left">C.Strand</th>
<th align="right">C.Length</th>
<th align="right">C.4x373</th>
<th align="right">C.95A014</th>
<th align="right">C.676</th>
<th align="right">C.554_2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000273443</td>
<td align="left">1;1;1</td>
<td align="left">175684;176538;176679</td>
<td align="left">176285;176555;176765</td>
<td align="left">-;-;-</td>
<td align="right">707</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">ENSG00000217801</td>
<td align="left">1;1;1;1;1;1;1</td>
<td align="left">176957;177673;177953;178638;179598;179901;180611</td>
<td align="left">177043;177864;178052;178718;179678;180035;180758</td>
<td align="left">+;+;+;+;+;+;+</td>
<td align="right">824</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000237330</td>
<td align="left">1</td>
<td align="left">186042</td>
<td align="left">186330</td>
<td align="left">-</td>
<td align="right">289</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ENSG00000223823</td>
<td align="left">1</td>
<td align="left">313681</td>
<td align="left">313774</td>
<td align="left">+</td>
<td align="right">94</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000207730</td>
<td align="left">1</td>
<td align="left">355180</td>
<td align="left">355274</td>
<td align="left">+</td>
<td align="right">95</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ENSG00000207607</td>
<td align="left">1</td>
<td align="left">355943</td>
<td align="left">356032</td>
<td align="left">+</td>
<td align="right">90</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198976</td>
<td align="left">1</td>
<td align="left">357158</td>
<td align="left">357240</td>
<td align="left">+</td>
<td align="right">83</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ENSG00000272141</td>
<td align="left">1</td>
<td align="left">357510</td>
<td align="left">358497</td>
<td align="left">+</td>
<td align="right">988</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000205231</td>
<td align="left">1;1;1;1</td>
<td align="left">361286;362148;362543;362711</td>
<td align="left">362104;362486;362644;362741</td>
<td align="left">-;-;-;-</td>
<td align="right">1291</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">ENSG00000186891</td>
<td align="left">1;1;1;1;1</td>
<td align="left">387554;388538;389740;389995;390258</td>
<td align="left">387860;388847;389827;390192;390717</td>
<td align="left">+;+;+;+;+</td>
<td align="right">1363</td>
<td align="right">45</td>
<td align="right">4</td>
<td align="right">8</td>
<td align="right">11</td>
</tr>
</tbody>
</table>
<pre class="r"><code>CountTableHumanFile &lt;- &#39;../output/PowerAnalysisCountTable.Human.25000000.subread.txt.gz&#39;
CountTableHuman &lt;- read.table(gzfile(CountTableHumanFile), header=T, check.names=FALSE, skip=1)
colnames(CountTableHuman) &lt;- paste0(&quot;H.&quot;, colnames(CountTableHuman))
kable(CountTableHuman[1:10,1:10])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">H.Geneid</th>
<th align="left">H.Chr</th>
<th align="left">H.Start</th>
<th align="left">H.End</th>
<th align="left">H.Strand</th>
<th align="right">H.Length</th>
<th align="right">H.SRR599380</th>
<th align="right">H.59511</th>
<th align="right">H.SRR1477033</th>
<th align="right">H.62606</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000188976</td>
<td align="left">1</td>
<td align="left">959215</td>
<td align="left">959309</td>
<td align="left">-</td>
<td align="right">95</td>
<td align="right">12</td>
<td align="right">10</td>
<td align="right">14</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">ENSG00000188157</td>
<td align="left">1</td>
<td align="left">1041478</td>
<td align="left">1041702</td>
<td align="left">+</td>
<td align="right">225</td>
<td align="right">12</td>
<td align="right">14</td>
<td align="right">6</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000273443</td>
<td align="left">1;1;1</td>
<td align="left">1062208;1063061;1063202</td>
<td align="left">1062808;1063078;1063288</td>
<td align="left">-;-;-</td>
<td align="right">706</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ENSG00000217801</td>
<td align="left">1;1;1;1;1;1;1</td>
<td align="left">1063480;1064210;1064490;1066587;1067566;1067869;1068577</td>
<td align="left">1063566;1064401;1064589;1066667;1067646;1068003;1068724</td>
<td align="left">+;+;+;+;+;+;+</td>
<td align="right">824</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000237330</td>
<td align="left">1</td>
<td align="left">1074016</td>
<td align="left">1074307</td>
<td align="left">-</td>
<td align="right">292</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ENSG00000223823</td>
<td align="left">1</td>
<td align="left">1137017</td>
<td align="left">1137110</td>
<td align="left">+</td>
<td align="right">94</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000207730</td>
<td align="left">1</td>
<td align="left">1167104</td>
<td align="left">1167198</td>
<td align="left">+</td>
<td align="right">95</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ENSG00000207607</td>
<td align="left">1</td>
<td align="left">1167863</td>
<td align="left">1167952</td>
<td align="left">+</td>
<td align="right">90</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198976</td>
<td align="left">1</td>
<td align="left">1169005</td>
<td align="left">1169087</td>
<td align="left">+</td>
<td align="right">83</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ENSG00000272141</td>
<td align="left">1</td>
<td align="left">1169357</td>
<td align="left">1170343</td>
<td align="left">+</td>
<td align="right">987</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>CombinedTable &lt;- inner_join(CountTableChimp[,c(1,7:length(CountTableChimp))], CountTableHuman[,c(1,7:length(CountTableHuman))], by=c(&quot;C.Geneid&quot;=&quot;H.Geneid&quot;)) %&gt;%
  column_to_rownames(&quot;C.Geneid&quot;) %&gt;% as.matrix()

#Plot depth per sample
CombinedTable %&gt;% colSums() %&gt;% as.data.frame() %&gt;%
  rownames_to_column(&quot;Sample&quot;) %&gt;%
  mutate(Species=substr(Sample, 1,1)) %&gt;%
  ggplot(aes(x=Sample, y=., fill=Species)) +
  geom_col() +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, size=6))</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/read-data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-read-data-1">
Past versions of read-data-1.png
</button>
</p>
<div id="fig-read-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/read-data-1.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cpm &lt;- cpm(CombinedTable, log=TRUE, prior.count=0.5)
kable(cpm[1:10,1:10])</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">C.4x373</th>
<th align="right">C.95A014</th>
<th align="right">C.676</th>
<th align="right">C.554_2</th>
<th align="right">C.456</th>
<th align="right">C.503</th>
<th align="right">C.4x0043</th>
<th align="right">C.570</th>
<th align="right">C.4X0095</th>
<th align="right">C.4x523</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000273443</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-2.0666967</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-2.772420</td>
<td align="right">-4.402683</td>
</tr>
<tr class="even">
<td>ENSG00000217801</td>
<td align="right">-1.659074</td>
<td align="right">-2.841283</td>
<td align="right">-0.6087941</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-0.5631552</td>
<td align="right">-2.198895</td>
<td align="right">-1.537341</td>
<td align="right">-3.016478</td>
</tr>
<tr class="odd">
<td>ENSG00000237330</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-2.7496282</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
</tr>
<tr class="even">
<td>ENSG00000223823</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
</tr>
<tr class="odd">
<td>ENSG00000207730</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
</tr>
<tr class="even">
<td>ENSG00000207607</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-2.8059958</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
</tr>
<tr class="odd">
<td>ENSG00000198976</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
</tr>
<tr class="even">
<td>ENSG00000272141</td>
<td align="right">-2.140163</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
</tr>
<tr class="odd">
<td>ENSG00000205231</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.4026833</td>
<td align="right">-2.8059958</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-2.1382998</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
<td align="right">-4.402683</td>
</tr>
<tr class="even">
<td>ENSG00000186891</td>
<td align="right">2.031303</td>
<td align="right">-1.264262</td>
<td align="right">-0.2199978</td>
<td align="right">0.1376718</td>
<td align="right">-1.624826</td>
<td align="right">-2.760991</td>
<td align="right">1.5922154</td>
<td align="right">-2.198895</td>
<td align="right">-1.537341</td>
<td align="right">1.315537</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Heatmap of correlation. Species segregate as expected.
SpeciesFactor &lt;- colnames(cpm) %&gt;% substr(1,1) %&gt;% factor() %&gt;% unclass() %&gt;% as.character()

cor(cpm, method = c(&quot;spearman&quot;)) %&gt;%
heatmap.2(trace=&quot;none&quot;, ColSideColors=SpeciesFactor)</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/read-data-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-read-data-2">
Past versions of read-data-2.png
</button>
</p>
<div id="fig-read-data-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/read-data-2.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Unsurprisingly, the samples with the lowest read depth in the human cohort are clear outliers. This might change once I filter out the more lowly expressed genes.</p>
<pre class="r"><code>d0 &lt;- DGEList(CombinedTable)

#Calculate normalization factors
d0 &lt;- calcNormFactors(d0)

#Note: calcNormFactors doesn’t normalize the data, it just calculates normalization factors for use downstream.
#Filter low-expressed genes

cutoff &lt;- 8
drop &lt;- which(apply(cpm(d0), 1, max) &lt; cutoff)
d &lt;- d0[-drop,] 
dim(d) # number of genes left</code></pre>
<pre><code>[1] 13097    88</code></pre>
<pre class="r"><code>plotMDS(d, col = as.numeric(SpeciesFactor))</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-1-1.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotMDS(d, col = as.numeric(SpeciesFactor), dim=c(3,4))</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-2">
Past versions of unnamed-chunk-1-2.png
</button>
</p>
<div id="fig-unnamed-chunk-1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-1-2.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cor(cpm(d, log=T, prior.count=0.5), method = c(&quot;spearman&quot;)) %&gt;%
heatmap.2(trace=&quot;none&quot;, ColSideColors=SpeciesFactor)</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-3">
Past versions of unnamed-chunk-1-3.png
</button>
</p>
<div id="fig-unnamed-chunk-1-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-1-3.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>mm &lt;- model.matrix(~0 + SpeciesFactor)
y &lt;- voom(d, mm, plot = T, normalize.method=&quot;cyclicloess&quot;)</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-1-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-4">
Past versions of unnamed-chunk-1-4.png
</button>
</p>
<div id="fig-unnamed-chunk-1-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-1-4.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In fact some of these samples seemed to have gotten worse. I’ll just throw these out of future analysis.</p>
<pre class="r"><code>HumanSamplesToDrop &lt;- c(&quot;SRR1507229&quot;,&quot;SRR603918&quot;, &quot;SRR1478149&quot;, &quot;SRR598509&quot;, &quot;SRR613186&quot;)
ChimpSamplesToDrop &lt;- c(&quot;Little_R&quot;)</code></pre>
<p>Anway, now that voom calculated mean/variance relationship from count values, I will now normalize for gene length differences between species (Convert cpm to rpkm). As per the voom publication (Law et al 2014) log-cpm values output by voom can be converted to log-rpkm by subtracting the log2 gene length in kilobases. For this I will make a matrix of gene lengths based on chimp orthologous exon length and human orthologous exon lengths to subtract the correct length for each species from the count matrix output by voom.</p>
<pre class="r"><code>rep.col&lt;-function(x,n){
   matrix(rep(x,each=n), ncol=n, byrow=TRUE)
}

GeneLengths &lt;- inner_join(CountTableChimp[,c(&quot;C.Geneid&quot;, &quot;C.Length&quot;)], CountTableHuman[,c(&quot;H.Geneid&quot;, &quot;H.Length&quot;)], by=c(&quot;C.Geneid&quot;=&quot;H.Geneid&quot;))
head(kable(GeneLengths))</code></pre>
<pre><code>[1] &quot;C.Geneid           C.Length   H.Length&quot;
[2] &quot;----------------  ---------  ---------&quot;
[3] &quot;ENSG00000273443         707        706&quot;
[4] &quot;ENSG00000217801         824        824&quot;
[5] &quot;ENSG00000237330         289        292&quot;
[6] &quot;ENSG00000223823          94         94&quot;</code></pre>
<pre class="r"><code>ggplot(GeneLengths, aes(x=log10(C.Length), y=log10(H.Length))) + geom_point()</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-3-1.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># accounting for the length differences probably will have negligble effect on results anyway. # Will probably calculate DE genes both ways (cpm and rpkm) to verify

GeneLengthMatrix &lt;- cbind(
  rep.col(log2(GeneLengths$C.Length/1000), length(CountTableChimp)-6),
  rep.col(log2(GeneLengths$H.Length/1000), length(CountTableHuman)-6))
rownames(GeneLengthMatrix) &lt;- GeneLengths$C.Geneid
kable(GeneLengthMatrix[1:10,1:10])</code></pre>
<table>
<tbody>
<tr class="odd">
<td align="left">ENSG00000273443</td>
<td align="right">-0.5002179</td>
<td align="right">-0.5002179</td>
<td align="right">-0.5002179</td>
<td align="right">-0.5002179</td>
<td align="right">-0.5002179</td>
<td align="right">-0.5002179</td>
<td align="right">-0.5002179</td>
<td align="right">-0.5002179</td>
<td align="right">-0.5002179</td>
<td align="right">-0.5002179</td>
</tr>
<tr class="even">
<td align="left">ENSG00000217801</td>
<td align="right">-0.2792838</td>
<td align="right">-0.2792838</td>
<td align="right">-0.2792838</td>
<td align="right">-0.2792838</td>
<td align="right">-0.2792838</td>
<td align="right">-0.2792838</td>
<td align="right">-0.2792838</td>
<td align="right">-0.2792838</td>
<td align="right">-0.2792838</td>
<td align="right">-0.2792838</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000237330</td>
<td align="right">-1.7908586</td>
<td align="right">-1.7908586</td>
<td align="right">-1.7908586</td>
<td align="right">-1.7908586</td>
<td align="right">-1.7908586</td>
<td align="right">-1.7908586</td>
<td align="right">-1.7908586</td>
<td align="right">-1.7908586</td>
<td align="right">-1.7908586</td>
<td align="right">-1.7908586</td>
</tr>
<tr class="even">
<td align="left">ENSG00000223823</td>
<td align="right">-3.4111954</td>
<td align="right">-3.4111954</td>
<td align="right">-3.4111954</td>
<td align="right">-3.4111954</td>
<td align="right">-3.4111954</td>
<td align="right">-3.4111954</td>
<td align="right">-3.4111954</td>
<td align="right">-3.4111954</td>
<td align="right">-3.4111954</td>
<td align="right">-3.4111954</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000207730</td>
<td align="right">-3.3959287</td>
<td align="right">-3.3959287</td>
<td align="right">-3.3959287</td>
<td align="right">-3.3959287</td>
<td align="right">-3.3959287</td>
<td align="right">-3.3959287</td>
<td align="right">-3.3959287</td>
<td align="right">-3.3959287</td>
<td align="right">-3.3959287</td>
<td align="right">-3.3959287</td>
</tr>
<tr class="even">
<td align="left">ENSG00000207607</td>
<td align="right">-3.4739312</td>
<td align="right">-3.4739312</td>
<td align="right">-3.4739312</td>
<td align="right">-3.4739312</td>
<td align="right">-3.4739312</td>
<td align="right">-3.4739312</td>
<td align="right">-3.4739312</td>
<td align="right">-3.4739312</td>
<td align="right">-3.4739312</td>
<td align="right">-3.4739312</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198976</td>
<td align="right">-3.5907449</td>
<td align="right">-3.5907449</td>
<td align="right">-3.5907449</td>
<td align="right">-3.5907449</td>
<td align="right">-3.5907449</td>
<td align="right">-3.5907449</td>
<td align="right">-3.5907449</td>
<td align="right">-3.5907449</td>
<td align="right">-3.5907449</td>
<td align="right">-3.5907449</td>
</tr>
<tr class="even">
<td align="left">ENSG00000272141</td>
<td align="right">-0.0174171</td>
<td align="right">-0.0174171</td>
<td align="right">-0.0174171</td>
<td align="right">-0.0174171</td>
<td align="right">-0.0174171</td>
<td align="right">-0.0174171</td>
<td align="right">-0.0174171</td>
<td align="right">-0.0174171</td>
<td align="right">-0.0174171</td>
<td align="right">-0.0174171</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000205231</td>
<td align="right">0.3684890</td>
<td align="right">0.3684890</td>
<td align="right">0.3684890</td>
<td align="right">0.3684890</td>
<td align="right">0.3684890</td>
<td align="right">0.3684890</td>
<td align="right">0.3684890</td>
<td align="right">0.3684890</td>
<td align="right">0.3684890</td>
<td align="right">0.3684890</td>
</tr>
<tr class="even">
<td align="left">ENSG00000186891</td>
<td align="right">0.4467856</td>
<td align="right">0.4467856</td>
<td align="right">0.4467856</td>
<td align="right">0.4467856</td>
<td align="right">0.4467856</td>
<td align="right">0.4467856</td>
<td align="right">0.4467856</td>
<td align="right">0.4467856</td>
<td align="right">0.4467856</td>
<td align="right">0.4467856</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#subtract gene log2(kb) from log2(cpm)
y$E &lt;- y$E - GeneLengthMatrix[rownames(y$E),]

#Now do model fitting, significance testing
fit&lt;- lmFit(y, mm)
plotSA(fit)</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>head(coef(fit))</code></pre>
<pre><code>                SpeciesFactor1 SpeciesFactor2
ENSG00000186891     -0.5042645     -2.9843307
ENSG00000186827      2.1923203      1.5926387
ENSG00000078808      4.9196758      5.2185157
ENSG00000176022      3.0372329      2.8850147
ENSG00000184163     -0.1085559     -0.5281716
ENSG00000160087      2.4507578      2.5740626</code></pre>
<pre class="r"><code>head(y$E[,1])</code></pre>
<pre><code>ENSG00000186891 ENSG00000186827 ENSG00000078808 ENSG00000176022 
      1.9909430       3.8369592       4.6769123       2.9626243 
ENSG00000184163 ENSG00000160087 
      0.4987694       2.4420370 </code></pre>
<pre class="r"><code>contr &lt;- makeContrasts(DE=SpeciesFactor1-SpeciesFactor2, levels = mm)

tmp &lt;- contrasts.fit(fit, contrasts=contr)
FC.NullInterval &lt;- log2(1.0)
True.efit &lt;- treat(tmp, lfc = FC.NullInterval)

summary(decideTests(True.efit))</code></pre>
<pre><code>         DE
Down   4402
NotSig 4279
Up     4416</code></pre>
<pre class="r"><code>TrueResponse &lt;- decideTests(True.efit)
plotMD(True.efit, column=1, zero.weights = F)</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, seems like the above workflow for identifying DE genes is set up reasonably… Now let’s repeat with less samples and see what changes. First, I will wrap all of the above analysis into a function with a sample size parameter:</p>
<pre class="r"><code>DE.Subsampled &lt;-function(ChimpCountTableFile, HumanCountTableFile, SubsampleSize, FC.NullInterval, drop, ChimpSampleDrop=NULL, HumanSampleDrop=NULL)
  #if SubsampleSize parameter == 0, use full table, otherwise, subsample from it
  {
   FullChimpData &lt;- read.table(gzfile(ChimpCountTableFile), header=T, check.names=FALSE, skip=1)
   FullHumanData &lt;- read.table(gzfile(HumanCountTableFile), header=T, check.names=FALSE, skip=1)
   
  if (!is.null(ChimpSampleDrop)){
     FullChimpData &lt;- FullChimpData %&gt;% dplyr::select(-ChimpSampleDrop)
   }   
  if (!is.null(HumanSampleDrop)){
     FullHumanData &lt;- FullHumanData %&gt;% dplyr::select(-HumanSampleDrop)
   }

   if (SubsampleSize==0){
     CountTableChimp &lt;- FullChimpData
     colnames(CountTableChimp) &lt;- paste0(&quot;C.&quot;, colnames(CountTableChimp))
     CountTableHuman &lt;- FullHumanData
     colnames(CountTableHuman) &lt;- paste0(&quot;H.&quot;, colnames(CountTableHuman))
     
   } else {
     CountTableChimp &lt;- FullChimpData %&gt;% dplyr::select(c(1:6, sample(7:length(FullChimpData), SubsampleSize)))
     colnames(CountTableChimp) &lt;- paste0(&quot;C.&quot;, colnames(CountTableChimp))
     
     CountTableHuman &lt;- FullHumanData %&gt;% dplyr::select(c(1:6, sample(7:length(FullHumanData), SubsampleSize)))
     colnames(CountTableHuman) &lt;- paste0(&quot;H.&quot;, colnames(CountTableHuman))
   }
   
   



CombinedTable &lt;- inner_join(CountTableChimp[,c(1,7:length(CountTableChimp))], CountTableHuman[,c(1,7:length(CountTableHuman))], by=c(&quot;C.Geneid&quot;=&quot;H.Geneid&quot;)) %&gt;%
  column_to_rownames(&quot;C.Geneid&quot;) %&gt;% as.matrix()

SpeciesFactor &lt;- colnames(CombinedTable) %&gt;% substr(1,1) %&gt;% factor() %&gt;% unclass() %&gt;% as.character()

d0 &lt;- DGEList(CombinedTable)
d0 &lt;- calcNormFactors(d0)
d &lt;- d0[-drop,] 
mm &lt;- model.matrix(~0 + SpeciesFactor)
y &lt;- voom(d, mm, normalize.method=&quot;cyclicloess&quot;, plot=F)

GeneLengths &lt;- inner_join(CountTableChimp[,c(&quot;C.Geneid&quot;, &quot;C.Length&quot;)], CountTableHuman[,c(&quot;H.Geneid&quot;, &quot;H.Length&quot;)], by=c(&quot;C.Geneid&quot;=&quot;H.Geneid&quot;))
GeneLengthMatrix &lt;- cbind(
  rep.col(log2(GeneLengths$C.Length/1000), length(CountTableChimp)-6),
  rep.col(log2(GeneLengths$H.Length/1000), length(CountTableHuman)-6))
rownames(GeneLengthMatrix) &lt;- GeneLengths$C.Geneid
y$E &lt;- y$E - GeneLengthMatrix[rownames(y$E),]

fit&lt;- lmFit(y, mm)
contr &lt;- makeContrasts(DE=SpeciesFactor1-SpeciesFactor2, levels = mm)
tmp &lt;- contrasts.fit(fit, contrasts=contr)
efit &lt;- treat(tmp, lfc = FC.NullInterval)
return(efit)
}</code></pre>
<p>Now that I have a function to do all of the DE gene analysis, use the function with a smaller sample size and check results:</p>
<pre class="r"><code>#20 samples
Subsampled.DE.results &lt;- DE.Subsampled(CountTableChimpFile,
              CountTableHumanFile,
              20, 0, drop)
summary(decideTests(Subsampled.DE.results))</code></pre>
<pre><code>         DE
Down   3848
NotSig 5494
Up     3755</code></pre>
<pre class="r"><code>RocCurveData &lt;- coords(roc(response=as.vector(abs(TrueResponse)), predictor=Subsampled.DE.results$p.value, plot=F))</code></pre>
<pre><code>Setting levels: control = 0, case = 1</code></pre>
<pre><code>Warning in roc.default(response = as.vector(abs(TrueResponse)), predictor
= Subsampled.DE.results$p.value, : Deprecated use a matrix as predictor.
Unexpected results may be produced, please pass a numeric vector.</code></pre>
<pre><code>Setting direction: controls &gt; cases</code></pre>
<pre><code>Warning in coords.roc(roc(response = as.vector(abs(TrueResponse)),
predictor = Subsampled.DE.results$p.value, : An upcoming version of pROC
will set the &#39;transpose&#39; argument to FALSE by default. Set transpose =
TRUE explicitly to keep the current behavior, or transpose = FALSE to
adopt the new one and silence this warning. Type help(coords_transpose) for
additional information.</code></pre>
<pre class="r"><code>plot(1-RocCurveData[&quot;specificity&quot;,], RocCurveData[&quot;sensitivity&quot;,])</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-5-1.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>SubSampledResponse &lt;- decideTests(Subsampled.DE.results)

# distribution of effect sizes for true positives
hist(abs(True.efit$coefficients[TrueResponse==SubSampledResponse &amp; SubSampledResponse!=0]), main=&quot;|effect size| of true positives&quot;)</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-5-2.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>median(abs(True.efit$coefficients[TrueResponse==SubSampledResponse &amp; SubSampledResponse!=0]))</code></pre>
<pre><code>[1] 0.7031364</code></pre>
<pre class="r"><code>#5 samples
Subsampled.DE.results &lt;- DE.Subsampled(CountTableChimpFile,
              CountTableHumanFile,
              5, 0, drop)
summary(decideTests(Subsampled.DE.results))</code></pre>
<pre><code>         DE
Down   1538
NotSig 9911
Up     1648</code></pre>
<pre class="r"><code>RocCurveData &lt;- coords(roc(response=as.vector(abs(TrueResponse)), predictor=Subsampled.DE.results$p.value, plot=F))</code></pre>
<pre><code>Setting levels: control = 0, case = 1</code></pre>
<pre><code>Warning in roc.default(response = as.vector(abs(TrueResponse)), predictor
= Subsampled.DE.results$p.value, : Deprecated use a matrix as predictor.
Unexpected results may be produced, please pass a numeric vector.</code></pre>
<pre><code>Setting direction: controls &gt; cases</code></pre>
<pre><code>Warning in coords.roc(roc(response = as.vector(abs(TrueResponse)),
predictor = Subsampled.DE.results$p.value, : An upcoming version of pROC
will set the &#39;transpose&#39; argument to FALSE by default. Set transpose =
TRUE explicitly to keep the current behavior, or transpose = FALSE to
adopt the new one and silence this warning. Type help(coords_transpose) for
additional information.</code></pre>
<pre class="r"><code>plot(1-RocCurveData[&quot;specificity&quot;,], RocCurveData[&quot;sensitivity&quot;,])</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-3">
Past versions of unnamed-chunk-5-3.png
</button>
</p>
<div id="fig-unnamed-chunk-5-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-5-3.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>SubSampledResponse &lt;- decideTests(Subsampled.DE.results)

# distribution of effect sizes for true positives
hist(abs(True.efit$coefficients[TrueResponse==SubSampledResponse &amp; SubSampledResponse!=0]), main=&quot;|effect size| of true positives&quot;)</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-4">
Past versions of unnamed-chunk-5-4.png
</button>
</p>
<div id="fig-unnamed-chunk-5-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-5-4.png" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>median(abs(True.efit$coefficients[TrueResponse==SubSampledResponse &amp; SubSampledResponse!=0]))</code></pre>
<pre><code>[1] 1.014835</code></pre>
<p>Now I will systematically do this for varying sample sizes and make some final plots</p>
<pre class="r"><code>#Note there is some randomness in subsampling samples. So much so that sometimes it effects results wherein 4 samples might yield more DE genes than 2 if you get unluck and pick &quot;bad&quot; samples in the 4
set.seed(1)


#Interval for null hypothesis
FC.NullInterval &lt;- log2(1.0)

#True results are those using all samples
True.efit &lt;- DE.Subsampled(CountTableChimpFile,
              CountTableHumanFile,
              0, FC.NullInterval, drop, ChimpSampleDrop=ChimpSamplesToDrop, HumanSampleDrop = HumanSamplesToDrop)

SampleSizes &lt;- c(2,4,8,16,24,32,38)
FDRLevels &lt;- c(0.01, 0.05, 0.1)


RocCurveDataToPlot &lt;- data.frame()
DEGeneCountToPlot &lt;- matrix(nrow=length(SampleSizes), ncol=length(FDRLevels))
rownames(DEGeneCountToPlot) &lt;- SampleSizes
EffectSizesToPlot &lt;- data.frame()

for (i in seq_along(SampleSizes)){
  paste0(&quot;processing &quot;, SampleSizes[i])
  Results &lt;- DE.Subsampled(CountTableChimpFile,
              CountTableHumanFile,
              SampleSizes[i], FC.NullInterval, drop, ChimpSampleDrop=ChimpSamplesToDrop, HumanSampleDrop = HumanSamplesToDrop)
  
  RocCurveData &lt;- as.data.frame(coords(roc(response=as.vector(abs(TrueResponse)), quiet=T, predictor=as.numeric(Results$p.value), plot=F), transpose=F))
  RocCurveData$samplesize &lt;- SampleSizes[i]
  RocCurveDataToPlot &lt;- rbind(RocCurveDataToPlot, RocCurveData)
  
  for (j in seq_along(FDRLevels)){
    
    SubSampledResponse &lt;- decideTests(Results, p.value=FDRLevels[j])
    
    DEGeneCountToPlot[i,j] &lt;- sum(table(SubSampledResponse)[c(&quot;-1&quot;,&quot;1&quot;)])
    
    if (length(table(TrueResponse==SubSampledResponse &amp; SubSampledResponse!=0)) &gt; 1){
      EffectSizes.df &lt;-data.frame(EffectSizes=abs(True.efit$coefficients[TrueResponse==SubSampledResponse &amp; SubSampledResponse!=0]), FDR=FDRLevels[j], SampleSize=SampleSizes[i])
      EffectSizesToPlot &lt;- rbind(EffectSizesToPlot, EffectSizes.df)
    }

    
  }
}

ggplot(RocCurveDataToPlot, aes(x=1-specificity, y=sensitivity, color=factor(samplesize))) +
  geom_line() +
  theme_bw()</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DEGeneCountToPlot.df&lt;-as.data.frame(DEGeneCountToPlot)
colnames(DEGeneCountToPlot.df) &lt;- FDRLevels
DEGeneCountToPlot.df$SampleSize &lt;- rownames(DEGeneCountToPlot.df)
DEGeneCountToPlot.df$SampleSize</code></pre>
<pre><code>[1] &quot;2&quot;  &quot;4&quot;  &quot;8&quot;  &quot;16&quot; &quot;24&quot; &quot;32&quot; &quot;38&quot;</code></pre>
<pre class="r"><code>DEGeneCountToPlot.df[is.na(DEGeneCountToPlot.df)] &lt;- 0
DEGeneCountToPlot.df %&gt;% melt() %&gt;%
  dplyr::rename(Number.DE.genes=value, FDR=variable) %&gt;%
ggplot(aes(x=as.numeric(SampleSize), y=Number.DE.genes, color=FDR)) +
  geom_line() +
  theme_bw()</code></pre>
<pre><code>Using SampleSize as id variables</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(EffectSizesToPlot, aes(x=factor(SampleSize), y=EffectSizes, color=factor(FDR))) +
  # geom_violin()
  geom_boxplot() +
  ylab(&quot;Effect Size of True positive DE genes\n|log2(FC)|&quot;) +
  # scale_y_continuous(limits=c(0,5))
  theme_bw()</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggplot(EffectSizesToPlot, aes(x=as.numeric(SampleSize), y=median(EffectSizes), color=factor(FDR), group=factor(FDR))) +
#   geom_line() +
#   # scale_y_continuous(limits=c(0,5))
#   theme_bw()

data.frame(coefficients=as.numeric(True.efit$coefficients), pval=as.numeric(True.efit$p.value), signif=decideTests(True.efit)) %&gt;%
ggplot(aes(x=coefficients, y=-log10(pval), color=factor(DE))) +
  geom_point() +
  scale_x_continuous(limits=c(-5,5))</code></pre>
<pre><code>Warning: Removed 116 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-6-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>I probably also want to explore how read depth plays into this, especially considering some samples were much more sparse than others. Here I’ll try to write a function to subsample without replacement the read counts from the original count table.</p>
<pre class="r"><code>sample_species &lt;- function(counts,n) {
  num_species &lt;- length(counts)
  total_count &lt;- sum(counts)
  samples &lt;- sample(1:total_count,n,replace=F)
  samples &lt;- samples[order(samples)]
  result &lt;- array(0,num_species)
  total &lt;- 0
  for (i in 1:num_species) {
    result[i] &lt;- length(which(samples &gt; total &amp; samples &lt;= total+counts[i]))
    total &lt;- total+counts[i]
  }
  return(result)
  # return(apply(t(counts),1,sample_species,1500) )
}

sample_count_table &lt;- function(counts, n){
  return(apply(t(counts),1,sample_species,n) )
}

A&lt;- sample_count_table(CombinedTable,2200000)</code></pre>
<p>This is still a work in progress… It seems that sampling without replcement from a large count table actually takes an unreasonably long time (and probably uses a lot of memory)… Sampling without replacement could be faster in theory, but I’m not sure if that is still reasonable. There are some third party packages out there that do this, maybe I will try them out. Though, I don’t see any that let you pick a number of reads to subsample as opposed to a proportion of reads to subsample (in other words, it wouldn’t be easy to normalize read depth across samples). Maybe it will actually just be easier to realign everything at various read depths and make new count table files for each read depth if I want to incorporate normalizing and adjusting read depths into this analysis.</p>
<p>Update: What I turned out doing is sampling from the original alignment bam file to varying depths, normalizing such that each sample has an equal number of aligned reads. Then count tables were generated (one table for chimp, one for human), and the block of code that makes roc curves and other plots for different numbers of samples was repeated below using those smaller count tables (but still using the full count tabe as to determine “true” responses)</p>
<pre class="r"><code>###25M reads for each sample

RocCurveDataToPlot &lt;- data.frame()
DEGeneCountToPlot &lt;- matrix(nrow=length(SampleSizes), ncol=length(FDRLevels))
rownames(DEGeneCountToPlot) &lt;- SampleSizes
EffectSizesToPlot &lt;- data.frame()

ReadDepthControlledChimpCountFile &lt;- &quot;../output/PowerAnalysisCountTable.Chimp.25000000.subread.txt.gz&quot;
ReadDepthControlledHumanCountFile &lt;- &quot;../output/PowerAnalysisCountTable.Human.25000000.subread.txt.gz&quot;

for (i in seq_along(SampleSizes)){
  paste0(&quot;processing &quot;, SampleSizes[i])
  Results &lt;- DE.Subsampled(ReadDepthControlledChimpCountFile,
              ReadDepthControlledHumanCountFile,
              SampleSizes[i], FC.NullInterval, drop, ChimpSampleDrop=ChimpSamplesToDrop, HumanSampleDrop = HumanSamplesToDrop)
  
  RocCurveData &lt;- as.data.frame(coords(roc(response=as.vector(abs(TrueResponse)), quiet=T, predictor=as.numeric(Results$p.value), plot=F), transpose=F))
  RocCurveData$samplesize &lt;- SampleSizes[i]
  RocCurveDataToPlot &lt;- rbind(RocCurveDataToPlot, RocCurveData)
  
  for (j in seq_along(FDRLevels)){
    
    SubSampledResponse &lt;- decideTests(Results, p.value=FDRLevels[j])
    
    DEGeneCountToPlot[i,j] &lt;- sum(table(SubSampledResponse)[c(&quot;-1&quot;,&quot;1&quot;)])
    
    if (length(table(TrueResponse==SubSampledResponse &amp; SubSampledResponse!=0)) &gt; 1){
      EffectSizes.df &lt;-data.frame(EffectSizes=abs(True.efit$coefficients[TrueResponse==SubSampledResponse &amp; SubSampledResponse!=0]), FDR=FDRLevels[j], SampleSize=SampleSizes[i])
      EffectSizesToPlot &lt;- rbind(EffectSizesToPlot, EffectSizes.df)
    }

    
  }
}

ggplot(RocCurveDataToPlot, aes(x=1-specificity, y=sensitivity, color=factor(samplesize))) +
  geom_line() +
  theme_bw()</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DEGeneCountToPlot.df&lt;-as.data.frame(DEGeneCountToPlot)
colnames(DEGeneCountToPlot.df) &lt;- FDRLevels
DEGeneCountToPlot.df$SampleSize &lt;- rownames(DEGeneCountToPlot.df)
DEGeneCountToPlot.df$SampleSize</code></pre>
<pre><code>[1] &quot;2&quot;  &quot;4&quot;  &quot;8&quot;  &quot;16&quot; &quot;24&quot; &quot;32&quot; &quot;38&quot;</code></pre>
<pre class="r"><code>DEGeneCountToPlot.df[is.na(DEGeneCountToPlot.df)] &lt;- 0
DEGeneCountToPlot.df %&gt;% melt() %&gt;%
  dplyr::rename(Number.DE.genes=value, FDR=variable) %&gt;%
ggplot(aes(x=as.numeric(SampleSize), y=Number.DE.genes, color=FDR)) +
  geom_line() +
  theme_bw()</code></pre>
<pre><code>Using SampleSize as id variables</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(EffectSizesToPlot, aes(x=factor(SampleSize), y=EffectSizes, color=factor(FDR))) +
  # geom_violin()
  geom_boxplot() +
  ylab(&quot;Effect Size of True positive DE genes\n|log2(FC)|&quot;) +
  # scale_y_continuous(limits=c(0,5))
  theme_bw()</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###5M reads for each sample

RocCurveDataToPlot &lt;- data.frame()
DEGeneCountToPlot &lt;- matrix(nrow=length(SampleSizes), ncol=length(FDRLevels))
rownames(DEGeneCountToPlot) &lt;- SampleSizes
EffectSizesToPlot &lt;- data.frame()

ReadDepthControlledChimpCountFile &lt;- &quot;../output/PowerAnalysisCountTable.Chimp.10000000.subread.txt.gz&quot;
ReadDepthControlledHumanCountFile &lt;- &quot;../output/PowerAnalysisCountTable.Human.10000000.subread.txt.gz&quot;

for (i in seq_along(SampleSizes)){
  paste0(&quot;processing &quot;, SampleSizes[i])
  Results &lt;- DE.Subsampled(ReadDepthControlledChimpCountFile,
              ReadDepthControlledHumanCountFile,
              SampleSizes[i], FC.NullInterval, drop, ChimpSampleDrop=ChimpSamplesToDrop, HumanSampleDrop = HumanSamplesToDrop)
  
  RocCurveData &lt;- as.data.frame(coords(roc(response=as.vector(abs(TrueResponse)), quiet=T, predictor=as.numeric(Results$p.value), plot=F), transpose=F))
  RocCurveData$samplesize &lt;- SampleSizes[i]
  RocCurveDataToPlot &lt;- rbind(RocCurveDataToPlot, RocCurveData)
  
  for (j in seq_along(FDRLevels)){
    
    SubSampledResponse &lt;- decideTests(Results, p.value=FDRLevels[j])
    
    DEGeneCountToPlot[i,j] &lt;- sum(table(SubSampledResponse)[c(&quot;-1&quot;,&quot;1&quot;)])
    
    if (length(table(TrueResponse==SubSampledResponse &amp; SubSampledResponse!=0)) &gt; 1){
      EffectSizes.df &lt;-data.frame(EffectSizes=abs(True.efit$coefficients[TrueResponse==SubSampledResponse &amp; SubSampledResponse!=0]), FDR=FDRLevels[j], SampleSize=SampleSizes[i])
      EffectSizesToPlot &lt;- rbind(EffectSizesToPlot, EffectSizes.df)
    }

    
  }
}

ggplot(RocCurveDataToPlot, aes(x=1-specificity, y=sensitivity, color=factor(samplesize))) +
  geom_line() +
  theme_bw()</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DEGeneCountToPlot.df&lt;-as.data.frame(DEGeneCountToPlot)
colnames(DEGeneCountToPlot.df) &lt;- FDRLevels
DEGeneCountToPlot.df$SampleSize &lt;- rownames(DEGeneCountToPlot.df)
DEGeneCountToPlot.df$SampleSize</code></pre>
<pre><code>[1] &quot;2&quot;  &quot;4&quot;  &quot;8&quot;  &quot;16&quot; &quot;24&quot; &quot;32&quot; &quot;38&quot;</code></pre>
<pre class="r"><code>DEGeneCountToPlot.df[is.na(DEGeneCountToPlot.df)] &lt;- 0
DEGeneCountToPlot.df %&gt;% melt() %&gt;%
  dplyr::rename(Number.DE.genes=value, FDR=variable) %&gt;%
ggplot(aes(x=as.numeric(SampleSize), y=Number.DE.genes, color=FDR)) +
  geom_line() +
  theme_bw()</code></pre>
<pre><code>Using SampleSize as id variables</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(EffectSizesToPlot, aes(x=factor(SampleSize), y=EffectSizes, color=factor(FDR))) +
  # geom_violin()
  geom_boxplot() +
  ylab(&quot;Effect Size of True positive DE genes\n|log2(FC)|&quot;) +
  # scale_y_continuous(limits=c(0,5))
  theme_bw()</code></pre>
<p><img src="figure/20190613_PowerAnalysis.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gridExtra_2.3   reshape2_1.4.3  qvalue_2.14.1   pROC_1.15.0    
 [5] gplots_3.0.1.1  corrplot_0.84   edgeR_3.24.3    limma_3.38.3   
 [9] knitr_1.23      forcats_0.4.0   stringr_1.4.0   dplyr_0.8.1    
[13] purrr_0.3.2     readr_1.3.1     tidyr_0.8.3     tibble_2.1.3   
[17] ggplot2_3.1.1   tidyverse_1.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1         locfit_1.5-9.1     lubridate_1.7.4   
 [4] lattice_0.20-38    gtools_3.8.1       assertthat_0.2.1  
 [7] rprojroot_1.3-2    digest_0.6.19      R6_2.4.0          
[10] cellranger_1.1.0   plyr_1.8.4         backports_1.1.4   
[13] evaluate_0.14      httr_1.4.0         highr_0.8         
[16] pillar_1.4.1       rlang_0.3.4        lazyeval_0.2.2    
[19] readxl_1.3.1       rstudioapi_0.10    gdata_2.18.0      
[22] whisker_0.3-2      rmarkdown_1.13     labeling_0.3      
[25] splines_3.5.1      munsell_0.5.0      broom_0.5.2       
[28] compiler_3.5.1     modelr_0.1.4       xfun_0.7          
[31] pkgconfig_2.0.2    htmltools_0.3.6    tidyselect_0.2.5  
[34] workflowr_1.4.0    crayon_1.3.4       withr_2.1.2       
[37] bitops_1.0-6       grid_3.5.1         nlme_3.1-140      
[40] jsonlite_1.6       gtable_0.3.0       git2r_0.25.2      
[43] magrittr_1.5       scales_1.0.0       KernSmooth_2.23-15
[46] cli_1.1.0          stringi_1.4.3      fs_1.3.1          
[49] xml2_1.2.0         generics_0.0.2     tools_3.5.1       
[52] glue_1.3.1         hms_0.4.2          yaml_2.2.0        
[55] colorspace_1.4-1   caTools_1.17.1.2   rvest_0.3.4       
[58] haven_2.1.0       </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
