<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>20190521_eQTL_enrichmentCrossSpecies</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20190521_eQTL_enrichmentCrossSpecies</h1>
<h4 class="author">Ben Fair</h4>
<h4 class="date">5/21/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-12
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
GO-2,
</li>
<li>
GO-3
</li>
<li>
GO-analysis
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>20190521_eQTL_CrossSpeciesEnrichment_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7targetblank7ef6fe2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7" target="_blank">7ef6fe2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7targetblank7ef6fe2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/20190521_eQTL_CrossSpeciesEnrichment_cache/
    Ignored:    analysis/figure/
    Ignored:    analysis_temp/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/snakemake_workflow/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    docs/assets/.DS_Store

Untracked files:
    Untracked:  analysis/20190606_eGene_Conservation_TopN.Rmd

Unstaged changes:
    Modified:   analysis/20190521_eQTL_CrossSpeciesEnrichment.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/analysis/20190521_eQTL_CrossSpeciesEnrichment.Rmd" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/20190521_eQTL_CrossSpeciesEnrichment.html" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
<td>
update site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(plyr)
library(tidyverse)
library(knitr)
library(data.table)
library(ggpmisc)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(gridExtra)
library(enrichplot)</code></pre>
<p>Here the dataset is eQTLs called from a model with the following pre-testing filters/transformations/checks:</p>
<ul>
<li>cis-window=250kB</li>
<li>MAF&gt;10%</li>
<li>genes tested must have &gt;6 reads in 80% of samples</li>
<li>lmm with genetic relatedness matrix produced by gemma</li>
<li>Gene expression is standardized and normalized</li>
<li>10PCs added as covariates</li>
<li>FDR estimated by Storey’s qvalue</li>
<li>Pvalues well calibrated under a permutated null</li>
</ul>
<p>First, Read in the data…</p>
<pre class="r"><code>eQTLs &lt;- read.table(gzfile(&quot;../data/PastAnalysesDataToKeep/20190521_eQTLs_250kB_10MAF.txt.gz&quot;), header=T)
kable(head(eQTLs))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">snps</th>
<th align="left">gene</th>
<th align="right">beta</th>
<th align="right">statistic</th>
<th align="right">pvalue</th>
<th align="right">FDR</th>
<th align="right">qvalue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ID.1.126459696.ACCCTAGTAAG.A</td>
<td align="left">ENSPTRG00000001061</td>
<td align="right">3.570039</td>
<td align="right">12.50828</td>
<td align="right">0</td>
<td align="right">3.5e-06</td>
<td align="right">3.5e-06</td>
</tr>
<tr class="even">
<td align="left">ID.1.126465687.TTGT.A</td>
<td align="left">ENSPTRG00000001061</td>
<td align="right">3.570039</td>
<td align="right">12.50828</td>
<td align="right">0</td>
<td align="right">3.5e-06</td>
<td align="right">3.5e-06</td>
</tr>
<tr class="odd">
<td align="left">ID.1.126465750.TG.CT</td>
<td align="left">ENSPTRG00000001061</td>
<td align="right">3.570039</td>
<td align="right">12.50828</td>
<td align="right">0</td>
<td align="right">3.5e-06</td>
<td align="right">3.5e-06</td>
</tr>
<tr class="even">
<td align="left">ID.1.126465756.T.C</td>
<td align="left">ENSPTRG00000001061</td>
<td align="right">3.570039</td>
<td align="right">12.50828</td>
<td align="right">0</td>
<td align="right">3.5e-06</td>
<td align="right">3.5e-06</td>
</tr>
<tr class="odd">
<td align="left">ID.1.126465766.C.A</td>
<td align="left">ENSPTRG00000001061</td>
<td align="right">3.570039</td>
<td align="right">12.50828</td>
<td align="right">0</td>
<td align="right">3.5e-06</td>
<td align="right">3.5e-06</td>
</tr>
<tr class="even">
<td align="left">ID.1.126465774.G.A</td>
<td align="left">ENSPTRG00000001061</td>
<td align="right">3.570039</td>
<td align="right">12.50828</td>
<td align="right">0</td>
<td align="right">3.5e-06</td>
<td align="right">3.5e-06</td>
</tr>
</tbody>
</table>
<pre class="r"><code># List of chimp tested genes
ChimpTestedGenes &lt;- rownames(read.table(&#39;../output/ExpressionMatrix.un-normalized.txt.gz&#39;, header=T, check.names=FALSE, row.names = 1))

ChimpToHumanGeneMap &lt;- read.table(&quot;../data/Biomart_export.Hsap.Ptro.orthologs.txt.gz&quot;, header=T, sep=&#39;\t&#39;, stringsAsFactors = F)
kable(head(ChimpToHumanGeneMap))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Gene.stable.ID</th>
<th align="left">Transcript.stable.ID</th>
<th align="left">Chimpanzee.gene.stable.ID</th>
<th align="left">Chimpanzee.gene.name</th>
<th align="left">Chimpanzee.protein.or.transcript.stable.ID</th>
<th align="left">Chimpanzee.homology.type</th>
<th align="right">X.id..target.Chimpanzee.gene.identical.to.query.gene</th>
<th align="right">X.id..query.gene.identical.to.target.Chimpanzee.gene</th>
<th align="right">dN.with.Chimpanzee</th>
<th align="right">dS.with.Chimpanzee</th>
<th align="right">Chimpanzee.orthology.confidence..0.low..1.high.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000198888</td>
<td align="left">ENST00000361390</td>
<td align="left">ENSPTRG00000042641</td>
<td align="left">MT-ND1</td>
<td align="left">ENSPTRP00000061407</td>
<td align="left">ortholog_one2one</td>
<td align="right">94.6541</td>
<td align="right">94.6541</td>
<td align="right">0.0267</td>
<td align="right">0.5455</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">ENSG00000198763</td>
<td align="left">ENST00000361453</td>
<td align="left">ENSPTRG00000042626</td>
<td align="left">MT-ND2</td>
<td align="left">ENSPTRP00000061406</td>
<td align="left">ortholog_one2one</td>
<td align="right">96.2536</td>
<td align="right">96.2536</td>
<td align="right">0.0185</td>
<td align="right">0.7225</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000210127</td>
<td align="left">ENST00000387392</td>
<td align="left">ENSPTRG00000042642</td>
<td align="left">MT-TA</td>
<td align="left">ENSPTRT00000076396</td>
<td align="left">ortholog_one2one</td>
<td align="right">100.0000</td>
<td align="right">100.0000</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000198804</td>
<td align="left">ENST00000361624</td>
<td align="left">ENSPTRG00000042657</td>
<td align="left">MT-CO1</td>
<td align="left">ENSPTRP00000061408</td>
<td align="left">ortholog_one2one</td>
<td align="right">98.8304</td>
<td align="right">98.8304</td>
<td align="right">0.0065</td>
<td align="right">0.5486</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198712</td>
<td align="left">ENST00000361739</td>
<td align="left">ENSPTRG00000042660</td>
<td align="left">MT-CO2</td>
<td align="left">ENSPTRP00000061402</td>
<td align="left">ortholog_one2one</td>
<td align="right">97.7974</td>
<td align="right">97.7974</td>
<td align="right">0.0106</td>
<td align="right">0.5943</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">ENSG00000228253</td>
<td align="left">ENST00000361851</td>
<td align="left">ENSPTRG00000042653</td>
<td align="left">MT-ATP8</td>
<td align="left">ENSPTRP00000061400</td>
<td align="left">ortholog_one2one</td>
<td align="right">94.1176</td>
<td align="right">94.1176</td>
<td align="right">0.0325</td>
<td align="right">0.3331</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Of this ortholog list, how many genes are one2one
table(ChimpToHumanGeneMap$Chimpanzee.homology.type)</code></pre>
<pre><code>
ortholog_many2many  ortholog_one2many   ortholog_one2one 
              2278              19917             140351 </code></pre>
<pre class="r"><code>OneToOneMap &lt;- ChimpToHumanGeneMap %&gt;%
  filter(Chimpanzee.homology.type==&quot;ortholog_one2one&quot;)

# Read gtex heart egene list
# Only consider those that were tested in both species and are one2one orthologs
GtexHeartEgenes &lt;- read.table(&quot;../data/Heart_Left_Ventricle.v7.egenes.txt.gz&quot;, header=T, sep=&#39;\t&#39;, stringsAsFactors = F) %&gt;%
  mutate(gene_id_stable = gsub(&quot;.\\d+$&quot;,&quot;&quot;,gene_id)) %&gt;%
  filter(gene_id_stable %in% OneToOneMap$Gene.stable.ID) %&gt;%
  mutate(chimp_id = plyr::mapvalues(gene_id_stable, OneToOneMap$Gene.stable.ID,  OneToOneMap$Chimpanzee.gene.stable.ID, warn_missing = F)) %&gt;%
  filter(chimp_id %in% ChimpTestedGenes)

ChimpToHuman.ID &lt;- function(Chimp.ID){
  #function to convert chimp ensembl to human ensembl gene ids
  return(
    plyr::mapvalues(Chimp.ID, OneToOneMap$Chimpanzee.gene.stable.ID, OneToOneMap$Gene.stable.ID, warn_missing = F)
  )}</code></pre>
<p>Now compare with GTEx by making 2x2 contigency table (eGene/not-eGene in Chimp/human). The odds ratio from this table is symetrical.</p>
<pre class="r"><code>HumanFDR &lt;- 0.1
ChimpFDR &lt;- 0.1

#Get chimp eQTLs
Chimp_eQTLs &lt;- eQTLs %&gt;%
  filter(qvalue&lt;ChimpFDR)

# Count chimp eGenes
length(unique(Chimp_eQTLs$gene))</code></pre>
<pre><code>[1] 336</code></pre>
<pre class="r"><code># Count human eGenes
length(GtexHeartEgenes %&gt;% filter(qval&lt; HumanFDR) %&gt;% pull(chimp_id))</code></pre>
<pre><code>[1] 5410</code></pre>
<pre class="r"><code># Count number genes tested in both species (already filtered for 1to1 orthologs)
length(GtexHeartEgenes$gene_id_stable)</code></pre>
<pre><code>[1] 11586</code></pre>
<p>The number of human eGenes is huge (about half of all tested genes) and GTEx over-powered compared to chimp. With huge power, everything is an eGene and the eGene classification becomes devoid of meaningful information. So I will play with different ways to classify human eGenes.</p>
<pre class="r"><code>#Change FDR thresholds or take top N eGenes by qvalue
HumanTopN &lt;- 600
HumanFDR &lt;- 0.1
ChimpFDR &lt;- 0.1

# Filter human eGenes by qval threshold
HumanSigGenes &lt;- GtexHeartEgenes %&gt;% filter(qval&lt;HumanFDR) %&gt;% pull(chimp_id)

# Filter human eGenes by topN qval
# HumanSigGenes &lt;- GtexHeartEgenes %&gt;% top_n(-HumanTopN, qval) %&gt;% pull(chimp_id)

# Filter human eGeness by qval threshold then topN betas 
# HumanSigGenes &lt;- GtexHeartEgenes %&gt;% filter(qval&lt;HumanFDR) %&gt;% top_n(1000, abs(slope)) %&gt;% pull(chimp_id)

HumanNonSigGenes &lt;- GtexHeartEgenes %&gt;%
  filter(!chimp_id %in% HumanSigGenes) %&gt;%
  pull(chimp_id)

ChimpSigGenes &lt;- GtexHeartEgenes %&gt;%
  filter(chimp_id %in% Chimp_eQTLs$gene) %&gt;%
  pull(chimp_id)

ChimpNonSigGenes &lt;- GtexHeartEgenes %&gt;%
  filter(! chimp_id %in% Chimp_eQTLs$gene) %&gt;%
  pull(chimp_id)



ContigencyTable &lt;- matrix( c( length(intersect(ChimpSigGenes,HumanSigGenes)),
                              length(intersect(HumanSigGenes,ChimpNonSigGenes)),
                              length(intersect(ChimpSigGenes,HumanNonSigGenes)),
                              length(intersect(ChimpNonSigGenes,HumanNonSigGenes))), 
                           nrow = 2)

rownames(ContigencyTable) &lt;- c(&quot;Chimp eGene&quot;, &quot;Not Chimp eGene&quot;)
colnames(ContigencyTable) &lt;- c(&quot;Human eGene&quot;, &quot;Not human eGene&quot;)

#what is qval threshold for human eGene classification in this contigency table
print(GtexHeartEgenes %&gt;% top_n(-HumanTopN, qval) %&gt;% top_n(1, qval) %&gt;% pull(qval))</code></pre>
<pre><code>[1] 5.83223e-12</code></pre>
<pre class="r"><code>#Contigency table of one to one orthologs tested in both chimps and humans of whether significant in humans, or chimps, or both, or neither
ContigencyTable</code></pre>
<pre><code>                Human eGene Not human eGene
Chimp eGene             145             135
Not Chimp eGene        5265            6041</code></pre>
<pre class="r"><code>#One-sided Fisher test for greater overlap than expected by chance
fisher.test(ContigencyTable, alternative=&quot;greater&quot;)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  ContigencyTable
p-value = 0.04779
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 1.002643      Inf
sample estimates:
odds ratio 
  1.232354 </code></pre>
<p>The above contingency table and one-sided fisher test indicated a greater-than-chance overlap between the sets of eGenes in chimp and human.</p>
<p>A few plots might give me a better intuition for thinking about the data…</p>
<pre class="r"><code># Plot betas vs qval for GTEx for qval&lt;0.1
GtexHeartEgenes %&gt;% filter(qval&lt;0.1) %&gt;%
ggplot(aes(x=slope, y=-log10(qval))) +
         geom_point(alpha=0.05)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/random-plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-random-plots-1">
Past versions of random-plots-1.png
</button>
</p>
<div id="fig-random-plots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/random-plots-1.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>HumanAndChimpEffectSizes &lt;- Chimp_eQTLs %&gt;%
    group_by(gene) %&gt;%
    dplyr::slice(which.min(qvalue)) %&gt;% 
    ungroup() %&gt;%
    left_join((GtexHeartEgenes %&gt;% filter(qval&lt;0.1)), by=c(&quot;gene&quot;=&quot;chimp_id&quot;)) %&gt;%
    dplyr::select(gene, slope, beta)</code></pre>
<pre><code>Warning: Column `gene`/`chimp_id` joining factor and character vector,
coercing into character vector</code></pre>
<pre class="r"><code># For shared eGenes, are scatter plot of absolute value of betas
ggplot(HumanAndChimpEffectSizes, aes(x=log10(abs(slope)), y=log10(abs(beta)))) +
  geom_point() +
  geom_smooth(method = &quot;lm&quot;) +
  theme_bw() +
  xlim(c(-1,1)) +
  ylim(c(-1,1)) +
  xlab(&quot;Human |beta|&quot;) +
  ylab(&quot;Chimp |beta|&quot;)</code></pre>
<pre><code>Warning: Removed 191 rows containing non-finite values (stat_smooth).</code></pre>
<pre><code>Warning: Removed 191 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/random-plots-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-random-plots-2">
Past versions of random-plots-2.png
</button>
</p>
<div id="fig-random-plots-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/random-plots-2.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cor.test(HumanAndChimpEffectSizes$slope, HumanAndChimpEffectSizes$beta,method=&#39;spearman&#39;)</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  HumanAndChimpEffectSizes$slope and HumanAndChimpEffectSizes$beta
S = 476920, p-value = 0.4633
alternative hypothesis: true rho is not equal to 0
sample estimates:
       rho 
0.06132499 </code></pre>
<p>Effect sizes for cross-species eGenes appear uncorrelated. This might at first appear at odds with the greater-than-chance overlap for the same set of eGenes across sepecies. However, one explanation is that the above effect sizes analysis didn’t consider non eGenes for which the null of effect-size=0 could not be rejected.</p>
<p>OK, so now I will perform the same Odds ratio estimate and Fisher exact test for different thresholds/quantiles of the human data ordered by significance.</p>
<pre class="r"><code>ChimpFDR &lt;- 0.1
HumanFDR &lt;- 0.1
Chimp_eQTLs &lt;- eQTLs %&gt;%
  filter(qvalue&lt;ChimpFDR)

ChimpSigGenes &lt;- GtexHeartEgenes %&gt;%
  filter(chimp_id %in% Chimp_eQTLs$gene) %&gt;%
  pull(chimp_id)

ChimpNonSigGenes &lt;- GtexHeartEgenes %&gt;%
  filter(! chimp_id %in% Chimp_eQTLs$gene) %&gt;%
  pull(chimp_id)

LengthOut=200
TopHumanEgeneCountList &lt;- round(seq(100,5410, length.out=LengthOut))
OddsRatiosList &lt;- rep(0,LengthOut)
CI.lower &lt;- rep(0,LengthOut)
CI.upper &lt;- rep(0,LengthOut)

for (i in seq_along(TopHumanEgeneCountList)){
  HumanTopN &lt;- TopHumanEgeneCountList[i]


# Filter human eGenes by qval threshold
# HumanSigGenes &lt;- GtexHeartEgenes %&gt;% filter(qval&lt;HumanFDR) %&gt;% pull(chimp_id)

# Filter human eGenes by topN qval
HumanSigGenes &lt;- GtexHeartEgenes %&gt;% top_n(-HumanTopN, qval) %&gt;% pull(chimp_id)

# Filter human eGeness by qval threshold then topN betas 
# HumanSigGenes &lt;- GtexHeartEgenes %&gt;% filter(qval&lt;HumanFDR) %&gt;% top_n(1000, abs(slope)) %&gt;% pull(chimp_id)

HumanNonSigGenes &lt;- GtexHeartEgenes %&gt;%
  filter(!chimp_id %in% HumanSigGenes) %&gt;%
  pull(chimp_id)

ContigencyTable &lt;- matrix( c( length(intersect(ChimpSigGenes,HumanSigGenes)),
                              length(intersect(HumanSigGenes,ChimpNonSigGenes)),
                              length(intersect(ChimpSigGenes,HumanNonSigGenes)),
                              length(intersect(ChimpNonSigGenes,HumanNonSigGenes))), 
                           nrow = 2)

#Contigency table of one to one orthologs tested in both chimps and humans of whether significant in humans, or chimps, or both, or neither
ContigencyTable
A&lt;-fisher.test(ContigencyTable, conf.int=T)
OddsRatiosList[i]&lt;- as.numeric(A$estimate)
CI.lower[i] &lt;- as.numeric(A$conf.int[1])
CI.upper[i] &lt;- as.numeric(A$conf.int[2])
}
HeartOddsRatios &lt;- data.frame(TopHumanEgeneCountList=TopHumanEgeneCountList, OddsRatio=OddsRatiosList, GTEx.Tissue=&quot;Heart_LeftVentricle&quot;, CI.lower=CI.lower, CI.upper=CI.upper)

ggplot(HeartOddsRatios, aes(x=TopHumanEgeneCountList, y=OddsRatiosList)) +
  geom_point() +
  geom_ribbon(aes(ymin=CI.lower, ymax=CI.upper), alpha=0.2) +
  theme_bw() +
  ylab(&quot;Odds Ratio&quot;) +
  xlab(&quot;Considering only top X human eGenes&quot;) +
  geom_hline(yintercept=1, linetype=&quot;dashed&quot;)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/CompareWithGtexEgenes_byquantiles-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-CompareWithGtexEgenes_byquantiles-1">
Past versions of CompareWithGtexEgenes_byquantiles-1.png
</button>
</p>
<div id="fig-CompareWithGtexEgenes_byquantiles-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/CompareWithGtexEgenes_byquantiles-1.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now do the same for a couple of GTEx tissues</p>
<pre class="r"><code>GTEx.eGene.Files &lt;- c(&quot;../data/Heart_Left_Ventricle.v7.egenes.txt.gz&quot;,
                      &quot;../data/Adipose_Subcutaneous.v7.egenes.txt.gz&quot;,
                      # &quot;../data/Cells_EBV-transformed_lymphocytes.v7.egenes.txt.gz&quot;,
                      &quot;../data/Thyroid.v7.egenes.txt.gz&quot;,
                      &quot;../data/Lung.v7.egenes.txt.gz&quot;)

GTEx.tissue.labels &lt;- c(&quot;Heart_Left_Ventricle&quot;,
                        &quot;Adipose_Subcutaneous&quot;,
                        # &quot;LCL&quot;,
                        &quot;Thyroid&quot;,
                        &quot;Lung&quot;)

OddsRatios.rbinded &lt;- data.frame()
                      
for (j in seq_along(GTEx.eGene.Files)){
  paste0(&quot;processing&quot;)
  GtexEgenes &lt;- read.table(GTEx.eGene.Files[j], header=T, sep=&#39;\t&#39;, stringsAsFactors = F) %&gt;%
  mutate(gene_id_stable = gsub(&quot;.\\d+$&quot;,&quot;&quot;,gene_id)) %&gt;%
  filter(gene_id_stable %in% OneToOneMap$Gene.stable.ID) %&gt;%
  mutate(chimp_id = plyr::mapvalues(gene_id_stable, OneToOneMap$Gene.stable.ID,  OneToOneMap$Chimpanzee.gene.stable.ID, warn_missing = F)) %&gt;%
  filter(chimp_id %in% ChimpTestedGenes)

LengthOut=200
TopHumanEgeneCountList &lt;- round(seq(100,5410, length.out=LengthOut))
OddsRatiosList &lt;- rep(0,LengthOut)
CI.lower &lt;- rep(0,LengthOut)
CI.upper &lt;- rep(0,LengthOut)

for (i in seq_along(TopHumanEgeneCountList)){
  HumanTopN &lt;- TopHumanEgeneCountList[i]


  # Filter human eGenes by qval threshold
  # HumanSigGenes &lt;- GtexHeartEgenes %&gt;% filter(qval&lt;HumanFDR) %&gt;% pull(chimp_id)
  
  # Filter human eGenes by topN qval
  HumanSigGenes &lt;- GtexEgenes %&gt;% top_n(-HumanTopN, qval) %&gt;% pull(chimp_id)
  
  HumanNonSigGenes &lt;- GtexEgenes %&gt;%
    filter(!chimp_id %in% HumanSigGenes) %&gt;%
    pull(chimp_id)
  
  ContigencyTable &lt;- matrix( c( length(intersect(ChimpSigGenes,HumanSigGenes)),
                                length(intersect(HumanSigGenes,ChimpNonSigGenes)),
                                length(intersect(ChimpSigGenes,HumanNonSigGenes)),
                                length(intersect(ChimpNonSigGenes,HumanNonSigGenes))), 
                             nrow = 2)
  
  #Contigency table of one to one orthologs tested in both chimps and humans of whether significant in humans, or chimps, or both, or neither
  A&lt;-fisher.test(ContigencyTable, conf.int=T)
  OddsRatiosList[i]&lt;- as.numeric(A$estimate)
  CI.lower[i] &lt;- as.numeric(A$conf.int[1])
  CI.upper[i] &lt;- as.numeric(A$conf.int[2])
}

OddsRatios &lt;- data.frame(HumanEgeneThreshold=TopHumanEgeneCountList, OR=OddsRatiosList, GTEx.Tissue=GTEx.tissue.labels[j], CI.lower=CI.lower, CI.upper=CI.upper)

OddsRatios.rbinded &lt;- rbind(OddsRatios.rbinded, OddsRatios)
  
}

ggplot(OddsRatios.rbinded, aes(x=HumanEgeneThreshold, y=OR, color=GTEx.Tissue)) +
  geom_line(size=2) +
  geom_ribbon(aes(ymin=CI.lower, ymax=CI.upper, fill = GTEx.Tissue), alpha=0.1,linetype = 3) +
  theme_bw() +
  ylab(&quot;Odds Ratio&quot;) +
  xlab(&quot;Considering only top X human eGenes&quot;) +
  geom_hline(yintercept=1, linetype=&quot;dashed&quot;) +
  ylim(c(0,4)) +
  theme(legend.position=&quot;top&quot;)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/CompareWithGtexEgenes_OtherTissue-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-CompareWithGtexEgenes_OtherTissue-1">
Past versions of CompareWithGtexEgenes_OtherTissue-1.png
</button>
</p>
<div id="fig-CompareWithGtexEgenes_OtherTissue-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/CompareWithGtexEgenes_OtherTissue-1.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Try one alternative way to present these odds ratios as a function of eGene stringency:</p>
<pre class="r"><code># Get qval ordered ordered list of Chimp genes (that were tested in both)
Chimp_OrderedGenes &lt;- eQTLs %&gt;%
    group_by(gene) %&gt;% 
    dplyr::slice(which.min(qvalue)) %&gt;%
    filter(gene %in% GtexHeartEgenes$chimp_id) %&gt;%
    arrange(qvalue) %&gt;%
    as.data.frame()

# Get qval ordered list of human genes 
Human_OrderedGenes &lt;- GtexHeartEgenes %&gt;%
  arrange(qval)


LengthOut=60

ChimpEgeneCount &lt;- Chimp_OrderedGenes %&gt;% filter(qvalue&lt;0.2) %&gt;% pull(qvalue) %&gt;% length()
HumanEgeneCount &lt;- Human_OrderedGenes %&gt;% filter(qval&lt;0.2) %&gt;% pull(qval) %&gt;% length()
TopHumanEgeneCountList &lt;- round(seq(1,HumanEgeneCount, by=50))[-1]
TopChimpEgeneCountList &lt;- round(seq(1,ChimpEgeneCount, by=25))[-1]

OddsRatioMatrix &lt;- matrix(nrow = length(TopHumanEgeneCountList), ncol = length(TopChimpEgeneCountList))

for (i in seq_along(TopHumanEgeneCountList)){
  HumanTopN &lt;- TopHumanEgeneCountList[i]

# Filter human eGenes by topN qval
HumanSigGenes &lt;- Human_OrderedGenes %&gt;% top_n(-HumanTopN, qval) %&gt;% pull(chimp_id)

HumanNonSigGenes &lt;- Human_OrderedGenes %&gt;%
  filter(!chimp_id %in% HumanSigGenes) %&gt;%
  pull(chimp_id)

for (j in seq_along(TopChimpEgeneCountList)){
  TopChimpN &lt;- TopChimpEgeneCountList[j]
  
  ChimpSigGenes &lt;- Chimp_OrderedGenes %&gt;% top_n(-TopChimpN, qvalue) %&gt;% pull(gene)
  ChimpNonSigGenes &lt;- GtexHeartEgenes %&gt;%
    filter(!chimp_id %in% ChimpSigGenes) %&gt;%
    pull(chimp_id)
  
  ContigencyTable &lt;- matrix( c( length(intersect(ChimpSigGenes,HumanSigGenes)),
                              length(intersect(HumanSigGenes,ChimpNonSigGenes)),
                              length(intersect(ChimpSigGenes,HumanNonSigGenes)),
                              length(intersect(ChimpNonSigGenes,HumanNonSigGenes))), 
                           nrow = 2)
  A&lt;-fisher.test(ContigencyTable, conf.int=F)
  OddsRatioMatrix[i,j] &lt;- (as.numeric(A$estimate))
}
}

row.names(OddsRatioMatrix) &lt;- TopHumanEgeneCountList
colnames(OddsRatioMatrix) &lt;- TopChimpEgeneCountList

melt(OddsRatioMatrix) %&gt;%
  dplyr::rename(OddsRatio=value) %&gt;%
ggplot(aes(x=as.numeric(Var1),y=as.numeric(Var2), fill=OddsRatio)) +
  geom_raster(interpolate=F) +
  xlab(&quot;Top N human eGenes&quot;) +
  ylab(&quot;Top N chimp eGenes&quot;) +
  scale_y_continuous(expand = c(0,0)) +
  scale_x_continuous(expand = c(0,0)) +
  scale_fill_distiller(palette = &quot;Spectral&quot;) +
  # coord_fixed(ratio = 1, xlim = NULL, ylim = NULL, expand = F, clip = &quot;on&quot;) +
  
  geom_vline(xintercept=7555, colour=&quot;grey&quot;) +
  geom_text(aes(x=7555, label=&quot;FDR=0.2&quot;, y=400), angle=90, vjust = 0, color=&quot;grey&quot;, size=4)+
  geom_vline(xintercept=5410, colour=&quot;grey&quot;) +
  geom_text(aes(x=5410, label=&quot;FDR=0.1&quot;, y=400), angle=90, vjust = 0, color=&quot;grey&quot;, size=4)+
  geom_vline(xintercept=4240, colour=&quot;grey&quot;) +
  geom_text(aes(x=4240, label=&quot;FDR=0.05&quot;, y=400), angle=90, vjust = 0, color=&quot;grey&quot;, size=4)+
  
    geom_hline(yintercept=731, colour=&quot;grey&quot;) +
  geom_text(aes(x=2000, label=&quot;FDR=0.2&quot;, y=731-50), vjust = 0, color=&quot;grey&quot;, size=4)+
  geom_hline(yintercept=280, colour=&quot;grey&quot;) +
  geom_text(aes(x=2000, label=&quot;FDR=0.1&quot;, y=280-50), vjust = 0, color=&quot;grey&quot;, size=4)+
  geom_hline(yintercept=98, colour=&quot;grey&quot;) +
  geom_text(aes(x=2000, label=&quot;FDR=0.05&quot;, y=98-50), vjust = 0, color=&quot;grey&quot;, size=4)+
  theme_bw() +
  theme(legend.position=&quot;bottom&quot;)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/unnamed-chunk-1-1.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Gene ontology overlap analysis of genes that are Chimp eGenes but not human eGenes. The interpretation is that this is a somewhat prioritized list of candidate genes under stabilizing selection in the human lineage.</p>
<pre class="r"><code>#Change FDR thresholds or take top N eGenes by qvalue
HumanFDR &lt;- 0.1
ChimpFDR &lt;- 0.1
HumanTopN &lt;- 600

Chimp_eQTLs &lt;- eQTLs %&gt;%
  filter(qvalue&lt;ChimpFDR)

# Filter human eGenes by qval threshold
HumanSigGenes &lt;- GtexHeartEgenes %&gt;% filter(qval&lt;HumanFDR) %&gt;% pull(chimp_id)
# HumanSigGenes &lt;- GtexHeartEgenes %&gt;% top_n(-HumanTopN, qval) %&gt;% pull(chimp_id)


HumanNonSigGenes &lt;- GtexHeartEgenes %&gt;%
  filter(!chimp_id %in% HumanSigGenes) %&gt;%
  pull(chimp_id)

ChimpSigGenes &lt;- GtexHeartEgenes %&gt;%
  filter(chimp_id %in% Chimp_eQTLs$gene) %&gt;%
  pull(chimp_id)

ChimpNonSigGenes &lt;- GtexHeartEgenes %&gt;%
  filter(! chimp_id %in% Chimp_eQTLs$gene) %&gt;%
  pull(chimp_id)



ContigencyTable &lt;- matrix( c( length(intersect(ChimpSigGenes,HumanSigGenes)),
                              length(intersect(HumanSigGenes,ChimpNonSigGenes)),
                              length(intersect(ChimpSigGenes,HumanNonSigGenes)),
                              length(intersect(ChimpNonSigGenes,HumanNonSigGenes))), 
                           nrow = 2)

rownames(ContigencyTable) &lt;- c(&quot;Chimp eGene&quot;, &quot;Not Chimp eGene&quot;)
colnames(ContigencyTable) &lt;- c(&quot;Human eGene&quot;, &quot;Not human eGene&quot;)


#Contigency table of one to one orthologs tested in both chimps and humans of whether significant in humans, or chimps, or both, or neither
ContigencyTable</code></pre>
<pre><code>                Human eGene Not human eGene
Chimp eGene             145             135
Not Chimp eGene        5265            6041</code></pre>
<p>Now that we have a contingency table of eGenes between chimp and human, lets do gene set enrichment analysis. This analysis uses <code>clusterProfiler</code> package to look for gene set enrichment in a foreground set of genes over a background set. For each foreground/background comparison, there will be a gene ontology enrichment for the GO subcategories of - biological process (bp) - cellular component (cc) - molecular function (mf)</p>
<p>Perhaps first most intuitive comparison to make is:</p>
<p>Foreground = Chimp eGene &amp; not human eGene</p>
<p>background = All genes tested in both species</p>
<pre class="r"><code>foreground = ChimpToHuman.ID(intersect(ChimpSigGenes,HumanNonSigGenes))
background = ChimpToHuman.ID(GtexHeartEgenes$chimp_id)
length(foreground)</code></pre>
<pre><code>[1] 135</code></pre>
<pre class="r"><code>length(background)</code></pre>
<pre><code>[1] 11586</code></pre>
<pre class="r"><code>ego.mf &lt;- enrichGO(gene = foreground,
                 universe = background,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 ont = &quot;MF&quot;, 
                 pAdjustMethod = &quot;BH&quot;, 
                 pvalueCutoff  = 0.01,
                 qvalueCutoff  = 0.05)
kable(head(as.data.frame(ego.mf)))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="left">geneID</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<pre class="r"><code>ego.bp &lt;- enrichGO(gene = foreground,
                 universe = background,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 ont = &quot;BP&quot;, 
                 pAdjustMethod = &quot;BH&quot;, 
                 pvalueCutoff  = 0.01,
                 qvalueCutoff  = 0.05)
kable(head(as.data.frame(ego.bp)))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="left">geneID</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<pre class="r"><code>ego.cc &lt;- enrichGO(gene = foreground,
                 universe = background,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 ont = &quot;CC&quot;, 
                 pAdjustMethod = &quot;BH&quot;, 
                 pvalueCutoff  = 0.01,
                 qvalueCutoff  = 0.05)
kable(head(as.data.frame(ego.cc)))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="left">geneID</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Nothing enriched by this comparison of gene sets. Perhaps a more appropriate foreground/background set is:</p>
<p>foreground = chimp specific eGenes</p>
<p>background = all chimp eGenes</p>
<pre class="r"><code>foreground = ChimpToHuman.ID(intersect(ChimpSigGenes,HumanNonSigGenes))
background = ChimpToHuman.ID(ChimpSigGenes)
length(foreground)</code></pre>
<pre><code>[1] 135</code></pre>
<pre class="r"><code>length(background)</code></pre>
<pre><code>[1] 280</code></pre>
<pre class="r"><code>ego.mf &lt;- enrichGO(gene = foreground,
                 universe = background,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 ont = &quot;MF&quot;, 
                 pAdjustMethod = &quot;BH&quot;, 
                 pvalueCutoff  = 0.01,
                 qvalueCutoff  = 0.05)
kable(head(as.data.frame(ego.mf)))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="left">geneID</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="GO:0003676" class="uri">GO:0003676</a></td>
<td align="left"><a href="GO:0003676" class="uri">GO:0003676</a></td>
<td align="left">nucleic acid binding</td>
<td align="left">40/127</td>
<td align="left">54/256</td>
<td align="right">3.91e-05</td>
<td align="right">0.0026993</td>
<td align="right">0.0025943</td>
<td align="left">ENSG00000159208/ENSG00000158796/ENSG00000153187/ENSG00000184898/ENSG00000163602/ENSG00000114127/ENSG00000072274/ENSG00000114503/ENSG00000151623/ENSG00000164342/ENSG00000124279/ENSG00000113643/ENSG00000037241/ENSG00000106261/ENSG00000165061/ENSG00000185122/ENSG00000196150/ENSG00000147854/ENSG00000107371/ENSG00000125484/ENSG00000150593/ENSG00000211450/ENSG00000170325/ENSG00000134283/ENSG00000075856/ENSG00000139718/ENSG00000061936/ENSG00000173575/ENSG00000178952/ENSG00000102974/ENSG00000102984/ENSG00000167513/ENSG00000125457/ENSG00000185504/ENSG00000141646/ENSG00000181007/ENSG00000128000/ENSG00000090971/ENSG00000204519/ENSG00000152454</td>
<td align="right">40</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ego.bp &lt;- enrichGO(gene = foreground,
                 universe = background,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 ont = &quot;BP&quot;, 
                 pAdjustMethod = &quot;BH&quot;, 
                 pvalueCutoff  = 0.01,
                 qvalueCutoff  = 0.05)
kable(head(as.data.frame(ego.bp)))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="left">geneID</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="GO:0051252" class="uri">GO:0051252</a></td>
<td align="left"><a href="GO:0051252" class="uri">GO:0051252</a></td>
<td align="left">regulation of RNA metabolic process</td>
<td align="left">39/122</td>
<td align="left">50/255</td>
<td align="right">1.4e-06</td>
<td align="right">0.0005097</td>
<td align="right">0.0003559</td>
<td align="left">ENSG00000159208/ENSG00000159388/ENSG00000153187/ENSG00000123609/ENSG00000163602/ENSG00000114127/ENSG00000114503/ENSG00000157404/ENSG00000118785/ENSG00000168743/ENSG00000151623/ENSG00000151718/ENSG00000164342/ENSG00000124279/ENSG00000152409/ENSG00000078401/ENSG00000106261/ENSG00000161057/ENSG00000164758/ENSG00000185122/ENSG00000196150/ENSG00000107371/ENSG00000171862/ENSG00000150593/ENSG00000170325/ENSG00000134283/ENSG00000094916/ENSG00000075856/ENSG00000061936/ENSG00000173575/ENSG00000102974/ENSG00000102984/ENSG00000167513/ENSG00000141646/ENSG00000181007/ENSG00000128000/ENSG00000090971/ENSG00000204519/ENSG00000152454</td>
<td align="right">39</td>
</tr>
<tr class="even">
<td><a href="GO:0019219" class="uri">GO:0019219</a></td>
<td align="left"><a href="GO:0019219" class="uri">GO:0019219</a></td>
<td align="left">regulation of nucleobase-containing compound metabolic process</td>
<td align="left">41/122</td>
<td align="left">54/255</td>
<td align="right">2.5e-06</td>
<td align="right">0.0005097</td>
<td align="right">0.0003559</td>
<td align="left">ENSG00000159208/ENSG00000159388/ENSG00000153187/ENSG00000123609/ENSG00000163602/ENSG00000114127/ENSG00000072274/ENSG00000114503/ENSG00000157404/ENSG00000118785/ENSG00000168743/ENSG00000151623/ENSG00000151718/ENSG00000164342/ENSG00000124279/ENSG00000152409/ENSG00000078401/ENSG00000106261/ENSG00000161057/ENSG00000164758/ENSG00000185122/ENSG00000196150/ENSG00000107371/ENSG00000170312/ENSG00000171862/ENSG00000150593/ENSG00000170325/ENSG00000134283/ENSG00000094916/ENSG00000075856/ENSG00000061936/ENSG00000173575/ENSG00000102974/ENSG00000102984/ENSG00000167513/ENSG00000141646/ENSG00000181007/ENSG00000128000/ENSG00000090971/ENSG00000204519/ENSG00000152454</td>
<td align="right">41</td>
</tr>
<tr class="odd">
<td><a href="GO:0051171" class="uri">GO:0051171</a></td>
<td align="left"><a href="GO:0051171" class="uri">GO:0051171</a></td>
<td align="left">regulation of nitrogen compound metabolic process</td>
<td align="left">57/122</td>
<td align="left">83/255</td>
<td align="right">3.0e-06</td>
<td align="right">0.0005097</td>
<td align="right">0.0003559</td>
<td align="left">ENSG00000121769/ENSG00000159208/ENSG00000159388/ENSG00000153187/ENSG00000172985/ENSG00000163083/ENSG00000123609/ENSG00000163359/ENSG00000163376/ENSG00000163602/ENSG00000114127/ENSG00000072274/ENSG00000114503/ENSG00000163975/ENSG00000157404/ENSG00000118785/ENSG00000168743/ENSG00000151623/ENSG00000151718/ENSG00000164342/ENSG00000124279/ENSG00000152409/ENSG00000055163/ENSG00000078401/ENSG00000106261/ENSG00000161057/ENSG00000164758/ENSG00000185122/ENSG00000196150/ENSG00000107140/ENSG00000107371/ENSG00000170312/ENSG00000171862/ENSG00000138166/ENSG00000150593/ENSG00000154127/ENSG00000170325/ENSG00000134283/ENSG00000094916/ENSG00000075856/ENSG00000061936/ENSG00000133116/ENSG00000173575/ENSG00000150281/ENSG00000102974/ENSG00000102984/ENSG00000167513/ENSG00000109062/ENSG00000125457/ENSG00000141646/ENSG00000181007/ENSG00000128000/ENSG00000090971/ENSG00000204519/ENSG00000152454/ENSG00000100985/ENSG00000099958</td>
<td align="right">57</td>
</tr>
<tr class="even">
<td><a href="GO:0010556" class="uri">GO:0010556</a></td>
<td align="left"><a href="GO:0010556" class="uri">GO:0010556</a></td>
<td align="left">regulation of macromolecule biosynthetic process</td>
<td align="left">41/122</td>
<td align="left">55/255</td>
<td align="right">6.0e-06</td>
<td align="right">0.0006226</td>
<td align="right">0.0004347</td>
<td align="left">ENSG00000159208/ENSG00000159388/ENSG00000153187/ENSG00000163083/ENSG00000123609/ENSG00000163376/ENSG00000163602/ENSG00000114127/ENSG00000114503/ENSG00000157404/ENSG00000118785/ENSG00000168743/ENSG00000151623/ENSG00000151718/ENSG00000164342/ENSG00000124279/ENSG00000152409/ENSG00000164219/ENSG00000078401/ENSG00000106261/ENSG00000161057/ENSG00000164758/ENSG00000185122/ENSG00000196150/ENSG00000170312/ENSG00000171862/ENSG00000150593/ENSG00000170325/ENSG00000134283/ENSG00000094916/ENSG00000173575/ENSG00000102974/ENSG00000102984/ENSG00000167513/ENSG00000125457/ENSG00000141646/ENSG00000181007/ENSG00000128000/ENSG00000090971/ENSG00000204519/ENSG00000152454</td>
<td align="right">41</td>
</tr>
<tr class="odd">
<td><a href="GO:0009889" class="uri">GO:0009889</a></td>
<td align="left"><a href="GO:0009889" class="uri">GO:0009889</a></td>
<td align="left">regulation of biosynthetic process</td>
<td align="left">42/122</td>
<td align="left">57/255</td>
<td align="right">7.4e-06</td>
<td align="right">0.0006226</td>
<td align="right">0.0004347</td>
<td align="left">ENSG00000121769/ENSG00000159208/ENSG00000159388/ENSG00000153187/ENSG00000163083/ENSG00000123609/ENSG00000163376/ENSG00000163602/ENSG00000114127/ENSG00000114503/ENSG00000157404/ENSG00000118785/ENSG00000168743/ENSG00000151623/ENSG00000151718/ENSG00000164342/ENSG00000124279/ENSG00000152409/ENSG00000164219/ENSG00000078401/ENSG00000106261/ENSG00000161057/ENSG00000164758/ENSG00000185122/ENSG00000196150/ENSG00000170312/ENSG00000171862/ENSG00000150593/ENSG00000170325/ENSG00000134283/ENSG00000094916/ENSG00000173575/ENSG00000102974/ENSG00000102984/ENSG00000167513/ENSG00000125457/ENSG00000141646/ENSG00000181007/ENSG00000128000/ENSG00000090971/ENSG00000204519/ENSG00000152454</td>
<td align="right">42</td>
</tr>
<tr class="even">
<td><a href="GO:0010468" class="uri">GO:0010468</a></td>
<td align="left"><a href="GO:0010468" class="uri">GO:0010468</a></td>
<td align="left">regulation of gene expression</td>
<td align="left">42/122</td>
<td align="left">57/255</td>
<td align="right">7.4e-06</td>
<td align="right">0.0006226</td>
<td align="right">0.0004347</td>
<td align="left">ENSG00000159208/ENSG00000159388/ENSG00000153187/ENSG00000163376/ENSG00000163602/ENSG00000114127/ENSG00000114503/ENSG00000163975/ENSG00000157404/ENSG00000118785/ENSG00000168743/ENSG00000151623/ENSG00000151718/ENSG00000164342/ENSG00000124279/ENSG00000152409/ENSG00000078401/ENSG00000106261/ENSG00000161057/ENSG00000164758/ENSG00000185122/ENSG00000196150/ENSG00000107371/ENSG00000170312/ENSG00000171862/ENSG00000150593/ENSG00000170325/ENSG00000134283/ENSG00000094916/ENSG00000075856/ENSG00000061936/ENSG00000173575/ENSG00000102974/ENSG00000102984/ENSG00000167513/ENSG00000125457/ENSG00000141646/ENSG00000181007/ENSG00000128000/ENSG00000090971/ENSG00000204519/ENSG00000152454</td>
<td align="right">42</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ego.cc &lt;- enrichGO(gene = foreground,
                 universe = background,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 ont = &quot;CC&quot;, 
                 pAdjustMethod = &quot;BH&quot;, 
                 pvalueCutoff  = 0.01,
                 qvalueCutoff  = 0.05)
kable(head(as.data.frame(ego.cc)))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="left">geneID</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<pre class="r"><code>dotplot(ego.mf, font.size=8)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-2,%20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GO-2,--1">
Past versions of “GO-2, -1.png”
</button>
</p>
<div id="fig-no-spaces-GO-2,--1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-2, -1.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dotplot(ego.bp, font.size=8)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-2,%20-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GO-2,--2">
Past versions of “GO-2, -2.png”
</button>
</p>
<div id="fig-no-spaces-GO-2,--2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-2, -2.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dotplot(ego.cc, font.size=8)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-2,%20-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GO-2,--3">
Past versions of “GO-2, -3.png”
</button>
</p>
<div id="fig-no-spaces-GO-2,--3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-2, -3.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Compare those that are shared species eGenes, with the interpretation that they would be enriched for things not under stabilizing selection in either, perhaps things like relatively useless pseudogenes.</p>
<p>foreground = shared eGenes</p>
<p>background = all genes tested in both species</p>
<pre class="r"><code>foreground = ChimpToHuman.ID(intersect(ChimpSigGenes,HumanSigGenes))
background = ChimpToHuman.ID(GtexHeartEgenes$chimp_id)
length(foreground)</code></pre>
<pre><code>[1] 145</code></pre>
<pre class="r"><code>length(background)</code></pre>
<pre><code>[1] 11586</code></pre>
<pre class="r"><code># foreground = ChimpToHuman.ID(intersect(ChimpSigGenes,(GtexHeartEgenes %&gt;% top_n(-600, qval) %&gt;% pull(chimp_id))))
# Could also use the more stringent intersection of chimp eGenes and top600 human eGenes. You get mostly the sae GO terms, including MHC complex

ego.mf &lt;- enrichGO(gene = foreground,
                 universe = background,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 ont = &quot;MF&quot;, 
                 pAdjustMethod = &quot;BH&quot;, 
                 pvalueCutoff  = 0.01,
                 qvalueCutoff  = 0.05)
kable(head(as.data.frame(ego.mf)))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="left">geneID</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="GO:0042605" class="uri">GO:0042605</a></td>
<td align="left"><a href="GO:0042605" class="uri">GO:0042605</a></td>
<td align="left">peptide antigen binding</td>
<td align="left">7/129</td>
<td align="left">19/10605</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000004</td>
<td align="right">0.0000004</td>
<td align="left">ENSG00000206503/ENSG00000234745/ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td><a href="GO:0003823" class="uri">GO:0003823</a></td>
<td align="left"><a href="GO:0003823" class="uri">GO:0003823</a></td>
<td align="left">antigen binding</td>
<td align="left">8/129</td>
<td align="left">37/10605</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000016</td>
<td align="right">0.0000016</td>
<td align="left">ENSG00000158473/ENSG00000206503/ENSG00000234745/ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td><a href="GO:0042277" class="uri">GO:0042277</a></td>
<td align="left"><a href="GO:0042277" class="uri">GO:0042277</a></td>
<td align="left">peptide binding</td>
<td align="left">11/129</td>
<td align="left">172/10605</td>
<td align="right">7.50e-06</td>
<td align="right">0.0007191</td>
<td align="right">0.0007172</td>
<td align="left">ENSG00000134184/ENSG00000158473/ENSG00000169760/ENSG00000163697/ENSG00000206503/ENSG00000234745/ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td><a href="GO:0033218" class="uri">GO:0033218</a></td>
<td align="left"><a href="GO:0033218" class="uri">GO:0033218</a></td>
<td align="left">amide binding</td>
<td align="left">11/129</td>
<td align="left">221/10605</td>
<td align="right">7.74e-05</td>
<td align="right">0.0055310</td>
<td align="right">0.0055167</td>
<td align="left">ENSG00000134184/ENSG00000158473/ENSG00000169760/ENSG00000163697/ENSG00000206503/ENSG00000234745/ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389</td>
<td align="right">11</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ego.bp &lt;- enrichGO(gene = foreground,
                 universe = background,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 ont = &quot;BP&quot;, 
                 pAdjustMethod = &quot;BH&quot;, 
                 pvalueCutoff  = 0.01,
                 qvalueCutoff  = 0.05)
kable(head(as.data.frame(ego.bp)))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="left">geneID</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="GO:0019884" class="uri">GO:0019884</a></td>
<td align="left"><a href="GO:0019884" class="uri">GO:0019884</a></td>
<td align="left">antigen processing and presentation of exogenous antigen</td>
<td align="left">9/133</td>
<td align="left">96/10512</td>
<td align="right">3.2e-06</td>
<td align="right">0.0065215</td>
<td align="right">0.0064726</td>
<td align="left">ENSG00000158473/ENSG00000206503/ENSG00000234745/ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389/ENSG00000161800</td>
<td align="right">9</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ego.cc &lt;- enrichGO(gene = foreground,
                 universe = background,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 ont = &quot;CC&quot;, 
                 pAdjustMethod = &quot;BH&quot;, 
                 pvalueCutoff  = 0.01,
                 qvalueCutoff  = 0.05)
kable(head(as.data.frame(ego.cc)))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="left">geneID</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="GO:0071556" class="uri">GO:0071556</a></td>
<td align="left"><a href="GO:0071556" class="uri">GO:0071556</a></td>
<td align="left">integral component of lumenal side of endoplasmic reticulum membrane</td>
<td align="left">8/136</td>
<td align="left">21/10999</td>
<td align="right">0.0e+00</td>
<td align="right">0.00e+00</td>
<td align="right">0.00e+00</td>
<td align="left">ENSG00000206503/ENSG00000234745/ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389/ENSG00000005206</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td><a href="GO:0098553" class="uri">GO:0098553</a></td>
<td align="left"><a href="GO:0098553" class="uri">GO:0098553</a></td>
<td align="left">lumenal side of endoplasmic reticulum membrane</td>
<td align="left">8/136</td>
<td align="left">21/10999</td>
<td align="right">0.0e+00</td>
<td align="right">0.00e+00</td>
<td align="right">0.00e+00</td>
<td align="left">ENSG00000206503/ENSG00000234745/ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389/ENSG00000005206</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td><a href="GO:0042611" class="uri">GO:0042611</a></td>
<td align="left"><a href="GO:0042611" class="uri">GO:0042611</a></td>
<td align="left">MHC protein complex</td>
<td align="left">7/136</td>
<td align="left">18/10999</td>
<td align="right">0.0e+00</td>
<td align="right">1.00e-07</td>
<td align="right">1.00e-07</td>
<td align="left">ENSG00000206503/ENSG00000234745/ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td><a href="GO:0042613" class="uri">GO:0042613</a></td>
<td align="left"><a href="GO:0042613" class="uri">GO:0042613</a></td>
<td align="left">MHC class II protein complex</td>
<td align="left">5/136</td>
<td align="left">12/10999</td>
<td align="right">2.0e-07</td>
<td align="right">1.31e-05</td>
<td align="right">1.17e-05</td>
<td align="left">ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td><a href="GO:0030660" class="uri">GO:0030660</a></td>
<td align="left"><a href="GO:0030660" class="uri">GO:0030660</a></td>
<td align="left">Golgi-associated vesicle membrane</td>
<td align="left">9/136</td>
<td align="left">84/10999</td>
<td align="right">9.0e-07</td>
<td align="right">4.61e-05</td>
<td align="right">4.10e-05</td>
<td align="left">ENSG00000206503/ENSG00000234745/ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389/ENSG00000005206/ENSG00000242247</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td><a href="GO:0012507" class="uri">GO:0012507</a></td>
<td align="left"><a href="GO:0012507" class="uri">GO:0012507</a></td>
<td align="left">ER to Golgi transport vesicle membrane</td>
<td align="left">7/136</td>
<td align="left">47/10999</td>
<td align="right">1.6e-06</td>
<td align="right">6.96e-05</td>
<td align="right">6.19e-05</td>
<td align="left">ENSG00000206503/ENSG00000234745/ENSG00000198502/ENSG00000196126/ENSG00000196735/ENSG00000237541/ENSG00000231389</td>
<td align="right">7</td>
</tr>
</tbody>
</table>
<pre class="r"><code>dotplot(ego.mf, font.size=8)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GO-3-1">
Past versions of GO-3-1.png
</button>
</p>
<div id="fig-GO-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-3-1.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dotplot(ego.bp, font.size=8)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GO-3-2">
Past versions of GO-3-2.png
</button>
</p>
<div id="fig-GO-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-3-2.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dotplot(ego.cc, font.size=8)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GO-3-3">
Past versions of GO-3-3.png
</button>
</p>
<div id="fig-GO-3-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/GO-3-3.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Interestingly, the MHC complex comes out of this analysis as enriched among the genes that are eGenes in both species, consistent with these genes being rapidly evolving targets of selection in between human/chimp, with high diversity in both species.</p>
<p>A complementart analysis to GO overlap is gene set enrichment analysis. Here I will perform GSEA on the chimp qvalue orded list of chimp/human shared eGenes and chimp specific eGenes based on FDR&lt;0.1 in the human dataset…</p>
<p>Actually, maybe a more sensible thing to do is consider the human qvalue ordered list and then separate based on if chimp eGene or not. This is more similar to what the previous odds-ratio plots are.</p>
<pre class="r"><code>HumanOrderedGenes &lt;- GtexHeartEgenes %&gt;%
  arrange(qval) %&gt;%
  filter(chimp_id %in% ChimpSigGenes)
GeneList &lt;- log10(HumanOrderedGenes$qval) * -1
names(GeneList) &lt;- as.character(ChimpToHuman.ID(HumanOrderedGenes$chimp_id))

gsego.cc.shared &lt;- gseGO(gene = GeneList,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 maxGSSize = 500,
                 # pvalueCutoff=0.8,
                 ont = &quot;CC&quot;)
kable(head(as.data.frame(gsego.cc.shared)))

dotplot(gsego.cc.shared, font.size=8, showCategory=15)
gseaplot(gsego.cc.shared, geneSetID = &quot;GO:0042611&quot;, pvalue_table = TRUE)</code></pre>
<p>That was still a work in progress that I may come back to one day… For now I will perform GSEA on ranked gene lists based on chimp FDR.</p>
<p>First, create chimp-qval-ordered gene list for genes that are human eGenes I expect this to return similar results as gene overlap analysis with forground as shared eGenes and background as all all genes tested. So this should return MHC complex among other things.</p>
<pre class="r"><code>Chimp_OrderedGenes &lt;- eQTLs %&gt;%
    group_by(gene) %&gt;% 
    dplyr::slice(which.min(qvalue)) %&gt;%
    filter(gene %in% GtexHeartEgenes$chimp_id) %&gt;%
    arrange(qvalue) %&gt;%
    filter(gene %in% HumanSigGenes)
GeneList &lt;- log10(Chimp_OrderedGenes$qvalue) * -1
names(GeneList) &lt;- as.character(ChimpToHuman.ID(Chimp_OrderedGenes$gene))



gsego.cc.shared &lt;- gseGO(gene = GeneList,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 maxGSSize = 500,
                 # pvalueCutoff=0.8,
                 ont = &quot;CC&quot;,
                 nPerm = 10000)</code></pre>
<pre><code>preparing geneSet collections...</code></pre>
<pre><code>GSEA analysis...</code></pre>
<pre><code>Warning in fgsea(pathways = geneSets, stats = geneList, nperm = nPerm, minSize = minGSSize, : There are ties in the preranked stats (38.71% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<pre><code>leading edge analysis...</code></pre>
<pre><code>done...</code></pre>
<pre class="r"><code>kable(head(as.data.frame(gsego.cc.shared)))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="right">setSize</th>
<th align="right">enrichmentScore</th>
<th align="right">NES</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalues</th>
<th align="right">rank</th>
<th align="left">leading_edge</th>
<th align="left">core_enrichment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="GO:0012507" class="uri">GO:0012507</a></td>
<td align="left"><a href="GO:0012507" class="uri">GO:0012507</a></td>
<td align="left">ER to Golgi transport vesicle membrane</td>
<td align="right">28</td>
<td align="right">0.7104864</td>
<td align="right">1.940217</td>
<td align="right">0.0001007</td>
<td align="right">0.0101562</td>
<td align="right">0.0096857</td>
<td align="right">436</td>
<td align="left">tags=39%, list=8%, signal=36%</td>
<td align="left">ENSG00000196735/ENSG00000196126/ENSG00000206503/ENSG00000198502/ENSG00000237541/ENSG00000234745/ENSG00000231389/ENSG00000204525/ENSG00000148396/ENSG00000179344/ENSG00000223865</td>
</tr>
<tr class="even">
<td><a href="GO:0071556" class="uri">GO:0071556</a></td>
<td align="left"><a href="GO:0071556" class="uri">GO:0071556</a></td>
<td align="left">integral component of lumenal side of endoplasmic reticulum membrane</td>
<td align="right">16</td>
<td align="right">0.8600123</td>
<td align="right">2.186654</td>
<td align="right">0.0001034</td>
<td align="right">0.0101562</td>
<td align="right">0.0096857</td>
<td align="right">436</td>
<td align="left">tags=69%, list=8%, signal=63%</td>
<td align="left">ENSG00000196735/ENSG00000196126/ENSG00000206503/ENSG00000198502/ENSG00000237541/ENSG00000005206/ENSG00000234745/ENSG00000231389/ENSG00000204525/ENSG00000179344/ENSG00000223865</td>
</tr>
<tr class="odd">
<td><a href="GO:0098553" class="uri">GO:0098553</a></td>
<td align="left"><a href="GO:0098553" class="uri">GO:0098553</a></td>
<td align="left">lumenal side of endoplasmic reticulum membrane</td>
<td align="right">16</td>
<td align="right">0.8600123</td>
<td align="right">2.186654</td>
<td align="right">0.0001034</td>
<td align="right">0.0101562</td>
<td align="right">0.0096857</td>
<td align="right">436</td>
<td align="left">tags=69%, list=8%, signal=63%</td>
<td align="left">ENSG00000196735/ENSG00000196126/ENSG00000206503/ENSG00000198502/ENSG00000237541/ENSG00000005206/ENSG00000234745/ENSG00000231389/ENSG00000204525/ENSG00000179344/ENSG00000223865</td>
</tr>
<tr class="even">
<td><a href="GO:0042611" class="uri">GO:0042611</a></td>
<td align="left"><a href="GO:0042611" class="uri">GO:0042611</a></td>
<td align="left">MHC protein complex</td>
<td align="right">14</td>
<td align="right">0.8498340</td>
<td align="right">2.108766</td>
<td align="right">0.0001048</td>
<td align="right">0.0101562</td>
<td align="right">0.0096857</td>
<td align="right">436</td>
<td align="left">tags=71%, list=8%, signal=66%</td>
<td align="left">ENSG00000196735/ENSG00000196126/ENSG00000206503/ENSG00000198502/ENSG00000237541/ENSG00000234745/ENSG00000231389/ENSG00000204525/ENSG00000179344/ENSG00000223865</td>
</tr>
<tr class="odd">
<td><a href="GO:0042613" class="uri">GO:0042613</a></td>
<td align="left"><a href="GO:0042613" class="uri">GO:0042613</a></td>
<td align="left">MHC class II protein complex</td>
<td align="right">10</td>
<td align="right">0.8404787</td>
<td align="right">1.960527</td>
<td align="right">0.0001085</td>
<td align="right">0.0101562</td>
<td align="right">0.0096857</td>
<td align="right">436</td>
<td align="left">tags=70%, list=8%, signal=64%</td>
<td align="left">ENSG00000196735/ENSG00000196126/ENSG00000198502/ENSG00000237541/ENSG00000231389/ENSG00000179344/ENSG00000223865</td>
</tr>
<tr class="even">
<td><a href="GO:0030669" class="uri">GO:0030669</a></td>
<td align="left"><a href="GO:0030669" class="uri">GO:0030669</a></td>
<td align="left">clathrin-coated endocytic vesicle membrane</td>
<td align="right">19</td>
<td align="right">0.7044431</td>
<td align="right">1.834206</td>
<td align="right">0.0002043</td>
<td align="right">0.0159346</td>
<td align="right">0.0151963</td>
<td align="right">567</td>
<td align="left">tags=47%, list=11%, signal=42%</td>
<td align="left">ENSG00000196735/ENSG00000196126/ENSG00000198502/ENSG00000237541/ENSG00000231389/ENSG00000179344/ENSG00000223865/ENSG00000174804/ENSG00000006125</td>
</tr>
</tbody>
</table>
<pre class="r"><code>dotplot(gsego.cc.shared, font.size=8, showCategory=15)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/unnamed-chunk-3-1.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gseaplot2(gsego.cc.shared, geneSetID = c(&quot;GO:0042611&quot;, &quot;GO:0042613&quot;, &quot;GO:0032588&quot;), title=&quot;Shared eGenes ordered by chimp -log(qvalue)&quot;, pvalue_table = TRUE)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/unnamed-chunk-3-2.png" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok that does return MHC as expect. Now I will create chimp-qval-ordered gene list for genes that are not human eGenes. I expect this to not return MHC. Instead,I expect this to return similar results as gene overlap analysis with forground as chimp-specific eGenes and background as all chimp-eGenes (nucleic acid binding, etc).</p>
<pre class="r"><code>Chimp_OrderedGenes &lt;- eQTLs %&gt;%
    group_by(gene) %&gt;% 
    dplyr::slice(which.min(qvalue)) %&gt;%
    filter(gene %in% GtexHeartEgenes$chimp_id) %&gt;%
    filter(qvalue&lt;0.2) %&gt;%
    arrange(qvalue) %&gt;%
    filter(!gene %in% HumanSigGenes)
GeneList &lt;- log10(Chimp_OrderedGenes$qvalue) * -1
names(GeneList) &lt;- as.character(ChimpToHuman.ID(Chimp_OrderedGenes$gene))

gsego.cc.chimpspecific &lt;- gseGO(gene = GeneList,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 maxGSSize = 500,
                 pvalueCutoff=0.8,
                 ont = &quot;BP&quot;)</code></pre>
<pre><code>preparing geneSet collections...</code></pre>
<pre><code>GSEA analysis...</code></pre>
<pre><code>Warning in fgsea(pathways = geneSets, stats = geneList, nperm = nPerm, minSize = minGSSize, : There are ties in the preranked stats (28.95% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<pre><code>no term enriched under specific pvalueCutoff...</code></pre>
<pre class="r"><code>kable(head(as.data.frame(gsego.cc.chimpspecific)))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ID</th>
<th align="left">Description</th>
<th align="right">setSize</th>
<th align="right">enrichmentScore</th>
<th align="right">NES</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalues</th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Ok that returned nothing, even at more relaxed pvalue thresholds (not shown) I can see the top terms do not have anything to do with nucleic acid binding like the previous gene overlap enrichment analysis results that compared chimp-specific-eGene (foreground) to all chimp eGenes (universe background).</p>
<p>By partitioning the GSEA analyses into “human eGene” ranked gene list and “not human eGene” ranked gene list, what I was really trying to get at is the semi-quantitative difference in eGene character. Essentially these are paired analyses, though GSEA does not have any natural way to run a paired analysis on two gene sets. Perhaps a better way to do this is to make a single gene list based on the difference in eGene ranks between the two species. Since GSEA is based on non-parametric rank orders, I think this is still somewhat reasonable to do. For example, if qvalues are ranked lowest to highest in both species (rank=1 being most significant), and I subtract the chimp ranking from the human ranking, I am left with the more chimp specific eGenes with a higher number. I will implement this idea below:</p>
<pre class="r"><code>Chimp_OrderedGenes &lt;- eQTLs %&gt;%
    group_by(gene) %&gt;% 
    dplyr::slice(which.min(qvalue)) %&gt;%
    filter(gene %in% GtexHeartEgenes$chimp_id) %&gt;%
    left_join(GtexHeartEgenes, by=c(&quot;gene&quot;=&quot;chimp_id&quot;)) %&gt;%
    dplyr::select(gene, qvalue, qval) %&gt;% as.data.frame() %&gt;%
    mutate(ChimpRank = dense_rank(qvalue)) %&gt;%
    mutate(HumanRank = dense_rank(qval)) %&gt;%
    mutate(RankDifference = HumanRank-ChimpRank) %&gt;%
    arrange(desc(RankDifference))</code></pre>
<pre><code>Warning: Column `gene`/`chimp_id` joining factor and character vector,
coercing into character vector</code></pre>
<pre class="r"><code>GeneList &lt;- Chimp_OrderedGenes$RankDifference
names(GeneList) &lt;- as.character(ChimpToHuman.ID(Chimp_OrderedGenes$gene))
    
gsego.cc.chimpspecific &lt;- gseGO(gene = GeneList,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 maxGSSize = 500,
                 pvalueCutoff=0.8,
                 ont = &quot;BP&quot;)</code></pre>
<pre><code>preparing geneSet collections...</code></pre>
<pre><code>GSEA analysis...</code></pre>
<pre><code>Warning in fgsea(pathways = geneSets, stats = geneList, nperm = nPerm, minSize = minGSSize, : There are ties in the preranked stats (35.5% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<pre><code>leading edge analysis...</code></pre>
<pre><code>done...</code></pre>
<pre class="r"><code>kable(head(as.data.frame(gsego.cc.chimpspecific)))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="right">setSize</th>
<th align="right">enrichmentScore</th>
<th align="right">NES</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalues</th>
<th align="right">rank</th>
<th align="left">leading_edge</th>
<th align="left">core_enrichment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="GO:0000904" class="uri">GO:0000904</a></td>
<td align="left"><a href="GO:0000904" class="uri">GO:0000904</a></td>
<td align="left">cell morphogenesis involved in differentiation</td>
<td align="right">473</td>
<td align="right">0.3623578</td>
<td align="right">1.318620</td>
<td align="right">0.000999</td>
<td align="right">0.0629941</td>
<td align="right">0.0555252</td>
<td align="right">4533</td>
<td align="left">tags=50%, list=41%, signal=31%</td>
<td align="left">ENSG00000171862/ENSG00000179242/ENSG00000073712/ENSG00000163359/ENSG00000179820/ENSG00000176749/ENSG00000105186/ENSG00000163110/ENSG00000173482/ENSG00000167461/ENSG00000100345/ENSG00000118046/ENSG00000125650/ENSG00000197702/ENSG00000082684/ENSG00000133110/ENSG00000143590/ENSG00000167193/ENSG00000117600/ENSG00000011009/ENSG00000173166/ENSG00000149930/ENSG00000147862/ENSG00000163975/ENSG00000132561/ENSG00000181790/ENSG00000169554/ENSG00000046889/ENSG00000115904/ENSG00000174989/ENSG00000087470/ENSG00000117400/ENSG00000197283/ENSG00000115232/ENSG00000119508/ENSG00000159674/ENSG00000137845/ENSG00000165802/ENSG00000168036/ENSG00000117707/ENSG00000196923/ENSG00000166313/ENSG00000142168/ENSG00000153707/ENSG00000068305/ENSG00000163785/ENSG00000174238/ENSG00000169379/ENSG00000118785/ENSG00000124171/ENSG00000114745/ENSG00000076356/ENSG00000160460/ENSG00000144857/ENSG00000104133/ENSG00000084733/ENSG00000170558/ENSG00000141646/ENSG00000144619/ENSG00000100320/ENSG00000160691/ENSG00000141522/ENSG00000084112/ENSG00000138660/ENSG00000160469/ENSG00000165819/ENSG00000130558/ENSG00000179915/ENSG00000112658/ENSG00000105649/ENSG00000138411/ENSG00000067141/ENSG00000117228/ENSG00000173786/ENSG00000137817/ENSG00000092964/ENSG00000120156/ENSG00000101126/ENSG00000051382/ENSG00000163531/ENSG00000205221/ENSG00000169862/ENSG00000132003/ENSG00000170017/ENSG00000132640/ENSG00000176248/ENSG00000146013/ENSG00000109062/ENSG00000197892/ENSG00000106278/ENSG00000106078/ENSG00000119608/ENSG00000177885/ENSG00000154096/ENSG00000113749/ENSG00000184347/ENSG00000124126/ENSG00000099250/ENSG00000140538/ENSG00000122877/ENSG00000198836/ENSG00000079215/ENSG00000077942/ENSG00000091129/ENSG00000132670/ENSG00000055163/ENSG00000162738/ENSG00000036257/ENSG00000179051/ENSG00000148948/ENSG00000169439/ENSG00000197555/ENSG00000100968/ENSG00000107404/ENSG00000010017/ENSG00000136040/ENSG00000173898/ENSG00000171055/ENSG00000149269/ENSG00000160007/ENSG00000111913/ENSG00000164488/ENSG00000136854/ENSG00000152767/ENSG00000071127/ENSG00000188026/ENSG00000021574/ENSG00000142599/ENSG00000007168/ENSG00000117632/ENSG00000169604/ENSG00000105711/ENSG00000169855/ENSG00000138069/ENSG00000150760/ENSG00000107758/ENSG00000127603/ENSG00000171517/ENSG00000167601/ENSG00000111859/ENSG00000145675/ENSG00000159167/ENSG00000064300/ENSG00000124782/ENSG00000109458/ENSG00000133026/ENSG00000106571/ENSG00000156642/ENSG00000169504/ENSG00000137872/ENSG00000092820/ENSG00000148498/ENSG00000153234/ENSG00000130203/ENSG00000081052/ENSG00000105245/ENSG00000081189/ENSG00000100503/ENSG00000214655/ENSG00000257923/ENSG00000138347/ENSG00000169032/ENSG00000079805/ENSG00000092929/ENSG00000166579/ENSG00000173706/ENSG00000067900/ENSG00000008324/ENSG00000077943/ENSG00000149557/ENSG00000111962/ENSG00000188906/ENSG00000179403/ENSG00000114270/ENSG00000147459/ENSG00000128656/ENSG00000125266/ENSG00000149781/ENSG00000135387/ENSG00000151892/ENSG00000136603/ENSG00000197694/ENSG00000139626/ENSG00000251322/ENSG00000169760/ENSG00000179218/ENSG00000182578/ENSG00000115738/ENSG00000108561/ENSG00000137825/ENSG00000198513/ENSG00000118257/ENSG00000100030/ENSG00000116106/ENSG00000166333/ENSG00000198492/ENSG00000182197/ENSG00000121879/ENSG00000080371/ENSG00000164050/ENSG00000108830/ENSG00000206052/ENSG00000185950/ENSG00000133104/ENSG00000101144/ENSG00000154133/ENSG00000114251/ENSG00000087250/ENSG00000074527/ENSG00000130449/ENSG00000077097/ENSG00000112715/ENSG00000243364/ENSG00000171608/ENSG00000139926/ENSG00000156299/ENSG00000116035/ENSG00000221866/ENSG00000107731/ENSG00000119328/ENSG00000164736/ENSG00000166963/ENSG00000154310/ENSG00000187688/ENSG00000161642/ENSG00000100393/ENSG00000033327/ENSG00000182580/ENSG00000149418/ENSG00000105976/ENSG00000177169/ENSG00000125875/ENSG00000115694/ENSG00000184144/ENSG00000125249/ENSG00000158321/ENSG00000150977</td>
</tr>
<tr class="even">
<td><a href="GO:0001501" class="uri">GO:0001501</a></td>
<td align="left"><a href="GO:0001501" class="uri">GO:0001501</a></td>
<td align="left">skeletal system development</td>
<td align="right">311</td>
<td align="right">0.3801804</td>
<td align="right">1.363580</td>
<td align="right">0.000999</td>
<td align="right">0.0629941</td>
<td align="right">0.0555252</td>
<td align="right">3391</td>
<td align="left">tags=41%, list=31%, signal=29%</td>
<td align="left">ENSG00000100985/ENSG00000163359/ENSG00000109906/ENSG00000147316/ENSG00000139329/ENSG00000143570/ENSG00000078401/ENSG00000157404/ENSG00000019549/ENSG00000182718/ENSG00000178573/ENSG00000141873/ENSG00000151388/ENSG00000125878/ENSG00000133110/ENSG00000173868/ENSG00000116183/ENSG00000143995/ENSG00000092969/ENSG00000147862/ENSG00000107372/ENSG00000151702/ENSG00000132561/ENSG00000124216/ENSG00000139567/ENSG00000172845/ENSG00000120254/ENSG00000163513/ENSG00000120149/ENSG00000168036/ENSG00000116539/ENSG00000165458/ENSG00000204262/ENSG00000101265/ENSG00000081237/ENSG00000173153/ENSG00000163785/ENSG00000131759/ENSG00000170577/ENSG00000168542/ENSG00000134013/ENSG00000145423/ENSG00000172071/ENSG00000104313/ENSG00000168646/ENSG00000077092/ENSG00000119699/ENSG00000164692/ENSG00000135111/ENSG00000141543/ENSG00000196159/ENSG00000184916/ENSG00000105664/ENSG00000108557/ENSG00000142453/ENSG00000104447/ENSG00000112658/ENSG00000164190/ENSG00000134250/ENSG00000128606/ENSG00000054598/ENSG00000075711/ENSG00000141627/ENSG00000120156/ENSG00000205221/ENSG00000155090/ENSG00000070814/ENSG00000121039/ENSG00000183688/ENSG00000107679/ENSG00000180228/ENSG00000174600/ENSG00000166949/ENSG00000185630/ENSG00000078246/ENSG00000177119/ENSG00000153162/ENSG00000126351/ENSG00000165804/ENSG00000109685/ENSG00000106483/ENSG00000179111/ENSG00000125845/ENSG00000164107/ENSG00000176842/ENSG00000172809/ENSG00000012223/ENSG00000158859/ENSG00000127418/ENSG00000007168/ENSG00000142871/ENSG00000141753/ENSG00000178882/ENSG00000110693/ENSG00000145431/ENSG00000138685/ENSG00000166224/ENSG00000089041/ENSG00000090539/ENSG00000159167/ENSG00000145545/ENSG00000106571/ENSG00000143867/ENSG00000167522/ENSG00000148516/ENSG00000072840/ENSG00000081189/ENSG00000020633/ENSG00000134853/ENSG00000163659/ENSG00000082293/ENSG00000163132/ENSG00000170776/ENSG00000166341/ENSG00000134253/ENSG00000011143/ENSG00000188783/ENSG00000100644/ENSG00000118523/ENSG00000088256/ENSG00000112293/ENSG00000134245/ENSG00000102977/ENSG00000186480/ENSG00000179403/ENSG00000154122/ENSG00000114270/ENSG00000197467</td>
</tr>
<tr class="odd">
<td><a href="GO:0001525" class="uri">GO:0001525</a></td>
<td align="left"><a href="GO:0001525" class="uri">GO:0001525</a></td>
<td align="left">angiogenesis</td>
<td align="right">351</td>
<td align="right">0.3623115</td>
<td align="right">1.306815</td>
<td align="right">0.000999</td>
<td align="right">0.0629941</td>
<td align="right">0.0555252</td>
<td align="right">4224</td>
<td align="left">tags=50%, list=38%, signal=32%</td>
<td align="left">ENSG00000171862/ENSG00000105974/ENSG00000164619/ENSG00000173482/ENSG00000100345/ENSG00000078401/ENSG00000182718/ENSG00000163638/ENSG00000069667/ENSG00000141376/ENSG00000105851/ENSG00000197702/ENSG00000164134/ENSG00000143995/ENSG00000143590/ENSG00000134871/ENSG00000092969/ENSG00000134954/ENSG00000154188/ENSG00000181790/ENSG00000139567/ENSG00000153904/ENSG00000137801/ENSG00000176170/ENSG00000163513/ENSG00000129173/ENSG00000233276/ENSG00000168036/ENSG00000173801/ENSG00000076706/ENSG00000138798/ENSG00000177464/ENSG00000176692/ENSG00000136848/ENSG00000106366/ENSG00000164342/ENSG00000134013/ENSG00000145423/ENSG00000123358/ENSG00000070495/ENSG00000172071/ENSG00000125730/ENSG00000141448/ENSG00000136158/ENSG00000048052/ENSG00000160691/ENSG00000205336/ENSG00000117461/ENSG00000112658/ENSG00000054598/ENSG00000140992/ENSG00000120156/ENSG00000051382/ENSG00000140199/ENSG00000074181/ENSG00000166292/ENSG00000174059/ENSG00000107014/ENSG00000065970/ENSG00000138080/ENSG00000163874/ENSG00000154096/ENSG00000099250/ENSG00000162434/ENSG00000105538/ENSG00000150782/ENSG00000130770/ENSG00000171951/ENSG00000091129/ENSG00000100219/ENSG00000214274/ENSG00000171812/ENSG00000138772/ENSG00000133639/ENSG00000100968/ENSG00000136960/ENSG00000173083/ENSG00000132170/ENSG00000170989/ENSG00000164107/ENSG00000239697/ENSG00000179348/ENSG00000142871/ENSG00000136826/ENSG00000169245/ENSG00000138685/ENSG00000160293/ENSG00000108691/ENSG00000168329/ENSG00000164283/ENSG00000126458/ENSG00000147649/ENSG00000101384/ENSG00000122679/ENSG00000111145/ENSG00000166086/ENSG00000171236/ENSG00000118503/ENSG00000142208/ENSG00000169855/ENSG00000187498/ENSG00000120063/ENSG00000101230/ENSG00000144476/ENSG00000157613/ENSG00000064300/ENSG00000166501/ENSG00000109458/ENSG00000162775/ENSG00000169429/ENSG00000169504/ENSG00000117020/ENSG00000140464/ENSG00000081052/ENSG00000117525/ENSG00000106511/ENSG00000073756/ENSG00000006638/ENSG00000074842/ENSG00000061273/ENSG00000134853/ENSG00000145632/ENSG00000111961/ENSG00000100644/ENSG00000115415/ENSG00000067900/ENSG00000168214/ENSG00000118523/ENSG00000112293/ENSG00000108840/ENSG00000126785/ENSG00000011405/ENSG00000006210/ENSG00000148926/ENSG00000112562/ENSG00000138061/ENSG00000125266/ENSG00000027869/ENSG00000140105/ENSG00000079385/ENSG00000123342/ENSG00000136720/ENSG00000106804/ENSG00000156427/ENSG00000113196/ENSG00000167772/ENSG00000161714/ENSG00000113721/ENSG00000185359/ENSG00000019991/ENSG00000160255/ENSG00000118257/ENSG00000115053/ENSG00000124212/ENSG00000135250/ENSG00000138639/ENSG00000064989/ENSG00000121879/ENSG00000087245/ENSG00000154133/ENSG00000127837/ENSG00000189058/ENSG00000114251/ENSG00000159399/ENSG00000143878/ENSG00000006831/ENSG00000135636/ENSG00000112715/ENSG00000091879/ENSG00000152270/ENSG00000004776/ENSG00000134318/ENSG00000110955/ENSG00000101109/ENSG00000107731/ENSG00000100292</td>
</tr>
<tr class="even">
<td><a href="GO:0001568" class="uri">GO:0001568</a></td>
<td align="left"><a href="GO:0001568" class="uri">GO:0001568</a></td>
<td align="left">blood vessel development</td>
<td align="right">472</td>
<td align="right">0.3523209</td>
<td align="right">1.281995</td>
<td align="right">0.000999</td>
<td align="right">0.0629941</td>
<td align="right">0.0555252</td>
<td align="right">4224</td>
<td align="left">tags=49%, list=38%, signal=31%</td>
<td align="left">ENSG00000171862/ENSG00000105974/ENSG00000148737/ENSG00000164619/ENSG00000173482/ENSG00000100345/ENSG00000078401/ENSG00000182718/ENSG00000163638/ENSG00000069667/ENSG00000141376/ENSG00000105851/ENSG00000197702/ENSG00000164134/ENSG00000143995/ENSG00000143590/ENSG00000134871/ENSG00000092969/ENSG00000134954/ENSG00000154188/ENSG00000181790/ENSG00000115904/ENSG00000174989/ENSG00000139567/ENSG00000153904/ENSG00000137801/ENSG00000176170/ENSG00000163513/ENSG00000129173/ENSG00000233276/ENSG00000168036/ENSG00000173801/ENSG00000076706/ENSG00000138798/ENSG00000117707/ENSG00000177464/ENSG00000176692/ENSG00000136848/ENSG00000106366/ENSG00000168542/ENSG00000169946/ENSG00000164342/ENSG00000134013/ENSG00000145423/ENSG00000123358/ENSG00000070495/ENSG00000172071/ENSG00000125730/ENSG00000104313/ENSG00000164692/ENSG00000141448/ENSG00000170558/ENSG00000136158/ENSG00000135111/ENSG00000133454/ENSG00000048052/ENSG00000086598/ENSG00000160691/ENSG00000205336/ENSG00000117461/ENSG00000112658/ENSG00000069122/ENSG00000054598/ENSG00000120738/ENSG00000140992/ENSG00000120156/ENSG00000116473/ENSG00000051382/ENSG00000185650/ENSG00000140199/ENSG00000074181/ENSG00000166292/ENSG00000174059/ENSG00000107014/ENSG00000065970/ENSG00000138080/ENSG00000163874/ENSG00000154096/ENSG00000099250/ENSG00000162434/ENSG00000105538/ENSG00000150782/ENSG00000198081/ENSG00000130770/ENSG00000107796/ENSG00000171951/ENSG00000091129/ENSG00000049323/ENSG00000100219/ENSG00000214274/ENSG00000171812/ENSG00000087088/ENSG00000138772/ENSG00000133639/ENSG00000100968/ENSG00000136960/ENSG00000105672/ENSG00000173083/ENSG00000132170/ENSG00000170989/ENSG00000164107/ENSG00000239697/ENSG00000179348/ENSG00000142871/ENSG00000136826/ENSG00000145715/ENSG00000169604/ENSG00000169245/ENSG00000138685/ENSG00000160293/ENSG00000166224/ENSG00000108691/ENSG00000113083/ENSG00000168329/ENSG00000164283/ENSG00000126458/ENSG00000147649/ENSG00000101384/ENSG00000122679/ENSG00000166135/ENSG00000111145/ENSG00000166086/ENSG00000171236/ENSG00000117713/ENSG00000118503/ENSG00000142208/ENSG00000137834/ENSG00000169855/ENSG00000187498/ENSG00000120063/ENSG00000101230/ENSG00000144476/ENSG00000157483/ENSG00000157613/ENSG00000064300/ENSG00000065534/ENSG00000166501/ENSG00000109458/ENSG00000162775/ENSG00000128918/ENSG00000133026/ENSG00000197603/ENSG00000169429/ENSG00000106571/ENSG00000169504/ENSG00000143867/ENSG00000117020/ENSG00000130203/ENSG00000140464/ENSG00000081052/ENSG00000107263/ENSG00000117525/ENSG00000081189/ENSG00000106511/ENSG00000073756/ENSG00000006638/ENSG00000074842/ENSG00000061273/ENSG00000134853/ENSG00000163659/ENSG00000145632/ENSG00000169032/ENSG00000079805/ENSG00000173706/ENSG00000111961/ENSG00000100644/ENSG00000115415/ENSG00000067900/ENSG00000168214/ENSG00000118523/ENSG00000112293/ENSG00000108840/ENSG00000126785/ENSG00000011405/ENSG00000006210/ENSG00000148926/ENSG00000112562/ENSG00000138061/ENSG00000125266/ENSG00000027869/ENSG00000163909/ENSG00000184557/ENSG00000140105/ENSG00000079385/ENSG00000183072/ENSG00000112655/ENSG00000123342/ENSG00000136720/ENSG00000106804/ENSG00000063176/ENSG00000156427/ENSG00000113196/ENSG00000167772/ENSG00000074219/ENSG00000161714/ENSG00000113721/ENSG00000185359/ENSG00000019991/ENSG00000160255/ENSG00000118257/ENSG00000115053/ENSG00000100030/ENSG00000124212/ENSG00000135250/ENSG00000138639/ENSG00000064989/ENSG00000121879/ENSG00000129038/ENSG00000087245/ENSG00000005981/ENSG00000114739/ENSG00000101144/ENSG00000154133/ENSG00000127837/ENSG00000189058/ENSG00000114251/ENSG00000159399/ENSG00000143878/ENSG00000006831/ENSG00000135636/ENSG00000118762/ENSG00000112715/ENSG00000091879/ENSG00000152270/ENSG00000107882/ENSG00000004776/ENSG00000134318/ENSG00000110955/ENSG00000101109/ENSG00000107731/ENSG00000100292</td>
</tr>
<tr class="odd">
<td><a href="GO:0001655" class="uri">GO:0001655</a></td>
<td align="left"><a href="GO:0001655" class="uri">GO:0001655</a></td>
<td align="left">urogenital system development</td>
<td align="right">224</td>
<td align="right">0.3804207</td>
<td align="right">1.351549</td>
<td align="right">0.000999</td>
<td align="right">0.0629941</td>
<td align="right">0.0555252</td>
<td align="right">4144</td>
<td align="left">tags=49%, list=38%, signal=31%</td>
<td align="left">ENSG00000171862/ENSG00000178585/ENSG00000109906/ENSG00000128059/ENSG00000164619/ENSG00000181690/ENSG00000168743/ENSG00000118046/ENSG00000170561/ENSG00000141449/ENSG00000092969/ENSG00000130707/ENSG00000154188/ENSG00000103494/ENSG00000132470/ENSG00000168036/ENSG00000117707/ENSG00000153094/ENSG00000177464/ENSG00000115884/ENSG00000117724/ENSG00000131759/ENSG00000170577/ENSG00000124145/ENSG00000070495/ENSG00000170348/ENSG00000104313/ENSG00000173473/ENSG00000077092/ENSG00000162849/ENSG00000187391/ENSG00000141646/ENSG00000196159/ENSG00000081181/ENSG00000115204/ENSG00000070404/ENSG00000164190/ENSG00000163348/ENSG00000054598/ENSG00000006016/ENSG00000120738/ENSG00000075711/ENSG00000120156/ENSG00000005884/ENSG00000112837/ENSG00000074181/ENSG00000174059/ENSG00000121039/ENSG00000169018/ENSG00000099250/ENSG00000100360/ENSG00000166949/ENSG00000107796/ENSG00000185630/ENSG00000162738/ENSG00000186350/ENSG00000087088/ENSG00000153162/ENSG00000171016/ENSG00000125845/ENSG00000164488/ENSG00000179348/ENSG00000025796/ENSG00000138685/ENSG00000166224/ENSG00000101384/ENSG00000005513/ENSG00000117318/ENSG00000137834/ENSG00000187498/ENSG00000157483/ENSG00000111276/ENSG00000128918/ENSG00000145545/ENSG00000197603/ENSG00000106571/ENSG00000143867/ENSG00000081052/ENSG00000081189/ENSG00000134853/ENSG00000163659/ENSG00000166341/ENSG00000115415/ENSG00000134245/ENSG00000116962/ENSG00000077943/ENSG00000102977/ENSG00000188906/ENSG00000125266/ENSG00000163909/ENSG00000137869/ENSG00000121361/ENSG00000091831/ENSG00000111087/ENSG00000115738/ENSG00000113721/ENSG00000196562/ENSG00000116106/ENSG00000166333/ENSG00000116455/ENSG00000114739/ENSG00000101144/ENSG00000114251/ENSG00000102900/ENSG00000118762/ENSG00000112715/ENSG00000115758/ENSG00000091879/ENSG00000205213</td>
</tr>
<tr class="even">
<td><a href="GO:0001701" class="uri">GO:0001701</a></td>
<td align="left"><a href="GO:0001701" class="uri">GO:0001701</a></td>
<td align="left">in utero embryonic development</td>
<td align="right">232</td>
<td align="right">0.3694125</td>
<td align="right">1.313362</td>
<td align="right">0.000999</td>
<td align="right">0.0629941</td>
<td align="right">0.0555252</td>
<td align="right">3705</td>
<td align="left">tags=45%, list=34%, signal=30%</td>
<td align="left">ENSG00000100345/ENSG00000143570/ENSG00000078401/ENSG00000141873/ENSG00000100601/ENSG00000112893/ENSG00000154188/ENSG00000107372/ENSG00000103494/ENSG00000124216/ENSG00000174989/ENSG00000139567/ENSG00000172845/ENSG00000142784/ENSG00000144357/ENSG00000163513/ENSG00000129173/ENSG00000137845/ENSG00000130726/ENSG00000126457/ENSG00000168036/ENSG00000153094/ENSG00000185122/ENSG00000169946/ENSG00000119699/ENSG00000132376/ENSG00000141448/ENSG00000131446/ENSG00000108773/ENSG00000135111/ENSG00000141646/ENSG00000133454/ENSG00000086598/ENSG00000184916/ENSG00000100146/ENSG00000254999/ENSG00000147536/ENSG00000198730/ENSG00000104375/ENSG00000112658/ENSG00000150961/ENSG00000163348/ENSG00000134057/ENSG00000054598/ENSG00000113163/ENSG00000185650/ENSG00000198026/ENSG00000166292/ENSG00000121039/ENSG00000177885/ENSG00000122257/ENSG00000119048/ENSG00000166949/ENSG00000186350/ENSG00000036257/ENSG00000105672/ENSG00000165916/ENSG00000125845/ENSG00000164107/ENSG00000196839/ENSG00000179348/ENSG00000142871/ENSG00000100814/ENSG00000110693/ENSG00000112578/ENSG00000005513/ENSG00000101003/ENSG00000142208/ENSG00000120063/ENSG00000197063/ENSG00000157483/ENSG00000115760/ENSG00000133026/ENSG00000106571/ENSG00000167522/ENSG00000100888/ENSG00000134853/ENSG00000101773/ENSG00000034053/ENSG00000110107/ENSG00000163132/ENSG00000169032/ENSG00000166579/ENSG00000136450/ENSG00000173706/ENSG00000100644/ENSG00000168214/ENSG00000113555/ENSG00000148926/ENSG00000141030/ENSG00000184557/ENSG00000136720/ENSG00000106804/ENSG00000136603/ENSG00000138336/ENSG00000107077/ENSG00000111596/ENSG00000171345/ENSG00000113196/ENSG00000169375/ENSG00000161714/ENSG00000105656/ENSG00000140416/ENSG00000011132</td>
</tr>
</tbody>
</table>
<pre class="r"><code># gseaplot(gsego.cc.chimpspecific, geneSetID = &quot;GO:0001501&quot;, pvalue_table = TRUE)
# dotplot(gsego.cc.chimpspecific, font.size=8, showCategory=15)</code></pre>
<p>Nothing is significant by any notable treshold by this method…</p>
<p>Perhaps I need to limit the gene set to genes which were called as eGenes in chimps, since many too many chimp non-eGenes may contribute to noise.</p>
<pre class="r"><code>Chimp_OrderedGenes &lt;- eQTLs %&gt;%
    group_by(gene) %&gt;% 
    dplyr::slice(which.min(qvalue)) %&gt;%
    filter(gene %in% GtexHeartEgenes$chimp_id) %&gt;%
    filter(qvalue &lt;0.1) %&gt;%
    left_join(GtexHeartEgenes, by=c(&quot;gene&quot;=&quot;chimp_id&quot;)) %&gt;%
    dplyr::select(gene, qvalue, qval) %&gt;% as.data.frame() %&gt;%
    mutate(ChimpRank = dense_rank(qvalue)) %&gt;%
    mutate(HumanRank = dense_rank(qval)) %&gt;%
    mutate(RankDifference = HumanRank-ChimpRank) %&gt;%
    arrange(desc(RankDifference))</code></pre>
<pre><code>Warning: Column `gene`/`chimp_id` joining factor and character vector,
coercing into character vector</code></pre>
<pre class="r"><code>GeneList &lt;- Chimp_OrderedGenes$RankDifference
names(GeneList) &lt;- as.character(ChimpToHuman.ID(Chimp_OrderedGenes$gene))
    
gsego.cc.chimpspecific &lt;- gseGO(gene = GeneList,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 maxGSSize = 500,
                 ont = &quot;BP&quot;,
                 nPerm = 10000)</code></pre>
<pre><code>preparing geneSet collections...</code></pre>
<pre><code>GSEA analysis...</code></pre>
<pre><code>Warning in fgsea(pathways = geneSets, stats = geneList, nperm = nPerm, minSize = minGSSize, : There are ties in the preranked stats (33.21% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<pre><code>leading edge analysis...</code></pre>
<pre><code>done...</code></pre>
<pre class="r"><code>kable(head(as.data.frame(gsego.cc.chimpspecific), 15))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="right">setSize</th>
<th align="right">enrichmentScore</th>
<th align="right">NES</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalues</th>
<th align="right">rank</th>
<th align="left">leading_edge</th>
<th align="left">core_enrichment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="GO:0044271" class="uri">GO:0044271</a></td>
<td align="left"><a href="GO:0044271" class="uri">GO:0044271</a></td>
<td align="left">cellular nitrogen compound biosynthetic process</td>
<td align="right">64</td>
<td align="right">0.5375722</td>
<td align="right">1.605074</td>
<td align="right">0.0003000</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">110</td>
<td align="left">tags=64%, list=39%, signal=50%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000128000/ENSG00000078401/ENSG00000163376/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000124279/ENSG00000178952/ENSG00000114503/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000163923/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000037241/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000127586/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000182899/ENSG00000170325/ENSG00000106261/ENSG00000113643/ENSG00000152402/ENSG00000185122/ENSG00000125457</td>
</tr>
<tr class="even">
<td><a href="GO:0051252" class="uri">GO:0051252</a></td>
<td align="left"><a href="GO:0051252" class="uri">GO:0051252</a></td>
<td align="left">regulation of RNA metabolic process</td>
<td align="right">50</td>
<td align="right">0.5679773</td>
<td align="right">1.667466</td>
<td align="right">0.0003003</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">101</td>
<td align="left">tags=66%, list=36%, signal=51%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000075856/ENSG00000167513/ENSG00000164758/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000124279/ENSG00000061936/ENSG00000114503/ENSG00000107371/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000170325/ENSG00000106261</td>
</tr>
<tr class="odd">
<td><a href="GO:0016070" class="uri">GO:0016070</a></td>
<td align="left"><a href="GO:0016070" class="uri">GO:0016070</a></td>
<td align="left">RNA metabolic process</td>
<td align="right">57</td>
<td align="right">0.5592330</td>
<td align="right">1.656604</td>
<td align="right">0.0004002</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">107</td>
<td align="left">tags=65%, list=38%, signal=50%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000075856/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000124279/ENSG00000061936/ENSG00000114503/ENSG00000107371/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000182899/ENSG00000170325/ENSG00000106261/ENSG00000113643/ENSG00000185122</td>
</tr>
<tr class="even">
<td><a href="GO:0019219" class="uri">GO:0019219</a></td>
<td align="left"><a href="GO:0019219" class="uri">GO:0019219</a></td>
<td align="left">regulation of nucleobase-containing compound metabolic process</td>
<td align="right">54</td>
<td align="right">0.5498605</td>
<td align="right">1.622555</td>
<td align="right">0.0004002</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">101</td>
<td align="left">tags=63%, list=36%, signal=50%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000075856/ENSG00000167513/ENSG00000164758/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000124279/ENSG00000061936/ENSG00000114503/ENSG00000107371/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000127586/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000170325/ENSG00000106261</td>
</tr>
<tr class="odd">
<td><a href="GO:0034654" class="uri">GO:0034654</a></td>
<td align="left"><a href="GO:0034654" class="uri">GO:0034654</a></td>
<td align="left">nucleobase-containing compound biosynthetic process</td>
<td align="right">52</td>
<td align="right">0.5455543</td>
<td align="right">1.605871</td>
<td align="right">0.0004003</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">107</td>
<td align="left">tags=63%, list=38%, signal=48%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000114503/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000127586/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000170325/ENSG00000106261/ENSG00000152402/ENSG00000185122</td>
</tr>
<tr class="even">
<td><a href="GO:0051171" class="uri">GO:0051171</a></td>
<td align="left"><a href="GO:0051171" class="uri">GO:0051171</a></td>
<td align="left">regulation of nitrogen compound metabolic process</td>
<td align="right">83</td>
<td align="right">0.5017512</td>
<td align="right">1.520043</td>
<td align="right">0.0005000</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">110</td>
<td align="left">tags=58%, list=39%, signal=50%</td>
<td align="left">ENSG00000173575/ENSG00000100985/ENSG00000163359/ENSG00000102974/ENSG00000171862/ENSG00000075856/ENSG00000167513/ENSG00000164758/ENSG00000128000/ENSG00000078401/ENSG00000163376/ENSG00000181007/ENSG00000159388/ENSG00000107140/ENSG00000094916/ENSG00000152454/ENSG00000138166/ENSG00000121769/ENSG00000124279/ENSG00000099958/ENSG00000061936/ENSG00000133116/ENSG00000114503/ENSG00000107371/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000154127/ENSG00000196150/ENSG00000172985/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000127586/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000109062/ENSG00000170325/ENSG00000106261/ENSG00000185122/ENSG00000034713/ENSG00000125457</td>
</tr>
<tr class="odd">
<td><a href="GO:0018130" class="uri">GO:0018130</a></td>
<td align="left"><a href="GO:0018130" class="uri">GO:0018130</a></td>
<td align="left">heterocycle biosynthetic process</td>
<td align="right">53</td>
<td align="right">0.5365747</td>
<td align="right">1.581872</td>
<td align="right">0.0005004</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">107</td>
<td align="left">tags=62%, list=38%, signal=47%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000114503/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000127586/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000170325/ENSG00000106261/ENSG00000152402/ENSG00000185122</td>
</tr>
<tr class="even">
<td><a href="GO:0019438" class="uri">GO:0019438</a></td>
<td align="left"><a href="GO:0019438" class="uri">GO:0019438</a></td>
<td align="left">aromatic compound biosynthetic process</td>
<td align="right">53</td>
<td align="right">0.5365747</td>
<td align="right">1.581872</td>
<td align="right">0.0005004</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">107</td>
<td align="left">tags=62%, list=38%, signal=47%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000114503/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000127586/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000170325/ENSG00000106261/ENSG00000152402/ENSG00000185122</td>
</tr>
<tr class="odd">
<td><a href="GO:0006351" class="uri">GO:0006351</a></td>
<td align="left"><a href="GO:0006351" class="uri">GO:0006351</a></td>
<td align="left">transcription, DNA-templated</td>
<td align="right">46</td>
<td align="right">0.5573156</td>
<td align="right">1.624813</td>
<td align="right">0.0005009</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">101</td>
<td align="left">tags=65%, list=36%, signal=50%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000114503/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000170325/ENSG00000106261</td>
</tr>
<tr class="even">
<td><a href="GO:0032774" class="uri">GO:0032774</a></td>
<td align="left"><a href="GO:0032774" class="uri">GO:0032774</a></td>
<td align="left">RNA biosynthetic process</td>
<td align="right">46</td>
<td align="right">0.5573156</td>
<td align="right">1.624813</td>
<td align="right">0.0005009</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">101</td>
<td align="left">tags=65%, list=36%, signal=50%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000114503/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000170325/ENSG00000106261</td>
</tr>
<tr class="odd">
<td><a href="GO:0097659" class="uri">GO:0097659</a></td>
<td align="left"><a href="GO:0097659" class="uri">GO:0097659</a></td>
<td align="left">nucleic acid-templated transcription</td>
<td align="right">46</td>
<td align="right">0.5573156</td>
<td align="right">1.624813</td>
<td align="right">0.0005009</td>
<td align="right">0.0229936</td>
<td align="right">0.0193151</td>
<td align="right">101</td>
<td align="left">tags=65%, list=36%, signal=50%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000114503/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000170325/ENSG00000106261</td>
</tr>
<tr class="even">
<td><a href="GO:0044260" class="uri">GO:0044260</a></td>
<td align="left"><a href="GO:0044260" class="uri">GO:0044260</a></td>
<td align="left">cellular macromolecule metabolic process</td>
<td align="right">127</td>
<td align="right">0.4959084</td>
<td align="right">1.521080</td>
<td align="right">0.0005999</td>
<td align="right">0.0233077</td>
<td align="right">0.0195789</td>
<td align="right">111</td>
<td align="left">tags=52%, list=40%, signal=57%</td>
<td align="left">ENSG00000173575/ENSG00000100985/ENSG00000163359/ENSG00000138641/ENSG00000102974/ENSG00000139718/ENSG00000171862/ENSG00000075856/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000167461/ENSG00000128000/ENSG00000078401/ENSG00000136930/ENSG00000163376/ENSG00000181007/ENSG00000159388/ENSG00000116183/ENSG00000107140/ENSG00000094916/ENSG00000152454/ENSG00000138166/ENSG00000124279/ENSG00000099958/ENSG00000178952/ENSG00000133116/ENSG00000114503/ENSG00000163833/ENSG00000107371/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000180776/ENSG00000154127/ENSG00000163923/ENSG00000196150/ENSG00000166479/ENSG00000172985/ENSG00000128731/ENSG00000150593/ENSG00000100626/ENSG00000204519/ENSG00000037241/ENSG00000090971/ENSG00000164342/ENSG00000185504/ENSG00000151623/ENSG00000163602/ENSG00000162434/ENSG00000153187/ENSG00000127586/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000182899/ENSG00000109062/ENSG00000170325/ENSG00000106261/ENSG00000113643/ENSG00000107789/ENSG00000185122/ENSG00000034713/ENSG00000125457/ENSG00000138081</td>
</tr>
<tr class="odd">
<td><a href="GO:0090304" class="uri">GO:0090304</a></td>
<td align="left"><a href="GO:0090304" class="uri">GO:0090304</a></td>
<td align="left">nucleic acid metabolic process</td>
<td align="right">66</td>
<td align="right">0.5231485</td>
<td align="right">1.565133</td>
<td align="right">0.0006000</td>
<td align="right">0.0233077</td>
<td align="right">0.0195789</td>
<td align="right">107</td>
<td align="left">tags=61%, list=38%, signal=49%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000075856/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000124279/ENSG00000061936/ENSG00000114503/ENSG00000107371/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000128731/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000185504/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000127586/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000182899/ENSG00000170325/ENSG00000106261/ENSG00000113643/ENSG00000185122</td>
</tr>
<tr class="even">
<td><a href="GO:0006139" class="uri">GO:0006139</a></td>
<td align="left"><a href="GO:0006139" class="uri">GO:0006139</a></td>
<td align="left">nucleobase-containing compound metabolic process</td>
<td align="right">72</td>
<td align="right">0.5131211</td>
<td align="right">1.543387</td>
<td align="right">0.0009001</td>
<td align="right">0.0284461</td>
<td align="right">0.0238953</td>
<td align="right">107</td>
<td align="left">tags=58%, list=38%, signal=49%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000075856/ENSG00000167513/ENSG00000164758/ENSG00000125484/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000124279/ENSG00000061936/ENSG00000114503/ENSG00000130313/ENSG00000107371/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000128731/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000185504/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000127586/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000182899/ENSG00000170325/ENSG00000106261/ENSG00000113643/ENSG00000152402/ENSG00000185122</td>
</tr>
<tr class="odd">
<td><a href="GO:1903506" class="uri">GO:1903506</a></td>
<td align="left"><a href="GO:1903506" class="uri">GO:1903506</a></td>
<td align="left">regulation of nucleic acid-templated transcription</td>
<td align="right">44</td>
<td align="right">0.5414930</td>
<td align="right">1.575646</td>
<td align="right">0.0009013</td>
<td align="right">0.0284461</td>
<td align="right">0.0238953</td>
<td align="right">101</td>
<td align="left">tags=64%, list=36%, signal=48%</td>
<td align="left">ENSG00000173575/ENSG00000102974/ENSG00000171862/ENSG00000167513/ENSG00000164758/ENSG00000128000/ENSG00000078401/ENSG00000181007/ENSG00000159388/ENSG00000094916/ENSG00000152454/ENSG00000161057/ENSG00000168743/ENSG00000157404/ENSG00000159208/ENSG00000196150/ENSG00000150593/ENSG00000204519/ENSG00000090971/ENSG00000164342/ENSG00000151623/ENSG00000163602/ENSG00000153187/ENSG00000102984/ENSG00000123609/ENSG00000141646/ENSG00000170325/ENSG00000106261</td>
</tr>
</tbody>
</table>
<pre class="r"><code>gseaplot2(gsego.cc.chimpspecific, geneSetID = c(&quot;GO:0006366&quot;,&quot;GO:0044260&quot;), title=&quot;Chimp eGenes ordered by difference in species rank&quot;, pvalue_table = TRUE)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dotplot(gsego.cc.chimpspecific, font.size=8, showCategory=38)</code></pre>
<p><img src="figure/20190521_eQTL_CrossSpeciesEnrichment.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /> Alright. Great. that analysis recapitulated the gene overlap analysis of [foreground=Chimp-specific-eGenes; background=all-chimp-eGenes] in the sense that terms like ‘gene transcription’ pop up.</p>
<p>Now let’s try the reciprocal; where we look for what is enriched in human-specific eGenes, (again where the metric to order the gene list is a difference in ranks.)… Still a work-in-progress</p>
<pre class="r"><code>Chimp_OrderedGenes &lt;- eQTLs %&gt;%
    group_by(gene) %&gt;% 
    dplyr::slice(which.min(qvalue)) %&gt;%
    filter(gene %in% GtexHeartEgenes$chimp_id) %&gt;%
    filter(qvalue &lt;0.1) %&gt;%
    left_join(GtexHeartEgenes, by=c(&quot;gene&quot;=&quot;chimp_id&quot;)) %&gt;%
    dplyr::select(gene, qvalue, qval) %&gt;% as.data.frame() %&gt;%
    mutate(ChimpRank = dense_rank(qvalue)) %&gt;%
    mutate(HumanRank = dense_rank(qval)) %&gt;%
    mutate(RankDifference = HumanRank-ChimpRank) %&gt;%
    arrange(desc(RankDifference))
GeneList &lt;- Chimp_OrderedGenes$RankDifference
names(GeneList) &lt;- as.character(ChimpToHuman.ID(Chimp_OrderedGenes$gene))
    
gsego.cc.chimpspecific &lt;- gseGO(gene = GeneList,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 maxGSSize = 500,
                 ont = &quot;BP&quot;)
kable(head(as.data.frame(gsego.cc.chimpspecific), 15))
gseaplot2(gsego.cc.chimpspecific, geneSetID = c(&quot;GO:0006357&quot;, &quot;GO:0006355&quot;), title=&quot;Chimp eGenes ordered by difference in species rank&quot;, pvalue_table = TRUE)

dotplot(gsego.cc.chimpspecific, font.size=8, showCategory=35)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] enrichplot_1.2.0       gridExtra_2.3          org.Hs.eg.db_3.7.0    
 [4] AnnotationDbi_1.44.0   IRanges_2.16.0         S4Vectors_0.20.1      
 [7] Biobase_2.42.0         BiocGenerics_0.28.0    clusterProfiler_3.10.1
[10] ggpmisc_0.3.1          data.table_1.12.2      knitr_1.23            
[13] forcats_0.4.0          stringr_1.4.0          dplyr_0.8.1           
[16] purrr_0.3.2            readr_1.3.1            tidyr_0.8.3           
[19] tibble_2.1.3           ggplot2_3.1.1          tidyverse_1.2.1       
[22] plyr_1.8.4            

loaded via a namespace (and not attached):
 [1] nlme_3.1-140        fs_1.3.1            lubridate_1.7.4    
 [4] bit64_0.9-7         progress_1.2.2      RColorBrewer_1.1-2 
 [7] httr_1.4.0          UpSetR_1.4.0        rprojroot_1.3-2    
[10] tools_3.5.1         backports_1.1.4     R6_2.4.0           
[13] DBI_1.0.0           lazyeval_0.2.2      colorspace_1.4-1   
[16] withr_2.1.2         prettyunits_1.0.2   tidyselect_0.2.5   
[19] bit_1.1-14          compiler_3.5.1      git2r_0.25.2       
[22] cli_1.1.0           rvest_0.3.4         xml2_1.2.0         
[25] labeling_0.3        triebeard_0.3.0     scales_1.0.0       
[28] ggridges_0.5.1      digest_0.6.19       rmarkdown_1.13     
[31] DOSE_3.8.2          pkgconfig_2.0.2     htmltools_0.3.6    
[34] highr_0.8           rlang_0.3.4         readxl_1.3.1       
[37] rstudioapi_0.10     RSQLite_2.1.1       gridGraphics_0.4-1 
[40] generics_0.0.2      farver_1.1.0        jsonlite_1.6       
[43] BiocParallel_1.16.6 GOSemSim_2.8.0      magrittr_1.5       
[46] ggplotify_0.0.3     GO.db_3.7.0         Matrix_1.2-17      
[49] Rcpp_1.0.1          munsell_0.5.0       viridis_0.5.1      
[52] stringi_1.4.3       whisker_0.3-2       yaml_2.2.0         
[55] ggraph_1.0.2        MASS_7.3-51.4       qvalue_2.14.1      
[58] grid_3.5.1          blob_1.1.1          ggrepel_0.8.1      
[61] DO.db_2.9           crayon_1.3.4        lattice_0.20-38    
[64] cowplot_0.9.4       haven_2.1.0         splines_3.5.1      
[67] hms_0.4.2           pillar_1.4.1        fgsea_1.8.0        
[70] igraph_1.2.4.1      reshape2_1.4.3      fastmatch_1.1-0    
[73] glue_1.3.1          evaluate_0.14       modelr_0.1.4       
[76] urltools_1.7.3      tweenr_1.0.1        cellranger_1.1.0   
[79] gtable_0.3.0        polyclip_1.10-0     assertthat_0.2.1   
[82] xfun_0.7            ggforce_0.2.2       europepmc_0.3      
[85] broom_0.5.2         viridisLite_0.3.0   rvcheck_0.1.3      
[88] memoise_1.1.0       workflowr_1.4.0    </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
