<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>20191001_GtexAnnotations</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20191001_GtexAnnotations</h1>
<h4 class="author">Ben Fair</h4>
<h4 class="date">10/1/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-10-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree65c8360ca271cb584c648059261c789592f5429ftargetblank65c8360a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/65c8360ca271cb584c648059261c789592f5429f" target="_blank">65c8360</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree65c8360ca271cb584c648059261c789592f5429ftargetblank65c8360a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/20190521_eQTL_CrossSpeciesEnrichment_cache/
    Ignored:    analysis_temp/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/snakemake_workflow/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    docs/assets/.DS_Store

Untracked files:
    Untracked:  analysis/20191001_GtexAnnotations.Rmd
    Untracked:  analysis_temp/20190716_VarianceInsteadOfEgenes_2.Rmd
    Untracked:  docs/figure/20190930_OverdispersionEstimates.Rmd/

Unstaged changes:
    Deleted:    analysis/20190716_VarianceInsteadOfEgenes_2.Rmd
    Modified:   analysis/20190930_OverdispersionEstimates.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>From my dataset of gene expression measurements across 39 chimp individuals (RNA-seq from heart tissue) and ~39 human individuals (~50 RNA-seq sampels randomly chosen from human GTEx datasets, then culled to exclude outliers and lowly sequenced samples) I am interested in quantifying gene expression dispersion within each population. I had <a href="">previously</a> made the observation that gene expression overdispersion is well correlated (<span class="math inline">\(R^2\approx0.5\)</span>) between the two species, even after regressing out the fact that variance is correlated to expression (due to both biological and statstical reasons). This amount of correlation is higher than I would have guessed, given how small the overlap is between eGenes that I could detect. However, I still don’t have a sense of what I should have expected. To gain some intuition on this, I would like to use other GTEx tissues to estimate gene-wise overdispersion parameters, and compare the correlation across tissues. First, let’s peruse the GTEx samples available to us, to pick ~40 reasonable samples per tissue for this analysis.</p>
<pre class="r"><code>library(tidyverse)

Samples &lt;- read.table(&quot;../data/GTExAnnotations/GTEx_Analysis_v8_Annotations_SampleAttributesDS.txt&quot;, header=T, sep=&#39;\t&#39;, quote=&quot;&quot;)
Subjects &lt;- read.table(&quot;../data/GTExAnnotations/GTEx_Analysis_v8_Annotations_SubjectPhenotypesDS.txt&quot;, header=T, sep=&#39;\t&#39;)
GtexColors &lt;- read.table(&quot;../data/GTEx_Analysis_TissueColorCodes.txt&quot;, sep=&#39;\t&#39;, header=T, stringsAsFactors = F) %&gt;% mutate(HEX=paste0(&quot;#&quot;,Color.code))

TissueColorVector &lt;- GtexColors$HEX
names(TissueColorVector)&lt;-GtexColors$Tissue

#Num samples per tissue. Note that I may want to filter out some samples to draw from ones where experimental protocols (like RNA extraction method) were kept reasonably constant
Samples %&gt;%
  filter(SMAFRZE==&quot;RNASEQ&quot;) %&gt;%
  count(SMTSD,SMNABTCHT) %&gt;%
  filter(SMTSD %in% GtexColors$Tissue) %&gt;% mutate(sum=sum(n)) %&gt;%
  ggplot(aes(x=SMTSD, y=n, fill=SMTSD, color=SMNABTCHT)) +
    geom_col(show.legend = T) +
    ylab(&quot;Number Samples&quot;) +
    theme_bw() +
    scale_fill_manual(values=TissueColorVector, guide=FALSE) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5)) +
    theme(legend.position=&quot;top&quot;) +
    guides(color=guide_legend(nrow=2,byrow=TRUE))</code></pre>
<p><img src="figure/20191001_GtexAnnotations.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Histogram of mapping rate
hist(Samples$SMMAPRT)</code></pre>
<p><img src="figure/20191001_GtexAnnotations.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Histogram of rRNA mapping rate
hist(Samples$SMRRNART)</code></pre>
<p><img src="figure/20191001_GtexAnnotations.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Histogram of exonic mapping rate
hist(Samples$SMEXNCRT)</code></pre>
<p><img src="figure/20191001_GtexAnnotations.Rmd/unnamed-chunk-1-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Histogram of mapped library size
hist(log10(Samples$SMMPPD))</code></pre>
<p><img src="figure/20191001_GtexAnnotations.Rmd/unnamed-chunk-1-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#RIN score
hist(Samples$SMRIN)</code></pre>
<p><img src="figure/20191001_GtexAnnotations.Rmd/unnamed-chunk-1-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#types of library preps
table(Samples$SMGEBTCHT)</code></pre>
<pre><code>
                Affymetrix Expression 
                                  837 
                             ChIP-Seq 
                                  356 
                            DroNc-seq 
                                   20 
   Exome Express (Agilent SureSelect) 
                                  100 
       Illumina Human Exome SNP Array 
                                  189 
              Illumina OMNI SNP Array 
                                  462 
PCR-Free 30x Coverage WGS v1 (HiSeqX) 
                                  249 
    PCR+ 30x Coverage WGS (HiSeq2000) 
                                   79 
 PCR+ 30x Coverage WGS v1 (HiSeq2000) 
                                    1 
    PCR+ 30x Coverage WGS v1 (HiSeqX) 
                                    1 
    PCR+ 30x Coverage WGS v2 (HiSeqX) 
                                  579 
                              RIP-Seq 
                                  226 
   Standard Exome Sequencing v2 (ICE) 
                                   80 
   Standard Exome Sequencing v3 (ICE) 
                                  510 
   Standard Exome Sequencing v5 (ICE) 
                                  315 
                             Tech Dev 
                                   24 
      Tech Dev - Large Insert RNA-seq 
                                   65 
                            TruSeq.v1 
                                18858 </code></pre>
<pre class="r"><code>#types of nucleic acid preps
table(Samples$SMNABTCHT)</code></pre>
<pre><code>
                                                       
                                                   376 
DNA Extraction from Paxgene-derived Lysate Plate Based 
                                                   117 
      DNA Isolation from Tissue via QIAgen Spin Column 
                                                    87 
      DNA Isolation of Compromised Blood with Autopure 
                                                    30 
  DNA isolation_Whole Blood_QIAGEN Puregene (Autopure) 
                                                    40 
    DNA isolation_Whole Blood_QIAGEN Puregene (Manual) 
                                                  2285 
RNA Extraction from Paxgene-derived Lysate Plate Based 
                                                 11760 
              RNA isolation_PAXgene Blood RNA (Manual) 
                                                   933 
                    RNA isolation_PAXgene Tissue miRNA 
                                                  6387 
             RNA isolation_Trizol Manual (Cell Pellet) 
                                                   936 </code></pre>
<pre class="r"><code>#sample type. RNASEQ type was deemed suitable for GTEx v8 eQTL calling
table(Samples$SMAFRZE)</code></pre>
<pre><code>
        EXCLUDE    OMNI  RNASEQ     WES     WGS 
    602    2700     450   17382     979     838 </code></pre>
<pre class="r"><code>#Now let&#39;s randomly pick ~40 samples that are  TruSeq.v1 and have coverage comparable to my ChimpSamples (~70M mapped reads), and are similar in other metrics (RNA Extraction from Paxgene-derived Lysate Plate Based).



Mode &lt;- function(x) {
  ux &lt;- unique(x)
  ux[which.max(tabulate(match(x, ux)))]
}

# Apply filters, like keeping RNA-extraction method consistent within tissue, minimum mapping rate, etc.
SamplesFilteredForTissuesSuitableForAnalysis &lt;- Samples %&gt;%
  mutate(IndidualID=gsub(&quot;GTEX-(\\w+)-.+&quot;, &quot;\\1&quot;, SAMPID, perl=T)) %&gt;%
  # filter(SMNABTCHT %in% c(&quot;RNA Extraction from Paxgene-derived Lysate Plate Based&quot;)) %&gt;%
  filter(SMAFRZE==&quot;RNASEQ&quot;) %&gt;%
  group_by(SMTSD) %&gt;%
  mutate(c=Mode(SMNABTCHT)) %&gt;%
  filter(quantile(SMMPPD, 0.1)&lt;SMMPPD &amp;
        quantile(SMMAPRT, 0.1)&lt;SMMAPRT &amp;
        quantile(SMRIN, 0.1)&lt;SMRIN) %&gt;%
  ungroup() %&gt;%
  filter(SMNABTCHT==c) %&gt;%
  add_count(SMTSD, name=&quot;CountBeforeSampling&quot;) %&gt;%
  filter(SMTSD %in% GtexColors$Tissue) %&gt;%
  filter(CountBeforeSampling&gt;40)

# Acceptable samples to draw from.
SamplesFilteredForTissuesSuitableForAnalysis %&gt;%
  group_by(SMTSD) %&gt;%
  count(SMTSD, SMNABTCHT) %&gt;%
  ggplot(aes(x=SMTSD, y=n, fill=SMTSD, color=SMNABTCHT)) +
    geom_col(show.legend = T) +
    ylab(&quot;Number Samples&quot;) +
    theme_bw() +
    scale_fill_manual(values=TissueColorVector, guide=FALSE) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5)) +
    theme(legend.position=&quot;top&quot;) +
    guides(color=guide_legend(nrow=2,byrow=TRUE))</code></pre>
<p><img src="figure/20191001_GtexAnnotations.Rmd/unnamed-chunk-1-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Samples drawn at random to keep sample size consistent
set.seed(0)
SuitableSubsample &lt;- SamplesFilteredForTissuesSuitableForAnalysis  %&gt;%
  group_by(SMTSD) %&gt;%
  sample_n(40) %&gt;% ungroup()
SuitableSubsample %&gt;%
  count(SMTSD, SMNABTCHT) %&gt;%
  ggplot(aes(x=SMTSD, y=n, fill=SMTSD, color=SMNABTCHT)) +
    geom_col(show.legend = T) +
    ylab(&quot;Number Samples&quot;) +
    theme_bw() +
    scale_fill_manual(values=TissueColorVector, guide=FALSE) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5)) +
    theme(legend.position=&quot;top&quot;) +
    guides(color=guide_legend(nrow=2,byrow=TRUE))</code></pre>
<p><img src="figure/20191001_GtexAnnotations.Rmd/unnamed-chunk-1-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#How many tissues exist for this analysis
SuitableSubsample %&gt;% distinct(SMTSD) %&gt;% dim()</code></pre>
<pre><code>[1] 48  1</code></pre>
<pre class="r"><code># How many times do the same individuals show up in multiple tissues for this subsample?
SuitableSubsample %&gt;% select(SAMPID, SMTSD, IndidualID) %&gt;%
  count(IndidualID) %&gt;% pull(n) %&gt;% hist(ylab=&quot;#individuals with X tissues in subsample&quot;)</code></pre>
<p><img src="figure/20191001_GtexAnnotations.Rmd/unnamed-chunk-1-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Could be worth also making a subsample of the same 40 people for all tissues.
#Unfortunately that wouldn&#39;t be possible. Would need to consider less tissues, or less individuals.
SamplesFilteredForTissuesSuitableForAnalysis %&gt;%
  count(IndidualID) %&gt;% pull(n) %&gt;% hist(ylab=&quot;#individuals with X tissues in subsample&quot;)</code></pre>
<p><img src="figure/20191001_GtexAnnotations.Rmd/unnamed-chunk-1-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>SuitableSubsample %&gt;% select(SAMPID, SMTSD, IndidualID) %&gt;%
  write.table(&quot;../data/OverdispersionGTExAnalysisSampleList.txt&quot;, sep=&quot;\t&quot;, quote=F, row.names = F)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.4.0   stringr_1.4.0   dplyr_0.8.1     purrr_0.3.2    
[5] readr_1.3.1     tidyr_0.8.3     tibble_2.1.3    ggplot2_3.1.1  
[9] tidyverse_1.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       cellranger_1.1.0 plyr_1.8.4       pillar_1.4.1    
 [5] compiler_3.5.1   git2r_0.25.2     workflowr_1.4.0  tools_3.5.1     
 [9] digest_0.6.19    lubridate_1.7.4  jsonlite_1.6     evaluate_0.14   
[13] nlme_3.1-140     gtable_0.3.0     lattice_0.20-38  pkgconfig_2.0.2 
[17] rlang_0.3.4      cli_1.1.0        rstudioapi_0.10  yaml_2.2.0      
[21] haven_2.1.0      xfun_0.7         withr_2.1.2      xml2_1.2.0      
[25] httr_1.4.0       knitr_1.23       hms_0.4.2        generics_0.0.2  
[29] fs_1.3.1         rprojroot_1.3-2  grid_3.5.1       tidyselect_0.2.5
[33] glue_1.3.1       R6_2.4.0         readxl_1.3.1     rmarkdown_1.13  
[37] modelr_0.1.4     magrittr_1.5     backports_1.1.4  scales_1.0.0    
[41] htmltools_0.3.6  rvest_0.3.4      assertthat_0.2.1 colorspace_1.4-1
[45] labeling_0.3     stringi_1.4.3    lazyeval_0.2.2   munsell_0.5.0   
[49] broom_0.5.2      crayon_1.3.4    </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
