<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>20190919_Overdispersion</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20190919_Overdispersion</h1>
<h4 class="author">Ben Fair</h4>
<h4 class="date">9/19/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-09-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree2ee2c50c93ba3c547976b6a805ff6ecb3e2eb0d6targetblank2ee2c50a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/2ee2c50c93ba3c547976b6a805ff6ecb3e2eb0d6" target="_blank">2ee2c50</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree2ee2c50c93ba3c547976b6a805ff6ecb3e2eb0d6targetblank2ee2c50a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/20190521_eQTL_CrossSpeciesEnrichment_cache/
    Ignored:    analysis/figure/
    Ignored:    analysis_temp/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/snakemake_workflow/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    docs/assets/.DS_Store

Untracked files:
    Untracked:  analysis/20190919_IsNegBinomialFitCallCorrectingForGeneLengthProperly.Rmd
    Untracked:  analysis/20190919_Overdispersion.Rmd
    Untracked:  code/CustomFunctions.R
    Untracked:  data/GTEx_Analysis_TissueColorCodes.txt
    Untracked:  data/wgEncodeBroadHmmHsmmHMM.hg19.bed.gz
    Untracked:  docs/figure/20190913_VarianceFromNB_model_Comparison.Rmd/
    Untracked:  docs/figure/20190919_Overdispersion.Rmd/
    Untracked:  output/1000GenesChimpCountTable.tsv
    Untracked:  output/1000GenesChimpGeneLenths.tsv

Unstaged changes:
    Modified:   analysis/20190913_VarianceFromNB_model_Comparison.Rmd
    Modified:   code/snakemake_workflow/Session.vim

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(&quot;edgeR&quot;)
library(stats)
library(corrplot)
library(gplots)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(enrichplot)
library(cowplot)

source(&quot;../code/CustomFunctions.R&quot;)</code></pre>
<p>Pick genes to run overdispersion analysis on… Pick genes that were tested for eGene in both species and in the ortho-exon count table.</p>
<pre class="r"><code>HumanSamplesToDrop &lt;- c(c(&quot;SRR1507229&quot;,&quot;SRR603918&quot;, &quot;SRR1478149&quot;, &quot;SRR598509&quot;, &quot;SRR613186&quot;), c(&quot;SRR1489693&quot;, &quot;SRR598148&quot;, &quot;59167&quot;, &quot;SRR1478900&quot;, &quot;SRR1474730&quot;, &quot;61317&quot;))
ChimpSamplesToDrop &lt;- c(&quot;Little_R&quot;)
CountTableChimpFile &lt;- &#39;../output/PowerAnalysisFullCountTable.Chimp.subread.txt.gz&#39;
CountTableHumanFile &lt;- &#39;../output/PowerAnalysisFullCountTable.Human.subread.txt.gz&#39;
eQTLs &lt;- read.table(gzfile(&quot;../data/PastAnalysesDataToKeep/20190521_eQTLs_250kB_10MAF.txt.gz&quot;), header=T)
# List of chimp tested genes
ChimpTestedGenes &lt;- rownames(read.table(&#39;../output/ExpressionMatrix.un-normalized.txt.gz&#39;, header=T, check.names=FALSE, row.names = 1))

ChimpToHumanGeneMap &lt;- read.table(&quot;../data/Biomart_export.Hsap.Ptro.orthologs.txt.gz&quot;, header=T, sep=&#39;\t&#39;, stringsAsFactors = F)

# Of this ortholog list, how many genes are one2one
OneToOneMap &lt;- ChimpToHumanGeneMap %&gt;%
  filter(Chimpanzee.homology.type==&quot;ortholog_one2one&quot;)

# Read gtex heart egene list
# Only consider those that were tested in both species and are one2one orthologs
GtexHeartEgenes &lt;- read.table(&quot;../data/Heart_Left_Ventricle.v7.egenes.txt.gz&quot;, header=T, sep=&#39;\t&#39;, stringsAsFactors = F) %&gt;%
  mutate(gene_id_stable = gsub(&quot;.\\d+$&quot;,&quot;&quot;,gene_id)) %&gt;%
  filter(gene_id_stable %in% OneToOneMap$Gene.stable.ID) %&gt;%
  mutate(chimp_id = plyr::mapvalues(gene_id_stable, OneToOneMap$Gene.stable.ID,  OneToOneMap$Chimpanzee.gene.stable.ID, warn_missing = F)) %&gt;%
  filter(chimp_id %in% ChimpTestedGenes)


EgenesTested &lt;- gsub(&quot;\\..+&quot;, &quot;&quot;, GtexHeartEgenes$gene_id, perl=T)
length(EgenesTested)</code></pre>
<pre><code>[1] 11586</code></pre>
<pre class="r"><code>GenesInDESet &lt;- read.table(gzfile(CountTableChimpFile), header=T, check.names=FALSE, skip=1)$Geneid
length(GenesInDESet)</code></pre>
<pre><code>[1] 44125</code></pre>
<pre class="r"><code>GeneList &lt;- intersect(as.character(GenesInDESet),EgenesTested)</code></pre>
<p>Load count tables and calculate overdispersion</p>
<pre class="r"><code>CountTables &lt;- GetCountTables(CountTableChimpFile,
                               CountTableHumanFile,
                               0, GeneList, ChimpSampleDrop=ChimpSamplesToDrop, HumanSampleDrop = HumanSamplesToDrop)

NumRowsToAnalyze=length(GeneList)
Chimp.NB.fit.parameters&lt;-GetParameterEstimatesOfUnderlyingGamma_lengthAdjusted_FromTable(CountTables$Chimp$Counts[1:NumRowsToAnalyze,], CountTables$Chimp$GeneLengths[1:NumRowsToAnalyze])
Human.NB.fit.parameters&lt;-GetParameterEstimatesOfUnderlyingGamma_lengthAdjusted_FromTable(CountTables$Human$Counts[1:NumRowsToAnalyze,], CountTables$Human$GeneLengths[1:NumRowsToAnalyze])</code></pre>
<p>Are overdispersion estimates between species correlated?</p>
<pre class="r"><code>ToPlot &lt;- cbind(Chimp.NB.fit.parameters,
                apply(CountTables$Chimp$log2RPKM, 1, mean),
                Human.NB.fit.parameters,
                apply(CountTables$Human$log2RPKM, 1, mean))

colnames(ToPlot) &lt;- c(&quot;Chimp.Mean.Expression&quot;, &quot;Chimp.Overdispersion&quot;, &quot;Chimp.Mean.Log2RPKM&quot;, &quot;Human.Mean.Expression&quot;, &quot;Human.Overdispersion&quot;, &quot;Human.Mean.Log2RPKM&quot;)

R&lt;-cor(log(ToPlot$Chimp.Overdispersion),log(ToPlot$Human.Overdispersion), use=&quot;complete.obs&quot;)
lb1 &lt;- paste(&quot;~R^2==~&quot;, round(R**2,2))
Overdispersion.Species.Correlation &lt;- ggplot(ToPlot, aes(x=Chimp.Overdispersion, y=Human.Overdispersion)) +
  geom_point(alpha=0.05) +
  scale_x_continuous(trans=&#39;log10&#39;, limits=c(1E-2,10), name=expression(paste(&quot;Chimp overdispersion estimate, &quot;, &quot;1&quot;,&quot;/&quot;, hat(theta)))) +
  scale_y_continuous(trans=&#39;log10&#39;, limits=c(1E-2,10), name=expression(paste(&quot;Human overdispersion estimate, &quot;, &quot;1&quot;,&quot;/&quot;, hat(theta)))) +
  annotate(&quot;text&quot;,x=5,y=0.1, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()</code></pre>
<p>Are overdispersion estimates correlated with mean expression?</p>
<pre class="r"><code>#Plot mu parameter estimate (estimate of log(mean expression)) vs logRPKM
R&lt;-cor(ToPlot$Chimp.Mean.Expression, ToPlot$Chimp.Mean.Log2RPKM, use=&quot;complete.obs&quot;)
lb1 &lt;- paste(&quot;~R^2==~&quot;, round(R**2,2))
ggplot(ToPlot, aes(x=Chimp.Mean.Expression, y=Chimp.Mean.Log2RPKM)) +
  geom_point(alpha=0.05) +
  # scale_x_continuous(name=&quot;Chimp mean expression\n(Factor of RPKM)&quot;) +
  # scale_y_continuous(trans=&#39;log10&#39;, limits=c(1E-2,10), name=expression(paste(&quot;Human overdispersion estimate, &quot;, &quot;1&quot;,&quot;/&quot;, hat(theta)))) +
  annotate(&quot;text&quot;,x=5,y=0.1, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Plot logRPKM vs calculated manually:
RPM&lt;-t(t(CountTables$Chimp$Counts) / colSums(CountTables$Chimp$Counts))
logRPKM&lt;-log((RPM)/CountTables$Chimp$GeneLengths)
MeanLogRPKM &lt;- apply(logRPKM, 1, mean)
plot(MeanLogRPKM, ToPlot$Chimp.Mean.Log2RPKM)</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(MeanLogRPKM, ToPlot$Chimp.Mean.Expression)</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(ToPlot$Chimp.Mean.Expression, ToPlot$Chimp.Mean.Log2RPKM)</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Chimp.NB.fit.parameters.CPM&lt;-GetParameterEstimatesOfUnderlyingGamma_lengthAdjusted_FromTable(CountTables$Chimp$Counts[1:NumRowsToAnalyze,], rep(1,NumRowsToAnalyze))
plot(Chimp.NB.fit.parameters.CPM$mu, log(apply(RPM, 1, mean)))
plot(Chimp.NB.fit.parameters.CPM$mu, log(apply(RPM, 1, mean)))</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-5-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Plot mu parameter estimate between species
R&lt;-cor(ToPlot$Chimp.Mean.Expression, ToPlot$Human.Mean.Expression, use=&quot;complete.obs&quot;)
lb1 &lt;- paste(&quot;~R^2==~&quot;, round(R**2,2))
ggplot(ToPlot, aes(x=Chimp.Mean.Expression, y=Human.Mean.Expression)) +
  geom_point(alpha=0.05) +
  # scale_x_continuous(name=&quot;Chimp mean expression\n(Factor of RPKM)&quot;) +
  # scale_y_continuous(trans=&#39;log10&#39;, limits=c(1E-2,10), name=expression(paste(&quot;Human overdispersion estimate, &quot;, &quot;1&quot;,&quot;/&quot;, hat(theta)))) +
  annotate(&quot;text&quot;,x=Inf,y=-Inf, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-5-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Plot overdispersion vs mu
R&lt;-cor(ToPlot$Chimp.Mean.Expression, log(ToPlot$Chimp.Overdispersion), use=&quot;complete.obs&quot;)
lb1 &lt;- paste(&quot;~R^2==~&quot;, round(R**2,2))
ggplot(ToPlot, aes(x=Chimp.Mean.Expression, y=Chimp.Overdispersion)) +
  geom_point(alpha=0.05) +
  scale_x_continuous(name=&quot;Chimp mean expression\n(Factor of RPKM)&quot;) +
  scale_y_continuous(trans=&#39;log10&#39;, limits=c(1E-2,10), name=expression(paste(&quot;Chimp overdispersion estimate, &quot;, &quot;1&quot;,&quot;/&quot;, hat(theta)))) +
  annotate(&quot;text&quot;,x=5,y=0.1, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-5-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>If I am looking for the difference in overdispersion between species, does it matter whether or not I regress out this mean dependence with a loess curve.</p>
<pre class="r"><code>GetLoessResidual &lt;- function(x, y){
  loess.fit &lt;- loess(y ~ x)
  loess.residual &lt;- y - predict(loess.fit, x)
  return(loess.residual)
}

ToPlot$Chimp.Residual &lt;- GetLoessResidual(ToPlot$Chimp.Mean.Expression, log(ToPlot$Chimp.Overdispersion))
R&lt;-cor(ToPlot$Chimp.Mean.Expression, log(ToPlot$Chimp.Overdispersion), use=&quot;complete.obs&quot;)
lb1 &lt;- paste(&quot;~R^2==~&quot;, round(R**2,2))
ggplot(ToPlot, aes(x=Chimp.Mean.Expression, y=Chimp.Overdispersion, color=Chimp.Residual)) +
  geom_point(alpha=0.05) +
  scale_x_continuous(name=&quot;Chimp mean expression\n(Factor of RPKM)&quot;) +
  scale_y_continuous(trans=&#39;log10&#39;, limits=c(1E-2,10), name=expression(paste(&quot;Chimp overdispersion estimate, &quot;, &quot;1&quot;,&quot;/&quot;, hat(theta)))) +
  scale_colour_gradient2(name = &quot;Residual from loess curve&quot;) +
  annotate(&quot;text&quot;,x=5,y=0.1, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ToPlot$Human.Residual &lt;- GetLoessResidual(ToPlot$Human.Mean.Expression, log(ToPlot$Human.Overdispersion))

ToPlot$DifferenceOfResiduals &lt;- ToPlot$Chimp.Residual - ToPlot$Human.Residual
ToPlot$DifferenceOfLogOverdispersion &lt;- log(ToPlot$Chimp.Overdispersion) - log(ToPlot$Human.Overdispersion)

ggplot(ToPlot, aes(x=DifferenceOfLogOverdispersion, y=DifferenceOfResiduals)) +
  geom_point(alpha=0.05) +
  theme_bw()</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>write.table(CountTables$Chimp$Counts[1:1000,], file=&quot;../output/1000GenesChimpCountTable.tsv&quot;, sep=&#39;\t&#39;, quote=F)

write.table(CountTables$Chimp$GeneLengths[1:1000], file=&quot;../output/1000GenesChimpGeneLenths.tsv&quot;, sep=&#39;\t&#39;, quote=F)</code></pre>
<p>Conclusion: For purposes of comparing species differences in overdispersion, it makes no difference.</p>
<p>Well I am also curious if it makes any difference to do all this negative binomial fitting, compared the simpler method of finding the residual of the trend between mean(log(RPKM)) and std(log(RPKM)).</p>
<pre class="r"><code>ToPlot$Human.SimpleResidual &lt;- GetLoessResidual(ToPlot$Human.Mean.Log2RPKM, sqrt(apply(CountTables$Human$log2RPKM, 1, var)))
ToPlot$Chimp.SimpleResidual &lt;- GetLoessResidual(ToPlot$Chimp.Mean.Log2RPKM, sqrt(apply(CountTables$Chimp$log2RPKM, 1, var)))
ToPlot$DifferenceOfSimpleResiduals &lt;- ToPlot$Chimp.SimpleResidual - ToPlot$Human.SimpleResidual

R&lt;-cor(ToPlot$DifferenceOfSimpleResiduals, ToPlot$DifferenceOfResiduals, use=&quot;complete.obs&quot;)
lb1 &lt;- paste(&quot;~R^2==~&quot;, round(R**2,2))
ggplot(ToPlot, aes(x=DifferenceOfSimpleResiduals, y=DifferenceOfResiduals)) +
  geom_point(alpha=0.05) +
    annotate(&quot;text&quot;,x=Inf,y=-Inf, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now let’s do with GSEA analysis and see if the same words come up as that both methods</p>
<pre class="r"><code># RankedGeneList&lt;-ToPlot$DifferenceOfSimpleResiduals
# names(RankedGeneList) &lt;- rownames(ToPlot)
# SortedGeneList &lt;- sort(RankedGeneList, decreasing=T)
# 
# chimp.high.var.Simple &lt;- gseGO(gene = SortedGeneList,
#                  OrgDb = org.Hs.eg.db,
#                  keyType = &#39;ENSEMBL&#39;,
#                  maxGSSize = 500,
#                  ont = &quot;BP&quot;,
#                  nPerm = 100000)
# as.data.frame(chimp.high.var.Simple) %&gt;% dplyr::select(Description, p.adjust, enrichmentScore) %&gt;% arrange(enrichmentScore) %&gt;% head(20)

RankedGeneList2&lt;-ToPlot$DifferenceOfLogOverdispersion
names(RankedGeneList2) &lt;- rownames(ToPlot)
SortedGeneList2 &lt;- sort(RankedGeneList2, decreasing=T)

chimp.high.var.Overdispersion &lt;- gseGO(gene = SortedGeneList2,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 maxGSSize = 500,
                 ont = &quot;BP&quot;,
                 nPerm = 100000)

as.data.frame(chimp.high.var.Overdispersion) %&gt;% dplyr::select(Description, p.adjust, enrichmentScore) %&gt;% arrange(enrichmentScore) %&gt;% head(20)</code></pre>
<pre><code>                                                      Description
1                                           DNA strand elongation
2     SRP-dependent cotranslational protein targeting to membrane
3                   cotranslational protein targeting to membrane
4  establishment of protein localization to endoplasmic reticulum
5                   protein localization to endoplasmic reticulum
6                            regulation of chromosome segregation
7                                    mitotic spindle organization
8                            mitotic sister chromatid segregation
9                                    sister chromatid segregation
10                                 nuclear chromosome segregation
11                                         chromosome segregation
12                   positive regulation of programmed cell death
13                       positive regulation of apoptotic process
14                              positive regulation of cell death
15                                 regulation of defense response
16                                   myeloid cell differentiation
17                           positive regulation of cell adhesion
18                                            cytokine production
19                              regulation of cytokine production
20                               regulation of cell-cell adhesion
      p.adjust enrichmentScore
1  0.030382696      -0.6748732
2  0.042678917      -0.4899624
3  0.027663509      -0.4885324
4  0.027422514      -0.4661413
5  0.011920527      -0.4643751
6  0.025832043      -0.4316608
7  0.028817843      -0.4199969
8  0.032403046      -0.3635242
9  0.028817843      -0.3489121
10 0.039622713      -0.3094772
11 0.049464386      -0.2924877
12 0.041348890       0.3346695
13 0.041348890       0.3351856
14 0.029862002       0.3362084
15 0.011920527       0.3524547
16 0.049464386       0.3590713
17 0.043164891       0.3605040
18 0.004873167       0.3617984
19 0.006751225       0.3633144
20 0.043164891       0.3651909</code></pre>
<pre class="r"><code>as.data.frame(chimp.high.var.Overdispersion) %&gt;% 
  dplyr::select(Description, p.adjust, enrichmentScore) %&gt;% 
  filter(enrichmentScore&lt;0) %&gt;%
  arrange(p.adjust) %&gt;%
  head(20)</code></pre>
<pre><code>                                                      Description
1                   protein localization to endoplasmic reticulum
2                            regulation of chromosome segregation
3  establishment of protein localization to endoplasmic reticulum
4                   cotranslational protein targeting to membrane
5                                    mitotic spindle organization
6                                    sister chromatid segregation
7                                           DNA strand elongation
8                            mitotic sister chromatid segregation
9                                  nuclear chromosome segregation
10    SRP-dependent cotranslational protein targeting to membrane
11                                         chromosome segregation
     p.adjust enrichmentScore
1  0.01192053      -0.4643751
2  0.02583204      -0.4316608
3  0.02742251      -0.4661413
4  0.02766351      -0.4885324
5  0.02881784      -0.4199969
6  0.02881784      -0.3489121
7  0.03038270      -0.6748732
8  0.03240305      -0.3635242
9  0.03962271      -0.3094772
10 0.04267892      -0.4899624
11 0.04946439      -0.2924877</code></pre>
<p>So either way you measure gene-wise dispersion, the Gene set enrichment for genes that are more dispersed in the chimp population are the generally the same: immune response genes.</p>
<p>Lastly, I am interested in asking what are the overdispersed genes in general. Let’s do another gsea analysis using the average of chimp_overdispersion and human_overdispersion.</p>
<pre class="r"><code>RankedGeneListMeanOverdispersion &lt;- (log(ToPlot$Chimp.Overdispersion) + log(ToPlot$Human.Overdispersion))/2
names(RankedGeneListMeanOverdispersion) &lt;- rownames(ToPlot)
SortedGeneListMeanOverdispersion &lt;- sort(RankedGeneListMeanOverdispersion, decreasing=T)

high.Overdispersion &lt;- gseGO(gene = SortedGeneListMeanOverdispersion,
                 OrgDb = org.Hs.eg.db,
                 keyType = &#39;ENSEMBL&#39;,
                 maxGSSize = 500,
                 ont = &quot;BP&quot;,
                 nPerm = 100000)

high.Overdispersion.simplified &lt;- simplify(high.Overdispersion)

as.data.frame(high.Overdispersion.simplified) %&gt;% dplyr::select(Description, p.adjust, enrichmentScore) %&gt;% arrange(enrichmentScore) %&gt;% head(30)</code></pre>
<pre><code>                                                                               Description
1  positive regulation of RNA polymerase II transcriptional preinitiation complex assembly
2                                                    respiratory chain complex IV assembly
3                             maintenance of protein localization in endoplasmic reticulum
4                                                                   SNARE complex assembly
5   regulation of viral-induced cytoplasmic pattern recognition receptor signaling pathway
6                             negative regulation of proteasomal protein catabolic process
7                                                                 autophagosome maturation
8                                                       mitotic G2/M transition checkpoint
9                                                             iron-sulfur cluster assembly
10                                                         metallo-sulfur cluster assembly
11                                                                   microtubule anchoring
12                                                               mitochondrial translation
13                                                 mitochondrial translational termination
14                                                  mitochondrial translational elongation
15                                                               translational termination
16                                                            COPII-coated vesicle budding
17                                                           mitochondrial gene expression
18                                                vesicle targeting, rough ER to cis-Golgi
19                                                                   COPII vesicle coating
20                                                                   transferrin transport
21       negative regulation of proteolysis involved in cellular protein catabolic process
22                                                  regulation of translational initiation
23                                         RNA splicing, via transesterification reactions
24                                        transcription-coupled nucleotide-excision repair
25                                                                G2 DNA damage checkpoint
26                                                                 snRNA metabolic process
27    RNA splicing, via transesterification reactions with bulged adenosine as nucleophile
28                                                          mRNA splicing, via spliceosome
29                                                                         vesicle coating
30                                transcription elongation from RNA polymerase II promoter
       p.adjust enrichmentScore
1  0.0114186286      -0.6772255
2  0.0008308334      -0.6599684
3  0.0235453374      -0.6440720
4  0.0226652078      -0.5780372
5  0.0148321265      -0.5766737
6  0.0004597997      -0.5505311
7  0.0184077684      -0.5208283
8  0.0214784592      -0.5135883
9  0.0304103359      -0.5126187
10 0.0304103359      -0.5126187
11 0.0290608819      -0.5093923
12 0.0004597997      -0.4978704
13 0.0004597997      -0.4953524
14 0.0004597997      -0.4951105
15 0.0004597997      -0.4937862
16 0.0004597997      -0.4916910
17 0.0004597997      -0.4836181
18 0.0004597997      -0.4801745
19 0.0004597997      -0.4801745
20 0.0214784592      -0.4784191
21 0.0004597997      -0.4769234
22 0.0004597997      -0.4743882
23 0.0004597997      -0.4740993
24 0.0004597997      -0.4736688
25 0.0416584647      -0.4722063
26 0.0004597997      -0.4711948
27 0.0004597997      -0.4711738
28 0.0004597997      -0.4711738
29 0.0004597997      -0.4700438
30 0.0004597997      -0.4692926</code></pre>
<p>Ok now make a nice plot of the results of what genes are more dispersed in humans and which are less:</p>
<pre class="r"><code>GSEA.overdispersion.simplified &lt;- simplify(chimp.high.var.Overdispersion)

GSEA.species.overdispersion.plot &lt;- GSEA.overdispersion.simplified %&gt;% 
  as.data.frame() %&gt;% 
  dplyr::select(Description, p.adjust, enrichmentScore, setSize) %&gt;% 
  top_n(20, abs(enrichmentScore)) %&gt;%
  ggplot(aes(x=enrichmentScore, y=Description, color=p.adjust, size=setSize)) +
  geom_point() +
  xlim(c(-1,1)) +
  xlab(&quot;Enrichment score\nMore overdispered in human&lt;----&gt;More overdispersed in chimp&quot;) +
  scale_colour_gradient(low=&quot;red&quot;, high=&quot;black&quot;) +
  labs(color = &quot;Adjusted P-value&quot;) +
  theme_bw()
Overdispersion.Species.Correlation</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(Overdispersion.Species.Correlation, GSEA.species.overdispersion.plot, labels = c(&#39;A&#39;, &#39;B&#39;), label_size = 12, nrow=2)</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>as.data.frame(high.Overdispersion.simplified) %&gt;% dplyr::select(Description, p.adjust, enrichmentScore, setSize) %&gt;% arrange(enrichmentScore) %&gt;% 
  top_n(20, abs(enrichmentScore)) %&gt;%
  ggplot(aes(x=enrichmentScore, y=Description, color=p.adjust, size=setSize)) +
  geom_point() +
  xlim(c(-1,1)) +
  xlab(&quot;Enrichment score\nMore overdispered in human&lt;----&gt;More overdispersed in chimp&quot;) +
  scale_colour_gradient(low=&quot;red&quot;, high=&quot;black&quot;) +
  labs(color = &quot;Adjusted P-value&quot;) +
  scale_y_discrete(labels = function(x) lapply(strwrap(x, width = 60, simplify = FALSE), paste, collapse=&quot;\n&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/20190919_Overdispersion.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] MASS_7.3-51.4          cowplot_0.9.4          enrichplot_1.2.0      
 [4] org.Hs.eg.db_3.7.0     AnnotationDbi_1.44.0   IRanges_2.16.0        
 [7] S4Vectors_0.20.1       Biobase_2.42.0         BiocGenerics_0.28.0   
[10] clusterProfiler_3.10.1 gplots_3.0.1.1         corrplot_0.84         
[13] edgeR_3.24.3           limma_3.38.3           knitr_1.23            
[16] forcats_0.4.0          stringr_1.4.0          dplyr_0.8.1           
[19] purrr_0.3.2            readr_1.3.1            tidyr_0.8.3           
[22] tibble_2.1.3           ggplot2_3.1.1          tidyverse_1.2.1       

loaded via a namespace (and not attached):
 [1] fgsea_1.8.0         colorspace_1.4-1    ggridges_0.5.1     
 [4] rprojroot_1.3-2     qvalue_2.14.1       fs_1.3.1           
 [7] rstudioapi_0.10     farver_1.1.0        urltools_1.7.3     
[10] ggrepel_0.8.1       bit64_0.9-7         lubridate_1.7.4    
[13] xml2_1.2.0          splines_3.5.1       GOSemSim_2.8.0     
[16] polyclip_1.10-0     jsonlite_1.6        workflowr_1.4.0    
[19] broom_0.5.2         GO.db_3.7.0         ggforce_0.2.2      
[22] compiler_3.5.1      httr_1.4.0          rvcheck_0.1.3      
[25] backports_1.1.4     assertthat_0.2.1    Matrix_1.2-17      
[28] lazyeval_0.2.2      cli_1.1.0           tweenr_1.0.1       
[31] htmltools_0.3.6     prettyunits_1.0.2   tools_3.5.1        
[34] igraph_1.2.4.1      gtable_0.3.0        glue_1.3.1         
[37] reshape2_1.4.3      DO.db_2.9           fastmatch_1.1-0    
[40] Rcpp_1.0.1          cellranger_1.1.0    gdata_2.18.0       
[43] nlme_3.1-140        ggraph_1.0.2        xfun_0.7           
[46] rvest_0.3.4         gtools_3.8.1        DOSE_3.8.2         
[49] europepmc_0.3       scales_1.0.0        hms_0.4.2          
[52] RColorBrewer_1.1-2  yaml_2.2.0          memoise_1.1.0      
[55] gridExtra_2.3       UpSetR_1.4.0        triebeard_0.3.0    
[58] stringi_1.4.3       RSQLite_2.1.1       caTools_1.17.1.2   
[61] BiocParallel_1.16.6 rlang_0.3.4         pkgconfig_2.0.2    
[64] bitops_1.0-6        evaluate_0.14       lattice_0.20-38    
[67] labeling_0.3        bit_1.1-14          tidyselect_0.2.5   
[70] plyr_1.8.4          magrittr_1.5        R6_2.4.0           
[73] generics_0.0.2      DBI_1.0.0           pillar_1.4.1       
[76] haven_2.1.0         withr_2.1.2         modelr_0.1.4       
[79] crayon_1.3.4        KernSmooth_2.23-15  rmarkdown_1.13     
[82] viridis_0.5.1       progress_1.2.2      locfit_1.5-9.1     
[85] grid_3.5.1          readxl_1.3.1        data.table_1.12.2  
[88] blob_1.1.1          git2r_0.25.2        digest_0.6.19      
[91] gridGraphics_0.4-1  munsell_0.5.0       viridisLite_0.3.0  
[94] ggplotify_0.0.3    </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
