<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>20190412_Check_eQTLs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20190412_Check_eQTLs</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-04-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree680e39ef06e316c56ecf87264155d5f31881839ctargetblank680e39ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/680e39ef06e316c56ecf87264155d5f31881839c" target="_blank">680e39e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree680e39ef06e316c56ecf87264155d5f31881839ctargetblank680e39ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/20190327_MakeCovariateFiles.Rmd
    Untracked:  analysis/20190421_RegressOutRNASeqPCs.Rmd
    Untracked:  docs/figure/20190424_Check_eQTLs.Rmd/

Unstaged changes:
    Modified:   analysis/20190412_Check_eQTLs.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/4c2b69f8e116f4a097dc4a235b944bc467d7b9e6/analysis/20190412_Check_eQTLs.Rmd" target="_blank">4c2b69f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-19
</td>
<td>
a messy commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(data.table)</code></pre>
<p>My first pass at eqtl mapping was as follows: Genotypes filtered for MAF&gt;0.5 &amp; GenotypingRate&gt;0.9. ~10.5M variants passed this filter. Gene expression as log(TPM), filtering for genes with &gt;0 TPM in all samples (~17,500 genes passed this filter). Association testing used the following linear mixed model for each cis-variant-gene-pair (cis definied as &lt;1Mb from gene):</p>
<p><span class="math display">\[
Y =Wα+xβ+u+ε
\]</span></p>
<p>where <span class="math inline">\(Y\)</span> is gene expression as <span class="math inline">\(log(TPM)\)</span>, <span class="math inline">\(W\)</span> covariates include first three genotype principal components (to account for population structure) as well as 3 RNA-seq PCs, Sex, and an intercept. <span class="math inline">\(x\)</span> is coded as 0,1,2, <span class="math inline">\(U \sim MVN(0,\sigma^2 K)\)</span> where <span class="math inline">\(K\)</span> a the kinship matrix calculated by KING-robust and negative cells are thresholded at 0 (This results in a symetric positive definite matrix). This wqs implemented in the R package ‘MatrixEQTL’. This resulted in ~7000 eSNP-gene pairs, across ~350eGenes at FDR&lt;0.1 (Benjamini Hodgeberg correction).</p>
<p>Here I want to check that the results of this analysis are reasonable, starting by checking boxplots of gene expression stratified by genotype for a handful of significant eSNP-gene pairs.</p>
<pre class="r"><code># Read in genotypes for eQTLs
Genotypes &lt;- read.table(&quot;../data/20190410_eqtls.sig_genotypes.raw&quot;, header=T, check.names = F)
# Genotypes[!duplicated(as.list(Genotypes))]
kable(Genotypes[1:10,1:10])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="right">PAT</th>
<th align="right">MAT</th>
<th align="right">SEX</th>
<th align="right">PHENOTYPE</th>
<th align="right">1:8120561:TATGAAGATCA:T_T</th>
<th align="right">1:12231268:G:A_A</th>
<th align="right">1:12260345:TCC:TG_TG</th>
<th align="right">1:12289164:CATCAACTG:GGGCCAC_GGGCCAC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">295</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.502512</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">317</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.933389</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">338</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-1.082890</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">389</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.779304</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">438</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.871399</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.515976</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">462</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.674267</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">476</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-1.087490</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">495</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.220741</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">4x0025</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.796774</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Make sure there aren&#39;t duplicate columns
length(colnames(Genotypes))</code></pre>
<pre><code>[1] 6619</code></pre>
<pre class="r"><code>length(unique(colnames(Genotypes)))</code></pre>
<pre><code>[1] 6619</code></pre>
<pre class="r"><code># Read in eQTLs from MatrixEQTL output (already filtered for FDR&lt;0.1)
eQTLs &lt;- read.table(&quot;../data/20190410_eqtls.txt&quot;, header=T)
kable(head(eQTLs))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SNP</th>
<th align="left">gene</th>
<th align="right">beta</th>
<th align="right">t.stat</th>
<th align="right">p.value</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1:223660942:GTGTTTATTTTAATATTGAG:GTGT</td>
<td align="left">ENSPTRT00000096266.1</td>
<td align="right">19.030028</td>
<td align="right">35.83869</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">11:32876022:CTCCATCTC:CTCCATCTCA</td>
<td align="left">ENSPTRT00000006565.4</td>
<td align="right">-4.833762</td>
<td align="right">-18.96827</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">11:32876022:CTCCATCTC:CTCCATCTCA</td>
<td align="left">ENSPTRT00000006565.4</td>
<td align="right">-4.833762</td>
<td align="right">-18.96827</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">17:30190097:CCACGGCCCGCTAAC:CTGCACCTGGCTGAA</td>
<td align="left">ENSPTRT00000084268.1</td>
<td align="right">16.315581</td>
<td align="right">14.89672</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">17:30191958:GGCAC:AGCGT</td>
<td align="left">ENSPTRT00000084268.1</td>
<td align="right">16.315581</td>
<td align="right">14.89672</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">15:65870468:C:T</td>
<td align="left">ENSPTRT00000013635.7</td>
<td align="right">8.335836</td>
<td align="right">14.76090</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Read in phenotypes, from count table
CountTable &lt;- read.table(gzfile(&#39;../output/CountTable.tpm.txt.gz&#39;), header=T, check.names=FALSE, row.names = 1) %&gt;% 
  t() %&gt;% 
  as.data.frame() %&gt;% 
  rownames_to_column(var = &quot;FID&quot;)
kable(CountTable[1:10, 1:10])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">FID</th>
<th align="right">ENSPTRT00000098376.1</th>
<th align="right">ENSPTRT00000091526.1</th>
<th align="right">ENSPTRT00000091354.1</th>
<th align="right">ENSPTRT00000080032.1</th>
<th align="right">ENSPTRT00000096913.1</th>
<th align="right">ENSPTRT00000079752.1</th>
<th align="right">ENSPTRT00000089496.1</th>
<th align="right">ENSPTRT00000077419.1</th>
<th align="right">ENSPTRT00000090315.1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">4X0095</td>
<td align="right">0.0377317</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="right">0.000000</td>
<td align="right">0.0781316</td>
<td align="right">0.0873579</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
</tr>
<tr class="even">
<td align="left">4X0212</td>
<td align="right">0.2722200</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.2363460</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
</tr>
<tr class="odd">
<td align="left">4X0267</td>
<td align="right">0.2832680</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
</tr>
<tr class="even">
<td align="left">4X0333</td>
<td align="right">0.0768681</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
</tr>
<tr class="odd">
<td align="left">4X0339</td>
<td align="right">0.0241551</td>
<td align="right">0.0279624</td>
<td align="right">0</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
<td align="right">0.0676831</td>
<td align="right">0.000000</td>
</tr>
<tr class="even">
<td align="left">4X0354</td>
<td align="right">0.4893560</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
</tr>
<tr class="odd">
<td align="left">4X0357</td>
<td align="right">0.1495150</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
</tr>
<tr class="even">
<td align="left">4X0550</td>
<td align="right">0.0338373</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
</tr>
<tr class="odd">
<td align="left">4x0025</td>
<td align="right">0.1593970</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="right">0.000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.176351</td>
<td align="right">0.0000000</td>
<td align="right">0.208957</td>
</tr>
<tr class="even">
<td align="left">4x0043</td>
<td align="right">0.2715600</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="right">0.126819</td>
<td align="right">0.0937206</td>
<td align="right">0.7335140</td>
<td align="right">0.000000</td>
<td align="right">0.1268190</td>
<td align="right">0.889987</td>
</tr>
</tbody>
</table>
<pre class="r"><code>MergedData &lt;- left_join(Genotypes, CountTable, by=&quot;FID&quot;)</code></pre>
<pre><code>Warning: Column `FID` joining factor and character vector, coercing into
character vector</code></pre>
<pre class="r"><code>#eqtls, ordered from most significant at top
kable(head(eQTLs))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SNP</th>
<th align="left">gene</th>
<th align="right">beta</th>
<th align="right">t.stat</th>
<th align="right">p.value</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1:223660942:GTGTTTATTTTAATATTGAG:GTGT</td>
<td align="left">ENSPTRT00000096266.1</td>
<td align="right">19.030028</td>
<td align="right">35.83869</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">11:32876022:CTCCATCTC:CTCCATCTCA</td>
<td align="left">ENSPTRT00000006565.4</td>
<td align="right">-4.833762</td>
<td align="right">-18.96827</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">11:32876022:CTCCATCTC:CTCCATCTCA</td>
<td align="left">ENSPTRT00000006565.4</td>
<td align="right">-4.833762</td>
<td align="right">-18.96827</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">17:30190097:CCACGGCCCGCTAAC:CTGCACCTGGCTGAA</td>
<td align="left">ENSPTRT00000084268.1</td>
<td align="right">16.315581</td>
<td align="right">14.89672</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">17:30191958:GGCAC:AGCGT</td>
<td align="left">ENSPTRT00000084268.1</td>
<td align="right">16.315581</td>
<td align="right">14.89672</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">15:65870468:C:T</td>
<td align="left">ENSPTRT00000013635.7</td>
<td align="right">8.335836</td>
<td align="right">14.76090</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code># 7682 eQTLs at FDR&lt;0.1
dim(eQTLs)</code></pre>
<pre><code>[1] 7682    6</code></pre>
<pre class="r"><code># betas for these
hist(eQTLs$beta)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Expression box plot, stratified by genotype. For a fe of top of the list snp-gene pairs (most significant)
data.frame(Genotype = MergedData$`1:223660942:GTGTTTATTTTAATATTGAG:GTGT`,
    Phenotype = MergedData$ENSPTRT00000096266.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data.frame(Genotype = MergedData$`17:30190097:CCACGGCCCGCTAAC:CTGCACCTGGCTGAA`,
    Phenotype = MergedData$ENSPTRT00000084268.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data.frame(Genotype = MergedData$`15:65870468:C:T`,
    Phenotype = MergedData$ENSPTRT00000013635.7,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-4.png" width="672" style="display: block; margin: auto;" /> Interpretation: This is concerning; these top hits seem to be all false positives, probably driven by genotyping errors as evidenced by a huge excess of heterozygotes. I checked the raw DNA-sequencing at some of these loci (not shown) and there doesn’t seem to be anything obviously wrong with the alignment or variant calling (~50% of non-duplicate, reads without any excess of mismatches and no strand bias indicate a mismatch/true snp at these locations). Paralogous genes can cause mismapping and false positive variant calls at positions like this. Probably the best way to filter them out will be to apply a Hardy-weinberg filter. This positions will deviate from HWE with very extreme Pvals, so applying a very stringent HWE filter (that still allows for some deviation from HWE, as expected because of population substructure and related individuals) may get rid of these problematic genotypes.</p>
<p>Let’s check some randomly sampled eqtls to see if all of the eqtls are like this.</p>
<pre class="r"><code>set.seed(80)
RandomSampleOfEqtls &lt;- eQTLs %&gt;% sample_n(10) %&gt;% select(SNP, gene, beta)
kable(RandomSampleOfEqtls)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SNP</th>
<th align="left">gene</th>
<th align="right">beta</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">3:64175024:T:C</td>
<td align="left">ENSPTRT00000097126.1</td>
<td align="right">-6.5789324</td>
</tr>
<tr class="even">
<td align="left">8:41999256:GCTGTGTTT:G</td>
<td align="left">ENSPTRT00000076576.1</td>
<td align="right">-0.8351890</td>
</tr>
<tr class="odd">
<td align="left">3:50210729:C:T</td>
<td align="left">ENSPTRT00000098400.1</td>
<td align="right">-1.7838139</td>
</tr>
<tr class="even">
<td align="left">12:65459594:T:G</td>
<td align="left">ENSPTRT00000048028.3</td>
<td align="right">-2.3928804</td>
</tr>
<tr class="odd">
<td align="left">12:65707861:T:C</td>
<td align="left">ENSPTRT00000048028.3</td>
<td align="right">-1.9414605</td>
</tr>
<tr class="even">
<td align="left">7:145137540:C:T</td>
<td align="left">ENSPTRT00000036621.4</td>
<td align="right">-5.5667535</td>
</tr>
<tr class="odd">
<td align="left">1:131341766:A:C</td>
<td align="left">ENSPTRT00000092453.1</td>
<td align="right">-0.9729193</td>
</tr>
<tr class="even">
<td align="left">12:66854411:A:G</td>
<td align="left">ENSPTRT00000048028.3</td>
<td align="right">-1.9414605</td>
</tr>
<tr class="odd">
<td align="left">12:64907042:G:A</td>
<td align="left">ENSPTRT00000048028.3</td>
<td align="right">-1.9414605</td>
</tr>
<tr class="even">
<td align="left">3:157318426:G:A</td>
<td align="left">ENSPTRT00000029000.5</td>
<td align="right">1.0544365</td>
</tr>
</tbody>
</table>
<pre class="r"><code>data.frame(Genotype = MergedData$`1:53393622:A:G`,
    Phenotype = MergedData$ENSPTRT00000108879.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-for-random-eqtls-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data.frame(Genotype = MergedData$`11:85739664:G:A`,
    Phenotype = MergedData$ENSPTRT00000089773.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-for-random-eqtls-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data.frame(Genotype = MergedData$`2A:70424303:C:G`,
    Phenotype = MergedData$ENSPTRT00000079798.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-for-random-eqtls-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data.frame(Genotype = MergedData$`16:58460835:C:T`,
    Phenotype = MergedData$ENSPTRT00000087116.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-for-random-eqtls-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data.frame(Genotype = MergedData$`10:124136336:C:A`,
    Phenotype = MergedData$ENSPTRT00000103391.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-for-random-eqtls-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data.frame(Genotype = MergedData$`16:58398631:C:A`,
    Phenotype = MergedData$ENSPTRT00000087116.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-for-random-eqtls-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># same gene as previous, snp in LD with previous plot
data.frame(Genotype = MergedData$`16:59259665:G:C`,
    Phenotype = MergedData$ENSPTRT00000087116.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-for-random-eqtls-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data.frame(Genotype = MergedData$`15:32018636:G:A`,
    Phenotype = MergedData$ENSPTRT00000081598.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-for-random-eqtls-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data.frame(Genotype = MergedData$`12:66710269:G:A`,
    Phenotype = MergedData$ENSPTRT00000048028.3,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-for-random-eqtls-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data.frame(Genotype = MergedData$`1:131341766:A:C`,
    Phenotype = MergedData$ENSPTRT00000092453.1,
    FID=MergedData$FID) %&gt;%
ggplot(aes(x=factor(Genotype), y=Phenotype, label=FID)) +
geom_boxplot(outlier.shape = NA) +
geom_text(position=position_jitter(width=0.25), alpha=1, size=2) +
scale_y_continuous(name=&quot;TPM&quot;, trans=&quot;log10&quot;)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/checks-for-random-eqtls-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>Interpretation: At least the first of the randomly selected eqtls seems reasonable. The others seem to be false positives, often driven by gene expression measurement for the MD_And sample. Perhaps this sample should be discarded completely, or alternatively some filtering for outlier gene expression measurements should be implemented on a gene-by-gene basis. This is not an extreme outlier for gene expression on the first PC plots, so another explanation is that this sample might be an outlier in the sense that it has very low heterozygosity, resulting in false positive calls at the gene-snp pairs for which this is the sole genotype2 (alt-homozygote) and for which this sample is a gene expression outlier.</p>
<p>Considerations for future: apply a hwe filter, check heterozygosity of samples, consider removing expression outliers on gene-by-gene basis and or normalization methods (eg quantile normalization would also dampen this outlier effect).</p>
<p>In the code block below I will choose a P-value filter for filtering genotypes by hwe as implemented with <code>plink --hwe midp</code>. The data comes from a subset of 1M snps (that passed undergone MAF&gt;0.5 and Geno&gt;0.9 filters) and their HWE pvals calculated with plink. I am looking for a threshold that takes care of these false positive calls that have an excess of heterozygotes</p>
<pre class="r"><code>hwe.results &lt;- read.table(gzfile(&#39;../data/20190416_hwe_tests.uncorrupted.txt.gz&#39;), header=T) %&gt;%
  separate(GENO, c(&quot;HomA1&quot;, &quot;Het&quot;, &quot;HomA2&quot;), sep=&quot;/&quot;, convert=T)
kable(head(hwe.results))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SNP</th>
<th align="right">HomA1</th>
<th align="right">Het</th>
<th align="right">HomA2</th>
<th align="right">P</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1:81:A:AGT</td>
<td align="right">0</td>
<td align="right">32</td>
<td align="right">7</td>
<td align="right">0.0000046</td>
</tr>
<tr class="even">
<td align="left">1:116:T:C</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">35</td>
<td align="right">0.5387000</td>
</tr>
<tr class="odd">
<td align="left">1:2849:TCAT:GCAC</td>
<td align="right">1</td>
<td align="right">9</td>
<td align="right">29</td>
<td align="right">0.3495000</td>
</tr>
<tr class="even">
<td align="left">1:2969:G:A</td>
<td align="right">0</td>
<td align="right">18</td>
<td align="right">21</td>
<td align="right">0.1249000</td>
</tr>
<tr class="odd">
<td align="left">1:2991:C:T</td>
<td align="right">0</td>
<td align="right">20</td>
<td align="right">19</td>
<td align="right">0.0245700</td>
</tr>
<tr class="even">
<td align="left">1:3693:G:C</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">34</td>
<td align="right">0.0973800</td>
</tr>
</tbody>
</table>
<pre class="r"><code>hist(hwe.results$P)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/Choose-hwe-filter-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#qqplot of Pvals
qplot(-log10(1:length(hwe.results$P)/length(hwe.results$P)), -log10(sort(hwe.results$P, decreasing = F))) +
  geom_abline(intercept = 0, slope = 1)</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/Choose-hwe-filter-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot -log10(P) vs num heterozygotes. Plot as hex with log(counts) to see density of points
ggplot(hwe.results, aes(x=Het, y=-log10(P))) +
  stat_binhex(aes(fill=log10(..count..))) +
  xlab(&quot;# heterozygote individuals&quot;)</code></pre>
<pre><code>Warning: Computation failed in `stat_binhex()`:
Package `hexbin` required for `stat_binhex`.
Please install and try again.</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/Choose-hwe-filter-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Interpretation: the vast majority of SNPs would pass a HWE filter thresholded at -log10(P)&lt;7.5. There are also a lot of SNPs that would fail this filter because of excess of heterozygotes. Not yet sure if those are genotyping errors or true snps that just don’t obey HWE (as might be expected with the relatedness and population structure in this cohort). Will stick with -log10(P)&lt;7.5 as the threshold. This would filter out those eqtls I suggested were false positives at the beginning of this analysis (-log10(P) is about 10 for those snps).</p>
<p>Below I will check heterozygosity by samples. Used the following plink command: <code>plink --het --maf 0.05 --bfile {MyFile} --geno --hwe 1e-7.5</code></p>
<p>The F statistic plink uses is: (<observed hom. count> - <expected count>) / (<total observations> - <expected count>))</p>
<pre class="r"><code>heterozygosity &lt;- read.table(&#39;../data/20190416_sample.heterozygosity.txt&#39;, header=T)
kable(head(heterozygosity))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="right">O.HOM.</th>
<th align="right">E.HOM.</th>
<th align="right">N.NM.</th>
<th align="right">F</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">295</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">7947394</td>
<td align="right">7569000</td>
<td align="right">10277099</td>
<td align="right">0.13970</td>
</tr>
<tr class="even">
<td align="left">317</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">7663288</td>
<td align="right">7569000</td>
<td align="right">10276542</td>
<td align="right">0.03489</td>
</tr>
<tr class="odd">
<td align="left">338</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">7921992</td>
<td align="right">7569000</td>
<td align="right">10277191</td>
<td align="right">0.13030</td>
</tr>
<tr class="even">
<td align="left">389</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">7991032</td>
<td align="right">7569000</td>
<td align="right">10276645</td>
<td align="right">0.15590</td>
</tr>
<tr class="odd">
<td align="left">438</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">7855254</td>
<td align="right">7569000</td>
<td align="right">10277192</td>
<td align="right">0.10560</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="left">Pan_troglodytes_ThisStudy</td>
<td align="right">7970770</td>
<td align="right">7569000</td>
<td align="right">10277040</td>
<td align="right">0.14830</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ggplot(heterozygosity, aes(x=FID, y=`F`)) +
  geom_bar(stat=&quot;identity&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5))</code></pre>
<p><img src="figure/20190412_Check_eQTLs.Rmd/check-heterozygosity-1.png" width="672" style="display: block; margin: auto;" /></p>
It is the case that MD_And sample has the highest homozygosity, though it’s F-statistic doesn’t stand out. On the basis of this, probably not justified to completely throw the sample out. Will consider phenotype filters on a gene-by-gene basis or quantile normalization to account for outliers that may drive false associations. <br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.12.0 knitr_1.22        forcats_0.4.0    
 [4] stringr_1.4.0     dplyr_0.8.0.1     purrr_0.3.2      
 [7] readr_1.3.1       tidyr_0.8.2       tibble_2.1.1     
[10] ggplot2_3.1.0     tidyverse_1.2.1  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       highr_0.8        cellranger_1.1.0 plyr_1.8.4      
 [5] pillar_1.3.1     compiler_3.5.1   git2r_0.24.0     workflowr_1.2.0 
 [9] tools_3.5.1      digest_0.6.18    lubridate_1.7.4  jsonlite_1.6    
[13] evaluate_0.13    nlme_3.1-137     gtable_0.3.0     lattice_0.20-38 
[17] pkgconfig_2.0.2  rlang_0.3.3      cli_1.1.0        rstudioapi_0.10 
[21] yaml_2.2.0       haven_2.1.0      xfun_0.6         withr_2.1.2     
[25] xml2_1.2.0       httr_1.4.0       hms_0.4.2        generics_0.0.2  
[29] fs_1.2.6         rprojroot_1.3-2  grid_3.5.1       tidyselect_0.2.5
[33] glue_1.3.1       R6_2.4.0         readxl_1.1.0     rmarkdown_1.11  
[37] modelr_0.1.4     magrittr_1.5     whisker_0.3-2    backports_1.1.3 
[41] scales_1.0.0     htmltools_0.3.6  rvest_0.3.2      assertthat_0.2.1
[45] colorspace_1.4-1 labeling_0.3     stringi_1.4.3    lazyeval_0.2.2  
[49] munsell_0.5.0    broom_0.5.1      crayon_1.3.4    </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
