<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>20190421_RegressOutRNASeqPC</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20190421_RegressOutRNASeqPC</h1>
<h4 class="author"><em>Ben Fair</em></h4>
<h4 class="date"><em>4/21/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-05-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree9a1ab6e1cf9db9868b66c811ca01373c419085detargetblank9a1ab6ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/9a1ab6e1cf9db9868b66c811ca01373c419085de" target="_blank">9a1ab6e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree9a1ab6e1cf9db9868b66c811ca01373c419085detargetblank9a1ab6ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis_temp/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    docs/.DS_Store

Untracked files:
    Untracked:  analysis/20190502_Check_eQTLs.Rmd
    Untracked:  analysis_temp/20190412_Check_eQTLs.Rmd
    Untracked:  data/PastAnalysesDataToKeep/20190428_log10TPM.txt.gz
    Untracked:  docs/assets/
    Untracked:  docs/figure/20190429_RNASeqSTAR_quantifications.Rmd/
    Untracked:  docs/figure/20190502_Check_eQTLs.Rmd/

Unstaged changes:
    Deleted:    analysis/20190412_Check_eQTLs.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/dee079accee958154bf71855e6ceeae231e690c2/analysis/20190421_RegressOutRNASeqPCs.Rmd" target="_blank">dee079a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
<td>
Organized analysis and data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/analysis/20190421_RegressOutRNASeqPCs.Rmd" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
<td>
updated site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/20190421_RegressOutRNASeqPCs.html" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
<td>
updated site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(corrplot)
library(ggfortify)
library(readxl)
library(tidyverse)
library(psych)
library(ggrepel)
library(knitr)
library(reshape2)
library(gplots)
library(matrixStats)</code></pre>
<p>RNA-seq data for each individual was pseudo-mapped/quantified by kallisto and merged into a matrix of TPM values. Genes were filtered by some criteria (see <a href="">link</a>) to generate a table of phenotypes for association testing. Here I will assess if there is still an obvious batch affect (compared to my <a href="">previous PCA analysis</a> of the TPM count table) after I regress out the first (or first few PCs), since I will be included some of these PCs as covariates in the gene expression model.</p>
<pre class="r"><code># Read in count table, filtering out rows that are all zeros. Here the count-table is log transformed TPM values, with genes filtered just as they would be used for association testing
CountTable &lt;- read.table(gzfile(&#39;../data/PastAnalysesDataToKeep/ForAssociationTesting.phenotypes.txt&#39;), header=T, check.names=FALSE, row.names = 1)

kable(head(CountTable))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">295</th>
<th align="right">317</th>
<th align="right">338</th>
<th align="right">389</th>
<th align="right">438</th>
<th align="right">456</th>
<th align="right">462</th>
<th align="right">476</th>
<th align="right">495</th>
<th align="right">4x0025</th>
<th align="right">4x0043</th>
<th align="right">4X0095</th>
<th align="right">4X0212</th>
<th align="right">4X0267</th>
<th align="right">4X0333</th>
<th align="right">4X0339</th>
<th align="right">4X0354</th>
<th align="right">4X0357</th>
<th align="right">4x0430</th>
<th align="right">4x0519</th>
<th align="right">4X0550</th>
<th align="right">4x373</th>
<th align="right">4x523</th>
<th align="right">503</th>
<th align="right">522</th>
<th align="right">529</th>
<th align="right">537</th>
<th align="right">549</th>
<th align="right">554</th>
<th align="right">554_2</th>
<th align="right">558</th>
<th align="right">570</th>
<th align="right">623</th>
<th align="right">676</th>
<th align="right">724</th>
<th align="right">88A020</th>
<th align="right">95A014</th>
<th align="right">Little_R</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSPTRT00000080965.1</td>
<td align="right">-0.5513734</td>
<td align="right">-0.2319409</td>
<td align="right">-0.3714275</td>
<td align="right">-0.2293842</td>
<td align="right">-0.2919820</td>
<td align="right">-0.5319812</td>
<td align="right">-0.4041485</td>
<td align="right">-0.4501398</td>
<td align="right">0.2158047</td>
<td align="right">0.6260347</td>
<td align="right">-0.4391315</td>
<td align="right">-0.4937964</td>
<td align="right">-0.2494888</td>
<td align="right">-0.2264954</td>
<td align="right">0.1223681</td>
<td align="right">-0.6938674</td>
<td align="right">0.1409699</td>
<td align="right">0.0916672</td>
<td align="right">0.0029157</td>
<td align="right">-0.0038353</td>
<td align="right">-0.2298674</td>
<td align="right">0.1747261</td>
<td align="right">0.2190873</td>
<td align="right">-0.8346923</td>
<td align="right">-0.3647676</td>
<td align="right">0.6106510</td>
<td align="right">0.1053955</td>
<td align="right">0.0129360</td>
<td align="right">-0.0639307</td>
<td align="right">0.1289823</td>
<td align="right">0.1163776</td>
<td align="right">0.1578495</td>
<td align="right">-0.3695865</td>
<td align="right">-0.1090986</td>
<td align="right">0.1366528</td>
<td align="right">-0.6287818</td>
<td align="right">-0.1732885</td>
<td align="right">-0.1616099</td>
</tr>
<tr class="even">
<td>ENSPTRT00000018164.6</td>
<td align="right">2.2754389</td>
<td align="right">2.3703465</td>
<td align="right">1.9695374</td>
<td align="right">2.2495018</td>
<td align="right">1.9178839</td>
<td align="right">1.7583578</td>
<td align="right">2.1810320</td>
<td align="right">2.3566092</td>
<td align="right">1.3547960</td>
<td align="right">2.3260770</td>
<td align="right">2.1149547</td>
<td align="right">2.3333568</td>
<td align="right">2.2967431</td>
<td align="right">2.4703343</td>
<td align="right">2.2777523</td>
<td align="right">0.7855623</td>
<td align="right">0.7971378</td>
<td align="right">1.6284265</td>
<td align="right">2.2682804</td>
<td align="right">2.7909994</td>
<td align="right">2.6200565</td>
<td align="right">1.9562169</td>
<td align="right">2.1513633</td>
<td align="right">2.5980121</td>
<td align="right">1.9711977</td>
<td align="right">0.9955122</td>
<td align="right">-0.2054019</td>
<td align="right">1.9623893</td>
<td align="right">2.0983639</td>
<td align="right">2.1229248</td>
<td align="right">0.4223026</td>
<td align="right">2.2237139</td>
<td align="right">2.4056289</td>
<td align="right">1.5448119</td>
<td align="right">2.6051829</td>
<td align="right">2.2298348</td>
<td align="right">2.3531736</td>
<td align="right">0.7679346</td>
</tr>
<tr class="odd">
<td>ENSPTRT00000035805.5</td>
<td align="right">1.3432720</td>
<td align="right">2.4229426</td>
<td align="right">1.6078026</td>
<td align="right">1.2802308</td>
<td align="right">1.2494890</td>
<td align="right">1.1620940</td>
<td align="right">2.4286273</td>
<td align="right">1.6570788</td>
<td align="right">0.8498175</td>
<td align="right">3.1629317</td>
<td align="right">1.5950225</td>
<td align="right">1.1084141</td>
<td align="right">1.5639927</td>
<td align="right">1.6094279</td>
<td align="right">1.9190644</td>
<td align="right">-0.1917987</td>
<td align="right">0.3655247</td>
<td align="right">0.6789823</td>
<td align="right">2.3933669</td>
<td align="right">2.8951216</td>
<td align="right">1.8714960</td>
<td align="right">1.5946856</td>
<td align="right">2.2466431</td>
<td align="right">0.9846744</td>
<td align="right">1.9114474</td>
<td align="right">1.1238251</td>
<td align="right">0.2178737</td>
<td align="right">1.0662062</td>
<td align="right">1.7802856</td>
<td align="right">1.4300887</td>
<td align="right">0.2772377</td>
<td align="right">1.4347964</td>
<td align="right">1.3889658</td>
<td align="right">1.2945518</td>
<td align="right">1.5695176</td>
<td align="right">1.7592466</td>
<td align="right">1.7533928</td>
<td align="right">-0.2615687</td>
</tr>
<tr class="even">
<td>ENSPTRT00000022320.6</td>
<td align="right">1.3718810</td>
<td align="right">1.2747034</td>
<td align="right">1.1767917</td>
<td align="right">1.0743333</td>
<td align="right">0.9092581</td>
<td align="right">1.2058840</td>
<td align="right">1.7636154</td>
<td align="right">1.3177987</td>
<td align="right">1.3897160</td>
<td align="right">2.0859516</td>
<td align="right">0.8509284</td>
<td align="right">1.5593460</td>
<td align="right">1.2497384</td>
<td align="right">1.7362168</td>
<td align="right">1.3808269</td>
<td align="right">0.8704771</td>
<td align="right">1.0551792</td>
<td align="right">1.3740041</td>
<td align="right">1.3269158</td>
<td align="right">1.5905079</td>
<td align="right">0.7792469</td>
<td align="right">1.7448775</td>
<td align="right">0.9564418</td>
<td align="right">0.6788403</td>
<td align="right">1.7180394</td>
<td align="right">0.4965543</td>
<td align="right">0.0468931</td>
<td align="right">1.3171747</td>
<td align="right">1.5570387</td>
<td align="right">1.1037093</td>
<td align="right">0.6737095</td>
<td align="right">3.1708698</td>
<td align="right">1.1041569</td>
<td align="right">2.3108608</td>
<td align="right">1.8350600</td>
<td align="right">1.5592324</td>
<td align="right">1.4415895</td>
<td align="right">-0.4678476</td>
</tr>
<tr class="odd">
<td>ENSPTRT00000100908.1</td>
<td align="right">1.4421358</td>
<td align="right">2.0734008</td>
<td align="right">0.8252932</td>
<td align="right">1.0169213</td>
<td align="right">1.4859383</td>
<td align="right">0.8487611</td>
<td align="right">0.8566172</td>
<td align="right">1.3527169</td>
<td align="right">1.3148950</td>
<td align="right">1.1739238</td>
<td align="right">1.6326035</td>
<td align="right">1.1883159</td>
<td align="right">0.9224676</td>
<td align="right">1.0194067</td>
<td align="right">1.5432319</td>
<td align="right">0.5191687</td>
<td align="right">-0.0180153</td>
<td align="right">1.3612381</td>
<td align="right">1.0813101</td>
<td align="right">1.4847405</td>
<td align="right">1.5176340</td>
<td align="right">1.4019585</td>
<td align="right">1.9364960</td>
<td align="right">1.1383008</td>
<td align="right">0.6865001</td>
<td align="right">-0.0887711</td>
<td align="right">-0.8953931</td>
<td align="right">0.0650946</td>
<td align="right">1.5840504</td>
<td align="right">0.7397874</td>
<td align="right">0.8659023</td>
<td align="right">1.7809041</td>
<td align="right">0.3450921</td>
<td align="right">1.0487881</td>
<td align="right">1.8017577</td>
<td align="right">1.3872814</td>
<td align="right">1.8891720</td>
<td align="right">-0.0173649</td>
</tr>
<tr class="even">
<td>ENSPTRT00000077951.1</td>
<td align="right">-1.0543007</td>
<td align="right">-0.7363178</td>
<td align="right">-1.1656307</td>
<td align="right">-1.0718136</td>
<td align="right">-0.9321051</td>
<td align="right">-1.1157300</td>
<td align="right">-1.4709289</td>
<td align="right">-1.1321205</td>
<td align="right">-1.7666708</td>
<td align="right">-0.1907550</td>
<td align="right">-1.1697292</td>
<td align="right">-0.8051094</td>
<td align="right">-1.0394673</td>
<td align="right">-0.8731307</td>
<td align="right">-1.0497993</td>
<td align="right">-1.6565767</td>
<td align="right">-1.6460708</td>
<td align="right">-1.4270497</td>
<td align="right">-0.6907740</td>
<td align="right">-0.6518344</td>
<td align="right">-0.9158408</td>
<td align="right">-1.1643035</td>
<td align="right">-1.2200819</td>
<td align="right">-1.1188476</td>
<td align="right">-1.1431537</td>
<td align="right">-1.4926465</td>
<td align="right">-3.2429892</td>
<td align="right">-1.0023308</td>
<td align="right">-1.2326055</td>
<td align="right">-0.9340805</td>
<td align="right">-2.3083708</td>
<td align="right">-1.0819587</td>
<td align="right">-0.5124136</td>
<td align="right">-0.3536809</td>
<td align="right">-0.9822676</td>
<td align="right">-1.3268375</td>
<td align="right">-1.3158769</td>
<td align="right">-3.0581078</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Read Covariates file
Covariates &lt;- read.table(&#39;../data/PastAnalysesDataToKeep/3RS_10GT.covariates.txt&#39;, header=T, check.names = F, stringsAsFactors = F, row.names=1)

# Read admixture coefficients (K=4), and first 3 principle components, since some form of population substructure will likely be included in the expression modeling as a covariate.
AdmixtureCoeff &lt;- read.table(&quot;../output/PopulationStructure/Admixture/MergedForAdmixture.4.Q.labelled&quot;) %&gt;% 
  dplyr::rename(Individual.ID=V2) %&gt;%
  select(-V1, -V3, -V4, -V5, -V6) %&gt;%
  dplyr::rename(Admix.Western=V9, Admix.Eastern=V10, Admix.Central=V8, Admix.NigeriaCameroon=V7) #Renaming the admixture clusters after looking at plots with known subspecies
kable(head(AdmixtureCoeff))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Individual.ID</th>
<th align="right">Admix.NigeriaCameroon</th>
<th align="right">Admix.Central</th>
<th align="right">Admix.Western</th>
<th align="right">Admix.Eastern</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">549</td>
<td align="right">1.0e-05</td>
<td align="right">0.000010</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">570</td>
<td align="right">1.3e-05</td>
<td align="right">0.059266</td>
<td align="right">0.940711</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">389</td>
<td align="right">1.0e-05</td>
<td align="right">0.000010</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="right">1.1e-05</td>
<td align="right">0.000010</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">623</td>
<td align="right">1.0e-05</td>
<td align="right">0.000010</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">438</td>
<td align="right">1.0e-05</td>
<td align="right">0.000010</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
</tbody>
</table>
<pre class="r"><code>GenotypePCs &lt;- read.table(&quot;../output/PopulationStructure/pca.eigenvec&quot;, header=T) %&gt;%
  select(IID, PC1, PC2, PC3) %&gt;%
  dplyr::rename(Individual.ID=IID, GenotypePC1=PC1, GenotypePC2=PC2, GenotypePC3=PC3)
kable(head(GenotypePCs))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Individual.ID</th>
<th align="right">GenotypePC1</th>
<th align="right">GenotypePC2</th>
<th align="right">GenotypePC3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">549</td>
<td align="right">-0.1085980</td>
<td align="right">-0.0184943</td>
<td align="right">0.0047314</td>
</tr>
<tr class="even">
<td align="left">570</td>
<td align="right">-0.0945827</td>
<td align="right">-0.0228220</td>
<td align="right">-0.0140236</td>
</tr>
<tr class="odd">
<td align="left">389</td>
<td align="right">-0.1101630</td>
<td align="right">-0.0206061</td>
<td align="right">0.0032451</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="right">-0.1080570</td>
<td align="right">-0.0178108</td>
<td align="right">0.0044597</td>
</tr>
<tr class="odd">
<td align="left">623</td>
<td align="right">-0.1098320</td>
<td align="right">-0.0206678</td>
<td align="right">0.0041903</td>
</tr>
<tr class="even">
<td align="left">438</td>
<td align="right">-0.1081420</td>
<td align="right">-0.0177823</td>
<td align="right">0.0045724</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Read in other metadata
OtherMetadata &lt;- as.data.frame(read_excel(&quot;../data/Metadata.xlsx&quot;))
kable(head(OtherMetadata))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Individual.ID</th>
<th align="left">Source</th>
<th align="left">Individual.Name</th>
<th align="right">Yerkes.ID</th>
<th align="left">Label</th>
<th align="left">Notes</th>
<th align="left">FileID.(Library_Species_CellType_FlowCell)</th>
<th align="left">SX</th>
<th align="right">RNA.Library.prep.batch</th>
<th align="right">RNA.Sequencing.Lane</th>
<th align="right">Sequencing.Barcode</th>
<th align="left">RNA.Extract_date</th>
<th align="left">DNASeq_FastqIdentifier</th>
<th align="right">DNA.library.prep.batch</th>
<th align="right">DNA.Sequencing.Lane</th>
<th align="left">DNA.Sequencin.Barcode</th>
<th align="left">DNA.Extract_date</th>
<th align="right">Age</th>
<th align="left">X__1</th>
<th align="left">Post.mortem.time.interval</th>
<th align="right">RIN</th>
<th align="left">Viral.status</th>
<th align="right">RNA.total.reads.mapped.to.genome</th>
<th align="right">RNA.total.reads.mapping.to.ortho.exons</th>
<th align="left">Subspecies</th>
<th align="left">DOB</th>
<th align="left">DOD</th>
<th align="left">DOB Estimated</th>
<th align="right">Age (DOD-DOB)</th>
<th align="left">OldLibInfo. RIN,RNA-extractdate,RNAbatch</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">295</td>
<td align="left">Yerkes</td>
<td align="left">Duncan</td>
<td align="right">295</td>
<td align="left">295</td>
<td align="left">NA</td>
<td align="left">24_CM_3_L006.bam</td>
<td align="left">M</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">18</td>
<td align="left">2018-10-10</td>
<td align="left">YG3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">40</td>
<td align="left">NA</td>
<td align="left">0.5</td>
<td align="right">7.3</td>
<td align="left">NA</td>
<td align="right">45.67002</td>
<td align="right">17.51562</td>
<td align="left">verus/ellioti</td>
<td align="left">24731</td>
<td align="left">39386</td>
<td align="left">NA</td>
<td align="right">40</td>
<td align="left">6.3,6/14/2016,2</td>
</tr>
<tr class="even">
<td align="left">317</td>
<td align="left">Yerkes</td>
<td align="left">Iyk</td>
<td align="right">317</td>
<td align="left">317</td>
<td align="left">NA</td>
<td align="left">11_CM_3_L004.bam</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="left">2016-06-07</td>
<td align="left">YG2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">44</td>
<td align="left">NA</td>
<td align="left">2.5</td>
<td align="right">7.6</td>
<td align="left">NA</td>
<td align="right">42.75617</td>
<td align="right">17.18811</td>
<td align="left">verus</td>
<td align="left">22859</td>
<td align="left">38832</td>
<td align="left">NA</td>
<td align="right">43</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">338</td>
<td align="left">Yerkes</td>
<td align="left">Maxine</td>
<td align="right">338</td>
<td align="left">338</td>
<td align="left">NA</td>
<td align="left">8_CF_3_L008.bam</td>
<td align="left">F</td>
<td align="right">3</td>
<td align="right">8</td>
<td align="right">6</td>
<td align="left">2016-06-07</td>
<td align="left">YG1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">53</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">7.2</td>
<td align="left">NA</td>
<td align="right">50.52632</td>
<td align="right">19.49295</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40179</td>
<td align="left">Yes</td>
<td align="right">53</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">389</td>
<td align="left">Yerkes</td>
<td align="left">Rogger</td>
<td align="right">389</td>
<td align="left">389</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">M</td>
<td align="right">4</td>
<td align="right">NA</td>
<td align="right">23</td>
<td align="left">2018-10-10</td>
<td align="left">YG39</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">NA</td>
<td align="left">2018-10-01</td>
<td align="right">45</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.7</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">verus</td>
<td align="left">25204</td>
<td align="left">41656</td>
<td align="left">NA</td>
<td align="right">45</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">438</td>
<td align="left">Yerkes</td>
<td align="left">Cheeta</td>
<td align="right">438</td>
<td align="left">438</td>
<td align="left">NA</td>
<td align="left">155_CF_3_L004.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">8</td>
<td align="left">2016-06-22</td>
<td align="left">YG22</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">55</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.6</td>
<td align="left">NA</td>
<td align="right">55.30614</td>
<td align="right">18.06375</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40909</td>
<td align="left">Yes</td>
<td align="right">55</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="left">Yerkes</td>
<td align="left">Mai</td>
<td align="right">456</td>
<td align="left">456</td>
<td align="left">NA</td>
<td align="left">156_CF_3_L001.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="left">2016-06-22</td>
<td align="left">YG23</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">49</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.5</td>
<td align="left">NA</td>
<td align="right">54.00665</td>
<td align="right">20.13760</td>
<td align="left">verus</td>
<td align="left">23377</td>
<td align="left">41275</td>
<td align="left">Yes</td>
<td align="right">49</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Merge all metadata tables
Metadata &lt;- OtherMetadata %&gt;%
  left_join(GenotypePCs, by=c(&quot;Individual.ID&quot;)) %&gt;%
  left_join(AdmixtureCoeff, by=c(&quot;Individual.ID&quot;))
kable(head(Metadata))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Individual.ID</th>
<th align="left">Source</th>
<th align="left">Individual.Name</th>
<th align="right">Yerkes.ID</th>
<th align="left">Label</th>
<th align="left">Notes</th>
<th align="left">FileID.(Library_Species_CellType_FlowCell)</th>
<th align="left">SX</th>
<th align="right">RNA.Library.prep.batch</th>
<th align="right">RNA.Sequencing.Lane</th>
<th align="right">Sequencing.Barcode</th>
<th align="left">RNA.Extract_date</th>
<th align="left">DNASeq_FastqIdentifier</th>
<th align="right">DNA.library.prep.batch</th>
<th align="right">DNA.Sequencing.Lane</th>
<th align="left">DNA.Sequencin.Barcode</th>
<th align="left">DNA.Extract_date</th>
<th align="right">Age</th>
<th align="left">X__1</th>
<th align="left">Post.mortem.time.interval</th>
<th align="right">RIN</th>
<th align="left">Viral.status</th>
<th align="right">RNA.total.reads.mapped.to.genome</th>
<th align="right">RNA.total.reads.mapping.to.ortho.exons</th>
<th align="left">Subspecies</th>
<th align="left">DOB</th>
<th align="left">DOD</th>
<th align="left">DOB Estimated</th>
<th align="right">Age (DOD-DOB)</th>
<th align="left">OldLibInfo. RIN,RNA-extractdate,RNAbatch</th>
<th align="right">GenotypePC1</th>
<th align="right">GenotypePC2</th>
<th align="right">GenotypePC3</th>
<th align="right">Admix.NigeriaCameroon</th>
<th align="right">Admix.Central</th>
<th align="right">Admix.Western</th>
<th align="right">Admix.Eastern</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">295</td>
<td align="left">Yerkes</td>
<td align="left">Duncan</td>
<td align="right">295</td>
<td align="left">295</td>
<td align="left">NA</td>
<td align="left">24_CM_3_L006.bam</td>
<td align="left">M</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">18</td>
<td align="left">2018-10-10</td>
<td align="left">YG3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">40</td>
<td align="left">NA</td>
<td align="left">0.5</td>
<td align="right">7.3</td>
<td align="left">NA</td>
<td align="right">45.67002</td>
<td align="right">17.51562</td>
<td align="left">verus/ellioti</td>
<td align="left">24731</td>
<td align="left">39386</td>
<td align="left">NA</td>
<td align="right">40</td>
<td align="left">6.3,6/14/2016,2</td>
<td align="right">-0.0819553</td>
<td align="right">0.0484042</td>
<td align="right">0.0151794</td>
<td align="right">0.184097</td>
<td align="right">1e-05</td>
<td align="right">0.815883</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">317</td>
<td align="left">Yerkes</td>
<td align="left">Iyk</td>
<td align="right">317</td>
<td align="left">317</td>
<td align="left">NA</td>
<td align="left">11_CM_3_L004.bam</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="left">2016-06-07</td>
<td align="left">YG2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">44</td>
<td align="left">NA</td>
<td align="left">2.5</td>
<td align="right">7.6</td>
<td align="left">NA</td>
<td align="right">42.75617</td>
<td align="right">17.18811</td>
<td align="left">verus</td>
<td align="left">22859</td>
<td align="left">38832</td>
<td align="left">NA</td>
<td align="right">43</td>
<td align="left">NA</td>
<td align="right">-0.1089490</td>
<td align="right">-0.0190248</td>
<td align="right">0.0051753</td>
<td align="right">0.000011</td>
<td align="right">1e-05</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">338</td>
<td align="left">Yerkes</td>
<td align="left">Maxine</td>
<td align="right">338</td>
<td align="left">338</td>
<td align="left">NA</td>
<td align="left">8_CF_3_L008.bam</td>
<td align="left">F</td>
<td align="right">3</td>
<td align="right">8</td>
<td align="right">6</td>
<td align="left">2016-06-07</td>
<td align="left">YG1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">53</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">7.2</td>
<td align="left">NA</td>
<td align="right">50.52632</td>
<td align="right">19.49295</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40179</td>
<td align="left">Yes</td>
<td align="right">53</td>
<td align="left">NA</td>
<td align="right">-0.1081500</td>
<td align="right">-0.0179254</td>
<td align="right">0.0048894</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">389</td>
<td align="left">Yerkes</td>
<td align="left">Rogger</td>
<td align="right">389</td>
<td align="left">389</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">M</td>
<td align="right">4</td>
<td align="right">NA</td>
<td align="right">23</td>
<td align="left">2018-10-10</td>
<td align="left">YG39</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">NA</td>
<td align="left">2018-10-01</td>
<td align="right">45</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.7</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">verus</td>
<td align="left">25204</td>
<td align="left">41656</td>
<td align="left">NA</td>
<td align="right">45</td>
<td align="left">NA</td>
<td align="right">-0.1101630</td>
<td align="right">-0.0206061</td>
<td align="right">0.0032451</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">438</td>
<td align="left">Yerkes</td>
<td align="left">Cheeta</td>
<td align="right">438</td>
<td align="left">438</td>
<td align="left">NA</td>
<td align="left">155_CF_3_L004.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">8</td>
<td align="left">2016-06-22</td>
<td align="left">YG22</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">55</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.6</td>
<td align="left">NA</td>
<td align="right">55.30614</td>
<td align="right">18.06375</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40909</td>
<td align="left">Yes</td>
<td align="right">55</td>
<td align="left">NA</td>
<td align="right">-0.1081420</td>
<td align="right">-0.0177823</td>
<td align="right">0.0045724</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="left">Yerkes</td>
<td align="left">Mai</td>
<td align="right">456</td>
<td align="left">456</td>
<td align="left">NA</td>
<td align="left">156_CF_3_L001.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="left">2016-06-22</td>
<td align="left">YG23</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">49</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.5</td>
<td align="left">NA</td>
<td align="right">54.00665</td>
<td align="right">20.13760</td>
<td align="left">verus</td>
<td align="left">23377</td>
<td align="left">41275</td>
<td align="left">Yes</td>
<td align="right">49</td>
<td align="left">NA</td>
<td align="right">-0.1080570</td>
<td align="right">-0.0178108</td>
<td align="right">0.0044597</td>
<td align="right">0.000011</td>
<td align="right">1e-05</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
</tr>
</tbody>
</table>
<p>Check some of the gene expression phenotype distributions</p>
<pre class="r"><code>#function to plot datapoints as violin plot with individual points labelled. I find this slightly more useful than histograms for plotting phenotype because I can see the distribution and also label points
MyPlot &lt;- function(DataIn){
  ggplot(mapping=aes(x=1, y=DataIn, label=names(DataIn))) +
  geom_violin() +
  geom_text(position=position_jitter(width=0.25), alpha=1, size=2)
}

#Plot expression phenotypes for a few genes
MyPlot(unlist(CountTable[1,]))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-1.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(unlist(CountTable[2,]))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-2">
Past versions of unnamed-chunk-1-2.png
</button>
</p>
<div id="fig-unnamed-chunk-1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-2.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(unlist(CountTable[3,]))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-3">
Past versions of unnamed-chunk-1-3.png
</button>
</p>
<div id="fig-unnamed-chunk-1-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-3.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(unlist(CountTable[4,]))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-4">
Past versions of unnamed-chunk-1-4.png
</button>
</p>
<div id="fig-unnamed-chunk-1-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-4.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(unlist(CountTable[5,]))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-5">
Past versions of unnamed-chunk-1-5.png
</button>
</p>
<div id="fig-unnamed-chunk-1-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-5.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(unlist(CountTable[6,]))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-6">
Past versions of unnamed-chunk-1-6.png
</button>
</p>
<div id="fig-unnamed-chunk-1-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-6.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(unlist(CountTable[7,]))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-7">
Past versions of unnamed-chunk-1-7.png
</button>
</p>
<div id="fig-unnamed-chunk-1-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-7.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(unlist(CountTable[8,]))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-8">
Past versions of unnamed-chunk-1-8.png
</button>
</p>
<div id="fig-unnamed-chunk-1-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-8.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(unlist(CountTable[9,]))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-9">
Past versions of unnamed-chunk-1-9.png
</button>
</p>
<div id="fig-unnamed-chunk-1-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-9.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(unlist(CountTable[10,]))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-10">
Past versions of unnamed-chunk-1-10.png
</button>
</p>
<div id="fig-unnamed-chunk-1-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-10.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#What is distribution of expression levels for all genes in the table
# Histogram of rowMeans. (Arithmetic mean of the log-transformed data)
hist(rowMeans(CountTable))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-11">
Past versions of unnamed-chunk-1-11.png
</button>
</p>
<div id="fig-unnamed-chunk-1-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-1-11.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plot correlation matrix… Should closely resemble the correlation matrix in this <a href="">previous analysis</a>, except that the previous correlation matrix included the top5000 expressed genes, while this one includes the top5000 genes of the genes that are included in the phenotype table for association testing.</p>
<pre class="r"><code>CorMatrix &lt;- CountTable %&gt;%
  mutate(sumVar = rowSums(.)) %&gt;%
  arrange(desc(sumVar)) %&gt;%
  head(5000) %&gt;%
  select(-sumVar) %&gt;%
  scale() %&gt;%
  cor(method = c(&quot;spearman&quot;))

RNAExtractionDate &lt;- as.character(unclass(factor(plyr::mapvalues(row.names(CorMatrix), from=Metadata$Individual.ID, to=Metadata$RNA.Extract_date))))</code></pre>
<pre><code>The following `from` values were not present in `x`: MD_And</code></pre>
<pre class="r"><code>RNA.Library.prep.batch &lt;- as.character(unclass(factor(plyr::mapvalues(row.names(CorMatrix), from=Metadata$Individual.ID, to=Metadata$RNA.Library.prep.batch))))</code></pre>
<pre><code>The following `from` values were not present in `x`: MD_And</code></pre>
<pre class="r"><code># Heatmap of correlation. Row colors for RNA extraction batch, column colors for RNA library prep batch
heatmap.2(CorMatrix, trace=&quot;none&quot;, ColSideColors=RNAExtractionDate, RowSideColors = RNA.Library.prep.batch)</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-2-1.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># What is mean correlation
mean(CorMatrix)</code></pre>
<pre><code>[1] 0.8059961</code></pre>
<p>Now regress out the first RNA-seq PC and recheck correlation matrix</p>
<pre class="r"><code># Regress out PC1

GetResiduals &lt;- function(Y, X){
  df &lt;- data.frame(c(Y),X)
  mylm &lt;- lm(Y ~ ., data=df)
  return(mylm$residuals)
}

ResidualCountTable &lt;- t(apply(CountTable,1,GetResiduals, t(Covariates[c(&quot;PC1.RS&quot;),])))

CorMatrix &lt;- ResidualCountTable %&gt;% as.data.frame() %&gt;%
  mutate(sumVar = rowSums(.)) %&gt;%
  arrange(desc(sumVar)) %&gt;%
  head(5000) %&gt;%
  select(-sumVar) %&gt;%
  scale() %&gt;%
  cor(method = c(&quot;spearman&quot;))

RNAExtractionDate &lt;- as.character(unclass(factor(plyr::mapvalues(row.names(CorMatrix), from=Metadata$Individual.ID, to=Metadata$RNA.Extract_date))))</code></pre>
<pre><code>The following `from` values were not present in `x`: MD_And</code></pre>
<pre class="r"><code>RNA.Library.prep.batch &lt;- as.character(unclass(factor(plyr::mapvalues(row.names(CorMatrix), from=Metadata$Individual.ID, to=Metadata$RNA.Library.prep.batch))))</code></pre>
<pre><code>The following `from` values were not present in `x`: MD_And</code></pre>
<pre class="r"><code># Heatmap of correlation. Row colors for RNA extraction batch, column colors for RNA library prep batch
heatmap.2(CorMatrix, trace=&quot;none&quot;, ColSideColors=RNAExtractionDate, RowSideColors = RNA.Library.prep.batch)</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-3-1.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># What is mean correlation
mean(CorMatrix)</code></pre>
<pre><code>[1] 0.2167298</code></pre>
<p>Interpretation: The regressing first PC1 seems to take away the obvious batch effect that was present in the lower left of the original correlation matrix (The samples that were in that batch no longer cluster together in the lower left), however, regressing out first PC alone still leaves 2 clear clusters of samples.</p>
<p>In my first pass at eQTL mapping I including 3 PCs, let’s see how the correlation matrix clusters after regressing out all of these PCs. (Not iteratively regressing out, but all at once; take residuals from this model Y~PC1+PC2+PC3)</p>
<pre class="r"><code># Regress out PC1, PC2, and PC3

ResidualCountTable &lt;- t(apply(CountTable,1,GetResiduals, t(Covariates[c(&quot;PC1.RS&quot;, &quot;PC2.RS&quot;, &quot;PC3.RS&quot;),])))

CorMatrix &lt;- ResidualCountTable %&gt;% as.data.frame() %&gt;%
  mutate(sumVar = rowSums(.)) %&gt;%
  arrange(desc(sumVar)) %&gt;%
  head(5000) %&gt;%
  select(-sumVar) %&gt;%
  scale() %&gt;%
  cor(method = c(&quot;spearman&quot;))

RNAExtractionDate &lt;- as.character(unclass(factor(plyr::mapvalues(row.names(CorMatrix), from=Metadata$Individual.ID, to=Metadata$RNA.Extract_date))))</code></pre>
<pre><code>The following `from` values were not present in `x`: MD_And</code></pre>
<pre class="r"><code>RNA.Library.prep.batch &lt;- as.character(unclass(factor(plyr::mapvalues(row.names(CorMatrix), from=Metadata$Individual.ID, to=Metadata$RNA.Library.prep.batch))))</code></pre>
<pre><code>The following `from` values were not present in `x`: MD_And</code></pre>
<pre class="r"><code># Heatmap of correlation. Row colors for RNA extraction batch, column colors for RNA library prep batch
heatmap.2(CorMatrix, trace=&quot;none&quot;, ColSideColors=RNAExtractionDate, RowSideColors = RNA.Library.prep.batch)</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-4-1.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># What is mean correlation
mean(CorMatrix)</code></pre>
<pre><code>[1] 0.1172888</code></pre>
<p>Clearly there is still residual structure in the data even after regressing out the top 3 PCs. Now I want to figure out if this residual structure is caused by any technical factors that I could justifiably adjust for (eg RIN score) or if it is genetic. I will now perform PCA on the residuals and see if they correlate to any of the observed technical factors or genetic population substructure that I <a href="">previously</a> looked at before regressing out PCs.</p>
<pre class="r"><code>#First get a list of the same genes that were used in the original PCA analysis (top 2500 expressed genes)

GenesForPCA &lt;- CountTable %&gt;%
  rownames_to_column(&#39;gene&#39;) %&gt;%
  mutate(sumVar = rowSums(select(.,-gene))) %&gt;%
  arrange(desc(sumVar)) %&gt;%
  head(2500) %&gt;%
  pull(gene)

pca_results &lt;- ResidualCountTable %&gt;% as.data.frame() %&gt;%
  rownames_to_column(&#39;gene&#39;) %&gt;%
  filter(gene %in% GenesForPCA) %&gt;%
  column_to_rownames(&#39;gene&#39;) %&gt;%
  mutate(Variance = rowVars(as.matrix(.))) %&gt;%
  filter(Variance&gt;0) %&gt;%
  select(-Variance) %&gt;%
  t() %&gt;%
  prcomp(center=T, scale. = T)

summary(pca_results)</code></pre>
<pre><code>Importance of components:
                           PC1     PC2     PC3     PC4     PC5     PC6
Standard deviation     42.6985 8.70349 7.77771 7.20705 6.58620 6.20179
Proportion of Variance  0.7322 0.03042 0.02429 0.02086 0.01742 0.01545
Cumulative Proportion   0.7322 0.76262 0.78691 0.80777 0.82519 0.84064
                           PC7     PC8     PC9    PC10   PC11    PC12
Standard deviation     5.38525 5.23972 5.02747 4.90940 4.7344 4.44945
Proportion of Variance 0.01165 0.01103 0.01015 0.00968 0.0090 0.00795
Cumulative Proportion  0.85228 0.86331 0.87346 0.88314 0.8921 0.90009
                          PC13    PC14   PC15    PC16    PC17    PC18
Standard deviation     4.25279 4.06900 4.0245 3.97194 3.82780 3.73073
Proportion of Variance 0.00726 0.00665 0.0065 0.00634 0.00588 0.00559
Cumulative Proportion  0.90736 0.91401 0.9205 0.92685 0.93273 0.93832
                          PC19    PC20    PC21   PC22    PC23    PC24
Standard deviation     3.70848 3.59657 3.57082 3.4581 3.40354 3.29980
Proportion of Variance 0.00552 0.00519 0.00512 0.0048 0.00465 0.00437
Cumulative Proportion  0.94384 0.94904 0.95416 0.9590 0.96362 0.96799
                          PC25    PC26    PC27    PC28    PC29    PC30
Standard deviation     3.23734 3.12198 3.02463 2.94894 2.85489 2.75365
Proportion of Variance 0.00421 0.00391 0.00367 0.00349 0.00327 0.00305
Cumulative Proportion  0.97220 0.97611 0.97979 0.98328 0.98655 0.98960
                          PC31    PC32    PC33    PC34     PC35      PC36
Standard deviation     2.67728 2.57653 2.50205 2.41617 2.01e-14 7.464e-15
Proportion of Variance 0.00288 0.00267 0.00251 0.00234 0.00e+00 0.000e+00
Cumulative Proportion  0.99248 0.99514 0.99766 1.00000 1.00e+00 1.000e+00
                            PC37      PC38
Standard deviation     5.681e-15 5.184e-15
Proportion of Variance 0.000e+00 0.000e+00
Cumulative Proportion  1.000e+00 1.000e+00</code></pre>
<pre class="r"><code>#Scree plot
screeplot(pca_results)</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/pca-on-residuals-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-on-residuals-1">
Past versions of pca-on-residuals-1.png
</button>
</p>
<div id="fig-pca-on-residuals-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/pca-on-residuals-1.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Merge with metadata
Merged &lt;- merge(pca_results$x, Metadata, by.x = &quot;row.names&quot;, by.y = &quot;Individual.ID&quot;, all=TRUE)
kable(head(Merged))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Row.names</th>
<th align="right">PC1</th>
<th align="right">PC2</th>
<th align="right">PC3</th>
<th align="right">PC4</th>
<th align="right">PC5</th>
<th align="right">PC6</th>
<th align="right">PC7</th>
<th align="right">PC8</th>
<th align="right">PC9</th>
<th align="right">PC10</th>
<th align="right">PC11</th>
<th align="right">PC12</th>
<th align="right">PC13</th>
<th align="right">PC14</th>
<th align="right">PC15</th>
<th align="right">PC16</th>
<th align="right">PC17</th>
<th align="right">PC18</th>
<th align="right">PC19</th>
<th align="right">PC20</th>
<th align="right">PC21</th>
<th align="right">PC22</th>
<th align="right">PC23</th>
<th align="right">PC24</th>
<th align="right">PC25</th>
<th align="right">PC26</th>
<th align="right">PC27</th>
<th align="right">PC28</th>
<th align="right">PC29</th>
<th align="right">PC30</th>
<th align="right">PC31</th>
<th align="right">PC32</th>
<th align="right">PC33</th>
<th align="right">PC34</th>
<th align="right">PC35</th>
<th align="right">PC36</th>
<th align="right">PC37</th>
<th align="right">PC38</th>
<th align="left">Source</th>
<th align="left">Individual.Name</th>
<th align="right">Yerkes.ID</th>
<th align="left">Label</th>
<th align="left">Notes</th>
<th align="left">FileID.(Library_Species_CellType_FlowCell)</th>
<th align="left">SX</th>
<th align="right">RNA.Library.prep.batch</th>
<th align="right">RNA.Sequencing.Lane</th>
<th align="right">Sequencing.Barcode</th>
<th align="left">RNA.Extract_date</th>
<th align="left">DNASeq_FastqIdentifier</th>
<th align="right">DNA.library.prep.batch</th>
<th align="right">DNA.Sequencing.Lane</th>
<th align="left">DNA.Sequencin.Barcode</th>
<th align="left">DNA.Extract_date</th>
<th align="right">Age</th>
<th align="left">X__1</th>
<th align="left">Post.mortem.time.interval</th>
<th align="right">RIN</th>
<th align="left">Viral.status</th>
<th align="right">RNA.total.reads.mapped.to.genome</th>
<th align="right">RNA.total.reads.mapping.to.ortho.exons</th>
<th align="left">Subspecies</th>
<th align="left">DOB</th>
<th align="left">DOD</th>
<th align="left">DOB Estimated</th>
<th align="right">Age (DOD-DOB)</th>
<th align="left">OldLibInfo. RIN,RNA-extractdate,RNAbatch</th>
<th align="right">GenotypePC1</th>
<th align="right">GenotypePC2</th>
<th align="right">GenotypePC3</th>
<th align="right">Admix.NigeriaCameroon</th>
<th align="right">Admix.Central</th>
<th align="right">Admix.Western</th>
<th align="right">Admix.Eastern</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">295</td>
<td align="right">-50.430132</td>
<td align="right">-2.3098435</td>
<td align="right">2.4369275</td>
<td align="right">-3.9747155</td>
<td align="right">4.1379063</td>
<td align="right">-15.0889535</td>
<td align="right">7.5684829</td>
<td align="right">-12.5989456</td>
<td align="right">9.1740419</td>
<td align="right">-0.1703849</td>
<td align="right">-2.9391712</td>
<td align="right">3.4073188</td>
<td align="right">-5.6781127</td>
<td align="right">-0.0438028</td>
<td align="right">-5.4085602</td>
<td align="right">2.3785468</td>
<td align="right">0.9952793</td>
<td align="right">1.9026303</td>
<td align="right">3.5125444</td>
<td align="right">-4.2340916</td>
<td align="right">-3.6697595</td>
<td align="right">-0.5426480</td>
<td align="right">-0.4313764</td>
<td align="right">-4.2924997</td>
<td align="right">0.7051881</td>
<td align="right">4.2004242</td>
<td align="right">3.6811031</td>
<td align="right">-4.4906383</td>
<td align="right">-0.3435784</td>
<td align="right">-1.1822404</td>
<td align="right">-0.0878580</td>
<td align="right">1.3669424</td>
<td align="right">0.4949519</td>
<td align="right">-0.3753532</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Duncan</td>
<td align="right">295</td>
<td align="left">295</td>
<td align="left">NA</td>
<td align="left">24_CM_3_L006.bam</td>
<td align="left">M</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">18</td>
<td align="left">2018-10-10</td>
<td align="left">YG3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">40</td>
<td align="left">NA</td>
<td align="left">0.5</td>
<td align="right">7.3</td>
<td align="left">NA</td>
<td align="right">45.67002</td>
<td align="right">17.51562</td>
<td align="left">verus/ellioti</td>
<td align="left">24731</td>
<td align="left">39386</td>
<td align="left">NA</td>
<td align="right">40</td>
<td align="left">6.3,6/14/2016,2</td>
<td align="right">-0.0819553</td>
<td align="right">0.0484042</td>
<td align="right">0.0151794</td>
<td align="right">0.184097</td>
<td align="right">1e-05</td>
<td align="right">0.815883</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">317</td>
<td align="right">241.532847</td>
<td align="right">-0.0316741</td>
<td align="right">0.5375320</td>
<td align="right">-0.7206688</td>
<td align="right">0.9010903</td>
<td align="right">-2.1914714</td>
<td align="right">1.0462675</td>
<td align="right">-1.7888688</td>
<td align="right">1.3112178</td>
<td align="right">0.1158945</td>
<td align="right">-0.4524644</td>
<td align="right">0.5905292</td>
<td align="right">-0.7663165</td>
<td align="right">0.0162050</td>
<td align="right">-0.7884457</td>
<td align="right">0.4414294</td>
<td align="right">0.0387473</td>
<td align="right">0.1202932</td>
<td align="right">0.3926779</td>
<td align="right">-0.4153662</td>
<td align="right">-0.4185246</td>
<td align="right">-0.1245901</td>
<td align="right">-0.1342599</td>
<td align="right">-0.7114382</td>
<td align="right">0.1637270</td>
<td align="right">0.6141035</td>
<td align="right">0.5338902</td>
<td align="right">-0.6323517</td>
<td align="right">-0.0367285</td>
<td align="right">-0.1392326</td>
<td align="right">-0.0173996</td>
<td align="right">0.1842514</td>
<td align="right">0.0924651</td>
<td align="right">-0.0445170</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Iyk</td>
<td align="right">317</td>
<td align="left">317</td>
<td align="left">NA</td>
<td align="left">11_CM_3_L004.bam</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="left">2016-06-07</td>
<td align="left">YG2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">44</td>
<td align="left">NA</td>
<td align="left">2.5</td>
<td align="right">7.6</td>
<td align="left">NA</td>
<td align="right">42.75617</td>
<td align="right">17.18811</td>
<td align="left">verus</td>
<td align="left">22859</td>
<td align="left">38832</td>
<td align="left">NA</td>
<td align="right">43</td>
<td align="left">NA</td>
<td align="right">-0.1089490</td>
<td align="right">-0.0190248</td>
<td align="right">0.0051753</td>
<td align="right">0.000011</td>
<td align="right">1e-05</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">338</td>
<td align="right">-2.144211</td>
<td align="right">-0.0669376</td>
<td align="right">-1.5453948</td>
<td align="right">-0.3322771</td>
<td align="right">0.9761652</td>
<td align="right">0.1501650</td>
<td align="right">0.4518967</td>
<td align="right">0.2665345</td>
<td align="right">-0.8105410</td>
<td align="right">-3.2601071</td>
<td align="right">-2.0590783</td>
<td align="right">7.3345467</td>
<td align="right">5.5370030</td>
<td align="right">-15.0025373</td>
<td align="right">2.5559130</td>
<td align="right">-6.5941064</td>
<td align="right">-2.0658545</td>
<td align="right">-10.7197849</td>
<td align="right">-0.2344986</td>
<td align="right">-4.7836382</td>
<td align="right">-2.8914650</td>
<td align="right">3.5479971</td>
<td align="right">-3.6397800</td>
<td align="right">0.5452169</td>
<td align="right">0.2185825</td>
<td align="right">0.5475979</td>
<td align="right">0.2768962</td>
<td align="right">-0.2863866</td>
<td align="right">0.4843528</td>
<td align="right">-0.6661467</td>
<td align="right">0.1334733</td>
<td align="right">1.5044987</td>
<td align="right">-0.4851243</td>
<td align="right">-0.0901153</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Maxine</td>
<td align="right">338</td>
<td align="left">338</td>
<td align="left">NA</td>
<td align="left">8_CF_3_L008.bam</td>
<td align="left">F</td>
<td align="right">3</td>
<td align="right">8</td>
<td align="right">6</td>
<td align="left">2016-06-07</td>
<td align="left">YG1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">53</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">7.2</td>
<td align="left">NA</td>
<td align="right">50.52632</td>
<td align="right">19.49295</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40179</td>
<td align="left">Yes</td>
<td align="right">53</td>
<td align="left">NA</td>
<td align="right">-0.1081500</td>
<td align="right">-0.0179254</td>
<td align="right">0.0048894</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">389</td>
<td align="right">-3.856348</td>
<td align="right">1.0343383</td>
<td align="right">1.7048629</td>
<td align="right">0.8954923</td>
<td align="right">1.4124154</td>
<td align="right">-1.1728185</td>
<td align="right">1.0234824</td>
<td align="right">-1.5591497</td>
<td align="right">0.7169038</td>
<td align="right">0.3727142</td>
<td align="right">-0.6452253</td>
<td align="right">-0.1899591</td>
<td align="right">0.3579922</td>
<td align="right">-0.5025431</td>
<td align="right">-1.3272068</td>
<td align="right">1.4198048</td>
<td align="right">1.3620939</td>
<td align="right">-1.1148167</td>
<td align="right">0.2188360</td>
<td align="right">1.5035396</td>
<td align="right">0.2172473</td>
<td align="right">-0.7683361</td>
<td align="right">0.5768306</td>
<td align="right">1.1989056</td>
<td align="right">-2.0854489</td>
<td align="right">1.5145974</td>
<td align="right">-4.7386290</td>
<td align="right">3.7445017</td>
<td align="right">4.1050593</td>
<td align="right">-9.4246426</td>
<td align="right">-7.3182332</td>
<td align="right">-5.0141238</td>
<td align="right">4.6001930</td>
<td align="right">-2.7149222</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Rogger</td>
<td align="right">389</td>
<td align="left">389</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">M</td>
<td align="right">4</td>
<td align="right">NA</td>
<td align="right">23</td>
<td align="left">2018-10-10</td>
<td align="left">YG39</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">NA</td>
<td align="left">2018-10-01</td>
<td align="right">45</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.7</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">verus</td>
<td align="left">25204</td>
<td align="left">41656</td>
<td align="left">NA</td>
<td align="right">45</td>
<td align="left">NA</td>
<td align="right">-0.1101630</td>
<td align="right">-0.0206061</td>
<td align="right">0.0032451</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="odd">
<td align="left">438</td>
<td align="right">-6.885783</td>
<td align="right">-2.4661120</td>
<td align="right">5.0472981</td>
<td align="right">3.5276778</td>
<td align="right">6.8895537</td>
<td align="right">1.2619018</td>
<td align="right">4.0897386</td>
<td align="right">-0.2801704</td>
<td align="right">1.9099370</td>
<td align="right">-2.2141640</td>
<td align="right">2.9491966</td>
<td align="right">-1.5915998</td>
<td align="right">-0.9030430</td>
<td align="right">-3.8399765</td>
<td align="right">0.0239256</td>
<td align="right">3.9599753</td>
<td align="right">-1.4000526</td>
<td align="right">2.3733713</td>
<td align="right">5.4107865</td>
<td align="right">-2.6253465</td>
<td align="right">0.6682401</td>
<td align="right">-1.9667922</td>
<td align="right">-0.4806345</td>
<td align="right">1.7904700</td>
<td align="right">5.7293414</td>
<td align="right">2.6928710</td>
<td align="right">-10.2441866</td>
<td align="right">3.8032137</td>
<td align="right">-4.3911520</td>
<td align="right">5.7353549</td>
<td align="right">1.3669328</td>
<td align="right">-1.4147645</td>
<td align="right">0.6556390</td>
<td align="right">-1.2714230</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Cheeta</td>
<td align="right">438</td>
<td align="left">438</td>
<td align="left">NA</td>
<td align="left">155_CF_3_L004.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">8</td>
<td align="left">2016-06-22</td>
<td align="left">YG22</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">55</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.6</td>
<td align="left">NA</td>
<td align="right">55.30614</td>
<td align="right">18.06375</td>
<td align="left">verus</td>
<td align="left">20821</td>
<td align="left">40909</td>
<td align="left">Yes</td>
<td align="right">55</td>
<td align="left">NA</td>
<td align="right">-0.1081420</td>
<td align="right">-0.0177823</td>
<td align="right">0.0045724</td>
<td align="right">0.000010</td>
<td align="right">1e-05</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
</tr>
<tr class="even">
<td align="left">456</td>
<td align="right">2.696852</td>
<td align="right">-0.0790156</td>
<td align="right">0.9954681</td>
<td align="right">-0.3353002</td>
<td align="right">2.2013716</td>
<td align="right">-0.9861758</td>
<td align="right">0.9860697</td>
<td align="right">-1.4269147</td>
<td align="right">0.7068544</td>
<td align="right">-0.9172136</td>
<td align="right">-0.0789697</td>
<td align="right">-1.1998594</td>
<td align="right">-0.2042522</td>
<td align="right">-0.0722823</td>
<td align="right">0.6281202</td>
<td align="right">1.4747030</td>
<td align="right">-0.0245971</td>
<td align="right">0.6281378</td>
<td align="right">1.9987541</td>
<td align="right">1.8467420</td>
<td align="right">0.4453291</td>
<td align="right">0.9543101</td>
<td align="right">0.9905206</td>
<td align="right">0.9298047</td>
<td align="right">-3.7077392</td>
<td align="right">-1.9989876</td>
<td align="right">3.4459438</td>
<td align="right">7.9691675</td>
<td align="right">-8.3976329</td>
<td align="right">-7.3413986</td>
<td align="right">6.3424025</td>
<td align="right">3.0488619</td>
<td align="right">0.4828621</td>
<td align="right">1.5717938</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Yerkes</td>
<td align="left">Mai</td>
<td align="right">456</td>
<td align="left">456</td>
<td align="left">NA</td>
<td align="left">156_CF_3_L001.bam</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="left">2016-06-22</td>
<td align="left">YG23</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">2018-09-01</td>
<td align="right">49</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">5.5</td>
<td align="left">NA</td>
<td align="right">54.00665</td>
<td align="right">20.13760</td>
<td align="left">verus</td>
<td align="left">23377</td>
<td align="left">41275</td>
<td align="left">Yes</td>
<td align="right">49</td>
<td align="left">NA</td>
<td align="right">-0.1080570</td>
<td align="right">-0.0178108</td>
<td align="right">0.0044597</td>
<td align="right">0.000011</td>
<td align="right">1e-05</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
</tr>
</tbody>
</table>
<pre class="r"><code>PCs_to_test &lt;- Merged[,2:11]
kable(head(PCs_to_test))</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">PC1</th>
<th align="right">PC2</th>
<th align="right">PC3</th>
<th align="right">PC4</th>
<th align="right">PC5</th>
<th align="right">PC6</th>
<th align="right">PC7</th>
<th align="right">PC8</th>
<th align="right">PC9</th>
<th align="right">PC10</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">-50.430132</td>
<td align="right">-2.3098435</td>
<td align="right">2.4369275</td>
<td align="right">-3.9747155</td>
<td align="right">4.1379063</td>
<td align="right">-15.0889535</td>
<td align="right">7.5684829</td>
<td align="right">-12.5989456</td>
<td align="right">9.1740419</td>
<td align="right">-0.1703849</td>
</tr>
<tr class="even">
<td align="right">241.532847</td>
<td align="right">-0.0316741</td>
<td align="right">0.5375320</td>
<td align="right">-0.7206688</td>
<td align="right">0.9010903</td>
<td align="right">-2.1914714</td>
<td align="right">1.0462675</td>
<td align="right">-1.7888688</td>
<td align="right">1.3112178</td>
<td align="right">0.1158945</td>
</tr>
<tr class="odd">
<td align="right">-2.144211</td>
<td align="right">-0.0669376</td>
<td align="right">-1.5453948</td>
<td align="right">-0.3322771</td>
<td align="right">0.9761652</td>
<td align="right">0.1501650</td>
<td align="right">0.4518967</td>
<td align="right">0.2665345</td>
<td align="right">-0.8105410</td>
<td align="right">-3.2601071</td>
</tr>
<tr class="even">
<td align="right">-3.856348</td>
<td align="right">1.0343383</td>
<td align="right">1.7048629</td>
<td align="right">0.8954923</td>
<td align="right">1.4124154</td>
<td align="right">-1.1728185</td>
<td align="right">1.0234824</td>
<td align="right">-1.5591497</td>
<td align="right">0.7169038</td>
<td align="right">0.3727142</td>
</tr>
<tr class="odd">
<td align="right">-6.885783</td>
<td align="right">-2.4661120</td>
<td align="right">5.0472981</td>
<td align="right">3.5276778</td>
<td align="right">6.8895537</td>
<td align="right">1.2619018</td>
<td align="right">4.0897386</td>
<td align="right">-0.2801704</td>
<td align="right">1.9099370</td>
<td align="right">-2.2141640</td>
</tr>
<tr class="even">
<td align="right">2.696852</td>
<td align="right">-0.0790156</td>
<td align="right">0.9954681</td>
<td align="right">-0.3353002</td>
<td align="right">2.2013716</td>
<td align="right">-0.9861758</td>
<td align="right">0.9860697</td>
<td align="right">-1.4269147</td>
<td align="right">0.7068544</td>
<td align="right">-0.9172136</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Grab potential continuous confounders that make sense to test
Continuous_confounders_to_test &lt;- Merged[, c(&quot;RIN&quot;, &quot;Age&quot;, &quot;GenotypePC1&quot;, &quot;GenotypePC2&quot;, &quot;GenotypePC3&quot;, &quot;Admix.Western&quot;, &quot;Admix.Eastern&quot;, &quot;Admix.Central&quot;, &quot;Admix.NigeriaCameroon&quot;)]
kable(head(Continuous_confounders_to_test))</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">RIN</th>
<th align="right">Age</th>
<th align="right">GenotypePC1</th>
<th align="right">GenotypePC2</th>
<th align="right">GenotypePC3</th>
<th align="right">Admix.Western</th>
<th align="right">Admix.Eastern</th>
<th align="right">Admix.Central</th>
<th align="right">Admix.NigeriaCameroon</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">7.3</td>
<td align="right">40</td>
<td align="right">-0.0819553</td>
<td align="right">0.0484042</td>
<td align="right">0.0151794</td>
<td align="right">0.815883</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.184097</td>
</tr>
<tr class="even">
<td align="right">7.6</td>
<td align="right">44</td>
<td align="right">-0.1089490</td>
<td align="right">-0.0190248</td>
<td align="right">0.0051753</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.000011</td>
</tr>
<tr class="odd">
<td align="right">7.2</td>
<td align="right">53</td>
<td align="right">-0.1081500</td>
<td align="right">-0.0179254</td>
<td align="right">0.0048894</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.000010</td>
</tr>
<tr class="even">
<td align="right">5.7</td>
<td align="right">45</td>
<td align="right">-0.1101630</td>
<td align="right">-0.0206061</td>
<td align="right">0.0032451</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.000010</td>
</tr>
<tr class="odd">
<td align="right">5.6</td>
<td align="right">55</td>
<td align="right">-0.1081420</td>
<td align="right">-0.0177823</td>
<td align="right">0.0045724</td>
<td align="right">0.999970</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.000010</td>
</tr>
<tr class="even">
<td align="right">5.5</td>
<td align="right">49</td>
<td align="right">-0.1080570</td>
<td align="right">-0.0178108</td>
<td align="right">0.0044597</td>
<td align="right">0.999969</td>
<td align="right">1e-05</td>
<td align="right">1e-05</td>
<td align="right">0.000011</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Test
Spearman_test_results &lt;- corr.test(Continuous_confounders_to_test, PCs_to_test, adjust=&quot;none&quot;, method=&quot;spearman&quot;)

MinP_floor &lt;- floor(log10(min(Spearman_test_results$p)))
# Plot
Spearman_test_results$p %&gt;%
  melt() %&gt;%
  dplyr::rename(Pvalue = value, Principle.Component=Var2, Potential.Confounder=Var1) %&gt;%
  ggplot(aes(x=Potential.Confounder, y=Principle.Component, fill=Pvalue)) + 
    geom_tile() +
    geom_text(aes(label = signif(Pvalue, 2))) +
    scale_fill_gradient(limits=c(10**MinP_floor, 1), breaks=10**seq(MinP_floor,0,1), trans = &#39;log&#39;, high=&quot;white&quot;, low=&quot;red&quot; ) +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/correlate-pcs-with-factors-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-correlate-pcs-with-factors-1">
Past versions of correlate-pcs-with-factors-1.png
</button>
</p>
<div id="fig-correlate-pcs-with-factors-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/correlate-pcs-with-factors-1.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Grab potential categorical confounders that make sense to test
Categorical_confounders_to_test &lt;- Merged[,c(&quot;Viral.status&quot;, &quot;SX&quot;,&quot;RNA.Extract_date&quot;, &quot;RNA.Library.prep.batch&quot;, &quot;Source&quot;)]
kable(head(Categorical_confounders_to_test))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Viral.status</th>
<th align="left">SX</th>
<th align="left">RNA.Extract_date</th>
<th align="right">RNA.Library.prep.batch</th>
<th align="left">Source</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
<td align="left">M</td>
<td align="left">2018-10-10</td>
<td align="right">5</td>
<td align="left">Yerkes</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">M</td>
<td align="left">2016-06-07</td>
<td align="right">3</td>
<td align="left">Yerkes</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">F</td>
<td align="left">2016-06-07</td>
<td align="right">3</td>
<td align="left">Yerkes</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">M</td>
<td align="left">2018-10-10</td>
<td align="right">4</td>
<td align="left">Yerkes</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">F</td>
<td align="left">2016-06-22</td>
<td align="right">2</td>
<td align="left">Yerkes</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">F</td>
<td align="left">2016-06-22</td>
<td align="right">2</td>
<td align="left">Yerkes</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Viral status will need to be reformatted to make factors that make sense for testing (example: HBV+, HBV- HCV+, HCV- are factors that make sense). Let&#39;s assume that NA means negative status.
Categorical_confounders_to_test$HBV_status &lt;- grepl(&quot;HBV+&quot;, Categorical_confounders_to_test$Viral.status)
Categorical_confounders_to_test$HCV_status &lt;- grepl(&quot;HCV+&quot;, Categorical_confounders_to_test$Viral.status)
Categorical_confounders_to_test &lt;- Categorical_confounders_to_test[, -1 ]
kable(head(Categorical_confounders_to_test))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SX</th>
<th align="left">RNA.Extract_date</th>
<th align="right">RNA.Library.prep.batch</th>
<th align="left">Source</th>
<th align="left">HBV_status</th>
<th align="left">HCV_status</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">M</td>
<td align="left">2018-10-10</td>
<td align="right">5</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">M</td>
<td align="left">2016-06-07</td>
<td align="right">3</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">F</td>
<td align="left">2016-06-07</td>
<td align="right">3</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">M</td>
<td align="left">2018-10-10</td>
<td align="right">4</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">F</td>
<td align="left">2016-06-22</td>
<td align="right">2</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="left">2016-06-22</td>
<td align="right">2</td>
<td align="left">Yerkes</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Do one-way anova test as a loop.
# First initialize results matrix
Pvalues &lt;- matrix(ncol = dim(PCs_to_test)[2], nrow = dim(Categorical_confounders_to_test)[2])
colnames(Pvalues) &lt;- colnames(PCs_to_test)
rownames(Pvalues) &lt;- colnames(Categorical_confounders_to_test)
for (confounder in seq_along(Categorical_confounders_to_test)) {
    for (PC in seq_along(PCs_to_test)) {
      res.aov &lt;- aov(PCs_to_test[[PC]] ~ Categorical_confounders_to_test[[confounder]])
      pval &lt;- summary(res.aov)[[1]][[&quot;Pr(&gt;F)&quot;]][1]
      Pvalues[confounder, PC] &lt;- pval
    }
}
# Plot
MinP_floor &lt;- floor(log10(min(Pvalues)))
Pvalues %&gt;%
  melt() %&gt;%
  dplyr::rename(Pvalue = value, Principle.Component=Var2, Potential.Confounder=Var1) %&gt;%
  ggplot(aes(x=Potential.Confounder, y=Principle.Component, fill=Pvalue)) + 
    geom_tile() +
    geom_text(aes(label = signif(Pvalue, 2))) +
    scale_fill_gradient(limits=c(10**MinP_floor, 1), breaks=10**seq(MinP_floor,0,1), trans = &#39;log&#39;, high=&quot;white&quot;, low=&quot;red&quot; ) +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/correlate-pcs-with-factors-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-correlate-pcs-with-factors-2">
Past versions of correlate-pcs-with-factors-2.png
</button>
</p>
<div id="fig-correlate-pcs-with-factors-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/correlate-pcs-with-factors-2.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Interpretation: The only observed technical factors (or measured population structure statistics) that correlate with any of the PCs on the residuals are Admix.Central and HBV status. Given that HBV status is correlated with such late PCs that explain so little variance (see scree plot above), and there are only 3 HBV+ observations, I think including HBV in the model is not justified. Also note that the correlation matrix of the residuals after regressing out 3PCs generally shows two clusters, neither of which segregate with the 3HBV+ observations. The other significant correlation is between PC9 (which explains little variance) and Admix.Central coefficient. This trait is already included implicitly in the eQTL calling model as a genotype PC covariate.</p>
<p>Lastly, I want to look at the residuals after regressing out those first three PCs.</p>
<pre class="r"><code>#Plot residuals for a few genes
MyPlot(ResidualCountTable[1,])</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-1">
Past versions of plot-residuals-1.png
</button>
</p>
<div id="fig-plot-residuals-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-1.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(ResidualCountTable[2,])</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-2">
Past versions of plot-residuals-2.png
</button>
</p>
<div id="fig-plot-residuals-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-2.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(ResidualCountTable[3,])</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-3">
Past versions of plot-residuals-3.png
</button>
</p>
<div id="fig-plot-residuals-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-3.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(ResidualCountTable[4,])</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-4">
Past versions of plot-residuals-4.png
</button>
</p>
<div id="fig-plot-residuals-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-4.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(ResidualCountTable[5,])</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-5">
Past versions of plot-residuals-5.png
</button>
</p>
<div id="fig-plot-residuals-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-5.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(ResidualCountTable[6,])</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-6">
Past versions of plot-residuals-6.png
</button>
</p>
<div id="fig-plot-residuals-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-6.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(ResidualCountTable[7,])</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-7">
Past versions of plot-residuals-7.png
</button>
</p>
<div id="fig-plot-residuals-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-7.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(ResidualCountTable[8,])</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-8">
Past versions of plot-residuals-8.png
</button>
</p>
<div id="fig-plot-residuals-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-8.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(ResidualCountTable[9,])</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-9">
Past versions of plot-residuals-9.png
</button>
</p>
<div id="fig-plot-residuals-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-9.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MyPlot(ResidualCountTable[10,])</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-10">
Past versions of plot-residuals-10.png
</button>
</p>
<div id="fig-plot-residuals-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-10.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Plot of first two PCs of residuals
pca_results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&#39;sample&#39;) %&gt;%
  ggplot(aes(x=PC1, y=PC2, label=sample)) +
  geom_text()</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-residuals-11">
Past versions of plot-residuals-11.png
</button>
</p>
<div id="fig-plot-residuals-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/plot-residuals-11.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Interpretation: After regressing out the PCs and looking at residuals, sample 317 seems to be consistently an outlier. This is also reflected in PC space of the residuals. It was not an outlier in the original phenotypes. Was it an outlier in the original PCs that were regressed out?</p>
<pre class="r"><code># The PCs were read in from the covariates file

t(Covariates) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&#39;sample&#39;) %&gt;%
  ggplot(aes(x=PC1.RS, y=PC2.RS, label=sample)) +
  geom_text()</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-5-1.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t(Covariates) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&#39;sample&#39;) %&gt;%
  ggplot(aes(x=PC2.RS, y=PC3.RS, label=sample)) +
  geom_text()</code></pre>
<p><img src="figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/figure/20190421_RegressOutRNASeqPCs.Rmd/unnamed-chunk-5-2.png" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
No it was not an outlier in the original phenotypes. <br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] matrixStats_0.54.0 gplots_3.0.1       reshape2_1.4.3    
 [4] knitr_1.22         ggrepel_0.8.0      psych_1.8.10      
 [7] forcats_0.4.0      stringr_1.4.0      dplyr_0.8.0.1     
[10] purrr_0.3.2        readr_1.3.1        tidyr_0.8.2       
[13] tibble_2.1.1       tidyverse_1.2.1    readxl_1.1.0      
[16] ggfortify_0.4.5    ggplot2_3.1.0      corrplot_0.84     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1         lubridate_1.7.4    lattice_0.20-38   
 [4] gtools_3.8.1       assertthat_0.2.1   rprojroot_1.3-2   
 [7] digest_0.6.18      R6_2.4.0           cellranger_1.1.0  
[10] plyr_1.8.4         backports_1.1.3    evaluate_0.13     
[13] highr_0.8          httr_1.4.0         pillar_1.3.1      
[16] rlang_0.3.3        lazyeval_0.2.2     rstudioapi_0.10   
[19] gdata_2.18.0       whisker_0.3-2      rmarkdown_1.11    
[22] labeling_0.3       foreign_0.8-71     munsell_0.5.0     
[25] broom_0.5.1        compiler_3.5.1     modelr_0.1.4      
[28] xfun_0.6           pkgconfig_2.0.2    mnormt_1.5-5      
[31] htmltools_0.3.6    tidyselect_0.2.5   gridExtra_2.3     
[34] workflowr_1.2.0    crayon_1.3.4       withr_2.1.2       
[37] bitops_1.0-6       grid_3.5.1         nlme_3.1-137      
[40] jsonlite_1.6       gtable_0.3.0       git2r_0.24.0      
[43] magrittr_1.5       scales_1.0.0       KernSmooth_2.23-15
[46] cli_1.1.0          stringi_1.4.3      fs_1.2.6          
[49] xml2_1.2.0         generics_0.0.2     tools_3.5.1       
[52] glue_1.3.1         hms_0.4.2          parallel_3.5.1    
[55] yaml_2.2.0         colorspace_1.4-1   caTools_1.17.1.1  
[58] rvest_0.3.2        haven_2.1.0       </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
