<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>ResponseToReviewer_Point6</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">ResponseToReviewer_Point6</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree52f85bbbd3f47b3f687d35fbfff41f8b53035e5btargetblank52f85bba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/52f85bbbd3f47b3f687d35fbfff41f8b53035e5b" target="_blank">52f85bb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree52f85bbbd3f47b3f687d35fbfff41f8b53035e5btargetblank52f85bba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    WorkingManuscript.zip
    Ignored:    WorkingManuscript/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis_temp/.DS_Store
    Ignored:    big_data/
    Ignored:    code/.DS_Store
    Ignored:    code/snakemake_workflow/.DS_Store
    Ignored:    code/snakemake_workflow/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    figures/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/20200907_Response_Point_02.Rmd
    Untracked:  analysis/20200907_Response_Point_04.Rmd
    Untracked:  data/c5.all.v7.1.symbols.gmt
    Untracked:  data/c5.all.v7.1.symbols.gmt.categories.tsv.gz
    Untracked:  data/h.all.v7.1.symbols.gmt

Unstaged changes:
    Modified:   analysis/20200907_Response_OriginalComments.Rmd
    Modified:   analysis/20200907_Response_Point_06.Rmd
    Modified:   analysis/20200907_Response_Point_09-2.Rmd
    Modified:   analysis/20200907_Response_Point_09.Rmd
    Modified:   analysis/20200907_Response_Point_11.Rmd
    Modified:   analysis/Final_2_DispersionPlots.Rmd
    Modified:   analysis_temp/TabulaMuris_analysis2.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>Original reviewer point:</p>
<blockquote>
<p>The fourth test authors conducted is to show that dn/ds and pn/ps ratios of genes are correlated with gene expression variability (variance). However, because of the existence of heterogeneity of cell-type composition in samples, any correlation observed may be utterly biased by this single uncontrollable confounding factor. Furthermore, heart tissues contain an over-abundant expression of genes encoded in the mitochondrial genome. The expression level of these mt-genes may vary substantially between samples and reflect the health status of primary sample donors. PEER normalization may have to take this into account as a covariant.</p>
</blockquote>
<ul>
<li>dn/ds and pn/ps are dna based measurements and should be totally orthogonal to gene expression variability, cell type heterogeneity between samples and such.</li>
<li>To address the reviewers second point, I will here investigate the extent to which PCA (since I use PCs as covariates in eQTL calling instead of PEER factors) is affected by the inclusion of mitochondrial genes which I originally excluded from analysis. To do this, I will compare the PCA results of the gene expression matrix with imputed MT gene expression values (imputed to the sample-wide median for each gene), and compare it to the results if I use actual expression values.</li>
</ul>
<p>First load necessary libraries for analysis…</p>
<pre class="r"><code>library(tidyverse)
library(data.table)
library(knitr)
library(edgeR)
library(gplots)
source(&quot;../code/CustomFunctions.R&quot;)</code></pre>
<p>Now load the some input files…</p>
<p>First the count table as used in the original manusript…</p>
<pre class="r"><code>#Get expression values for the 
CountTableChimpFile &lt;- &#39;../output/PowerAnalysisFullCountTable.Chimp.subread.txt.gz&#39;
CountTableHumanFile &lt;- &#39;../output/PowerAnalysisFullCountTable.Human.subread.txt.gz&#39;
OutputDE &lt;- &#39;../output/Final/TableS2.tab&#39;
DropFileName &lt;- &#39;../data/DE_SamplesToDrop.txt&#39;

DropFile &lt;- read.delim(DropFileName, sep=&#39;\t&#39;, col.names = c(&quot;Sample&quot;, &quot;Species&quot;), stringsAsFactors = F)
HumanSamplesToDrop &lt;- DropFile %&gt;% filter(Species==&quot;Human&quot;) %&gt;% pull(Sample)
ChimpSamplesToDrop &lt;- DropFile %&gt;% filter(Species==&quot;Chimp&quot;) %&gt;% pull(Sample)
DE.results &lt;- read.delim(OutputDE, sep=&#39;\t&#39;, stringsAsFactors = F)
GeneListForOverdispersionCalculation &lt;- DE.results$Ensembl_geneID

CountTables &lt;- GetCountTables(CountTableChimpFile,
                              CountTableHumanFile,
                              0, GeneListForOverdispersionCalculation, ChimpSampleDrop=ChimpSamplesToDrop, HumanSampleDrop = HumanSamplesToDrop)</code></pre>
<p>Now a fuller count table so we can find expression values from MT genes.</p>
<pre class="r"><code>FullTable &lt;- fread(&quot;../output/STAR.RawCountTable.txt.gz&quot;)

FullTable &lt;- FullTable %&gt;%
  separate(2, into=paste0(&quot;C.&quot;, strsplit(colnames(FullTable)[2], &#39; &#39;)[[1]])) %&gt;%
  dplyr::select(-2) %&gt;%
  mutate_at(-1, as.integer)

GeneNames &lt;- read.delim(&quot;../data/Biomart_export.Hsap.Ptro.orthologs.txt.gz&quot;)


MT.Genes &lt;- GeneNames %&gt;%
  filter(str_detect(Chimpanzee.gene.name,pattern=&quot;^MT-&quot;)) %&gt;%
  dplyr::select(Chimpanzee.gene.name, Chimpanzee.gene.stable.ID)

#Count table of MT genes
MT.CountTable &lt;- FullTable %&gt;%
  filter(GENE %in% MT.Genes$Chimpanzee.gene.stable.ID) %&gt;%
  column_to_rownames(&quot;GENE&quot;)
MT.CountTable %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">C.4X0212</th>
<th align="right">C.4X0267</th>
<th align="right">C.4X0333</th>
<th align="right">C.4X0339</th>
<th align="right">C.4X0354</th>
<th align="right">C.4X0357</th>
<th align="right">C.4X0550</th>
<th align="right">C.4x0025</th>
<th align="right">C.4x0043</th>
<th align="right">C.4x373</th>
<th align="right">C.4x0430</th>
<th align="right">C.4x0519</th>
<th align="right">C.4x523</th>
<th align="right">C.88A020</th>
<th align="right">C.95A014</th>
<th align="right">C.295</th>
<th align="right">C.317</th>
<th align="right">C.338</th>
<th align="right">C.389</th>
<th align="right">C.438</th>
<th align="right">C.456</th>
<th align="right">C.462</th>
<th align="right">C.476</th>
<th align="right">C.495</th>
<th align="right">C.503</th>
<th align="right">C.522</th>
<th align="right">C.529</th>
<th align="right">C.537</th>
<th align="right">C.549</th>
<th align="right">C.554</th>
<th align="right">C.554_2</th>
<th align="right">C.558</th>
<th align="right">C.570</th>
<th align="right">C.623</th>
<th align="right">C.676</th>
<th align="right">C.724</th>
<th align="right">C.Little_R</th>
<th align="right">C.MD_And</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSPTRG00000042641</td>
<td align="right">1225139</td>
<td align="right">1395362</td>
<td align="right">626059</td>
<td align="right">1817795</td>
<td align="right">1314663</td>
<td align="right">1487905</td>
<td align="right">1414239</td>
<td align="right">724262</td>
<td align="right">1117381</td>
<td align="right">844931</td>
<td align="right">1069735</td>
<td align="right">1131150</td>
<td align="right">907877</td>
<td align="right">1386558</td>
<td align="right">1119113</td>
<td align="right">2671514</td>
<td align="right">945443</td>
<td align="right">1068457</td>
<td align="right">1948692</td>
<td align="right">1630818</td>
<td align="right">1345994</td>
<td align="right">1853598</td>
<td align="right">2578236</td>
<td align="right">2049414</td>
<td align="right">1903301</td>
<td align="right">1609196</td>
<td align="right">1905046</td>
<td align="right">2436447</td>
<td align="right">2232543</td>
<td align="right">923291</td>
<td align="right">989955</td>
<td align="right">1613475</td>
<td align="right">872765</td>
<td align="right">1255842</td>
<td align="right">1174224</td>
<td align="right">1322953</td>
<td align="right">1635983</td>
<td align="right">1313148</td>
</tr>
<tr class="even">
<td>ENSPTRG00000042626</td>
<td align="right">595124</td>
<td align="right">1184959</td>
<td align="right">752864</td>
<td align="right">1346396</td>
<td align="right">1330544</td>
<td align="right">999018</td>
<td align="right">1204887</td>
<td align="right">491613</td>
<td align="right">920392</td>
<td align="right">929378</td>
<td align="right">658610</td>
<td align="right">1006395</td>
<td align="right">829517</td>
<td align="right">1565618</td>
<td align="right">472789</td>
<td align="right">2922113</td>
<td align="right">644442</td>
<td align="right">717594</td>
<td align="right">1644331</td>
<td align="right">834599</td>
<td align="right">790593</td>
<td align="right">1191411</td>
<td align="right">1430169</td>
<td align="right">2115597</td>
<td align="right">1230627</td>
<td align="right">1670707</td>
<td align="right">1786107</td>
<td align="right">2345625</td>
<td align="right">1314135</td>
<td align="right">629247</td>
<td align="right">540243</td>
<td align="right">1507712</td>
<td align="right">478228</td>
<td align="right">1081295</td>
<td align="right">531279</td>
<td align="right">871002</td>
<td align="right">1188400</td>
<td align="right">901150</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000042642</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>ENSPTRG00000042657</td>
<td align="right">10919560</td>
<td align="right">11207253</td>
<td align="right">7621373</td>
<td align="right">18913511</td>
<td align="right">9783877</td>
<td align="right">11991378</td>
<td align="right">10591085</td>
<td align="right">5063262</td>
<td align="right">8021281</td>
<td align="right">3180011</td>
<td align="right">7576705</td>
<td align="right">9036370</td>
<td align="right">6778170</td>
<td align="right">11438819</td>
<td align="right">8809947</td>
<td align="right">18821780</td>
<td align="right">7776345</td>
<td align="right">11662175</td>
<td align="right">15657083</td>
<td align="right">14310164</td>
<td align="right">12434341</td>
<td align="right">10838457</td>
<td align="right">24634805</td>
<td align="right">9917607</td>
<td align="right">18374093</td>
<td align="right">12572345</td>
<td align="right">15123590</td>
<td align="right">20209687</td>
<td align="right">13005903</td>
<td align="right">11287173</td>
<td align="right">10533101</td>
<td align="right">14801818</td>
<td align="right">6833776</td>
<td align="right">10489326</td>
<td align="right">17177721</td>
<td align="right">11693458</td>
<td align="right">11372879</td>
<td align="right">8920509</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000042660</td>
<td align="right">2281051</td>
<td align="right">2217094</td>
<td align="right">2471250</td>
<td align="right">4976683</td>
<td align="right">2925848</td>
<td align="right">3028701</td>
<td align="right">2507750</td>
<td align="right">1875989</td>
<td align="right">2243275</td>
<td align="right">1257402</td>
<td align="right">1780727</td>
<td align="right">1638230</td>
<td align="right">2535655</td>
<td align="right">2428793</td>
<td align="right">1890506</td>
<td align="right">5237640</td>
<td align="right">1499587</td>
<td align="right">2337969</td>
<td align="right">2993579</td>
<td align="right">2563361</td>
<td align="right">3326678</td>
<td align="right">2441279</td>
<td align="right">5198371</td>
<td align="right">3013554</td>
<td align="right">3532133</td>
<td align="right">2980545</td>
<td align="right">3154789</td>
<td align="right">6547681</td>
<td align="right">2696388</td>
<td align="right">1829700</td>
<td align="right">1572366</td>
<td align="right">3265725</td>
<td align="right">1615535</td>
<td align="right">2169811</td>
<td align="right">2283633</td>
<td align="right">2243143</td>
<td align="right">3440372</td>
<td align="right">2732462</td>
</tr>
<tr class="even">
<td>ENSPTRG00000042653</td>
<td align="right">120</td>
<td align="right">69</td>
<td align="right">58</td>
<td align="right">803</td>
<td align="right">73</td>
<td align="right">65</td>
<td align="right">60</td>
<td align="right">43</td>
<td align="right">49</td>
<td align="right">84918</td>
<td align="right">54</td>
<td align="right">112447</td>
<td align="right">76</td>
<td align="right">110</td>
<td align="right">47</td>
<td align="right">907</td>
<td align="right">70</td>
<td align="right">61</td>
<td align="right">133</td>
<td align="right">101</td>
<td align="right">274</td>
<td align="right">101</td>
<td align="right">246</td>
<td align="right">963</td>
<td align="right">358</td>
<td align="right">104</td>
<td align="right">306149</td>
<td align="right">331</td>
<td align="right">120</td>
<td align="right">77</td>
<td align="right">95</td>
<td align="right">148</td>
<td align="right">68</td>
<td align="right">558</td>
<td align="right">91</td>
<td align="right">150</td>
<td align="right">95</td>
<td align="right">49</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000042650</td>
<td align="right">1948</td>
<td align="right">894410</td>
<td align="right">1856</td>
<td align="right">12672</td>
<td align="right">1257519</td>
<td align="right">1386395</td>
<td align="right">1104507</td>
<td align="right">521</td>
<td align="right">869441</td>
<td align="right">936150</td>
<td align="right">709</td>
<td align="right">1015588</td>
<td align="right">889</td>
<td align="right">1114737</td>
<td align="right">855</td>
<td align="right">2116107</td>
<td align="right">1262</td>
<td align="right">1223</td>
<td align="right">1313614</td>
<td align="right">971</td>
<td align="right">1571</td>
<td align="right">900153</td>
<td align="right">6123</td>
<td align="right">884452</td>
<td align="right">2803</td>
<td align="right">1190496</td>
<td align="right">2244503</td>
<td align="right">5392</td>
<td align="right">1198</td>
<td align="right">2675</td>
<td align="right">2198</td>
<td align="right">3999</td>
<td align="right">747</td>
<td align="right">780231</td>
<td align="right">1890</td>
<td align="right">2579</td>
<td align="right">2157</td>
<td align="right">978152</td>
</tr>
<tr class="even">
<td>ENSPTRG00000042661</td>
<td align="right">1797</td>
<td align="right">491013</td>
<td align="right">1505</td>
<td align="right">4202</td>
<td align="right">211135</td>
<td align="right">663966</td>
<td align="right">478544</td>
<td align="right">456</td>
<td align="right">471689</td>
<td align="right">470307</td>
<td align="right">557</td>
<td align="right">589501</td>
<td align="right">707</td>
<td align="right">573991</td>
<td align="right">690</td>
<td align="right">1060849</td>
<td align="right">928</td>
<td align="right">1185</td>
<td align="right">679592</td>
<td align="right">941</td>
<td align="right">1287</td>
<td align="right">381282</td>
<td align="right">3715</td>
<td align="right">389361</td>
<td align="right">1704</td>
<td align="right">695032</td>
<td align="right">1126484</td>
<td align="right">3957</td>
<td align="right">935</td>
<td align="right">1519</td>
<td align="right">7551</td>
<td align="right">2634</td>
<td align="right">627</td>
<td align="right">446728</td>
<td align="right">1182</td>
<td align="right">2421</td>
<td align="right">1227</td>
<td align="right">524811</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000042628</td>
<td align="right">399168</td>
<td align="right">268875</td>
<td align="right">300273</td>
<td align="right">647370</td>
<td align="right">394313</td>
<td align="right">268190</td>
<td align="right">335091</td>
<td align="right">205321</td>
<td align="right">198786</td>
<td align="right">223935</td>
<td align="right">294631</td>
<td align="right">219204</td>
<td align="right">311540</td>
<td align="right">301228</td>
<td align="right">253407</td>
<td align="right">598135</td>
<td align="right">306587</td>
<td align="right">438166</td>
<td align="right">331364</td>
<td align="right">427547</td>
<td align="right">375668</td>
<td align="right">287757</td>
<td align="right">713317</td>
<td align="right">368139</td>
<td align="right">505874</td>
<td align="right">401212</td>
<td align="right">278580</td>
<td align="right">1052447</td>
<td align="right">509227</td>
<td align="right">268682</td>
<td align="right">329197</td>
<td align="right">487375</td>
<td align="right">210474</td>
<td align="right">262422</td>
<td align="right">298908</td>
<td align="right">290496</td>
<td align="right">495081</td>
<td align="right">312884</td>
</tr>
<tr class="even">
<td>ENSPTRG00000042631</td>
<td align="right">390948</td>
<td align="right">176607</td>
<td align="right">357965</td>
<td align="right">762341</td>
<td align="right">179459</td>
<td align="right">362027</td>
<td align="right">179424</td>
<td align="right">133679</td>
<td align="right">219514</td>
<td align="right">200934</td>
<td align="right">255022</td>
<td align="right">315252</td>
<td align="right">275124</td>
<td align="right">476205</td>
<td align="right">287551</td>
<td align="right">958120</td>
<td align="right">400544</td>
<td align="right">245808</td>
<td align="right">545295</td>
<td align="right">341233</td>
<td align="right">345938</td>
<td align="right">146246</td>
<td align="right">757070</td>
<td align="right">294020</td>
<td align="right">658181</td>
<td align="right">383394</td>
<td align="right">705494</td>
<td align="right">1003542</td>
<td align="right">582681</td>
<td align="right">417394</td>
<td align="right">397242</td>
<td align="right">730114</td>
<td align="right">219884</td>
<td align="right">374371</td>
<td align="right">364539</td>
<td align="right">417088</td>
<td align="right">628863</td>
<td align="right">264270</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000042639</td>
<td align="right">1392110</td>
<td align="right">1965775</td>
<td align="right">1284030</td>
<td align="right">3176466</td>
<td align="right">1307718</td>
<td align="right">2642693</td>
<td align="right">1817170</td>
<td align="right">665078</td>
<td align="right">1337400</td>
<td align="right">1166799</td>
<td align="right">1107350</td>
<td align="right">1745780</td>
<td align="right">1164278</td>
<td align="right">2092694</td>
<td align="right">1199804</td>
<td align="right">4856059</td>
<td align="right">1449623</td>
<td align="right">1011518</td>
<td align="right">2420903</td>
<td align="right">1544513</td>
<td align="right">1628723</td>
<td align="right">1351917</td>
<td align="right">2777542</td>
<td align="right">1999679</td>
<td align="right">2330633</td>
<td align="right">2281329</td>
<td align="right">3190582</td>
<td align="right">3301221</td>
<td align="right">2306915</td>
<td align="right">1439562</td>
<td align="right">1527431</td>
<td align="right">2274414</td>
<td align="right">971233</td>
<td align="right">1888972</td>
<td align="right">1698314</td>
<td align="right">1282767</td>
<td align="right">2065563</td>
<td align="right">1820314</td>
</tr>
<tr class="even">
<td>ENSPTRG00000042651</td>
<td align="right">441717</td>
<td align="right">783666</td>
<td align="right">241447</td>
<td align="right">394041</td>
<td align="right">360707</td>
<td align="right">420344</td>
<td align="right">776491</td>
<td align="right">94651</td>
<td align="right">440235</td>
<td align="right">247369</td>
<td align="right">225283</td>
<td align="right">755897</td>
<td align="right">197704</td>
<td align="right">1021927</td>
<td align="right">463430</td>
<td align="right">1252668</td>
<td align="right">204724</td>
<td align="right">303717</td>
<td align="right">1059799</td>
<td align="right">388412</td>
<td align="right">454757</td>
<td align="right">372984</td>
<td align="right">925955</td>
<td align="right">453910</td>
<td align="right">566058</td>
<td align="right">750072</td>
<td align="right">857840</td>
<td align="right">690980</td>
<td align="right">433615</td>
<td align="right">183957</td>
<td align="right">260048</td>
<td align="right">418771</td>
<td align="right">161251</td>
<td align="right">630265</td>
<td align="right">253936</td>
<td align="right">569209</td>
<td align="right">341702</td>
<td align="right">620303</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000042630</td>
<td align="right">287</td>
<td align="right">155145</td>
<td align="right">152</td>
<td align="right">269</td>
<td align="right">55666</td>
<td align="right">95115</td>
<td align="right">184778</td>
<td align="right">55</td>
<td align="right">119504</td>
<td align="right">29564</td>
<td align="right">79</td>
<td align="right">138895</td>
<td align="right">45</td>
<td align="right">200023</td>
<td align="right">2708</td>
<td align="right">133743</td>
<td align="right">98</td>
<td align="right">165</td>
<td align="right">177571</td>
<td align="right">94</td>
<td align="right">193</td>
<td align="right">50630</td>
<td align="right">952</td>
<td align="right">41034</td>
<td align="right">186</td>
<td align="right">125266</td>
<td align="right">83507</td>
<td align="right">360</td>
<td align="right">113</td>
<td align="right">172</td>
<td align="right">154</td>
<td align="right">194</td>
<td align="right">50</td>
<td align="right">89916</td>
<td align="right">125</td>
<td align="right">305</td>
<td align="right">124</td>
<td align="right">183474</td>
</tr>
<tr class="even">
<td>ENSPTRG00000042625</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>ENSPTRG00000042637</td>
<td align="right">38464</td>
<td align="right">991272</td>
<td align="right">24835</td>
<td align="right">70907</td>
<td align="right">96148</td>
<td align="right">1334270</td>
<td align="right">1061224</td>
<td align="right">198674</td>
<td align="right">734818</td>
<td align="right">652169</td>
<td align="right">27719</td>
<td align="right">680740</td>
<td align="right">33438</td>
<td align="right">1073477</td>
<td align="right">40952</td>
<td align="right">2091075</td>
<td align="right">19850</td>
<td align="right">45846</td>
<td align="right">1166749</td>
<td align="right">51464</td>
<td align="right">311207</td>
<td align="right">128121</td>
<td align="right">600603</td>
<td align="right">105627</td>
<td align="right">54572</td>
<td align="right">861677</td>
<td align="right">1060310</td>
<td align="right">83196</td>
<td align="right">67202</td>
<td align="right">15619</td>
<td align="right">43277</td>
<td align="right">50951</td>
<td align="right">27461</td>
<td align="right">716071</td>
<td align="right">31770</td>
<td align="right">42275</td>
<td align="right">47629</td>
<td align="right">949398</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Add MT genes to count table, and convert to logCPM
Log.Cpm &lt;- bind_rows(
  FullTable %&gt;%
  filter(GENE %in% MT.Genes$Chimpanzee.gene.stable.ID),
  CountTables$Chimp$Counts %&gt;% rownames_to_column(&quot;GENE&quot;)
) %&gt;%
  column_to_rownames(&quot;GENE&quot;) %&gt;%
  dplyr::select(-C.4X0095) %&gt;% 
  DGEList() %&gt;%
  calcNormFactors() %&gt;% cpm(log=T) %&gt;% as.data.frame()

#Get median expression of all genes
Median.Log.Cpm &lt;- apply(Log.Cpm, 1, median)

#Plot median expression of all genes, and MT genes
data.frame(Med=Median.Log.Cpm) %&gt;%
  rownames_to_column(&quot;GENE&quot;) %&gt;%
  mutate(Is.MT.gene = GENE %in% MT.Genes$Chimpanzee.gene.stable.ID) %&gt;%
  ggplot(aes(x=Med, color=Is.MT.gene)) +
  geom_histogram() +
  facet_wrap(~Is.MT.gene, scales=&quot;free_y&quot;)</code></pre>
<p><img src="figure/20200907_Response_Point_04.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Make &quot;imputed&quot; count table of MT genes based on median expression of those genes across all samples
MT.Med.Log.Cpm.df &lt;- data.frame(Med=Median.Log.Cpm) %&gt;%
  rownames_to_column(&quot;GENE&quot;) %&gt;%
  filter(GENE %in% MT.Genes$Chimpanzee.gene.stable.ID) %&gt;%
  pull(Med) %&gt;%
  matrix(nrow = nrow(MT.CountTable),ncol = ncol(MT.CountTable)) %&gt;%
  as.data.frame(row.names = rownames(MT.CountTable))
colnames(MT.Med.Log.Cpm.df) &lt;- colnames(Log.Cpm)

#Make count table with the MT expression from the imputed values
Log.Cpm.MT.Genes.Imputed.To.Med &lt;- Log.Cpm %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;GENE&quot;) %&gt;%
  filter(!GENE %in% MT.Genes$Chimpanzee.gene.stable.ID) %&gt;%
  bind_rows(MT.Med.Log.Cpm.df %&gt;% rownames_to_column(&quot;GENE&quot;)) %&gt;%
  column_to_rownames(&quot;GENE&quot;)

Log.Cpm.MT.Genes.Imputed.To.Med &lt;- bind_rows(
  MT.Med.Log.Cpm.df %&gt;% rownames_to_column(&quot;GENE&quot;),
  Log.Cpm %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;GENE&quot;) %&gt;%
  filter(!GENE %in% MT.Genes$Chimpanzee.gene.stable.ID)
) %&gt;%
  column_to_rownames(&quot;GENE&quot;)
  
qplot(Log.Cpm.MT.Genes.Imputed.To.Med$C.4X0212, Log.Cpm$C.4X0212)</code></pre>
<p><img src="figure/20200907_Response_Point_04.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#PCA for both count tables
PCA.real &lt;- Log.Cpm %&gt;% t() %&gt;% prcomp()
PCA.imputed &lt;- Log.Cpm.MT.Genes.Imputed.To.Med %&gt;% t() %&gt;% prcomp()


summary(PCA.real)</code></pre>
<pre><code>Importance of components:
                           PC1      PC2      PC3      PC4      PC5     PC6
Standard deviation     48.5612 27.36447 24.50734 21.36360 19.60195 18.9923
Proportion of Variance  0.2863  0.09092  0.07292  0.05542  0.04665  0.0438
Cumulative Proportion   0.2863  0.37724  0.45017  0.50558  0.55223  0.5960
                            PC7      PC8      PC9     PC10     PC11    PC12
Standard deviation     17.04537 15.62519 15.40365 15.21628 13.94564 13.3682
Proportion of Variance  0.03528  0.02964  0.02881  0.02811  0.02361  0.0217
Cumulative Proportion   0.63131  0.66095  0.68976  0.71787  0.74149  0.7632
                           PC13     PC14     PC15     PC16     PC17     PC18
Standard deviation     12.70993 12.04979 11.59453 11.16545 10.74883 10.52678
Proportion of Variance  0.01961  0.01763  0.01632  0.01514  0.01403  0.01345
Cumulative Proportion   0.78280  0.80043  0.81675  0.83189  0.84591  0.85937
                           PC19    PC20    PC21    PC22    PC23    PC24    PC25
Standard deviation     10.17483 9.52356 9.18588 8.96669 8.72231 8.51636 8.25668
Proportion of Variance  0.01257 0.01101 0.01025 0.00976 0.00924 0.00881 0.00828
Cumulative Proportion   0.87194 0.88295 0.89320 0.90296 0.91220 0.92100 0.92928
                          PC26    PC27    PC28    PC29    PC30    PC31    PC32
Standard deviation     8.17220 7.83975 7.70779 7.41979 7.09841 6.90808 6.68419
Proportion of Variance 0.00811 0.00746 0.00721 0.00668 0.00612 0.00579 0.00542
Cumulative Proportion  0.93739 0.94485 0.95206 0.95875 0.96487 0.97066 0.97609
                          PC33    PC34    PC35    PC36    PC37      PC38
Standard deviation     6.65103 6.39709 6.26032 6.20019 5.84560 4.751e-14
Proportion of Variance 0.00537 0.00497 0.00476 0.00467 0.00415 0.000e+00
Cumulative Proportion  0.98146 0.98642 0.99118 0.99585 1.00000 1.000e+00</code></pre>
<pre class="r"><code>summary(PCA.imputed)</code></pre>
<pre><code>Importance of components:
                          PC1      PC2      PC3      PC4      PC5      PC6
Standard deviation     48.536 27.35180 24.28489 21.31422 19.57805 18.88980
Proportion of Variance  0.289  0.09178  0.07235  0.05573  0.04702  0.04378
Cumulative Proportion   0.289  0.38079  0.45314  0.50888  0.55590  0.59968
                            PC7      PC8      PC9     PC10     PC11     PC12
Standard deviation     17.01909 15.60581 15.20284 15.07386 13.85255 13.34940
Proportion of Variance  0.03554  0.02988  0.02836  0.02788  0.02354  0.02186
Cumulative Proportion   0.63521  0.66509  0.69345  0.72132  0.74487  0.76673
                           PC13     PC14     PC15    PC16     PC17     PC18
Standard deviation     12.68231 11.96573 11.47176 11.0944 10.61526 10.29805
Proportion of Variance  0.01973  0.01757  0.01615  0.0151  0.01382  0.01301
Cumulative Proportion   0.78646  0.80403  0.82017  0.8353  0.84910  0.86211
                          PC19    PC20    PC21    PC22   PC23    PC24    PC25
Standard deviation     9.70311 9.43585 8.96692 8.79606 8.7052 8.44992 8.24773
Proportion of Variance 0.01155 0.01092 0.00986 0.00949 0.0093 0.00876 0.00835
Cumulative Proportion  0.87366 0.88458 0.89445 0.90394 0.9132 0.92200 0.93034
                          PC26    PC27   PC28    PC29    PC30    PC31    PC32
Standard deviation     7.99613 7.68151 7.6064 7.21507 7.08926 6.84747 6.67435
Proportion of Variance 0.00784 0.00724 0.0071 0.00639 0.00617 0.00575 0.00547
Cumulative Proportion  0.93819 0.94542 0.9525 0.95891 0.96507 0.97083 0.97629
                          PC33    PC34    PC35    PC36    PC37      PC38
Standard deviation     6.55250 6.34532 6.23390 6.13808 5.78836 4.589e-14
Proportion of Variance 0.00527 0.00494 0.00477 0.00462 0.00411 0.000e+00
Cumulative Proportion  0.98156 0.98650 0.99127 0.99589 1.00000 1.000e+00</code></pre>
<p>So the fraction of total variance explained by the first 10 PCs in both cases in virtually the same. I will use this to argue that the PC covariates we have already used are similarly effective at capturing cellular state. Let’s make a screeplot to show this visually.</p>
<pre class="r"><code>bind_rows(
  summary(PCA.real)$importance %&gt;% t() %&gt;% data.frame() %&gt;% mutate(GeneExpressionMatrix=&quot;With MT genes&quot;) %&gt;% rownames_to_column(&quot;PC&quot;),
  summary(PCA.imputed)$importance %&gt;% t() %&gt;% data.frame() %&gt;% mutate(GeneExpressionMatrix=&quot;Without MT genes&quot;) %&gt;% rownames_to_column(&quot;PC&quot;)) %&gt;% 
  mutate(PC=as.numeric(PC)) %&gt;%
  ggplot(aes(x=PC, y=Cumulative.Proportion, color=GeneExpressionMatrix)) +
  geom_line() +
  ylab(&quot;Cumulative fraction\nvariance explained&quot;) +
  scale_x_continuous(breaks=1:13, limits = c(1,13)) +
  scale_color_discrete(guide = guide_legend(title.position = &quot;top&quot;, nrow = 2)) +
  theme_bw() +
  theme(panel.grid.minor = element_blank(), legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/20200907_Response_Point_04.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>And write out the plot…</p>
<pre class="r"><code>ggsave(&quot;../figures/OriginalArt/ResponseToReviewers.MT.PCA.pdf&quot;, height=3, width=3)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] cowplot_1.0.0     gridExtra_2.3     MASS_7.3-51.4     gplots_3.0.1.1   
 [5] edgeR_3.26.8      limma_3.40.6      knitr_1.26        data.table_1.12.8
 [9] forcats_0.4.0     stringr_1.4.0     dplyr_0.8.3       purrr_0.3.3      
[13] readr_1.3.1       tidyr_1.0.0       tibble_2.1.3      ggplot2_3.2.1    
[17] tidyverse_1.3.0  

loaded via a namespace (and not attached):
 [1] httr_1.4.1         jsonlite_1.6       R.utils_2.9.0      modelr_0.1.5      
 [5] gtools_3.8.1       assertthat_0.2.1   highr_0.8          cellranger_1.1.0  
 [9] yaml_2.2.0         pillar_1.4.2       backports_1.1.5    lattice_0.20-38   
[13] glue_1.3.1         digest_0.6.23      promises_1.1.0     rvest_0.3.5       
[17] colorspace_1.4-1   htmltools_0.4.0    httpuv_1.5.2       R.oo_1.23.0       
[21] pkgconfig_2.0.3    broom_0.5.2        haven_2.2.0        scales_1.1.0      
[25] gdata_2.18.0       later_1.0.0        git2r_0.26.1       generics_0.0.2    
[29] farver_2.0.1       ellipsis_0.3.0     withr_2.1.2        lazyeval_0.2.2    
[33] cli_2.0.0          magrittr_1.5       crayon_1.3.4       readxl_1.3.1      
[37] evaluate_0.14      R.methodsS3_1.7.1  fs_1.3.1           fansi_0.4.0       
[41] nlme_3.1-143       xml2_1.2.2         tools_3.6.1        hms_0.5.2         
[45] lifecycle_0.1.0    munsell_0.5.0      reprex_0.3.0       locfit_1.5-9.1    
[49] compiler_3.6.1     caTools_1.17.1.3   rlang_0.4.1        grid_3.6.1        
[53] rstudioapi_0.10    bitops_1.0-6       labeling_0.3       rmarkdown_1.18    
[57] gtable_0.3.0       DBI_1.0.0          R6_2.4.1           lubridate_1.7.4   
[61] zeallot_0.1.0      workflowr_1.5.0    rprojroot_1.3-2    KernSmooth_2.23-16
[65] stringi_1.4.3      Rcpp_1.0.5         vctrs_0.2.0        dbplyr_1.4.2      
[69] tidyselect_0.2.5   xfun_0.11         </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
