<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>ResponseToReviewer_Point6</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">ResponseToReviewer_Point6</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-10-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190319code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190319code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190319)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree9ea713978506fb284367d2d422c39f04f29d2519targetblank9ea7139a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/9ea713978506fb284367d2d422c39f04f29d2519" target="_blank">9ea7139</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtree9ea713978506fb284367d2d422c39f04f29d2519targetblank9ea7139a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    WorkingManuscript.zip
    Ignored:    WorkingManuscript/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis_temp/.DS_Store
    Ignored:    big_data/
    Ignored:    code/.DS_Store
    Ignored:    code/snakemake_workflow/.DS_Store
    Ignored:    code/snakemake_workflow/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    figures/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/20200907_Response_Point_02.Rmd
    Untracked:  analysis/20200907_Response_Point_04.Rmd
    Untracked:  analysis/20200907_Response_Point_07.Rmd
    Untracked:  analysis/20200907_Response_Point_08.Rmd
    Untracked:  analysis/20200907_Response_Point_11_2.Rmd
    Untracked:  analysis/20201012_MakeSourceData.Rmd
    Untracked:  data/FinalSourceData/
    Untracked:  data/c5.all.v7.1.symbols.gmt
    Untracked:  data/c5.all.v7.1.symbols.gmt.categories.tsv.gz
    Untracked:  data/h.all.v7.1.symbols.gmt

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/20200907_Response_OriginalComments.Rmd
    Modified:   analysis/20200907_Response_Point_06.Rmd
    Modified:   analysis/20200907_Response_Point_09-2.Rmd
    Modified:   analysis/20200907_Response_Point_09.Rmd
    Modified:   analysis/20200907_Response_Point_11.Rmd
    Modified:   analysis/Final_2_DispersionPlots.Rmd
    Modified:   analysis/about.Rmd
    Modified:   analysis_temp/TabulaMuris_analysis2.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>Original reviewer point:</p>
<blockquote>
<p>What do the authors think these findings mean for study systems outside of humans and captive chimpanzees? Both on the technical level (e.g. sample size), and for how their approach could be helpful outside of these species. Generalizing this approach would broaden the impact and audience of the paper.</p>
</blockquote>
<ul>
<li>Referring to our analysis of DE power, this is an excellent question, but empirically answering this question is outside the scope of this paper so we can only speculate. Though, we reason that our findings may depend on the species comparison (species which have diverged more may have greater differences, and thus a similarly sized study may identify more differences), and the tissue type (tissues with high inter-individual variability due to technical or environmental factors may have less power). We have added a discussion of this in the revised manuscript.</li>
<li>Further, here we provide an analysis of GTEx tissues which quantifies the variability (gene-wide overdispersion parameter estimate measured from a negative binomial) across GTEx samples for different tissues. We note that this correlates with the number of eGenes detected in GTEx, and may be serve as a reference as to which tissues may have more power in DE analysis or eQTL analysis due to the degree of inter-individual variability.</li>
</ul>
<p>First load necessary libraries for analysis…</p>
<pre class="r"><code>library(tidyverse)
library(CorShrink)
library(gplots)
library(readxl)
library(&quot;org.Hs.eg.db&quot;)
library(ggrepel)
library(ggpmisc)


source(&quot;../code/CustomFunctions.R&quot;)</code></pre>
<p>Now, read in data and do analysis… In the <code>code</code> section of this repository, the snakemake has already used the same negative binomial fit model used to calculate overdispersion as described in the mansucript, but for all GTEx tissues using all GTEx v8 samples.</p>
<p>First, let’s read in the data of mean expression estimates and overdispersion estimates from that model fit across all genes across all GTEx tissues and also calculate residual overdispersion</p>
<pre class="r"><code>GTEx.Overdispersion.Estimates.fn &lt;- &quot;../output/GtexTissueMatrix.overdispersion.AllModeExtractionSamples.txt.gz&quot;
GTEx.Expression.Estimates.fn &lt;- &quot;../output/GtexTissueMatrix.mu.AllModeExtractionSamples.txt.gz&quot;



MuTable &lt;- read.table(GTEx.Expression.Estimates.fn, header=T, sep=&#39;\t&#39;, check.names = F, row.names = 1) %&gt;%
  rownames_to_column(&quot;X&quot;) %&gt;%
  mutate(gene = gsub(&quot;\\.\\d+&quot;, &quot;&quot;, X, perl=T)) %&gt;% dplyr::select(-X)

OverdispersionTable &lt;- read.table(GTEx.Overdispersion.Estimates.fn, header=T, sep=&#39;\t&#39;, check.names = F, row.names = 1) %&gt;%
  rownames_to_column(&quot;X&quot;)  %&gt;% 
  mutate(gene = gsub(&quot;\\.\\d+&quot;, &quot;&quot;, X, perl=T)) %&gt;% dplyr::select(-X)

ChimpOverdispersionTable &lt;- read.table(&quot;../output/OverdispersionEstimatesFromChimp.txt&quot;, header=T, sep=&#39;\t&#39;)

OverdispersionCombined &lt;- ChimpOverdispersionTable %&gt;%
  dplyr::select(ThisStudy_ChimpHeart=Chimp.Overdispersion,
         ThisStudy_HumanHeart=Human.Overdispersion,
         gene) %&gt;%
  left_join(OverdispersionTable, by=&quot;gene&quot;) %&gt;%
  select_if(~sum(!is.na(.)) &gt; 0) %&gt;% tibble::column_to_rownames(&quot;gene&quot;)

MuCombined &lt;- ChimpOverdispersionTable %&gt;%
  dplyr::select(ThisStudy_ChimpHeart=Chimp.Mean.Expression,
         ThisStudy_HumanHeart=Human.Mean.Expression,
         gene) %&gt;%
  left_join(MuTable, by=&quot;gene&quot;) %&gt;%
  select_if(~sum(!is.na(.)) &gt; 0) %&gt;% tibble::column_to_rownames(&quot;gene&quot;)


GtexColors &lt;- read.table(&quot;../data/GTEx_Analysis_TissueColorCodes.txt&quot;, sep=&#39;\t&#39;, header=T, stringsAsFactors = F) %&gt;% mutate(HEX=paste0(&quot;#&quot;,Color.code))
GtexColors[nrow(GtexColors) + 1,] = c(&quot;ThisStudy_ChimpHeart&quot;,&quot;black&quot;, &quot;#000000&quot;)
GtexColors[nrow(GtexColors) + 1,] = c(&quot;ThisStudy_HumanHeart&quot;,&quot;black&quot;, &quot;#000000&quot;)

TissueColorVector &lt;- GtexColors$HEX
names(TissueColorVector)&lt;-GtexColors$Tissue</code></pre>
<p>Do tissues generally differ in their levels of overdispersion</p>
<pre class="r"><code>#Box whisker plot of overdispersion estimates across all genes for all GTEx tissues
Boxplot &lt;- OverdispersionCombined %&gt;%
  gather() %&gt;% mutate(TissueColor=TissueColorVector[key]) %&gt;%
  ggplot(aes(x=key, y=log(value))) +
    geom_boxplot(outlier.shape = NA, aes(fill = key)) +
    ylab(&quot;overdispersion&quot;) +
    ylim(c(-6,2)) +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5)) +
    theme(legend.position=&quot;none&quot;, axis.title.x=element_blank()) +
    scale_fill_manual(values=TissueColorVector) +
    guides(color=guide_legend(nrow=2,byrow=TRUE))
Boxplot</code></pre>
<p><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, we can see whole blood is the most overdispersed. This could be consistent with the idea that whole blood is a hetergeneous mix of cell types that is different person to person.</p>
<p>Do these different levels of overdispersion relate to power in eQTL studies?</p>
<p>Here, let’s read in the number of egenes per Gtex tissue from the GTEx website</p>
<pre class="r"><code># Do these different levels of overdispersion relate to power in eQTL studies
# First plot the number eGenes for each tissue, versus number of samples
GtexEgeneCounts &lt;- read_excel(&quot;../data/NumberEgenesPerGtexTissue.xlsx&quot;) %&gt;%
  filter(!Percent.eGenes.Over.Total.expressed==0)
Lm.Plot1 &lt;- GtexEgeneCounts %&gt;%
  ggplot(aes(x=`# RNASeq and Genotyped samples`, y=as.numeric(Percent.eGenes.Over.Total.expressed), label=Tissue)) +
  geom_point(aes(color=Tissue)) +
  geom_smooth(method=&quot;lm&quot;, formula=y~x, color=&quot;red&quot;, size=0.5) +
  scale_color_manual(values=TissueColorVector) +
  geom_text_repel(size=1, aes(color=Tissue)) +
  ylab(&quot;Fraction eGenes\n(# eGenes / # genes expressed)&quot;) +
    stat_fit_glance(method = &#39;lm&#39;,
                       method.args = list(formula = formula),
                       geom = &#39;text&#39;,
                       aes(label = paste(&quot;P-value = &quot;, signif(..p.value.., digits = 3), sep = &quot;&quot;)),
                  label.x.npc = &quot;right&quot;, label.x= &quot;middle&quot;, size = 3, label.y=&quot;top&quot;) +
    stat_poly_eq(
               aes(label = paste(..eq.label.., ..rr.label.., sep = &quot;*plain(\&quot;,\&quot;)~&quot;)), 
               label.x.npc = &quot;middle&quot;, label.y=&quot;top&quot;, label.x=&quot;right&quot;,
               formula = formula, parse = TRUE, size = 3) +

  theme_bw() +
  theme(legend.position = &quot;none&quot;)
Lm.Plot1</code></pre>
<p><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Well it sort of looks like whole blood and testis and some cerebellum and spleen are outliers in number of eGenes, and in directions that are concordant with the idea that more overdispersed tissues have less eQTL power. Let’s test this more explicitly with a linear model using sampleCount and MedianOverdispersion as predictors of eGene count</p>
<pre class="r"><code>MedianOverdispersion &lt;- apply(log(OverdispersionCombined), 2, FUN = median, na.rm=T)

DataForLm &lt;- as.data.frame(MedianOverdispersion) %&gt;%
  rownames_to_column(&quot;Tissue&quot;) %&gt;%
  right_join(GtexEgeneCounts, by=&quot;Tissue&quot;) %&gt;%
  drop_na()

My.lm &lt;- lm(Percent.eGenes.Over.Total.expressed ~ `# RNASeq and Genotyped samples` + MedianOverdispersion, DataForLm)
summary(My.lm)</code></pre>
<pre><code>
Call:
lm(formula = Percent.eGenes.Over.Total.expressed ~ `# RNASeq and Genotyped samples` + 
    MedianOverdispersion, data = DataForLm)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.076886 -0.023411 -0.000178  0.023472  0.078091 

Coefficients:
                                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                      -1.614e-01  3.453e-02  -4.675 2.69e-05 ***
`# RNASeq and Genotyped samples`  8.022e-04  3.174e-05  25.277  &lt; 2e-16 ***
MedianOverdispersion             -1.082e-01  1.424e-02  -7.603 1.31e-09 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.0371 on 45 degrees of freedom
Multiple R-squared:  0.9385,    Adjusted R-squared:  0.9357 
F-statistic: 343.2 on 2 and 45 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>plot(My.lm)</code></pre>
<p><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok that model says MedianOverdispersion is a significant predictor for eGene count, wherein more overdispersed tissues have less eGenes. The diagnostic plots for the model fit look reasonable. Let’s also try regressing out the sampleCount effect and fitting the MedianOverdispersion to the residuals.</p>
<pre class="r"><code>#Fit model using just sample count as predictor
My.simpler.lm &lt;- lm(Percent.eGenes.Over.Total.expressed ~ `# RNASeq and Genotyped samples`, DataForLm)

#Save residuals
DataForLm$ResidualAfterRegressingOutSampleCount &lt;- My.simpler.lm$residuals

#Fit MedianOverdispersion to the residuals 
LmOnResidual &lt;- lm(ResidualAfterRegressingOutSampleCount ~ MedianOverdispersion, DataForLm)

summary(LmOnResidual)</code></pre>
<pre><code>
Call:
lm(formula = ResidualAfterRegressingOutSampleCount ~ MedianOverdispersion, 
    data = DataForLm)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.077908 -0.023497  0.000139  0.022630  0.077540 

Coefficients:
                     Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          -0.24628    0.03250  -7.578 1.25e-09 ***
MedianOverdispersion -0.10816    0.01408  -7.680 8.79e-10 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.03671 on 46 degrees of freedom
Multiple R-squared:  0.5619,    Adjusted R-squared:  0.5523 
F-statistic: 58.99 on 1 and 46 DF,  p-value: 8.785e-10</code></pre>
<pre class="r"><code>plot(LmOnResidual)</code></pre>
<p><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-6-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Lm.Plot2 &lt;- DataForLm %&gt;%
  ggplot(aes(x=MedianOverdispersion, y=ResidualAfterRegressingOutSampleCount)) +
  geom_point(aes(color=Tissue)) +
  scale_color_manual(values=TissueColorVector) +
  geom_text_repel(size=1, aes(label=Tissue, color=Tissue)) +
  stat_smooth(method = &quot;lm&quot;, col = &quot;red&quot;, size=0.5) +
  ylab(&quot;Residuals&quot;) +
    stat_poly_eq(
               aes(label = paste(..eq.label.., ..rr.label.., sep = &quot;*plain(\&quot;,\&quot;)~&quot;)), 
               label.x.npc = &quot;middle&quot;, label.y=&quot;top&quot;, label.x=&quot;right&quot;,
               formula = formula, parse = TRUE, size = 3) +
  stat_fit_glance(method = &#39;lm&#39;,
                       method.args = list(formula = formula),
                       geom = &#39;text&#39;,
                       aes(label = paste(&quot;\n\nP-value = &quot;, signif(..p.value.., digits = 3), sep = &quot;&quot;)),
                  label.x.npc = &quot;right&quot;, label.x= &quot;middle&quot;, size = 3, label.y=&quot;top&quot;) +
  theme_bw() +
  theme(legend.position = &quot;none&quot;)
Lm.Plot2</code></pre>
<p><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-6-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, yeah, same result. The trend is significant, and the diagnostic plots point to one outlier, (testis) which clearly isn’t the sole driver of the trend.</p>
<p>So dispersion is a predictor of eQTL mapping power. Let’s also see how it is a predictor of DE analysis power. Let’s compare dispersion estimates</p>
<pre class="r"><code>HumanEgenes &lt;- read.delim(&quot;../data/GTEX_v8_eGenes/Heart_Left_Ventricle.v8.egenes.txt.gz&quot;) %&gt;%
  mutate(gene=str_remove(gene_id, &quot;\\.\\d+$&quot;))

OutputDE &lt;- &#39;../output/Final/TableS2.tab&#39;
DE.results &lt;- read.delim(OutputDE, sep=&#39;\t&#39;, stringsAsFactors = F)


ChimpOverdispersionTable %&gt;%
  mutate(Mean=Human.Residual+Chimp.Residual) %&gt;%
  left_join(HumanEgenes) %&gt;%
  dplyr::select(Mean,Human.Residual, gene, qval) %&gt;%
  ggplot(aes(x=Mean, y=-log10(qval))) +
  geom_point(alpha=0.05) +
  # geom_hex(bins=100) +
  scale_fill_viridis_c() +
  theme_bw()</code></pre>
<p><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>formula &lt;- y ~ x  
DispersionVsDE &lt;- ChimpOverdispersionTable %&gt;%
  mutate(Mean=Human.Residual+Chimp.Residual) %&gt;%
  # mutate(Mean=log(Human.Overdispersion) + log(Chimp.Overdispersion)) %&gt;%
  left_join(DE.results, by=c(&quot;gene&quot;=&quot;Ensembl_geneID&quot;)) %&gt;%
  dplyr::select(Mean, Chimp.Residual, gene, adj.P.Val) %&gt;%
  ggplot(aes(y=Mean, x=-log10(adj.P.Val))) +
  geom_point(alpha=0.05) +
  # geom_hex(bins=100) +
  geom_smooth(method=&#39;rlm&#39;, formula=formula) +
  ylim(c(-6,6)) +
  ylab(&quot;Genewise dispersion&quot;) +
  xlab(&quot;Interspecies DE -log10(P)&quot;) +
      stat_poly_eq(
               aes(label = paste(..eq.label.., ..rr.label.., sep = &quot;*plain(\&quot;,\&quot;)~&quot;)), 
               label.x.npc = &quot;middle&quot;, label.y=&quot;top&quot;, label.x=&quot;right&quot;,
               formula = formula, parse = TRUE, size = 3) +
  stat_fit_glance(method = &#39;lm&#39;,
                       method.args = list(formula = formula),
                       geom = &#39;text&#39;,
                       aes(label = paste(&quot;\n\nP-value = &quot;, signif(..p.value.., digits = 3), sep = &quot;&quot;)),
                  label.x.npc = &quot;right&quot;, label.x= &quot;middle&quot;, size = 3, label.y=&quot;top&quot;) +
  theme_bw()

DispersionVsDE</code></pre>
<p><img src="figure/20200907_Response_Point_07.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Make plots</p>
<pre class="r"><code>ggsave(&quot;../figures/OriginalArt/ResponseToReviewers.Overdispersion.boxplots.pdf&quot;, Boxplot, height=4, width=8)
ggsave(&quot;../figures/OriginalArt/ResponseToReviewers.Overdispersion.lm.2.pdf&quot;, Lm.Plot2, height=3, width=4)
ggsave(&quot;../figures/OriginalArt/ResponseToReviewers.Overdispersion.lm.1.pdf&quot;, Lm.Plot1, height=3, width=4)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] cowplot_1.0.0        gridExtra_2.3        edgeR_3.26.8        
 [4] limma_3.40.6         MASS_7.3-53          ggpmisc_0.3.6       
 [7] ggrepel_0.8.1        org.Hs.eg.db_3.8.2   AnnotationDbi_1.46.1
[10] IRanges_2.18.3       S4Vectors_0.22.1     Biobase_2.44.0      
[13] BiocGenerics_0.30.0  readxl_1.3.1         gplots_3.0.1.1      
[16] CorShrink_0.1-6      forcats_0.4.0        stringr_1.4.0       
[19] dplyr_1.0.2          purrr_0.3.3          readr_1.3.1         
[22] tidyr_1.0.0          tibble_3.0.3         ggplot2_3.3.2       
[25] tidyverse_1.3.0     

loaded via a namespace (and not attached):
 [1] colorspace_1.4-1   ellipsis_0.3.0     rprojroot_1.3-2    corpcor_1.6.9     
 [5] fs_1.3.1           rstudioapi_0.10    farver_2.0.3       bit64_0.9-7       
 [9] fansi_0.4.0        lubridate_1.7.9    xml2_1.2.2         splines_3.6.1     
[13] codetools_0.2-16   R.methodsS3_1.7.1  pscl_1.5.2         doParallel_1.0.15 
[17] knitr_1.26         polynom_1.4-0      jsonlite_1.6       workflowr_1.5.0   
[21] broom_0.7.0        Rmpfr_0.7-2        ashr_2.2-39        dbplyr_1.4.2      
[25] R.oo_1.23.0        compiler_3.6.1     httr_1.4.1         backports_1.1.5   
[29] assertthat_0.2.1   Matrix_1.2-18      cli_2.0.0          later_1.0.0       
[33] htmltools_0.4.0    tools_3.6.1        gmp_0.5-13.5       gtable_0.3.0      
[37] glue_1.4.2         reshape2_1.4.3     Rcpp_1.0.5         cellranger_1.1.0  
[41] vctrs_0.3.4        nlme_3.1-143       gdata_2.18.0       iterators_1.0.12  
[45] xfun_0.11          CVXR_0.99-7        scs_1.3-2          rvest_0.3.5       
[49] lifecycle_0.2.0    ECOSolveR_0.5.3    gtools_3.8.1       scales_1.1.1      
[53] hms_0.5.2          promises_1.1.0     yaml_2.2.0         memoise_1.1.0     
[57] stringi_1.4.3      RSQLite_2.1.4      SQUAREM_2017.10-1  corrplot_0.84     
[61] foreach_1.4.7      caTools_1.17.1.3   truncnorm_1.0-8    shape_1.4.5       
[65] rlang_0.4.7        pkgconfig_2.0.3    bitops_1.0-6       evaluate_0.14     
[69] lattice_0.20-38    labeling_0.3       bit_1.1-14         tidyselect_1.1.0  
[73] plyr_1.8.6         magrittr_1.5       R6_2.4.1           generics_0.0.2    
[77] DBI_1.0.0          mgcv_1.8-31        pillar_1.4.6       haven_2.2.0       
[81] withr_2.1.2        mixsqp_0.2-2       modelr_0.1.5       crayon_1.3.4      
[85] KernSmooth_2.23-16 rmarkdown_1.18     locfit_1.5-9.1     grid_3.6.1        
[89] blob_1.2.0         git2r_0.26.1       reprex_0.3.0       digest_0.6.23     
[93] httpuv_1.5.2       R.utils_2.9.0      munsell_0.5.0      glmnet_3.0-2      
[97] viridisLite_0.3.0 </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
