<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Comparative_eQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bfairkun/Comparative_eQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-10-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Comparative_eQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtreeaae75ef6dd8879e73aac54684d7fdd20628c2d79targetblankaae75efa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/Comparative_eQTL/tree/aae75ef6dd8879e73aac54684d7fdd20628c2d79" target="_blank">aae75ef</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunComparativeeQTLtreeaae75ef6dd8879e73aac54684d7fdd20628c2d79targetblankaae75efa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    WorkingManuscript.zip
    Ignored:    WorkingManuscript/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis_temp/.DS_Store
    Ignored:    big_data/
    Ignored:    code/.DS_Store
    Ignored:    code/snakemake_workflow/.DS_Store
    Ignored:    code/snakemake_workflow/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    data/FinalSourceData/.DS_Store
    Ignored:    data/PastAnalysesDataToKeep/.DS_Store
    Ignored:    figures/
    Ignored:    output/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/aae75ef6dd8879e73aac54684d7fdd20628c2d79/analysis/index.Rmd" target="_blank">aae75ef</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-10-26
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7fc5e9c429452ff664bd2a9a33df02ae705f25ca/analysis/index.Rmd" target="_blank">7fc5e9c</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-09-09
</td>
<td>
update site, address reviewers
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/7fc5e9c429452ff664bd2a9a33df02ae705f25ca/docs/index.html" target="_blank">7fc5e9c</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-09-09
</td>
<td>
update site, address reviewers
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/bfa790a7a733489e6b3ae25150aa7487edd2e005/analysis/index.Rmd" target="_blank">bfa790a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-10-10
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/bfa790a7a733489e6b3ae25150aa7487edd2e005/docs/index.html" target="_blank">bfa790a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-10-10
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/71db33a2a4e18c3fed650e74c106b7aa32810f74/analysis/index.Rmd" target="_blank">71db33a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-10-03
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/71db33a2a4e18c3fed650e74c106b7aa32810f74/docs/index.html" target="_blank">71db33a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-10-03
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/cf50e71ecd3367076842e675c942f3b824c92b5c/analysis/index.Rmd" target="_blank">cf50e71</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-21
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/cf50e71ecd3367076842e675c942f3b824c92b5c/docs/index.html" target="_blank">cf50e71</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-21
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/b62b0894dcb31cc9678396fd614158b111871254/analysis/index.Rmd" target="_blank">b62b089</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-21
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/b62b0894dcb31cc9678396fd614158b111871254/docs/index.html" target="_blank">b62b089</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-21
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/fdbae1200f387d38a61503dd2f2ce00b206ed3ae/analysis/index.Rmd" target="_blank">fdbae12</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-08-14
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/641b1d9e3bb86a49be929bf1d1f640c9a0744c25/analysis/index.Rmd" target="_blank">641b1d9</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/641b1d9e3bb86a49be929bf1d1f640c9a0744c25/docs/index.html" target="_blank">641b1d9</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/34e6b4f4c30464c6c32619d985b9663b392de4fb/analysis/index.Rmd" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/34e6b4f4c30464c6c32619d985b9663b392de4fb/docs/index.html" target="_blank">34e6b4f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-07-11
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/c3dbe5a11f848a94299edef3e9f2117f256fe270/analysis/index.Rmd" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/c3dbe5a11f848a94299edef3e9f2117f256fe270/docs/index.html" target="_blank">c3dbe5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-19
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/22eda3af4c5359691dddf7d4d2d577b41b068de5/analysis/index.Rmd" target="_blank">22eda3a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-12
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/22eda3af4c5359691dddf7d4d2d577b41b068de5/docs/index.html" target="_blank">22eda3a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-12
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/analysis/index.Rmd" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/7ef6fe2b4cac728106b0582bf0ddf2c49c0a06b7/docs/index.html" target="_blank">7ef6fe2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-06-11
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/8ab5bbfa4ea16595aba2ac3f91dd3294605851b2/docs/index.html" target="_blank">8ab5bbf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-05-02
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/f47ec35c04da477a94bfe2482822041a31e61ec5/analysis/index.Rmd" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
<td>
updated site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/f47ec35c04da477a94bfe2482822041a31e61ec5/docs/index.html" target="_blank">f47ec35</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-25
</td>
<td>
updated site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/e5923352fa1aeb42041e564534e8e145011d505d/analysis/index.Rmd" target="_blank">e592335</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-04
</td>
<td>
build site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/e5923352fa1aeb42041e564534e8e145011d505d/docs/index.html" target="_blank">e592335</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-04-04
</td>
<td>
build site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/905723c66cb641274c19e3d5e81cc44c99b9f2fd/docs/index.html" target="_blank">905723c</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-03-20
</td>
<td>
updated link
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8f7c7373514a9631e3771854004805697df89d83/analysis/index.Rmd" target="_blank">8f7c737</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-03-20
</td>
<td>
updated link
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/8dd795fa81ffe40134fc952f4249f95d3efbefff/analysis/index.Rmd" target="_blank">8dd795f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-03-20
</td>
<td>
First analysis on workflowr
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/8dd795fa81ffe40134fc952f4249f95d3efbefff/docs/index.html" target="_blank">8dd795f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-03-20
</td>
<td>
First analysis on workflowr
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/Comparative_eQTL/5b477a35615ae185890e9a1c136f147b97a1b789/docs/index.html" target="_blank">5b477a3</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-03-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/Comparative_eQTL/blob/b1207ee02c0e7532688dcf3a46d924aa61259917/analysis/index.Rmd" target="_blank">b1207ee</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-03-19
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This site contains exploratory analysis notebook and R code to make most of the figures for <a href="https://elifesciences.org/articles/59929">DOI: 10.7554/eLife.59929</a>. This site only contains R code (Rmarkdowns), including code to make most all of the figures from the paper. Many of these Rmarkdowns read data included in the <a href="https://github.com/bfairkun/Comparative_eQTL">associated repository</a> in the <code>data</code> and <code>output</code> directories. To run through R code on this site, clone the repository, and run Rmarkdowns from the <code>analysis</code> directory. Besides the figure source data and supplemental tables in the publication, other processed or semi-processed data (eg count tables) is available in the associated repository. Reproducible code for other aspects of this project is in a snakemake pipeline in <code>code</code>. See the <code>code/snakemake_workflow/README</code> for more info on how to run that.</p>
<div id="abstract" class="section level2">
<h2>Abstract</h2>
<p>Inter-individual variation in gene expression has been shown to be heritable and is often associated with differences in disease susceptibility between individuals. Many studies focused on mapping associations between genetic and gene regulatory variation, yet much less attention has been paid to the evolutionary processes that shape the observed differences in gene regulation between individuals in humans or any other primate. To begin addressing this gap, we performed a comparative analysis of gene expression variability and expression quantitative trait loci (eQTLs) in humans and chimpanzees, using gene expression data from primary heart samples. We found that expression variability in both species is often determined by non-genetic sources, such as cell-type heterogeneity. However, we also provide evidence that inter-individual variation in gene regulation can be genetically controlled, and that the degree of such variability is generally conserved in humans and chimpanzees. In particular, we found a significant overlap of orthologous genes associated with eQTLs in both species. We conclude that gene expression variability in humans and chimpanzees often evolves under similar evolutionary pressures.</p>
</div>
<div id="raw-data" class="section level2">
<h2>Raw data</h2>
<p>Raw data for this project includes post mortem heart-tissues:</p>
<ul>
<li>RNA-seq of 10 human and 10 chimp samples <a href="https://www.nature.com/articles/s41598-018-33478-9">Pavlovic et al</a> (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE110471">GEO accession GSE110471</a>)</li>
<li>RNA-seq of ~39 human samples (GTEx v7, Heart left ventricle, see Fig1 source data in paper for exact samples, requires dbGaP to access raw data)</li>
<li>RNA-seq of 29 new chimp samples (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE151397">GEO accession GSE151397</a>). This GEO entry also contains a count table for all RNA-seq samples analyzed in this project</li>
<li>Whole genome sequencing of 39 chimp samples (Raw reads at <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA635393/">SRA accession PRJNA635393</a>, genotype calls vcf at <a href="https://www.ebi.ac.uk/eva/?eva-study=PRJEB39475">EVA accession PRJEB39475</a>).</li>
</ul>
<div id="code-for-qc" class="section level3">
<h3>Code for QC</h3>
<ul>
<li><a href="../output/multiqc.html">fastqc</a></li>
</ul>
<div id="genotype-data" class="section level4">
<h4>Genotype data</h4>
<ul>
<li><a href="">Genotype call rates</a></li>
<li><a href="20190321_Check-Kinship-And-PopulationStructure.html">kinship</a></li>
<li><a href="20190326_PCA.html">PCA</a></li>
<li><a href="20190326_Admixture.html">Admixture</a></li>
</ul>
</div>
<div id="expression-data" class="section level4">
<h4>Expression data</h4>
<ul>
<li><a href="20190325_MergingRNASeqLanes.html">Check that RNA-seq data segregates by biological sample over sequencing flow cell</a></li>
<li><a href="20190320_Check-RNAseq-PCs.html">Checking RNA-seq data for covariates</a></li>
<li><a href="20190429_RNASeqSTAR_quantifications.html">STAR Aligner vs kallisto pseudoaligner for gene quantification</a></li>
</ul>
</div>
<div id="prepare-data-for-eqtl-testing" class="section level4">
<h4>Prepare data for eQTL testing</h4>
<ul>
<li><a href="20190327_MakeFamPhenotypeFile.html">Make phenotype table for testing</a></li>
<li><a href="20190327_MakeCovariateFiles.html">Make covariate table for testing</a></li>
</ul>
</div>
</div>
<div id="code-for-results-and-exploration" class="section level3">
<h3>Code for results and exploration</h3>
<div id="association-testing-with-various-models" class="section level4">
<h4>Association testing with various models</h4>
<ul>
<li><a href="20190412_Check_eQTLs.html">first iteration</a>: description: lmm with KING-robust GRM thresholded at 0, and 3 genotype PCs</li>
<li><a href="20190421_RegressOutRNASeqPCs.html">Check residuals after regressing out some covariates</a></li>
<li><a href="20190424_Check_eQTLs.html">second iteration</a>: description: lm with 5 genotype PCs (PCs 4 and 5 takes into account some first hand relatedness) and more stringent genotype filtering. Also, outlier sample MD_And dropped from analysis</li>
<li><a href="20190428_Check_eQTLs.html">Third iteration</a></li>
<li><a href="20190502_Check_eQTLs.html">fourth iteration</a>: description, lmm with 4 PCs and 3 Genotype PCs, used STAR RNA-seq CPM for less outliers. Fixed big bug that was permuting samples, resulting in no true hits in previous iterations. Here I used standardization and qqnorm.</li>
<li>Final iteration: see paper for methods details. In short, I used lmm with kinship matrix (implented in MatrixEQTL) with 10 expression PCs as covariates. QQ-plots of permutation control and real samples indicates some eQTLs and well calibrated P-values.</li>
</ul>
</div>
<div id="conservation-and-gogsea-analysis" class="section level4">
<h4>Conservation and GO/GSEA analysis</h4>
<ul>
<li><a href="20190521_eQTL_CrossSpeciesEnrichment.html">GO analysis, FDR=0.1</a> overlap enrichment analysis of eGenes across humans and chumps, and gene ontology analysis of eGenes based on eGene classification defined at FDR=0.1 threshold. Also, a set of similar analyses using GSEA methodology, which is based on relative ranking of eGenes between species.</li>
<li><a href="20190606_eGene_Conservation.html">Conservation analysis, FDR=0.1</a> analysis of conservation of coding sequence (percent identity and dN/dS) based on eGene classification defined at FDR=0.1</li>
<li><a href="20190606_eGene_Conservation_TopN.html">Conservation analysis, HumanTop600_eGenes</a> analysis of conservation of coding sequence (percent identity and dN/dS) based on eGene classification defined at FDR=0.1 for chimp and top600 qvalue genes for human.</li>
<li><a href="20190821_eGeneTissueCount.html">Tissue sharing, both FDR=0.1 and Top600</a> analysis of number of GTEx tissues eGenes are detected in for shared and species specific eGenes. As before, I classify species specific and shared eGenes using either FDR=0.1 for both species, or using FDR-0.1 for chimp and top600 qvalue for human.</li>
<li><a href="20190617_ExpressionVarianceOfEgenes.html">Conservation analysis, most high-variance genes</a> analysis of conservation of coding sequence (percent identity and dN/dS) based on within species variance of expression. A useful comparison for the similar analysis above.</li>
<li><a href="20190716_VarianceInsteadOfEgenes.html">More thourough analyses looking at gene variance</a> within and between species after adjusting for expression.</li>
<li><a href="20190930_OverdispersionEstimates.html">Different model of gene variance character</a> using advice from Abhishek</li>
<li><a href="20190930_OverdispersionEstimates_HBV_excluded.html">Different model of gene variance character</a> using advice from Abhishek. But this time leaving out the virus challenged chimps</li>
</ul>
</div>
<div id="power-analysis-for-inter-species-differential-expression" class="section level4">
<h4>Power analysis for inter-species differential expression</h4>
<ul>
<li><a href="20190613_PowerAnalysis.html">PowerAnalysisFromOrinalDataset</a> DE gene analysis based on subsampling ~39 chimp samples &amp; 50 human samples (mostly GTEx)… Note that there are some outlier samples that I want to purge in later iterations of this analysis.</li>
</ul>
</div>
<div id="differential-contacts" class="section level4">
<h4>Differential contacts</h4>
<ul>
<li>Do differential DNA contacts between species explain differences in eGene character between species.</li>
<li><a href="20190627_DiffContactsEgenes.html">First crude analysis</a> In this analysis I asked whether the there is a correlation between the rank difference in eGenes between species, and the difference in the sum of contacts in each species’ cis-window for each gene</li>
<li>There was a slight but significant correlation, but I worry about a potential bias introduced by the fact that eGene significance for humans was based on cis-eQTL calling with a 1MB window and chimp was a 250kB window. I should control for this.</li>
<li><a href="20190808_HumanLeadSnps.html">Check that human lead snps in 250kB window are reasonable</a> As expected, the p-value for lead snps within 250kb is well correlated for the eGene p-value, with the exception that there are many genes with much lower eGene pvalues probably because there is a better SNP &gt;250kb away</li>
<li><a href="20190708_DiffContactsEgenes_CisWindowControlled.html">After controlling for cis-window-size</a></li>
</ul>
</div>
<div id="shared-polymorphic-loci" class="section level4">
<h4>Shared polymorphic loci</h4>
<ul>
<li><a href="20190708_SpeciesSharedPolymorphisms_InitialQQPlot.html">QQ Plot of species shared polymorphisms</a> Of all the shared polymorphic loci (putative targets of balancing selection), do they have inflated cis-eQTL P-values compared to a random set of snp-gene tests.</li>
</ul>
</div>
</div>
<div id="code-for-final-figures-and-analysis" class="section level3">
<h3>Code for final figures and analysis</h3>
<ul>
<li>todo… haven’t build these markdowns yet.</li>
</ul>
</div>
</div>
<div id="reviewer-comments" class="section level2">
<h2>Reviewer comments</h2>
<ul>
<li>See <a href="20200907_Response_OriginalComments.html">reviewer comments</a> and associated analyses to address.</li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
