---
title: "20190913_VarianceFromNB_Model_Comparison"
author: "Ben Fair"
date: "9/13/2019"
output: html_document
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)
```

As per Ahbishek's advice, I should try using a different method of estimating gene variance. That is: I should fit the counts data to a negative binomial and to obtain an estimate the variance of each gene's underlying gamma distribbution. Here I will try implementing that method, comparing to the previous method, which was to take the residual of a loess fit of the trend between mean(log(CPM)) and standard deviation(log(CPM)).


```{r load-libraries, message=F, warning=F}
library(tidyverse)
library(knitr)
library("edgeR")
library(stats)
library(corrplot)
library(gplots)
library("clusterProfiler")
library("org.Hs.eg.db")
library(enrichplot)
library(MASS)




# Helper function reference in body of later function
rep.col<-function(x,n){
   matrix(rep(x,each=n), ncol=n, byrow=TRUE)
}

# Modified from the function in the PowerAnalysis Rmarkdown.
# Function to return RPKM table from chimp and human datasets (n=38 each)
# Use GenesToKeep argument to subset the same list that I compared for eQTLs
GetCPMCountTable <-function(ChimpCountTableFile, HumanCountTableFile, SubsampleSize, GenesToKeep, ChimpSampleDrop=NULL, HumanSampleDrop=NULL)
  #if SubsampleSize parameter == 0, use full table, otherwise, subsample from it
  {
   FullChimpData <- read.table(gzfile(ChimpCountTableFile), header=T, check.names=FALSE, skip=1)
   FullHumanData <- read.table(gzfile(HumanCountTableFile), header=T, check.names=FALSE, skip=1)
   
  if (!is.null(ChimpSampleDrop)){
     FullChimpData <- FullChimpData %>% dplyr::select(-ChimpSampleDrop)
   }   
  if (!is.null(HumanSampleDrop)){
     FullHumanData <- FullHumanData %>% dplyr::select(-HumanSampleDrop)
   }

   if (SubsampleSize==0){
     CountTableChimp <- FullChimpData
     colnames(CountTableChimp) <- paste0("C.", colnames(CountTableChimp))
     CountTableHuman <- FullHumanData
     colnames(CountTableHuman) <- paste0("H.", colnames(CountTableHuman))
     
   } else {
     CountTableChimp <- FullChimpData %>% dplyr::select(c(1:6, sample(7:length(FullChimpData), SubsampleSize)))
     colnames(CountTableChimp) <- paste0("C.", colnames(CountTableChimp))
     
     CountTableHuman <- FullHumanData %>% dplyr::select(c(1:6, sample(7:length(FullHumanData), SubsampleSize)))
     colnames(CountTableHuman) <- paste0("H.", colnames(CountTableHuman))
   }
   
CombinedTable <- inner_join(CountTableChimp[,c(1,7:length(CountTableChimp))], CountTableHuman[,c(1,7:length(CountTableHuman))], by=c("C.Geneid"="H.Geneid")) %>%
  column_to_rownames("C.Geneid") %>% as.matrix()

SpeciesFactor <- colnames(CombinedTable) %>% substr(1,1) %>% factor() %>% unclass() %>% as.character()

d0 <- DGEList(CombinedTable)
d0 <- calcNormFactors(d0)



d <- d0[GenesToKeep,]
mm <- model.matrix(~0 + SpeciesFactor)
# y <- voom(d, mm, normalize.method="cyclicloess", plot=F)

# GeneLengths <- inner_join(CountTableChimp[,c("C.Geneid", "C.Length")], CountTableHuman[,c("H.Geneid", "H.Length")], by=c("C.Geneid"="H.Geneid"))
# GeneLengthMatrix <- cbind(
#   rep.col(log2(GeneLengths$C.Length/1000), length(CountTableChimp)-6),
#   rep.col(log2(GeneLengths$H.Length/1000), length(CountTableHuman)-6))
# rownames(GeneLengthMatrix) <- GeneLengths$C.Geneid
# y$E <- y$E - GeneLengthMatrix[rownames(y$E),]

return(cpm(d, log=T))
# return(d0)

}


# Modified from the function in the PowerAnalysis Rmarkdown.
# Function to return list of count tables (in either log(RPKM), or raw counts) from chimp and human datasets (n=38 each)
# Use GenesToKeep argument to subset the same list that I compared for eQTLs
GetCountTable <-function(ChimpCountTableFile, HumanCountTableFile, SubsampleSize, GenesToKeep, ChimpSampleDrop=NULL, HumanSampleDrop=NULL)
  #if SubsampleSize parameter == 0, use full table, otherwise, subsample from it
  {
   FullChimpData <- read.table(gzfile(ChimpCountTableFile), header=T, check.names=FALSE, skip=1)
   FullHumanData <- read.table(gzfile(HumanCountTableFile), header=T, check.names=FALSE, skip=1)
   
  if (!is.null(ChimpSampleDrop)){
     FullChimpData <- FullChimpData %>% dplyr::select(-ChimpSampleDrop)
   }   
  if (!is.null(HumanSampleDrop)){
     FullHumanData <- FullHumanData %>% dplyr::select(-HumanSampleDrop)
   }

   if (SubsampleSize==0){
     CountTableChimp <- FullChimpData
     colnames(CountTableChimp) <- paste0("C.", colnames(CountTableChimp))
     CountTableHuman <- FullHumanData
     colnames(CountTableHuman) <- paste0("H.", colnames(CountTableHuman))
     
   } else {
     CountTableChimp <- FullChimpData %>% dplyr::select(c(1:6, sample(7:length(FullChimpData), SubsampleSize)))
     colnames(CountTableChimp) <- paste0("C.", colnames(CountTableChimp))
     
     CountTableHuman <- FullHumanData %>% dplyr::select(c(1:6, sample(7:length(FullHumanData), SubsampleSize)))
     colnames(CountTableHuman) <- paste0("H.", colnames(CountTableHuman))
   }
   
CombinedTable <- inner_join(CountTableChimp[,c(1,7:length(CountTableChimp))], CountTableHuman[,c(1,7:length(CountTableHuman))], by=c("C.Geneid"="H.Geneid")) %>%
  column_to_rownames("C.Geneid") %>% as.matrix()

SpeciesFactor <- colnames(CombinedTable) %>% substr(1,1) %>% factor() %>% unclass() %>% as.character()

d0 <- DGEList(CombinedTable)
d0 <- calcNormFactors(d0)



d <- d0[GenesToKeep,]
mm <- model.matrix(~0 + SpeciesFactor)
# y <- voom(d, mm, normalize.method="cyclicloess", plot=F)

# GeneLengths <- inner_join(CountTableChimp[,c("C.Geneid", "C.Length")], CountTableHuman[,c("H.Geneid", "H.Length")], by=c("C.Geneid"="H.Geneid"))
# GeneLengthMatrix <- cbind(
#   rep.col(log2(GeneLengths$C.Length/1000), length(CountTableChimp)-6),
#   rep.col(log2(GeneLengths$H.Length/1000), length(CountTableHuman)-6))
# rownames(GeneLengthMatrix) <- GeneLengths$C.Geneid
# y$E <- y$E - GeneLengthMatrix[rownames(y$E),]

return(d)
# return(d0)

}

```


Get list of cis-eQTL tested genes that were tested in both chimps and humans and one-to-one orthologs and also in DE gene count tables (from reads mapped to ortho exons):

```{r}

CountTableChimpFile <- '../output/PowerAnalysisFullCountTable.Chimp.subread.txt.gz'
CountTableHumanFile <- '../output/PowerAnalysisFullCountTable.Human.subread.txt.gz'

eQTLs <- read.table(gzfile("../data/PastAnalysesDataToKeep/20190521_eQTLs_250kB_10MAF.txt.gz"), header=T)

# List of chimp tested genes
ChimpTestedGenes <- rownames(read.table('../output/ExpressionMatrix.un-normalized.txt.gz', header=T, check.names=FALSE, row.names = 1))

ChimpToHumanGeneMap <- read.table("../data/Biomart_export.Hsap.Ptro.orthologs.txt.gz", header=T, sep='\t', stringsAsFactors = F)

# Of this ortholog list, how many genes are one2one
OneToOneMap <- ChimpToHumanGeneMap %>%
  filter(Chimpanzee.homology.type=="ortholog_one2one")

# Read gtex heart egene list
# Only consider those that were tested in both species and are one2one orthologs
GtexHeartEgenes <- read.table("../data/Heart_Left_Ventricle.v7.egenes.txt.gz", header=T, sep='\t', stringsAsFactors = F) %>%
  mutate(gene_id_stable = gsub(".\\d+$","",gene_id)) %>%
  filter(gene_id_stable %in% OneToOneMap$Gene.stable.ID) %>%
  mutate(chimp_id = plyr::mapvalues(gene_id_stable, OneToOneMap$Gene.stable.ID,  OneToOneMap$Chimpanzee.gene.stable.ID, warn_missing = F)) %>%
  filter(chimp_id %in% ChimpTestedGenes)


EgenesTested <- gsub("\\..+", "", GtexHeartEgenes$gene_id, perl=T)
length(EgenesTested)
GenesInDESet <- read.table(gzfile(CountTableChimpFile), header=T, check.names=FALSE, skip=1)$Geneid
length(GenesInDESet)

GeneList <- intersect(as.character(GenesInDESet),EgenesTested)
kable(head(GeneList))
length(GeneList)
```

Ok, now get CountTable of 38 chimps, 38 humans (based on remapped data that maps to orthologous exons) of RPKM, and filtered for the outlier samples that I also left out of the DE gene power analysis

```{r}
HumanSamplesToDrop <- c(c("SRR1507229","SRR603918", "SRR1478149", "SRR598509", "SRR613186"), c("SRR1489693", "SRR598148", "59167", "SRR1478900", "SRR1474730", "61317"))
ChimpSamplesToDrop <- c("Little_R")




CountTable <- GetCPMCountTable(CountTableChimpFile,
              CountTableHumanFile,
              0, GeneList, ChimpSampleDrop=ChimpSamplesToDrop, HumanSampleDrop = HumanSamplesToDrop)

CountTable.Counts <- GetCountTable(CountTableChimpFile,
              CountTableHumanFile,
              0, GeneList, ChimpSampleDrop=ChimpSamplesToDrop, HumanSampleDrop = HumanSamplesToDrop)




qplot(apply(CountTable,1,mean), sqrt(apply(CountTable,1,var)), alpha=0.05) +
  geom_smooth(method="loess", show_guide = FALSE, se=F) +
  xlab("Mean expression (log(CPM)") +
  ylab("Standard deviation") +
  theme_bw() +
  theme(legend.position = "none")

scatter.smooth(apply(CountTable,1,mean), sqrt(apply(CountTable,1,var)), color="red")



```

Ok now write functions to estimate the variance of the underlying gamma distribution for each gene, using a negative binomial fit to the count data.

```{r}
fit <- MASS::glm.nb(CountTable.Counts$counts[1,] ~ offset(log(CountTable.Counts$samples$lib.size))  + 1)
var <- fit$coefficients**2/fit$theta

GetVarianceOfUnderlyingGamma <- function(Y, libsizes, genesize=1){
  tryCatch(
    expr = {
      GeneSizeVector <- rep(genesize, length(libsizes))
      fit <- MASS::glm.nb(Y ~ offset(log(libsizes)-log(GeneSizeVector))  + 1)
      # var <- fit$coefficients**2/fit$theta
      var <- 1/fit$theta

    },
    error = function(e){var <- NA},
    finally = {
      return(var)
    }
  )
}



GetInterceptOfNB.fit <- function(Y, libsizes){
  tryCatch(
    expr = {
      fit <- MASS::glm.nb(Y ~ offset(log(libsizes))  + 1)
      intercept <- fit$coefficients
    },
    error = function(e){intercept <- NA},
    finally = {
      return(intercept)
    }
  )
}





# Check that the intercept of the NB fit corresponds to mean
Means<- apply(CountTable.Counts$counts[1:100,], 1, GetInterceptOfNB.fit, CountTable.Counts$samples$lib.size)
MeansCPM <- apply(CountTable[1:100,],1,mean)
qplot(Means, MeansCPM) +
  xlab('Mean (intercept of the NB fit)') +
  ylab('Mean of log(CPM)') +
  theme_bw()


A<- apply(CountTable.Counts$counts[1:1000,], 1, GetVarianceOfUnderlyingGamma, CountTable.Counts$samples$lib.size, 400)

Gene.summarystats <- data.frame(
                     SDlogexpression = sqrt(apply(CountTable,1,var)),
                     logmean = apply(CountTable,1,mean))
Gene.summarystats.lo <- loess(SDlogexpression ~ logmean, Gene.summarystats)
Gene.summarystats$SD.minus.loess <- Gene.summarystats$SDlogexpression - predict(Gene.summarystats.lo, Gene.summarystats$logmean)

B<- data.frame(NB=A, LoessWay=Gene.summarystats$SD.minus.loess[1:1000])
R<-cor(B$NB, B$LoessWay, method='pearson')
lb1 <- paste("~R^2==~", round(R**2,2))
qplot(B$NB, B$LoessWay) +
  xlab(expression(paste("Variance of underlying gamma... I think?, ", mu^"2","/", theta))) +
  ylab("Residual from loess curve of\nmean-SD-trend after log-transform") +
  annotate("text",x=Inf,y=-Inf, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()

```

Ok now see there is a strong correlation between human variance and chimp variance.

```{r}

GetVarianceOfUnderlyingGamma_lengthAdjusted_FromTable <- function(MyCountTable, genesizes){
  Out <- matrix(nrow=nrow(MyCountTable))
  LibSize <- colSums(MyCountTable)
  for(i in 1:nrow(MyCountTable)) {
    Var <- GetVarianceOfUnderlyingGamma(t(MyCountTable[i,]), LibSize, genesizes[i])
    Out[i,1] <- Var
    rownames(Out) <- rownames(MyCountTable)
  }
  return(Out)
}

# GetVarianceOfUnderlyingGamma(t(ChimpCountTable[1,]), colSums(ChimpCountTable), 50)

ChimpCountTable <- CountTable.Counts$counts %>% as.data.frame() %>% 
  dplyr::select(contains("C."))
ChimpGeneLengths <- read.table(gzfile('../output/PowerAnalysisFullCountTable.Chimp.subread.txt.gz'), header=T, check.names=FALSE, skip=1) %>% dplyr::select(Geneid, Length) %>%
  right_join((ChimpCountTable %>% rownames_to_column('Geneid')), by='Geneid') %>% pull(Length)
GetVarianceOfUnderlyingGamma(t(ChimpCountTable[3,]), colSums(ChimpCountTable), ChimpGeneLengths[3])

Chimp.var <- GetVarianceOfUnderlyingGamma_lengthAdjusted_FromTable(ChimpCountTable, ChimpGeneLengths)


HumanCountTable <- CountTable.Counts$counts %>% as.data.frame() %>% 
  dplyr::select(contains("H."))
HumanGeneLengths <- read.table(gzfile('../output/PowerAnalysisFullCountTable.Human.subread.txt.gz'), header=T, check.names=FALSE, skip=1) %>% dplyr::select(Geneid, Length) %>%
  right_join((HumanCountTable %>% rownames_to_column('Geneid')), by='Geneid') %>% pull(Length)

Human.var <- GetVarianceOfUnderlyingGamma_lengthAdjusted_FromTable(HumanCountTable, HumanGeneLengths)



ToPlot <- data.frame(HumanVar=Human.var, 
                     ChimpVar=Chimp.var, 
                     ChimpMean=apply(log(ChimpCountTable/colSums(ChimpCountTable)/ChimpGeneLengths*1000) ,1,mean))
ToPlot[mapply(is.infinite, ToPlot)] <- NA
R<-cor(log(ToPlot$HumanVar), log(ToPlot$ChimpVar), method='pearson')
lb1 <- paste("~R^2==~", round(R**2,2))
ggplot(ToPlot, aes(x=Human.var, y=Chimp.var)) +
  geom_point(alpha=0.05) +
  scale_x_continuous(trans='log10', limits=c(1E-7, 1E3), name=expression(paste("Human variance, ", hat(mu)^"2","/", hat(theta)))) +
  scale_y_continuous(trans='log10', limits=c(1E-7, 1E3), name=expression(paste("Chimp variance, ", hat(mu)^"2","/", hat(theta)))) +
  annotate("text",x=1E1,y=1E-7, label=lb1, hjust=0, vjust=-1, parse=TRUE) +
  coord_fixed() +
  theme_bw()

R<-cor(log(ToPlot$HumanVar), log(ToPlot$ChimpVar), method='pearson')
lb1 <- paste("~R^2==~", round(R**2,2))
ggplot(ToPlot, aes(x=Human.var, y=Chimp.var)) +
  geom_point(alpha=0.05) +
  scale_x_continuous(trans='log10', limits=c(1E-2, 1E1), name=expression(paste("Human overdispersion estimate, ", "1","/", hat(theta)))) +
  scale_y_continuous(trans='log10', limits=c(1E-2, 1E1), name=expression(paste("Chimp overdispersion estimate, ", "1","/", hat(theta)))) +
  annotate("text",x=1E-1,y=1E-2, label=lb1, hjust=0, vjust=-1, parse=TRUE) +
  coord_fixed() +
  theme_bw()

#Plot of this statistic vs expression level:
R<-cor(ToPlot$ChimpMean, log(ToPlot$ChimpVar), method="pearson", use="pairwise.complete.obs")
lb1 <- paste("~R^2==~", round(R**2,2))
ggplot(ToPlot, aes(x=ChimpMean, y=Chimp.var)) +
  geom_point(alpha=0.05) +
  xlab("Chimp mean expression (log(RPKM), though scale is off)") +
  scale_y_continuous(trans='log10', limits=c(1E-2, 1E1), name=expression(paste("Chimp overdispersion estimate, ", "1","/", hat(theta)))) +
  annotate("text",x=-6,y=0.1, label=lb1, hjust=1, vjust=-1, parse=TRUE) +
  theme_bw()

```

