---
title: "Check RNA-seq PCs"
author: "Ben Fair"
date: "3/20/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r load-libraries, message=F, warning=F}
library(plyr)
library(reshape2)
library(tidyverse)
library(readxl)
library(knitr)
library(ggrepel)
```

```{r make-tidy-data}
# Read in kinship data
YerkesKinship_from_pedigree <- as.data.frame(read_excel("../data/Metadata.xlsx", sheet="Yerkes.coan", col_types = c("text", "text", "numeric")))
kable(head(YerkesKinship_from_pedigree))

Kinship_from_KING.WGS <- read.table("../output/PopulationStructure/king.kin", header=T, stringsAsFactors = F)
kable(head(Kinship_from_KING.WGS))
```
Plot the kinship matrix obtained from Yerkes pedigree data, as well as the matrix obtained from whole genome SNP data...

```{r}
Yerkes.Matrix <-acast(YerkesKinship_from_pedigree, ind1 ~ ind2, value.var="coeff", fill=0)
melt(Yerkes.Matrix) %>%
  ggplot(aes(x=Var1, y=Var2, fill=value)) +
    geom_tile() +
    scale_fill_gradient(low="blue", high="red", limits=c(-0.5, 0.5)) +
    theme(text = element_text(size=4), axis.text.x = element_text(angle=90, hjust=1))

WGS.Matrix <- acast(Kinship_from_KING.WGS, ID1 ~ ID2, value.var="Kinship", fill=0) + acast(Kinship_from_KING.WGS, ID2 ~ ID1, value.var="Kinship", fill=0)
melt(WGS.Matrix) %>%
  ggplot(aes(x=Var1, y=Var2, fill=value)) +
    geom_tile() +
    scale_fill_gradient(low="blue", high="red", limits=c(-0.5, 0.5)) +
    theme(text = element_text(size=7), axis.text.x = element_text(angle=90, hjust=1))
```


Let's fix a known sample mislabel... 554_2 is actually 554. And 554 is actually unknown.
```{r}
Kinship_from_KING.WGS$ID1 <- mapvalues(Kinship_from_KING.WGS$ID1, from=c("554_2"), to=c("554"))
Kinship_from_KING.WGS$ID2 <- mapvalues(Kinship_from_KING.WGS$ID2, from=c("554_2"), to=c("554"))
```


Now I want to correlate the kinship-coefficients from Yerkes pedigree to those from the whole genome SNP data (KING algorithm). Easiest way I could think to match up the pairwise kinship coefficients is to make a new field that is the sorted ID pair and then merge the Yerkes coefficient-table to the KING-table by the sorted ID pairs...

```{r}
KSort <- Kinship_from_KING.WGS %>%
  mutate(Teams = paste(pmin(ID1, ID2), pmax(ID1, ID2), sep= " - "))
YSort <- YerkesKinship_from_pedigree %>%
  mutate(Teams = paste(pmin(ind1, ind2), pmax(ind1, ind2), sep= " - ")) %>%
  distinct(Teams, .keep_all=T)
# Merge and plot
Merged <- merge(KSort, YSort, by="Teams")
Merged$label <- Merged$Team
Merged$label[Merged$coeff==0] <- ""

ggplot(Merged, aes(x=coeff, y=Kinship, label=label)) +
  geom_point() +
  geom_abline(slope=1, intercept=0, color="red") +
  geom_text_repel(size=2.5) +
  xlab("Kinship from Yerkes pedigree") +
  ylab("Kinship from WGS (KING algorithm)")
  # geom_jitter()
```

The expected value of Kinships coefficients from KING seem to match pedigree info, with the exception of sample 554 (not to be confused with 554_2 which for purposes of matching KING coefficients to pedigree coefficients we relabelled as 554) which we had prior knowledge to be a mislabelled sample... Hence why 554 is unrelated to a point labelled 554 (actually 554_2). Assuming sample 554 came from Yerkes, we might be able to correlate its kinship coefficients to other Yerkes chimps to help identify it. Bryan Pavlovic already did this for me by looking at the kinship matrices by hand and concluded that 554 is most likely Booka, but here I will repeat that analysis...

```{r}
# Find who 554 is related to from whole genome SNP data.
WGS.Matrix['554',] %>%
  sort(decreasing=T) %>%
  head() %>%
  t() %>% kable()
```
...554 looks related to 495 (Amos), perhaps 2nd or 3rd degree relationship

```{r}
# Find who to Amos is related from Yerkes Pedigree
Yerkes.Matrix['495',] %>%
  sort(decreasing=T) %>%
  head(20) %>%
  t() %>% kable()
```
Can eliminate 495, 724, 462 as they are all also part of my cohort, and not related to 554.



## Update for presentation:
make same scatter plot (kinship estimated from wgs vs kinship from pedigree) based on the gemma-derived kinship matrix that i actually used for eqtl calling

```{r}
SampleLabels <- read.table('../output/ForAssociationTesting.temp.fam', stringsAsFactors = F)$V2
GemmaMatrix <- as.matrix(read.table('../output/GRM.cXX.txt'))
colnames(GemmaMatrix) <- SampleLabels
row.names(GemmaMatrix) <- SampleLabels

KSort <- melt(GemmaMatrix) %>%
  mutate(ID1=as.character(Var1), ID2=as.character(Var2)) %>%
  mutate(Teams = paste(pmin(ID1, ID2), pmax(ID1, ID2), sep= " - ")) %>%
  distinct(Teams, .keep_all=T)
YSort <- YerkesKinship_from_pedigree %>%
  mutate(Teams = paste(pmin(ind1, ind2), pmax(ind1, ind2), sep= " - ")) %>%
  distinct(Teams, .keep_all=T)
# Merge and plot
Merged <- merge(KSort, YSort, by="Teams")
Merged$label <- Merged$Team
Merged$label[Merged$coeff==0] <- ""

Merged %>%
  filter(!ID1==ID2) %>%
ggplot(aes(x=coeff, y=value*2, label=label)) +
  geom_point() +
  geom_abline(slope=1, intercept=0, color="red") +
  geom_text_repel(size=2.5) +
  xlab("Kinship from Yerkes pedigree") +
  ylab("Kinship from WGS (GEMMA)") +
  theme_bw()
  # geom_jitter()


```

