---
title: "Final_plots_1_CumDist"
author: "Benjamin Fair"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
  html_document:
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)
```

## Introduction

This markdown will be used to generate publication quality figures that I have already made at some point in my exploratory Rmarkdown analyses. This markdown will just make plots that i anticipate for publication that are shown as empricial cdf plots for various unrelated things.


First, load necessary libraries
```{r}
library(tidyverse)
library("org.Hs.eg.db")
library(readxl)
library(biomaRt)
library(broom)

source("../code/CustomFunctions.R")
```

Load eGene data for humans and chimp.

```{r}
ChimpOverdispersionTable <- read.table("../output/OverdispersionEstimatesFromChimp.NoLengthNorm.txt", header=T, sep='\t')

EgenesTested <- TsvToCombinedEgenes(Chimp.tsv = "../output/ChimpEgenes.eigenMT.txt.gz", Human.tsv = "../data/GTEX_v8_eGenes/Heart_Left_Ventricle.v8.egenes.txt.gz", SysToID.tsv = "../data/Biomart_export.Hsap.Ptro.orthologs.txt.gz", HumanTsvType = "GTEx")

DE.results <- read.table("../data/DE_genes.NoVirusChallangedInds.txt", sep='\t', header=T)

HumanEgeneCount <- NULL
HumanEgeneCount <- 495 #Use 495 instead of 500 because of ties in ranks. 495 will get you 500.
EgenesTested.grouped <- AddGroups(EgenesTested, HumanEgeneCount=HumanEgeneCount)

#hypergeom test (using one sided fisher test function) for overlap of eGenes
EgenesTested.grouped %>%
  dplyr::count(group, name="freq") %>%
  pull(freq) %>%
  matrix(ncol=2, nrow=2) %>% fisher.test(alternative="greater")

#Same but using FDR=0.1 threshold in both species
AddGroups(EgenesTested, HumanEgeneCount=NULL) %>%
  dplyr::count(group, name="freq") %>%
  pull(freq) %>%
  matrix(ncol=2, nrow=2) %>% fisher.test(alternative="greater")
```

## Make plots.

Note that plots make come out with odd looking dimensions, that only look better through my experimentation with ggsave that saves the final figure to a file not included in the rendered markdown.

### DE by eGene group

First plot interspecies DE effect size by eGene group. Only save the plot for the ~8000 genes that are significantly DE. (Similar results are obtained if you make the plot for all genes as shown first). eGenes are more likely to be differentially expressed between species. this is even more true for species shared eGenes. All consistent with most neutral selection. Traits that are varied within a species, should be varied between species as well.

```{r}

DE.results %>% pull(DE) %>% table()


Plot.Interpecies.DE.byGroup(EgenesTested.grouped, DE.results)$plot
Plot.Interpecies.DE.byGroup(EgenesTested.grouped, DE.results)$PvalTable

DE.By.Group <- Plot.Interpecies.DE.byGroup(EgenesTested.grouped, (DE.results %>% filter(DE != 0)))

DE.Fig <- ggdraw(DE.By.Group$plot +                            guides(color=guide_legend(nrow=2,byrow=TRUE))
) +
  draw_grob(tableGrob(DE.By.Group$PvalTable, rows=NULL, theme=ttheme_default(colhead=list(fg_params = list(parse=TRUE)))), x=0.6, y=0.3, width=0.3, height=0.3)
DE.Fig
ggsave(paste0("../figures/OriginalArt/eGeneGroup.", HumanEgeneCount, ".DE.pdf"), DE.Fig, height=120, width=120, units="mm")
```


### Cell specificity by group

```{r}

Plot.tau.byGroup <- function(ToPlot, tauFile){
  CellSpecificity <- read.delim(tauFile, col.names=c("gene", "tau"))
  ToPlot <- left_join(ToPlot, CellSpecificity, by=c("H.gene"="gene"))
  
  SharedLabel <- "both"
  NeitherLabel <- "neither"
  ChimpLabel <- "chimp"
  HumanLabel <- "human"
  AlternativeHypothesis <- c(
    "chimp > neither",
    "human > neither",
    "both > chimp",
    "both > human")
  Pvalues <- c(
    format.pval(wilcox.test(data=ToPlot %>% filter(group %in% c(ChimpLabel,NeitherLabel)), tau ~ group, alternative="greater")$p.value, 2),
    format.pval(wilcox.test(data=ToPlot %>% filter(group %in% c(HumanLabel,NeitherLabel)), tau ~ group, alternative="greater")$p.value, 2),
    format.pval(wilcox.test(data=ToPlot %>% filter(group %in% c(SharedLabel,ChimpLabel)), tau ~ group, alternative="greater")$p.value, 2),
    format.pval(wilcox.test(data=ToPlot %>% filter(group %in% c(SharedLabel,HumanLabel)), tau ~ group, alternative="greater")$p.value, 2)
  )
  PvalTable <- data.frame(AlternativeHypothesis,Pvalues)
  colnames(PvalTable) <- c('"H"[a]', "`P-value`")
  tau.byGroup.plot <- ggplot(ToPlot, aes(color=group,x=tau)) +
    stat_ecdf(geom = "step") +
    ylab("Cumulative frequency") +
    xlab("Cell type specificity, tau") +
    labs(color = "eGene discovered in") +
    theme_bw() +
    theme(legend.position="bottom")

  return(list(plot=tau.byGroup.plot, PvalTable=PvalTable))
}

A<-Plot.tau.byGroup(EgenesTested.grouped, "../output/TissueSpecificity/tau.txt")
A$plot
A$PvalTable
```



### Percent identity by eGene group

plot percent identity by eGene group. Hypothesis being that eGenes are genes which are evolving neautrally, so they will have less conservation at coding level (more percent non-identity)

```{r}
Identity.By.Group <- Plot.PercentNonIdentity.byGroup(EgenesTested.grouped)

Identity.Fig <- ggdraw(Identity.By.Group$plot +                            guides(color=guide_legend(nrow=2,byrow=TRUE))
) +
  draw_grob(tableGrob(Identity.By.Group$PvalTable, rows=NULL, theme=ttheme_default(colhead=list(fg_params = list(parse=TRUE)))), x=0.6, y=0.3, width=0.3, height=0.3)
Identity.Fig
ggsave(paste0("../figures/OriginalArt/eGeneGroup.", HumanEgeneCount, ".identity.pdf"), Identity.Fig, height=120, width=120, units="mm")
```

### plot dN/dS by group

plot dN/dS ratio by group. Hypothesis being that eGenes are genes which are evolving neautrally, so they will have less conservation at coding level (higher dN/dS)
```{r}
dNdS.By.Group <- Plot.dNdS.byGroup(EgenesTested.grouped)

dNdS.Fig <- ggdraw(dNdS.By.Group$plot +                                       guides(color=guide_legend(nrow=2,byrow=TRUE))
) +
  draw_grob(tableGrob(dNdS.By.Group$PvalTable, rows=NULL, theme=ttheme_default(colhead=list(fg_params = list(parse=TRUE)), base_size = 10)), x=0.65, y=0.3, width=0.25, height=0.25)
dNdS.Fig
ggsave(paste0("../figures/OriginalArt/eGeneGroup.", HumanEgeneCount, ".dNdS.pdf"), dNdS.Fig, height=120, width=120, units="mm")
```

### Plot difference in dispersion by group

plot difference in dispersion by group. Hypothesis being that chimp eGenes will be more dispersed in chimp, human eGenes will be more dispersed in human, consistent with there being a measurable genetic component to dispersion.

```{r}
Dispersion.By.Group <-Plot.DispersionDifference.byGroup(EgenesTested.grouped, ChimpOverdispersionTable)


Dispersion.Fig <- ggdraw(Dispersion.By.Group$plot +
                           guides(color=guide_legend(nrow=2,byrow=TRUE))
                         ) +
  draw_grob(tableGrob(Dispersion.By.Group$PvalTable, rows=NULL, theme=ttheme_default(colhead=list(fg_params = list(parse=TRUE)))), x=0.6, y=0.25, width=0.3, height=0.3)
Dispersion.Fig
ggsave(paste0("../figures/OriginalArt/eGeneGroup.", HumanEgeneCount, ".dispersion.pdf"), Dispersion.Fig, height=120, width=120, units="mm")
```

### Difference in fdr rank vs difference in dispersion

Another way to look at the same thing. This is important investigate since considered using this difference in ranks as a less arbitrary way of categorizing eGenes vs non eGenes and doing gene overlap analysis.

```{r}
DifferenceInFDR.vs.DiffInDispersion <- EgenesTested.grouped %>%
  mutate(H.rank = dense_rank(H.FDR),
         C.rank = dense_rank(C.FDR)) %>%
  filter(C.FDR <= 0.1) %>%
  mutate(RankDiff=C.rank-H.rank) %>%
  left_join(ChimpOverdispersionTable, by=c("H.gene"="gene")) %>%
  mutate(DispersionDiff = Chimp.Residual-Human.Residual ) %>%
  filter(DispersionDiff < 2 & DispersionDiff > -2)

cor.test(x=DifferenceInFDR.vs.DiffInDispersion$RankDiff, y=DifferenceInFDR.vs.DiffInDispersion$DispersionDiff)

ggplot(DifferenceInFDR.vs.DiffInDispersion, aes(x=RankDiff, y=DispersionDiff)) +
  geom_point()
```

Ok so maybe the gene categorization makes more sense

### essentiality by group

Not a plot but just make some contingency tables and do some tests to see if eGenes (in human, and or chimp) are more to be non essential. Perhaps the more appropriate way to test this but for now I will just stick to 2x2 contingency tables and doing a Fisher test with the expectation that chimp eGenes are more likely to be non essential than genes that are not eGenes in either species. Same for human eGenes. Furthermore, genes that are eGenes in both will be even more likely to be non essential than eGenes in just one species.


```{r}
pLI.data <- as.data.frame(read_excel("../data/Lek_etal_nature19057-SI Table 13.xlsx", sheet=2))

human = useMart("ensembl", dataset = "hsapiens_gene_ensembl")
Symbols <- getBM(attributes=c("hgnc_symbol", "ensembl_gene_id"), filters="ensembl_gene_id", values=ChimpOverdispersionTable$gene, mart=human)

LargeContingencyTable <- EgenesTested.grouped %>%
  left_join(Symbols, by=c("H.gene" ="ensembl_gene_id")) %>%
  left_join(pLI.data, by=c("hgnc_symbol"="gene")) %>%
  mutate(pLI.Score.essentiality=case_when(
    pLI>=0.1  ~ "Essential",
    pLI<=0.1 ~ "NonEssential")) %>%
  filter(pLI.Score.essentiality %in% c("Essential", "NonEssential")) %>%
  group_by(group, pLI.Score.essentiality) %>%
  dplyr::summarise(n = n()) %>% 
  spread(key=group, value=n) %>%
  column_to_rownames("pLI.Score.essentiality")

LargeContingencyTable

fisher.test(LargeContingencyTable[,c("chimp", "neither")], alternative="less")

fisher.test(LargeContingencyTable[,c("both", "chimp")], alternative="less")
fisher.test(LargeContingencyTable[,c("both", "human")], alternative="less")

ChimpContingencyTable <- LargeContingencyTable %>%
  rownames_to_column("essentiality") %>%
  mutate(eGene=both+chimp,
         not.eGene=human+neither) %>%
  dplyr::select(essentiality, eGene, not.eGene) %>%
  column_to_rownames("essentiality")

HumanContingencyTable <- LargeContingencyTable %>%
  rownames_to_column("essentiality") %>%
  mutate(eGene=both+human,
         not.eGene=chimp+neither) %>%
  dplyr::select(essentiality, eGene, not.eGene) %>%
  column_to_rownames("essentiality")

# Are chimp eGenes (not chimp specific eGenes, but all chimp eGenes) more likely to be non-essential
fisher.test(ChimpContingencyTable, alternative="less")

# Are human eGenes (not human specific eGenes, but all human eGenes, top500) more likely to be non-essential
fisher.test(HumanContingencyTable, alternative="less")

```

Seems the paradigm of eGenes are more likely non-essential is clearly true in human, and the pointing that way for chimp but could be due to chance.

### Number of tissues eGene in GTEx by group.

Are chimp eGenes more likely to be eGenes across many GTEx tissues?

```{r}
GTEx <- read.table("../data/AllGTExTissues.egenes.txt", header=T, sep='\t')

Threshold=0.1
TissueEgeneCount <- data.frame(TissueCount=rowSums(GTEx[,-1]<=Threshold, na.rm=T), Gene.stable.ID=gsub("\\.\\d+", "", GTEx$gene_id, perl=T))

EgenesTested.grouped %>%
  left_join(TissueEgeneCount, by=c("H.gene"="Gene.stable.ID")) %>%
  # filter(group %in% c("both", "human")) %>%
  ggplot(aes(color=group,x=TissueCount)) +
  stat_ecdf(geom = "step") +
  ylab("Cumulative frequency") +
  xlab("TissueCount") +
  theme_bw() +
  theme(legend.position = c(.80, .2), legend.title=element_blank())

```

Ok that was silly. Obviously human eGenes will be eGenes in more tissues. Although it is worth noting that the both eGene group doesn't look different than the human eGene group.

Perhaps the more relevant grouping for this question is comparing chimp eGene and not chimp eGene, and asking how many GTEx tissues it is an eGene in..

```{r}
ToPlot <- EgenesTested.grouped %>%
  left_join(TissueEgeneCount, by=c("H.gene"="Gene.stable.ID")) %>%
  mutate(Chimp.eGene = case_when(
    group == "both" ~ "chimp eGene",
    group == "chimp" ~ "chimp eGene",
    group == "neither" ~ "not chimp eGene",
    group == "human" ~ "not chimp eGene"
  ))

Test <- wilcox.test(data=ToPlot, TissueCount ~ Chimp.eGene)
lb1 = paste0('P==', format.pval(Test$p.value, 2))

  ggplot(ToPlot, aes(color=Chimp.eGene,x=TissueCount)) +
  stat_ecdf(geom = "step") +
  ylab("Cumulative frequency") +
  xlab("TissueCount") +
  theme_bw() +
  annotate("text",x=Inf,y=-Inf, label=lb1, hjust=1.1, vjust=-0.1, parse=TRUE) +
  theme(legend.position="bottom")


ggsave("../figures/OriginalArt/eGeneGroup.num.GTEx.eGenes.pdf", units="mm", height=120, width=120)
```

### Plots about relating dn/ds to dispersion.

```{r}

A<-EgenesTested.grouped %>%
  left_join(ChimpOverdispersionTable, by=c("H.gene"="gene")) %>%
  mutate(MeanDisp=(Chimp.Residual+Human.Residual)/2,
         DispDifference=Chimp.Residual-Human.Residual) %>%
  mutate(dn.ds=dN/dS)

A %>%
  mutate(dn.ds.Quintile = ntile(dn.ds,10)) %>%
  # filter(dn.ds.Quintile==1) %>% pull(dn.ds) %>% median(na.rm=T)
  filter(dn.ds.Quintile %in% c(4,10)) %>%
  ggplot(aes(x=MeanDisp, color=as.factor(dn.ds.Quintile))) +
  stat_ecdf(geom = "step") +
  # scale_x_continuous(trans="log10", limits=c(0.01,10)) +
  ylab("Cumulative frequency") +
  theme_bw()
  


A %>% 
  do(tidy(cor.test(.$MeanDisp, log(.$dn.ds), method='spearman', alternative = 'less')))

A %>% 
  # filter(DispDifference<1) %>%
  do(tidy(cor.test(.$MeanDisp, .$PercentIdentitiyHumanToChimp, method='spearman')))

A %>% 
  # filter(DispDifference<1) %>%
  do(tidy(cor.test(.$dn.ds, .$PercentIdentitiyHumanToChimp, method='spearman')))

```

Ok I'm a little confused here. It seems I see the opposite of the expected effect with dn/ds. That is, higher dispersion genes have lower dn/ds. But I see the expected effect with percent identity (high dispersion genes are have less % identity). And there is the expected relationship with dn/ds (higher dn/ds is lower %identity.)... Maybe instead of using dn/ds between human and chimp I should be using a dn/ds that counts differences between many mammals as in [Lindblad-Toh et al](https://www.nature.com/articles/nature10530).

```{r}
PanMammal.dnds <- read.delim("../data/Overall_dN_dS.bed", col.names = c("chrom", 'start', "stop", "gene", "score", "strand", "blockstart", "blockstop", "color", "ensemblprotein", "dn.ds"))

#Check correlation between chimp/human dn/ds and pan-mammal dn/ds
PanMammal.dnds %>%
  left_join(Symbols, by=c("gene"="hgnc_symbol")) %>%
  inner_join(EgenesTested.grouped, by=c("ensembl_gene_id"="H.gene")) %>%
  ggplot(aes(x=log10(dn.ds), y=log10(dN/dS))) +
  geom_point()

#Check correlation between dispersion and pan-mammal dn/ds
Correlation <- PanMammal.dnds %>%
  left_join(Symbols, by=c("gene"="hgnc_symbol")) %>%
  inner_join(ChimpOverdispersionTable, by=c("ensembl_gene_id"="gene")) %>%
  do(tidy(cor.test(.$Human.Residual, .$dn.ds, method='spearman')))
Correlation
lb1 = paste0('atop(rho==', signif(Correlation$estimate,2), ', P==', format.pval(Correlation$p.value, 2), ')')
```

Ok. now I see the expected correlation. Let's plot by breaking dn/ds into quantiles. maybe deciles. could show as a scatter but it the direction of the effect will be more visually obvious by plotting the ecdf of some extreme quantiles (like top 10% and bottom 10%)

```{r}
PanMammal.dnds %>%
  left_join(Symbols, by=c("gene"="hgnc_symbol")) %>%
  inner_join(ChimpOverdispersionTable, by=c("ensembl_gene_id"="gene")) %>%
  mutate(dn.ds.ntile=ntile(dn.ds,10)) %>%
  filter(dn.ds.ntile %in% c(1,10)) %>%
  ggplot(aes(x=Human.Residual, color=as.factor(dn.ds.ntile))) +
  stat_ecdf(geom = "step") +
  scale_x_continuous(limits=c(-2.5,2.5)) +
  ylab("Cumulative frequency") +
  xlab("Dispersion (Human)") +
  scale_color_discrete(name = "dN/dS decile", labels = c("0-0.1; Low coding divergence", "0.9-1.0; High coding divergence")) +
  theme_bw() +
  annotate("text",x=Inf,y=-Inf, label=lb1, hjust=1, vjust=-0.1, parse=TRUE) +
  theme(legend.position = c(.5, .8),
        legend.key = element_rect(colour = "transparent", fill = "transparent"),
        legend.background = element_rect(fill=NA))
ggsave("../figures/OriginalArt/DispersionByDnDs.pdf", units="mm", height=70, width=70)

```

Let's make the analogous plot for percent identity, since I have it elsewhere in the manuscript.

```{r}

Correlation <- EgenesTested.grouped %>%
  inner_join(ChimpOverdispersionTable, by=c("H.gene"="gene")) %>%
  do(tidy(cor.test(.$Human.Residual, .$PercentIdentitiyHumanToChimp, method='spearman')))
Correlation
lb1 = paste0('atop(rho==', signif(Correlation$estimate,2), ', P==', format.pval(Correlation$p.value, 2), ')')

EgenesTested.grouped %>%
  inner_join(ChimpOverdispersionTable, by=c("H.gene"="gene")) %>%
  mutate(dn.ds.ntile=ntile(PercentIdentitiyHumanToChimp*-1,10)) %>%
  filter(dn.ds.ntile %in% c(1,10)) %>%
  ggplot(aes(x=Human.Residual, color=as.factor(dn.ds.ntile))) +
  stat_ecdf(geom = "step") +
  scale_x_continuous(limits=c(-2.5,2.5)) +
  ylab("Cumulative frequency") +
  xlab("Dispersion (Human)") +
  scale_color_discrete(name = "%amino acid identity decile", labels = c("0.9-1.0; Low coding divergence", "0.0-0.1; High coding divergence")) +
  theme_bw() +
  annotate("text",x=Inf,y=-Inf, label=lb1, hjust=1, vjust=-0.1, parse=TRUE) +
  theme(legend.position = c(.5, .8),
        legend.key = element_rect(colour = "transparent", fill = "transparent"),
        legend.background = element_rect(fill=NA))
ggsave("../figures/OriginalArt/DispersionByPercentIdentity.pdf", units="mm", height=70, width=70)
```


### TATA box versus dispersion
```{r}
TSS.annotations <- read.table("../data/TSS_annotations/TssClassificationAnnotations.txt.gz", sep='\t', header=T, stringsAsFactors = F, fill=T)
TATA.annotations <- read.table("../data/TSS_annotations/TataAnnotations.txt.gz", sep='\t', header=T, stringsAsFactors=F, quote = "", fill=T)


TSS.filtered.annotations <- TSS.annotations %>% filter(TSSclassification=="yes") %>%
  left_join(TATA.annotations, by=c("refTSSID")) %>%
  dplyr::select(refTSSID, Chr, Start, End, Strand, Distance.to.TSS, Nearest.Ensembl, TATA=TATA.Box.TBP..Promoter.Homer.Distance.From.Peak.sequence.strand.conservation.) %>%
  mutate(TATA.dist = as.numeric(gsub("^(.+?)\\(.+", "\\1", TATA, perl=T)))

#TATA boxes are ~35bp upstream of TSS
qplot(as.numeric(TSS.filtered.annotations$TATA.dist), xlim=c(-50,10))

#annotated genes TSS are close distance to TSS from this database (based on CAGE and other data), but slightly biased
qplot(TSS.filtered.annotations$Distance.to.TSS, xlim=c(-400,400))

#let's keep only the TATA box genes where the TATA motif is within -40 to -30 of the TSS, assuming that the others are false positives in the functional sense. Also get rid of TSS where distance of gene to TSS is unreasonably far
TATA.GeneList <- TSS.filtered.annotations %>%
  filter(TATA.dist >= -40 & TATA.dist <= -30) %>%
  filter(Distance.to.TSS >= -200 & Distance.to.TSS <= 500) %>%
  distinct(Nearest.Ensembl, .keep_all = T) %>%
  pull(Nearest.Ensembl)
Not.TATA.GeneList <- TSS.filtered.annotations %>%
  distinct(Nearest.Ensembl, .keep_all = T) %>%
  filter(!Nearest.Ensembl %in% TATA.GeneList) %>%
  pull(Nearest.Ensembl)

TATA.disp <- ChimpOverdispersionTable %>%
  mutate(PromoterType=case_when(
    gene %in% TATA.GeneList ~ "TATA box",
    gene %in% Not.TATA.GeneList ~ "No TATA box"
  )) %>%
  filter(!is.na(PromoterType))

Test <- wilcox.test(TATA.disp %>%
                    filter(PromoterType=="TATA box") %>%
                    pull(Human.Residual),
                    TATA.disp %>%
                    filter(PromoterType=="No TATA box") %>%
                    pull(Human.Residual))
lb1 = paste0('P', format.pval(Test$p.value, 2))
ggplot(TATA.disp, aes(x=Human.Residual, color=PromoterType)) +
  stat_ecdf(geom = "step") +
  xlim(c(-2.5,2.5)) +
  ylab("Cumulative frequency") +
  theme_bw() +
  annotate("text",x=Inf,y=-Inf, label=lb1, hjust=1.1, vjust=-0.1, parse=TRUE) +
  xlab("Dispersion (Human)") +
  scale_color_discrete(name = "Promoter type") +
  theme(legend.position = c(.3, .8),
        legend.key = element_rect(colour = "transparent", fill = "transparent"),
        legend.background = element_rect(fill=NA))

ggsave("../figures/OriginalArt/DispersionByTATABox.pdf", units="mm", height=70, width=70)

```


### Essentiality versus dispersion
```{r}
Essentiality <- pLI.data %>%
  left_join(Symbols, by=c("gene"="hgnc_symbol")) %>% 
  inner_join(ChimpOverdispersionTable, by=c("ensembl_gene_id"="gene")) %>%
  mutate(Essentiality=case_when(
    pLI > 0.9 ~ "Essential; pLI>0.9",
    pLI < 0.1 ~ "Non-essential; pLI<0.1",
    )) %>%
  filter(!is.na(Essentiality))

Test <- wilcox.test(Essentiality %>%
                    filter(Essentiality=="Essential; pLI>0.9") %>%
                    pull(Human.Residual),
                    Essentiality %>%
                    filter(Essentiality=="Non-essential; pLI<0.1") %>%
                    pull(Human.Residual))
lb1 = paste0('P==', format.pval(Test$p.value, 2))

ggplot(Essentiality, aes(x=Human.Residual, color=Essentiality)) +
  stat_ecdf(geom = "step") +
  xlim(c(-2.5,2.5)) +
  ylab("Cumulative frequency") +
  theme_bw() +
  scale_color_discrete(name = "ExAC score", labels = c(">0.9; LoF intolerant", "<0.1; LoF tolerant")) +
  xlab("Dispersion (Human)") +
  theme(legend.position = c(.4, .8),
        legend.key = element_rect(colour = "transparent", fill = "transparent"),
        legend.background = element_rect(fill=NA)) +
  annotate("text",x=Inf,y=-Inf, label=lb1, hjust=1.1, vjust=-0.1, parse=TRUE)
ggsave("../figures/OriginalArt/DispersionByEssentiality.pdf", units="mm", height=70, width=70)

```

### dN/dS by group plot2
...Remaking this plot with this new dN.dS calculation from 26 mammals.

```{r}
dNdS.By.Group2 <- PanMammal.dnds %>%
  left_join(Symbols, by=c("gene"="hgnc_symbol")) %>%
  inner_join(EgenesTested.grouped, by=c("ensembl_gene_id"="H.gene")) %>%
  mutate(dN.dS=dn.ds) %>%
  Plot.dNdS.byGroup2()

dNdS.Fig2 <- ggdraw(dNdS.By.Group2$plot +                                       guides(color=guide_legend(nrow=2,byrow=TRUE))
) +
  draw_grob(tableGrob(dNdS.By.Group2$PvalTable, rows=NULL, theme=ttheme_default(colhead=list(fg_params = list(parse=TRUE)), base_size = 10)), x=0.65, y=0.3, width=0.25, height=0.25)
dNdS.Fig2
ggsave(paste0("../figures/OriginalArt/eGeneGroup.", HumanEgeneCount, ".dNdS_2.pdf"), dNdS.Fig2, height=120, width=120, units="mm")
```


## Conclusions


