---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
---

Welcome to my research website.

## QC
- [fastqc](../output/multiqc.html)

### Genotype data
- [Genotype call rates]()
- [kinship](20190321_Check-Kinship-And-PopulationStructure.html)
- [PCA](20190326_PCA.html)
- [Admixture](20190326_Admixture.html)

### Expression data
- [Check that RNA-seq data segregates by biological sample over sequencing flow cell](20190325_MergingRNASeqLanes.html)
- [Checking RNA-seq data for covariates](20190320_Check-RNAseq-PCs.html)
- [STAR Aligner vs kallisto pseudoaligner for gene quantification](20190429_RNASeqSTAR_quantifications.html)

### Prepare data for eQTL testing
- [Make phenotype table for testing](20190327_MakeFamPhenotypeFile.html)
- [Make covariate table for testing](20190327_MakeCovariateFiles.html)

## Exploration

### Association testing with various models
- [first iteration](20190412_Check_eQTLs.html): description: lmm with KING-robust GRM thresholded at 0, and 3 genotype PCs
- [Check residuals after regressing out some covariates](20190421_RegressOutRNASeqPCs.html)
- [second iteration](20190424_Check_eQTLs.html): description: lm with 5 genotype PCs (PCs 4 and 5 takes into account some first hand relatedness) and more stringent genotype filtering. Also, outlier sample MD_And dropped from analysis
- [Third iteration](20190428_Check_eQTLs.html)
- [fourth iteration](20190502_Check_eQTLs.html): description, lmm with 4 PCs and 3 Genotype PCs, used STAR RNA-seq CPM for less outliers. Fixed big bug that was permuting samples, resulting in no true hits in previous iterations. Here I used standardization and qqnorm.

### Conservation and GO/GSEA analysis
- [GO analysis, FDR=0.1](20190521_eQTL_CrossSpeciesEnrichment.html) overlap enrichment analysis of eGenes across humans and chumps, and gene ontology analysis of eGenes based on eGene classification defined at FDR=0.1 threshold. Also, a set of similar analyses using GSEA methodology, which is based on relative ranking of eGenes between species.
- [Conservation analysis, FDR=0.1](20190606_eGene_Conservation.html) analysis of conservation of coding sequence (percent identity and dN/dS) based on eGene classification defined at FDR=0.1
- [Conservation analysis, HumanTop600_eGenes](20190606_eGene_Conservation_TopN.html) analysis of conservation of coding sequence (percent identity and dN/dS) based on eGene classification defined at FDR=0.1 for chimp and top600 qvalue genes for human.
- [Tissue sharing, both FDR=0.1 and Top600](20190821_eGeneTissueCount.html) analysis of number of GTEx tissues eGenes are detected in for shared and species specific eGenes. As before, I classify species specific and shared eGenes using either FDR=0.1 for both species, or using FDR-0.1 for chimp and top600 qvalue for human.
- [Conservation analysis, most high-variance genes](20190617_ExpressionVarianceOfEgenes.html) analysis of conservation of coding sequence (percent identity and dN/dS) based on within species variance of expression. A useful comparison for the similar analysis above.
- [More thourough analyses looking at gene variance](20190716_VarianceInsteadOfEgenes.html) within and between species after adjusting for expression.
- [Different model of gene variance character](20190930_OverdispersionEstimates.html) using advice from Abhishek
- [Different model of gene variance character](20190930_OverdispersionEstimates_HBV_excluded.html) using advice from Abhishek. But this time leaving out the virus challenged chimps


### Power analysis for inter-species differential expression
- [PowerAnalysisFromOrinalDataset](20190613_PowerAnalysis.html) DE gene analysis based on subsampling ~39 chimp samples & 50 human samples (mostly GTEx)... Note that there are some outlier samples that I want to purge in later iterations of this analysis.

### Differential contacts
- Do differential DNA contacts between species explain differences in eGene character between species.
- [First crude analysis](20190627_DiffContactsEgenes.html) In this analysis I asked whether the there is a correlation between the rank difference in eGenes between species, and the difference in the sum of contacts in each species' cis-window for each gene
- There was a slight but significant correlation, but I worry about a potential bias introduced by the fact that eGene significance for humans was based on cis-eQTL calling with a 1MB window and chimp was a 250kB window. I should control for this.
- [Check that human lead snps in 250kB window are reasonable](20190808_HumanLeadSnps.html) As expected, the p-value for lead snps within 250kb is well correlated for the eGene p-value, with the exception that there are many genes with much lower eGene pvalues probably because there is a better SNP >250kb away
- [After controlling for cis-window-size](20190708_DiffContactsEgenes_CisWindowControlled.html)

### Shared polymorphic loci
- [QQ Plot of species shared polymorphisms](20190708_SpeciesSharedPolymorphisms_InitialQQPlot.html) Of all the shared polymorphic loci (putative targets of balancing selection), do they have inflated cis-eQTL P-values compared to a random set of snp-gene tests.

## Final figures

- todo... haven't build these markdowns yet.

## Reviewer comments

- See [reviewer comments](20200907_Response_OriginalComments.html)

